# Genome Assembly and Annotation

```{r, echo=FALSE, include=FALSE}
# load required libraries
up_packages = c("circlize", "clusterProfiler", "ComplexHeatmap", "ggplot2", "grid", "gridExtra", "magrittr", "RColorBrewer")
lapply(up_packages, require, character.only = TRUE)

# set working directory
mainDir = "/home/Andrea/Desktop/kveiks/manuscript/plotting_data/"
knitr::opts_knit$set(root.dir = mainDir)
```

To determine the genes in Kveiks samples, we need to do a _de novo_ assembly and then to run an _ab initio_ prediction of genes (using _Saccharomyes_ proteins and genes as guide). We will use SPAdes assembler, follower by Redundans pipeline to collapse redundant contigs.


## Genome Assembly

```{bash, eval=FALSE}
## SPAdes assembly 
while read line ; do
	python ~/bin/SPAdes-3.13.0-Linux/bin/spades.py \
		-1 ../00_trim_reads/"${line}".R1.tr.fq.gz \
		-2 ../00_trim_reads/"${line}".R2.tr.fq.gz \
		-o "${line}"_SPAdes \
		--threads 72 \
		-k 21,29,39,59,79,99,119,127
done < ../sample.lst

## redundans
docker run -v /home/andrea/03_KVEIK/:/mydata:rw -it lpryszcz/redundan2

while read line ; do
	/root/src/redundans/redundans.py \
		--verbose \
		--fastq /mydata/00_trim_reads/"${line}".R1.tr.fq.gz \
		/mydata/00_trim_reads/"${line}".R2.tr.fq.gz \
		--fasta /mydata/03_assemblies/"${line}".SPAdes.fa \
		--outdir /mydata/04_redundans/"${line}" \
		--threads 72 \
		--log /mydata/04_redundans/"${line}".SPAdes.redundans.log
done < mydata/sample.lst 

## Generate assembly stats
for file in *redundans.fa; do 
	perl ~/scripts/Nstat.pl $file > $file.Nstat;
done
echo "Sample"$'\t'"Total length (bp)"$'\t'"# contigs"$'\t'"longest (bp)"$'\t'"N50 (bp)" \
	> Vikings.assembly.stats.txt;
for file in *.Nstat; do
	NAME=$(basename $file .SPAdes.redundans.fa.Nstat);
	LONGEST=$(tail $file | head -n 1 | tr ':' '\t' | cut -f 1);
	TOT_LEN=$(grep "Total length" $file | sed 's/Total length of sequence://g');
	NCONT=$(grep "Total number" $file | sed 's/Total number of sequences://g');
	N50=$(grep "N50 stats:" $file | sed 's/.*sequences >= //g');
	echo $NAME$'\t'$TOT_LEN$'\t'$NCONT$'\t'$LONGEST$'\t'$N50; done |\
    	sed 's/ bp//g' >> Vikings.assembly.stats.txt;
done
```



```{r, echo=FALSE}
matrix.d = cbind(c("14R30", "14R6", "16R23", "16R37", "17P5", "19R18", "1R16", "21P1", "21R38",
                   "27R17", "28P1", "28P6", "28R21", "28R33", "28R8", "2R23", "38R16", "39R20", 
                   "3R11", "40R14", "40R1", "40R20", "41R10", "41R15", "42R20", "42R31", "44R32", "44R7", 
                   "45P5", "45R11", "46R12", "46R37", "6R15", "7R7", "8R19", "9R40", "Granvin1", 
                   "Hornindal1", "Hornindal2", "k7R25", "Laerdal2", "Muri", "SortdalEbbe1", "Voss1"),
                 c("12,414,423", "12,298,670", "11,582,806", "11,736,494", "12,617,870", "12,678,102", "12,277,498", "12,619,711", 
                   "11,572,760", "12,382,460", "11,529,812", "11,538,139", "11,505,145", "11,553,531", "11,532,904", "12,727,977", 
                   "12,143,606", "11,641,798", "11,548,489", "11,685,486", "11,704,488", "12,744,802", "11,540,301", "11,573,148", 
                   "11,703,681", "11,591,540", "12,693,136", "12,215,896", "11,627,693", "11,628,113", "11,704,521", "11,636,081", 
                   "12,950,275", "12,607,735", "12,422,210", "12,360,883", "15,983,330", "12,928,993", "12,899,850", "16,359,314", 
                   "11,889,233", "17,101,719", "12,443,720", "12,968,222"),
                 c("3,362", "3047", "183", "254", "3,805", "4,047", "4,194", "3,784", 
                   "144", "3,342", "149", "160", "155", "134", "154", "4,180", 
                   "1,558", "171", "128", "164", "200", "3,831", "158", "113", 
                   "281", "171", "3,929", "1,939", "158", "168", "202", "158", 
                   "4,606", "3,898", "3,398", "3,050", "6,647", "3,477", "3,480", "685", 
                   "933", "281", "2,328", "3,703"),
                 c("247,500", "248,000", "416,400", "524,500", "231,700", "113,400", "178,600", "279,100", 
                   "885,100", "170,700", "860,500", "594,200", "426,900", "543,200", "555,700", "254,400", 
                   "425,400", "663,700", "732,300", "832,500", "559,500", "161,000", "729,200", "725,000", 
                   "478,100", "798,200", "176,500", "493,000", "1,006,900", "662,400", "553,100", "569,500", 
                   "166,600", "176,500", "229,300", "278,500", "318,100", "333,200", "350,200", "913,200", 
                   "454,900", "975,000", "340,500", "264,600"),
                 c("34,894", "35,540", "243,962", "178,599", "35,905", "33,065", "32,500", "37,907", 
                   "279,921", "35,066", "270,301", "224,961", "243,601", "303,859", "318,176", "36,043", 
                   "117,315", "255,254", "403,046", "260,029", "157,697", "36,853", "264,547", "358,684", 
                   "174,279", "262,472", "38,027", "108,200", "398,201", "306,033", "225,706", "336,362", 
                   "33,539", "30,940", "33,882", "43,130", "81,053", "105,047", "92,521", "279,420", 
                   "161,156", "285,562", "102,847", "68,133"))
colnames(matrix.d) = c("Sample", "Total length (bp)", "# contigs", "longest (bp)", "N50 (bp)")

# print table
DT::datatable(matrix.d, 
              extensions = c("FixedColumns", "FixedHeader"),
              caption = "Genome Assembly Statistics", 
              options = list(scrollX = TRUE, paging = TRUE, fixedHeader = FALSE))
```




Despite being more or less fragmented, all assembled genomes are in the range of _Saccharomyces_ sizes. Notable exceptions are: Granvin1 (???), Muri and k7R25 (which are _S. cerevisiae_, _ X _S. eubayanus_ X _S. uvarum_ triple hybrids).



## Genome Annotation

We will perform _ab initio_ annotation of the assembled genomes with the MAKER pipeline. We will use _Saccharomyces_ RepeatMasker to mask repetitive regions, _ab intio_ gene models are predicted with SNAP and augustus with the corresponding _S. cerevisiae_ HMM models, ORFs from S288C_reference_genome_R64-2-1_20150113 reference genome and 1011 _S. cerevisiae_ genomes as ESTs evidences, proteins from S288C_reference_genome_R64-2-1_20150113 genome and the following yeast proteomes as proteins evidences.


```{r, echo=FALSE}
matrix.d = cbind(c("Candida glabrata", "Eremothecium cymbalariae", "Eremothecium gossypii",  "Kazachstania africana",
                   "Kazachstania naganishii", "Kluyveromyces lactis", "Komagataella phaffii", "Lachancea kluyveri",
                   "Lachancea thermotolerans", "Naumovozyma castellii", "Naumovozyma dairenensis", "Saccharomyces bayanus", 
                   "Saccharomyces cerevisiae", "Saccharomyces kudrazevii", "Saccharomyces mikatae", "Saccharomyces paradoxus",
                   "Saccharomyces uvarum", "Tetrapisispora blattae", "Torulaspora delbrueckii", "Zygosaccharomyces rouxii"),
                 c("Cglab", "Ecymb", "Egoss", "Kafri", "Knaga", "Klact", "Kphaf", "Lkluy", "Lther", "Ncast",
                   "Ndair", "Sbaya", "Scere", "Skudr", "Smika", "Spara", "Suvar", "Tblat", "Tdelb", "Zroux"),
                 c("PROT", "PROT", "PROT", "PROT", "PROT", "PROT", "PROT", "GENdev", "PROT", "PROT",
                   "PROT", "GENdev", "PROT", "PROT", "GENdev", "GENdev", "GENdev", "PROT", "PROT", "PROT"),
                 c("ASM254v2", "ASM23536v1", "ASM9102v4", "CBS2517", "ASM34898v1", "ASM251v1", "ASM2700v1",
                   "ASM14922v1", "ASM14280v1", "ASM23734v1", "ASM22711v2", "ASM16703v1", "R64", "NA",
                   "AABZ00000000.1", "AABY00000000.1", "ASM224264v1", "ASM31591v1", "ASM24337v1", "ASM2636v1" ))
colnames(matrix.d) = c("Species Name", "Label", "Proteome/Genome Derived", "Accession Number")

# print table
DT::datatable(matrix.d, 
              extensions = c("FixedColumns", "FixedHeader"),
              caption = "Yeast Reference Proteomes for Gene Annotation", 
              options = list(scrollX = TRUE, paging = TRUE, fixedHeader = FALSE))
```



```{bash, eval=FALSE}
# run the MAKER pipeline
for file in *.redundans.fa; do
	~/bin/maker/bin/maker -genome $file \
		maker_bopts.ctl \
		maker_opts.ctl \
		maker_exe.ctl; 
done 

# summary of transcripts and proteins
for DIR in *.maker.output; do
	~/bin/maker/bin/fasta_merge -d ./$DIR/*master_datastore_index.log;
	~/bin/maker/bin/gff3_merge -d ./$DIR/*master_datastore_index.log;
done

# create gene IDs
for file in *.all.gff ; do
	~/bin/maker/bin/maker_map_ids \
		--prefix $(basename $file | sed 's/\..*//g') \
		$file > $(basename $file .gff).id.map;
	~/bin/maker/bin/map_gff_ids \
		$(basename $file .gff).id.map \
		$file;
	~/bin/maker/bin/map_fasta_ids \
		$(basename $file .gff).id.map \
		$(basename $file .gff).maker.transcripts.fasta;
	~/bin/maker/bin/map_fasta_ids \
		$(basename $file .gff).id.map \
		$(basename $file .gff).maker.proteins.fasta;
done

## Generate annotation stats
echo "Sample"$'\t'"# Transcripts"$'\t'"# Proteins" > Vikings.annotation.stats.txt;
while read line; do
	NAME=$line;
	TRANS=$(grep ">" $line.SPAdes.redundans.all.maker.transcripts.fasta | wc -l);
	PROT=$(grep ">" $line.SPAdes.redundans.all.maker.proteins.fasta | wc -l);
	echo $NAME$'\t'$TRANS$'\t'$PROT >> Vikings.annotation.stats.txt;
done < ../sample.lst

```


Below are the number of genes predicted for each _de novo_ assembled farmhouse genome.


```{r, max.height='100px', echo=FALSE}
matrix.d = cbind(c("14R30", "14R6", "16R23", "16R37", "17P5", "19R18", "1R16", "21P1", "21R38",
                   "27R17", "28P1", "28P6", "28R21", "28R33", "28R8", "2R23", "38R16", "39R20", 
                   "3R11", "40R14", "40R1", "40R20", "41R10", "41R15", "42R20", "42R31", "44R32", "44R7", 
                   "45P5", "45R11", "46R12", "46R37", "6R15", "7R7", "8R19", "9R40", "Granvin1", 
                   "Hornindal1", "Hornindal2", "k7R25", "Laerdal2", "Muri", "SortdalEbbe1", "Voss1"),
                 c("6,728", "6,641", "6,072", "6,157", "6,764", "6,835", "6,810", "6,771", "6,072", "6,681", "6,074",
                   "6,077", "6,079", "6,121", "6,069", "6,802", "6,290", "6,138", "6,095", "6,121", "6,159", "6,853",
                   "6,059", "6,073", "6,155", "6,127", "6,750", "6,371", "6,115", "6,108", "6,137", "6,107", "6,892",
                   "6,786", "6,703", "6,628", "6,518", "6,270", "6,286", "8,044", "6,163", "8,454", "6,244", "6,318"))
colnames(matrix.d) = c("Kveik strain", "Number of predicted genes")

# print table
DT::datatable(matrix.d, 
              extensions = c("FixedColumns", "FixedHeader"),
              caption = "Farmhouse yeasts genes", 
              options = list(scrollX = TRUE, paging = TRUE, fixedHeader = FALSE))
```



## Identification of Farmhouse-specific Gene Families

Do kveiks harbor kveik-specific gene families? To answer this question we build protein orthogroups using proteomes from Kveiks, 302 industrail strains (from Gallone et al. 2019) and reference proteomes for _S. cerevisiae_, _S. kudriavzevii_, _S. eubayanus_, _S. uvarum_ using the Orthofinder pipeline. This approach resulted to be successful for clustering HGT, so hopefully we can identify Orthogroups specific for kveik strains.


```{bash, eval=FALSE}
# create Gene Families
~/bin/OrthoFinder-2.3.3/orthofinder -f 00_aa_start/ \
  -t 72 \
  -M msa \
  -S diamond \
  -A mafft \
  -T iqtree \
  -I 2
  
# filter for Orthogroups enriched for kveiks
python3.8 Vikings.GeneFamiliesFilter.py \
  --in Vikings.Orthogroups.GeneCount.tsv \
  > Vikings.Orthoclusters.counts.tab
```


We select Orthogroups with 10 genes or more (6,811 out of 13,508 groups), and we look for gene families specific for kveiks or enriched in kveiks.


```{r, fig.cap="Kveiks-specific gene families", warning=FALSE, message=FALSE}
# get yeasts gene families
Orthogroups = read.delim(paste(mainDir, "Vikings.Orthoclusters.counts.tab", sep = ""), header = FALSE)
colnames(Orthogroups) = Orthogroups[1, ]
rownames(Orthogroups) = Orthogroups$Orthogroups
Orthogroups = Orthogroups[-which(Orthogroups$Orthogroup == "Orthogroups"), ]
Orthogroups = Orthogroups[, -which(colnames(Orthogroups) == "Orthogroups")]
Orthogroups = dplyr::mutate_all(Orthogroups, function(x) as.numeric(as.character(x)))

# drop columns if empty
Orthogroups = Orthogroups[, colSums(Orthogroups != 0) > 0]

# reorder columns
kveiks = c("14R30", "14R6", "16R23", "16R37", "17P5", "19R18", "1R16", "21P1", "21R38",
           "27R17", "28P1", "28P6", "28R21", "28R33", "28R8", "2R23", "38R16", "39R20", 
           "3R11", "40R14", "40R1", "40R20", "41R10", "41R15", "42R20", "42R31", "44R32", "44R7", 
           "45P5", "45R11", "46R12", "46R37", "6R15", "7R7", "8R19", "9R40", "Granvin1", 
           "Hornindal1", "Hornindal2", "k7R25", "Laerdal2", "Muri", "SortdalEbbe1", "Voss1")
Orthogroups_k = Orthogroups[, which(colnames(Orthogroups) %in% kveiks)]
Orthogroups_nk = Orthogroups[, -which(colnames(Orthogroups) %in% kveiks)]
Orthogroups = cbind(Orthogroups_k, Orthogroups_nk)

# plot heatmap
ComplexHeatmap::Heatmap(Orthogroups,
                        cluster_rows = TRUE,
                        cluster_columns = FALSE,
                        column_dend_reorder = TRUE,
                        col = colorRamp2(c(-1, 0, 1), rev(brewer.pal(n = 3, name = "RdBu"))),
                        na_col = "grey75",
                        show_row_names = FALSE,
                        show_column_names = TRUE,
                        show_row_dend = TRUE,
                        show_heatmap_legend = FALSE,
                        row_title = "Orthogroups",
                        column_title_side = "bottom")
```


**We could not identify no evidences for Farmhouse-specific Orthogroups.**



