# Variant Calling

```{r, echo=FALSE, include=FALSE}
# load required libraries
up_packages = c("ggplot2", "ggpubr", "grid", "gridExtra", "RColorBrewer")
lapply(up_packages, require, character.only = TRUE)

# set working directory
mainDir = "/home/Andrea/Desktop/kveiks/manuscript/plotting_data/"
knitr::opts_knit$set(root.dir = mainDir)
```


We will now analyse the DNA variants between the sequenced farmhouse yeasts and the reference _S. cerevisiae_ lab strain S288C. Genetic variants will be combined with information from the 151 yeast strains sequenced in Gallone et al 2019 used to recostruct (kveiks) population structures, levels of heterozygosities, puatative regions with Loss Of Heterozygosity (LOH) and the eventaul genes affected by LOH.


## Variant Calling

### Reads alignment

Here the steps are the standard I always do:

- align reads to reference genome with bwa mem;
- sort the bam file by read position;
- mark reads duplicates with what used to be picard tools;
- assign read groups, so that each sequencing library is assigned to the right sample (necessary for when I merge the vcf files)


```{bash, eval=FALSE}
## align reads to reference genome S288C
# align reads
DB=~/00_Brigida_starting_files/variants/Saccharomyces_cerevisiae.EF4.73.dna.chromosome.all.fa

while read line; do
  bwa mem -t 72 \
    -K 100000000 $DB \
    ../00_trim_reads/$line.R1.tr.fq.gz \
    ../00_trim_reads/$line.R2.tr.fq.gz \
    > $line.S288C.align.sam;
  samtools view -@ 72 -Sb \
    $line.S288C.align.sam > $line.S288C.align.bam;
  samtools sort -@ 72 \
    $line.S288C.align.bam $line.S288C.align.sort;
  rm $line.S288C.align.sam $line.S288C.align.bam;
done < ../sample.lst

## mark duplicates
for file in *.bam; do
  ~/bin/gatk-4.0.11.0/gatk MarkDuplicates \
    --INPUT=$file \
    --OUTPUT=$(basename $file .bam).md.bam \
    --METRICS_FILE=$(basename $file .bam).md.log \
    --CREATE_INDEX=true;
done

# assign read group
while read line; do
  ~/bin/gatk-4.0.11.0/gatk AddOrReplaceReadGroups \
    --INPUT $line.S288C.align.sort.md.bam \
    --OUTPUT $line.S288C.align.sort.md.r.bam \
    --RGID $line --RGPL $line --RGLB $line \
    --RGPU $line --RGSM $line;
done < ../sample.lst
```


### Variant Calling

I run HaplotypeCaller in gVCF mode, sample ploidy 2 and calling all the kveiks positions at the same time to reduce the number of false positives.

```{bash, eval=FALSE}
## run haplotype caller
DB=~/00_Brigida_starting_files/variants/Saccharomyces_cerevisiae.EF4.73.dna.chromosome.all.fa

for i in chr1 chr2 chr3 chr4 chr5 chr6 chr7 chr8 chr9 chr10 chr11 chr12 chr13 chr14 chr15 chr16; do
  java -Xmx8g \
    -jar ~/bin/GenomeAnalysisTK-3.3-0-g37228af/GenomeAnalysisTK.jar \
    --logging_level INFO --num_threads 4 \
    --analysis_type UnifiedGenotyper \
    --reference_sequence $DB \
    -I bam_files.list -o S288C.variants.$i.gvcf \
    --intervals Saccharomyces_cerevisiae.EF4.73.dna.chromosome.$i.interval_list \
    --standard_min_confidence_threshold_for_calling 20 \
    -stand_emit_conf 30.0 --min_base_quality_score 20 \
    -glm BOTH --sample_ploidy 2 \
    --validation_strictness SILENT --output_mode EMIT_ALL_SITES &
done

# merge kveiks with with brigida's chromosomes
for i in chr1 chr2 chr3 chr4 chr5 chr6 chr7 chr8 chr9 chr10 chr11 chr12 chr13 chr14 chr15 chr16; do
  bcftools merge --threads 4 \
    --output-type z --output all_plusKV.$i.vcf.gz \
    kveiks.S288C.variants.$i.gvcf.gz ../03_Bri_chr/Scer.$i.raw.vcf.gz &
done

# merge all chromosomes together
bcftools concat all_plusKV.chr1.vcf.gz all_plusKV.chr2.vcf.gz \
  all_plusKV.chr3.vcf.gz all_plusKV.chr4.vcf.gz all_plusKV.chr5.vcf.gz \
  all_plusKV.chr6.vcf.gz all_plusKV.chr7.vcf.gz all_plusKV.chr8.vcf.gz \
  all_plusKV.chr9.vcf.gz all_plusKV.chr10.vcf.gz all_plusKV.chr11.vcf.gz \
  all_plusKV.chr12.vcf.gz all_plusKV.chr13.vcf.gz all_plusKV.chr14.vcf.gz \
  all_plusKV.chr15.vcf.gz all_plusKV.chr16.vcf.gz \
  --output-type z --output all_plusKV.vcf.gz --threads 64
```


### Variant filtering and convert to plink bed input

Here I filter for SNPs that are informative for Population Structure analysis
What I do:

- apply hard filters to flag low quality positions;
- select biallelic and high frequency SNPs, SNPs in linkage disequilibrium.

Here I select biallelic SNPs and filter out minor allele frequency. Then I merge my vcf file with the
vcf files from your dataset. After, I just invoke plink for generate the input files for FastStructure

```{bash, eval=FALSE}
# hard filter low quality variants
java -jar ~/bin/GenomeAnalysisTK-3.3-0-g37228af/GenomeAnalysisTK.jar \
  --analysis_type VariantFiltration \
  --variant all_plusKV.vcf.gz --out all_plusKV.fltr.vcf \
  --reference_sequence $DB \
  --filterExpression "QD < 2.0" --filterName "QD_Low" \
  --filterExpression "MQ < 40.0" --filterName "MQ_Low" \
  --filterExpression "GQ < 30" --filterName "GQ_Low" \
  --filterExpression "AD < 5" --filterName "AD_Low"

# select biallelic loci, filter for MAF and linkage disequilibrium
~/bin/plink_linux_x86_64_20181202/plink \
  --make-bed --double-id --allow-extra-chr \
  --indep-pairwise 50 5 0.5 --maf 0.05 \
  --biallelic-only --geno 0.25 \
  --vcf all_plusKV.fltr.vcf.gz --out all_plusKV.fltr.bi
```



## Heterozigous positions

Farmhouse yeasts contains a high degree of Heterozygosity, comparable to yeast strains from the "mixed" Clade.

```{r, fig.cap="Heterozygosity of yeasts from different clades", warning=FALSE, message=FALSE}
# import dataset
kveik = read.delim(paste(mainDir, "all_plusKV.heterozygous.positions.txt", sep = ""), header = TRUE)

# sort 
kveik$Pop = factor(
  kveik$Pop,
  levels = c("Farmhouse", "Asian", "Beer1", "Beer2", "Bread",
             "Mixed", "Mosaic", "Sake", "Wild", "Wine")
)

# prepare count labels
stat_box_data <- function(y) {
  return( 
    data.frame(
      y = 6,
      #sign =tests1[which(tests1$group2 == y),]$p.signif,
      label = paste('n =', length(y))
    )
  )
}

# plot
ggplot(kveik, aes(x = Pop, y = HET_CT)) +
  geom_point(position = position_jitter(width = 0.25),
             shape = 21, fill = "grey50", 
             alpha = 0.25) +
  geom_boxplot(fill = NA, outlier.shape = NA) +
  geom_vline(xintercept = 1.5) +
  stat_summary(fun.data = stat_box_data, geom = "text", hjust = 0.5, vjust = 0.9) +
  stat_compare_means(method = "anova", label.y = 0.01) +
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "Farmhouse") +
  scale_y_log10(labels = scales::comma, limits = c(1, 1300000)) +
  labs(title = "Heterozygosity in yeast populations",
       x = "Populations",
       y = "log10 heterozygous positions") +
  theme(plot.title = element_text(hjust = 0.5, size = 24),
        axis.text.x = element_text(hjust = 0.95, angle = 30, size = 16),
        axis.text.y = element_text(size = 16),
        axis.title = element_text(size = 18),
        legend.position = "none",
        panel.background = element_rect(colour = "black", fill = NA),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(),
        panel.grid.major.y = element_line(colour = "grey75"),
        panel.grid.minor.y = element_blank())

# plot
ggplot(kveik, aes(x = Pop, y = HET_CT)) +
  geom_point(shape = "-", size = 3.5) +
  geom_boxplot(fill = "grey95", alpha = 0.75, width = 0.25,
               position = position_nudge(x = -0.2),
               outlier.colour = NA) +
  ggdist::stat_halfeye(fill = "grey95", alpha = 0.75, slab_color = "grey45", slab_size = 0.25, 
                       adjust = 5, justification = -0.05, .width = 0, point_colour = NA) +
  geom_vline(xintercept = 1.5) +
  stat_summary(fun.data = stat_box_data, geom = "text", hjust = 0.5, vjust = 0.9) +
  stat_compare_means(method = "anova", label.y = 0.01) +
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "Farmhouse") +
  scale_y_log10(labels = scales::comma, limits = c(1, 1300000)) +
  labs(title = "Heterozygosity in yeast populations",
       x = "Populations",
       y = "log10 heterozygous positions") +
  theme(plot.title = element_text(hjust = 0.5, size = 24),
        axis.text.x = element_text(hjust = 0.95, angle = 30, size = 16),
        axis.text.y = element_text(size = 16),
        axis.title = element_text(size = 18),
        legend.position = "none",
        panel.background = element_rect(colour = "black", fill = NA),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(),
        panel.grid.major.y = element_line(colour = "grey75"),
        panel.grid.minor.y = element_blank())
```



## LOH







## Population Structure

Here I simply run FastStructure for a range of k, and then identify the best kmer. I have run FastStructure as well giving logarithmic priors, but the results where comparable

### Farmhouse yeasts only

```{bash, eval=FALSE}
# merge all chromosomes together
~/bin/bcftools-1.9/bcftools concat \
  kveiks.S288C.variants.chr1.gvcf.gz kveiks.S288C.variants.chr2.gvcf.gz \
  kveiks.S288C.variants.chr3.gvcf.gz kveiks.S288C.variants.chr4.gvcf.gz \
  kveiks.S288C.variants.chr5.gvcf.gz kveiks.S288C.variants.chr6.gvcf.gz \
  kveiks.S288C.variants.chr7.gvcf.gz kveiks.S288C.variants.chr8.gvcf.gz \
  kveiks.S288C.variants.chr9.gvcf.gz kveiks.S288C.variants.chr10.gvcf.gz \
  kveiks.S288C.variants.chr11.gvcf.gz kveiks.S288C.variants.chr12.gvcf.gz \
  kveiks.S288C.variants.chr13.gvcf.gz kveiks.S288C.variants.chr14.gvcf.gz \
  kveiks.S288C.variants.chr15.gvcf.gz kveiks.S288C.variants.chr16.gvcf.gz \
  --output-type z --output kveiks.S288C.variants.vcf.gz --threads 16

# hard filter low quality variants
java -jar ~/bin/GenomeAnalysisTK-3.3-0-g37228af/GenomeAnalysisTK.jar \
  --analysis_type VariantFiltration \
  --variant kveiks.S288C.variants.vcf.gz \
  --out kveiks.S288C.variants.fltr.vcf \
  --reference_sequence $DB \
  --filterExpression "QD < 2.0" --filterName "QD_Low" \
  --filterExpression "MQ < 40.0" --filterName "MQ_Low" \
  --filterExpression "GQ < 30" --filterName "GQ_Low" \
  --filterExpression "AD < 5" --filterName "AD_Low"

# select biallelic loci, filter for MAF and linkage disequilibrium
~/bin/plink_linux_x86_64_20181202/plink \
  --make-bed --double-id --allow-extra-chr \
  --indep-pairwise 50 5 0.5 --maf 0.05 \
  --biallelic-only --geno 0.25 \
  --vcf kveiks.S288C.variants.fltr.vcf.gz \
  --out kveiks.S288C.variants.fltr.bi

# run fastStructure simple
for k in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20; do
  python2.7 ~/bin/faststructure/fastStructure-1.0/structure.py \
    -K "${k}" --input=kveiks.S288C.variants.fltr.bi \
    --output=kveiks.S288C.variants.fltr.bi.structure \
    --prior=simple &
done

# choose the best K-mer
python2.7 ~/bin/faststructure/fastStructure-1.0/chooseK.py \
  --input=all_plusKV.fltr.bi.structure
```


==> Model complexity that maximizes marginal likelihood = 4
==> Model components used to explain structure in data = 5



### Farmhouse yeasts + Gallone 2019

```{bash, eval=FALSE}
# run fastStructure simple
for k in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20; do
  python2.7 ~/bin/faststructure/fastStructure-1.0/structure.py \
    -K "${k}" --input=all_plusKV.fltr.bi \
    --output=all_plusKV.fltr.bi.structure \
    --prior=simple &
done

# choose the best K-mer
python2.7 ~/bin/faststructure/fastStructure-1.0/chooseK.py \
  --input=all_plusKV.fltr.bi.structure
```


==> Model complexity that maximizes marginal likelihood = 20
==> Model components used to explain structure in data = 15


**CODE BELOW TO BE DOUBLECHECKED**

```{bash, eval=FALSE}
# print distruct plots
for k in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20; do python2.7
~/bin/faststructure/fastStructure-1.0/distruct.py
 -K "${k}" --
input=all_plusKV.fltr.bi.structure --
output=all_plusKV.fltr.bi.structure."${k}".svg --popfile=all_plusKV.poplabels;
done
The population labels file all_plusKV.poplabels looks like:
```










