<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Phylogenetic reconstruction | Vikings Project</title>
  <meta name="description" content="Exploring yeast phenotypic diversity in traditional European farmhouse ale cultures as a source of novel industrial traits" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Phylogenetic reconstruction | Vikings Project" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Exploring yeast phenotypic diversity in traditional European farmhouse ale cultures as a source of novel industrial traits" />
  <meta name="github-repo" content="19ADC99/VikingsProject" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Phylogenetic reconstruction | Vikings Project" />
  
  <meta name="twitter:description" content="Exploring yeast phenotypic diversity in traditional European farmhouse ale cultures as a source of novel industrial traits" />
  

<meta name="author" content="Andrea Del Cortona" />


<meta name="date" content="2023-05-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="horizontal-gene-transfer.html"/>
<link rel="next" href="variant-calling.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Vikings Project</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="1" data-path="qc-and-genome-composition-assessment.html"><a href="qc-and-genome-composition-assessment.html"><i class="fa fa-check"></i><b>1</b> QC and Genome Composition assessment</a>
<ul>
<li class="chapter" data-level="1.1" data-path="qc-and-genome-composition-assessment.html"><a href="qc-and-genome-composition-assessment.html#reads-qc-and-filtering"><i class="fa fa-check"></i><b>1.1</b> Reads QC and Filtering</a></li>
<li class="chapter" data-level="1.2" data-path="qc-and-genome-composition-assessment.html"><a href="qc-and-genome-composition-assessment.html#genome-composition"><i class="fa fa-check"></i><b>1.2</b> Genome Composition</a></li>
<li class="chapter" data-level="1.3" data-path="qc-and-genome-composition-assessment.html"><a href="qc-and-genome-composition-assessment.html#ploydies-estimation"><i class="fa fa-check"></i><b>1.3</b> Ploydies estimation</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="copy-number-variants.html"><a href="copy-number-variants.html"><i class="fa fa-check"></i><b>2</b> Copy Number Variants</a>
<ul>
<li class="chapter" data-level="2.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#call-cnvs"><i class="fa fa-check"></i><b>2.1</b> Call CNVs</a></li>
<li class="chapter" data-level="2.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#identify-common-cnvs-in-farmhouse-yeasts"><i class="fa fa-check"></i><b>2.2</b> Identify common CNVs in Farmhouse yeasts</a></li>
<li class="chapter" data-level="2.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv-functional-enrichment"><i class="fa fa-check"></i><b>2.3</b> CNV Functional Enrichment</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---i1-160000"><i class="fa fa-check"></i><b>2.3.1</b> CNV - I:1-160,000</a>
<ul>
<li class="chapter" data-level="2.3.1.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms"><i class="fa fa-check"></i><b>2.3.1.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.1.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms"><i class="fa fa-check"></i><b>2.3.1.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.1.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms"><i class="fa fa-check"></i><b>2.3.1.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.1.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-kegg-pathways"><i class="fa fa-check"></i><b>2.3.1.4</b> Over-represented / Enriched KEGG Pathways</a></li>
<li class="chapter" data-level="2.3.1.5" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways"><i class="fa fa-check"></i><b>2.3.1.5</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---i160501-230500"><i class="fa fa-check"></i><b>2.3.2</b> CNV - I:160,501-230,500</a>
<ul>
<li class="chapter" data-level="2.3.2.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-1"><i class="fa fa-check"></i><b>2.3.2.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.2.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-1"><i class="fa fa-check"></i><b>2.3.2.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.2.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-1"><i class="fa fa-check"></i><b>2.3.2.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.2.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-kegg-pathways-1"><i class="fa fa-check"></i><b>2.3.2.4</b> Over-represented / Enriched KEGG Pathways</a></li>
<li class="chapter" data-level="2.3.2.5" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-1"><i class="fa fa-check"></i><b>2.3.2.5</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---iii1001-317000"><i class="fa fa-check"></i><b>2.3.3</b> CNV - III:1,001-317,000</a>
<ul>
<li class="chapter" data-level="2.3.3.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-2"><i class="fa fa-check"></i><b>2.3.3.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.3.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-2"><i class="fa fa-check"></i><b>2.3.3.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.3.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-2"><i class="fa fa-check"></i><b>2.3.3.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.3.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-kegg-pathways-2"><i class="fa fa-check"></i><b>2.3.3.4</b> Over-represented / Enriched KEGG Pathways</a></li>
<li class="chapter" data-level="2.3.3.5" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-2"><i class="fa fa-check"></i><b>2.3.3.5</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---iv1-18500"><i class="fa fa-check"></i><b>2.3.4</b> CNV - IV:1-18,500</a>
<ul>
<li class="chapter" data-level="2.3.4.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-3"><i class="fa fa-check"></i><b>2.3.4.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.4.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-3"><i class="fa fa-check"></i><b>2.3.4.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.4.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-3"><i class="fa fa-check"></i><b>2.3.4.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.4.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-kegg-pathways-3"><i class="fa fa-check"></i><b>2.3.4.4</b> Over-represented / Enriched KEGG Pathways</a></li>
<li class="chapter" data-level="2.3.4.5" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-3"><i class="fa fa-check"></i><b>2.3.4.5</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.5" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---iv527001-540000"><i class="fa fa-check"></i><b>2.3.5</b> CNV - IV:527,001-540,000</a>
<ul>
<li class="chapter" data-level="2.3.5.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-4"><i class="fa fa-check"></i><b>2.3.5.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.5.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-4"><i class="fa fa-check"></i><b>2.3.5.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.5.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-4"><i class="fa fa-check"></i><b>2.3.5.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.5.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-kegg-pathways-4"><i class="fa fa-check"></i><b>2.3.5.4</b> Over-represented / Enriched KEGG Pathways</a></li>
<li class="chapter" data-level="2.3.5.5" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-4"><i class="fa fa-check"></i><b>2.3.5.5</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.6" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---vi4001-270500"><i class="fa fa-check"></i><b>2.3.6</b> CNV - VI:4,001-270,500</a>
<ul>
<li class="chapter" data-level="2.3.6.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-5"><i class="fa fa-check"></i><b>2.3.6.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.6.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-5"><i class="fa fa-check"></i><b>2.3.6.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.6.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-5"><i class="fa fa-check"></i><b>2.3.6.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.6.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-kegg-pathways-5"><i class="fa fa-check"></i><b>2.3.6.4</b> Over-represented / Enriched KEGG Pathways</a></li>
<li class="chapter" data-level="2.3.6.5" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-5"><i class="fa fa-check"></i><b>2.3.6.5</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.7" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---viii85501-556000"><i class="fa fa-check"></i><b>2.3.7</b> CNV - VIII:85,501-556,000</a>
<ul>
<li class="chapter" data-level="2.3.7.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-6"><i class="fa fa-check"></i><b>2.3.7.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.7.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-6"><i class="fa fa-check"></i><b>2.3.7.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.7.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-6"><i class="fa fa-check"></i><b>2.3.7.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.7.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-kegg-pathways-6"><i class="fa fa-check"></i><b>2.3.7.4</b> Over-represented / Enriched KEGG Pathways</a></li>
<li class="chapter" data-level="2.3.7.5" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-6"><i class="fa fa-check"></i><b>2.3.7.5</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.8" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---xi645001-667000"><i class="fa fa-check"></i><b>2.3.8</b> CNV - XI:645,001-667,000</a>
<ul>
<li class="chapter" data-level="2.3.8.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-7"><i class="fa fa-check"></i><b>2.3.8.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.8.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-7"><i class="fa fa-check"></i><b>2.3.8.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.8.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-7"><i class="fa fa-check"></i><b>2.3.8.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.8.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-kegg-pathways-7"><i class="fa fa-check"></i><b>2.3.8.4</b> Over-represented / Enriched KEGG Pathways</a></li>
<li class="chapter" data-level="2.3.8.5" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-7"><i class="fa fa-check"></i><b>2.3.8.5</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.9" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---xii451001-490000"><i class="fa fa-check"></i><b>2.3.9</b> CNV - XII:451,001-490,000</a>
<ul>
<li class="chapter" data-level="2.3.9.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-8"><i class="fa fa-check"></i><b>2.3.9.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.9.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-8"><i class="fa fa-check"></i><b>2.3.9.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.9.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-8"><i class="fa fa-check"></i><b>2.3.9.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.9.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-kegg-pathways-8"><i class="fa fa-check"></i><b>2.3.9.4</b> Over-represented / Enriched KEGG Pathways</a></li>
<li class="chapter" data-level="2.3.9.5" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-8"><i class="fa fa-check"></i><b>2.3.9.5</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.10" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---xv1-38000"><i class="fa fa-check"></i><b>2.3.10</b> CNV - XV:1-38,000</a>
<ul>
<li class="chapter" data-level="2.3.10.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-9"><i class="fa fa-check"></i><b>2.3.10.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.10.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-9"><i class="fa fa-check"></i><b>2.3.10.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.10.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-9"><i class="fa fa-check"></i><b>2.3.10.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.10.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-kegg-pathways-9"><i class="fa fa-check"></i><b>2.3.10.4</b> Over-represented / Enriched KEGG Pathways</a></li>
<li class="chapter" data-level="2.3.10.5" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-9"><i class="fa fa-check"></i><b>2.3.10.5</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.11" data-path="copy-number-variants.html"><a href="copy-number-variants.html#summary-of-enriched-or-depleted-functions"><i class="fa fa-check"></i><b>2.3.11</b> Summary of enriched or depleted functions</a>
<ul>
<li class="chapter" data-level="2.3.11.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#enriched-go-terms"><i class="fa fa-check"></i><b>2.3.11.1</b> Enriched GO terms</a></li>
<li class="chapter" data-level="2.3.11.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#enriched-kegg-pathways"><i class="fa fa-check"></i><b>2.3.11.2</b> Enriched KEGG pathways</a></li>
<li class="chapter" data-level="2.3.11.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#enriched-reactome-pathways"><i class="fa fa-check"></i><b>2.3.11.3</b> Enriched Reactome pathways</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="genome-assembly-and-annotation.html"><a href="genome-assembly-and-annotation.html"><i class="fa fa-check"></i><b>3</b> Genome Assembly and Annotation</a>
<ul>
<li class="chapter" data-level="3.1" data-path="genome-assembly-and-annotation.html"><a href="genome-assembly-and-annotation.html#genome-assembly"><i class="fa fa-check"></i><b>3.1</b> Genome Assembly</a></li>
<li class="chapter" data-level="3.2" data-path="genome-assembly-and-annotation.html"><a href="genome-assembly-and-annotation.html#genome-annotation"><i class="fa fa-check"></i><b>3.2</b> Genome Annotation</a></li>
<li class="chapter" data-level="3.3" data-path="genome-assembly-and-annotation.html"><a href="genome-assembly-and-annotation.html#identification-of-farmhouse-specific-gene-families"><i class="fa fa-check"></i><b>3.3</b> Identification of Farmhouse-specific Gene Families</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html"><i class="fa fa-check"></i><b>4</b> Horizontal Gene Transfer</a>
<ul>
<li class="chapter" data-level="4.1" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#taxonomic-annotation-of-predicted-genes"><i class="fa fa-check"></i><b>4.1</b> Taxonomic annotation of predicted genes</a></li>
<li class="chapter" data-level="4.2" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#evidence-for-bacterial-hgt"><i class="fa fa-check"></i><b>4.2</b> Evidence for bacterial HGT</a></li>
<li class="chapter" data-level="4.3" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#evidence-for-hgt-from-ascomycota"><i class="fa fa-check"></i><b>4.3</b> Evidence for HGT from Ascomycota</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#zygosaccharomyces-parabailli-transferred-operon"><i class="fa fa-check"></i><b>4.3.1</b> <em>Zygosaccharomyces parabailli</em> transferred operon</a>
<ul>
<li class="chapter" data-level="4.3.1.1" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#overrepresented-go-bp-terms"><i class="fa fa-check"></i><b>4.3.1.1</b> Overrepresented GO BP terms</a></li>
<li class="chapter" data-level="4.3.1.2" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#overrepresented-go-cc-terms"><i class="fa fa-check"></i><b>4.3.1.2</b> Overrepresented GO CC terms</a></li>
<li class="chapter" data-level="4.3.1.3" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#overrepresented-go-mf-terms"><i class="fa fa-check"></i><b>4.3.1.3</b> Overrepresented GO MF terms</a></li>
<li class="chapter" data-level="4.3.1.4" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#overrepresented-kegg-pathways"><i class="fa fa-check"></i><b>4.3.1.4</b> Overrepresented KEGG Pathways</a></li>
<li class="chapter" data-level="4.3.1.5" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#overrepresented-reactome-pathways"><i class="fa fa-check"></i><b>4.3.1.5</b> Overrepresented Reactome Pathways</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="phylogenetic-reconstruction.html"><a href="phylogenetic-reconstruction.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic reconstruction</a>
<ul>
<li class="chapter" data-level="5.1" data-path="phylogenetic-reconstruction.html"><a href="phylogenetic-reconstruction.html#identify-quasi-single-copy-genes"><i class="fa fa-check"></i><b>5.1</b> Identify quasi single-copy genes</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="phylogenetic-reconstruction.html"><a href="phylogenetic-reconstruction.html#create-animo-acid-and-codon-alignments"><i class="fa fa-check"></i><b>5.1.1</b> Create animo acid and codon alignments</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="phylogenetic-reconstruction.html"><a href="phylogenetic-reconstruction.html#run-phylogenetic-reconstruction"><i class="fa fa-check"></i><b>5.2</b> Run phylogenetic reconstruction</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="phylogenetic-reconstruction.html"><a href="phylogenetic-reconstruction.html#create-ml-tree"><i class="fa fa-check"></i><b>5.2.1</b> Create ML tree</a></li>
<li class="chapter" data-level="5.2.2" data-path="phylogenetic-reconstruction.html"><a href="phylogenetic-reconstruction.html#create-coalescence-based-tree"><i class="fa fa-check"></i><b>5.2.2</b> Create coalescence-based tree</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="variant-calling.html"><a href="variant-calling.html"><i class="fa fa-check"></i><b>6</b> Variant Calling</a>
<ul>
<li class="chapter" data-level="6.1" data-path="variant-calling.html"><a href="variant-calling.html#variant-calling-1"><i class="fa fa-check"></i><b>6.1</b> Variant Calling</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="variant-calling.html"><a href="variant-calling.html#reads-alignment"><i class="fa fa-check"></i><b>6.1.1</b> Reads alignment</a></li>
<li class="chapter" data-level="6.1.2" data-path="variant-calling.html"><a href="variant-calling.html#variant-calling-2"><i class="fa fa-check"></i><b>6.1.2</b> Variant Calling</a></li>
<li class="chapter" data-level="6.1.3" data-path="variant-calling.html"><a href="variant-calling.html#variant-filtering-and-convert-to-plink-bed-input"><i class="fa fa-check"></i><b>6.1.3</b> Variant filtering and convert to plink bed input</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="variant-calling.html"><a href="variant-calling.html#heterozigous-positions"><i class="fa fa-check"></i><b>6.2</b> Heterozigous positions</a></li>
<li class="chapter" data-level="6.3" data-path="variant-calling.html"><a href="variant-calling.html#loh"><i class="fa fa-check"></i><b>6.3</b> LOH</a></li>
<li class="chapter" data-level="6.4" data-path="variant-calling.html"><a href="variant-calling.html#population-structure"><i class="fa fa-check"></i><b>6.4</b> Population Structure</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="variant-calling.html"><a href="variant-calling.html#farmhouse-yeasts-only"><i class="fa fa-check"></i><b>6.4.1</b> Farmhouse yeasts only</a></li>
<li class="chapter" data-level="6.4.2" data-path="variant-calling.html"><a href="variant-calling.html#farmhouse-yeasts-gallone-2019"><i class="fa fa-check"></i><b>6.4.2</b> Farmhouse yeasts + Gallone 2019</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="farmhouse-yeasts-culturess-heterogeneity.html"><a href="farmhouse-yeasts-culturess-heterogeneity.html"><i class="fa fa-check"></i><b>7</b> Farmhouse Yeasts Culturess Heterogeneity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="farmhouse-yeasts-culturess-heterogeneity.html"><a href="farmhouse-yeasts-culturess-heterogeneity.html#vaaa"><i class="fa fa-check"></i><b>7.1</b> Vaaa</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Vikings Project</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="phylogenetic-reconstruction" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Phylogenetic reconstruction<a href="phylogenetic-reconstruction.html#phylogenetic-reconstruction" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="identify-quasi-single-copy-genes" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Identify quasi single-copy genes<a href="phylogenetic-reconstruction.html#identify-quasi-single-copy-genes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Uniform header sequences, since they come from multiple sources and they are all messed up and incosistent.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="phylogenetic-reconstruction.html#cb1-1" tabindex="-1"></a><span class="co"># rename protein sequences</span></span>
<span id="cb1-2"><a href="phylogenetic-reconstruction.html#cb1-2" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>maker.aa.fa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb1-3"><a href="phylogenetic-reconstruction.html#cb1-3" tabindex="-1"></a>  <span class="va">NAME</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .maker.aa.fa<span class="va">)</span>_<span class="kw">;</span></span>
<span id="cb1-4"><a href="phylogenetic-reconstruction.html#cb1-4" tabindex="-1"></a>  <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&quot;s/&gt;/&gt;</span><span class="va">$NAME</span><span class="st">/g&quot;</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb1-5"><a href="phylogenetic-reconstruction.html#cb1-5" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb1-6"><a href="phylogenetic-reconstruction.html#cb1-6" tabindex="-1"></a></span>
<span id="cb1-7"><a href="phylogenetic-reconstruction.html#cb1-7" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*[</span><span class="ss">0</span><span class="pp">-</span><span class="ss">9</span><span class="pp">]</span>.aa.fa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb1-8"><a href="phylogenetic-reconstruction.html#cb1-8" tabindex="-1"></a>  <span class="va">NAME</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .aa.fa<span class="va">)</span>_<span class="kw">;</span></span>
<span id="cb1-9"><a href="phylogenetic-reconstruction.html#cb1-9" tabindex="-1"></a>  <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&quot;s/&gt;/&gt;</span><span class="va">$NAME</span><span class="st">/g&quot;</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb1-10"><a href="phylogenetic-reconstruction.html#cb1-10" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb1-11"><a href="phylogenetic-reconstruction.html#cb1-11" tabindex="-1"></a></span>
<span id="cb1-12"><a href="phylogenetic-reconstruction.html#cb1-12" tabindex="-1"></a><span class="co"># concatenate all files together</span></span>
<span id="cb1-13"><a href="phylogenetic-reconstruction.html#cb1-13" tabindex="-1"></a><span class="fu">cat</span> <span class="pp">*</span>.aa.fa <span class="op">&gt;</span> all_plusKV.aa.fa</span></code></pre></div>
<p>We will create a set of non-redundant <em>S. cerevisiae</em> proteins, by clustering them based on a 70% sequence similarity groups.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="phylogenetic-reconstruction.html#cb2-1" tabindex="-1"></a><span class="co"># cluster sequences</span></span>
<span id="cb2-2"><a href="phylogenetic-reconstruction.html#cb2-2" tabindex="-1"></a><span class="ex">cd-hit</span> <span class="at">-i</span> all_plusKV.aa.fa <span class="at">-o</span> all_plusKV.aa.0.7.fa <span class="at">-T</span> 24 <span class="at">-M</span> 16000 <span class="at">-c</span> 0.7 <span class="at">-d</span> 60</span></code></pre></div>
<p>We now check protein clusters having only a single copy gene per yeast strain.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="phylogenetic-reconstruction.html#cb3-1" tabindex="-1"></a><span class="co"># filter clusters into small one gene ortholog clusters</span></span>
<span id="cb3-2"><a href="phylogenetic-reconstruction.html#cb3-2" tabindex="-1"></a><span class="ex">python3.5</span> Vikings.filterCDHITclusters.py <span class="at">--clstr</span> all_plusKV.aa.0.7.fa.clstr</span>
<span id="cb3-3"><a href="phylogenetic-reconstruction.html#cb3-3" tabindex="-1"></a></span>
<span id="cb3-4"><a href="phylogenetic-reconstruction.html#cb3-4" tabindex="-1"></a><span class="co"># clean folder</span></span>
<span id="cb3-5"><a href="phylogenetic-reconstruction.html#cb3-5" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> 01_filter_clusters/01_clusters_311-350</span>
<span id="cb3-6"><a href="phylogenetic-reconstruction.html#cb3-6" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> 01_filter_clusters/02_clusters_311-350_noDuplicates</span>
<span id="cb3-7"><a href="phylogenetic-reconstruction.html#cb3-7" tabindex="-1"></a><span class="fu">mv</span> all_plusKV.aa.0.7.fa.clstr.fltr.clstr.<span class="pp">*</span> 01_filter_clusters</span>
<span id="cb3-8"><a href="phylogenetic-reconstruction.html#cb3-8" tabindex="-1"></a></span>
<span id="cb3-9"><a href="phylogenetic-reconstruction.html#cb3-9" tabindex="-1"></a><span class="co"># remove duplicated genes from orthologs clusters</span></span>
<span id="cb3-10"><a href="phylogenetic-reconstruction.html#cb3-10" tabindex="-1"></a><span class="ex">python3.5</span> Vikings.filterCDHITclustersDeduplicates.py <span class="dt">\</span></span>
<span id="cb3-11"><a href="phylogenetic-reconstruction.html#cb3-11" tabindex="-1"></a>    <span class="at">--indir</span> 01_filter_clusters/01_clusters_311-350 <span class="dt">\</span></span>
<span id="cb3-12"><a href="phylogenetic-reconstruction.html#cb3-12" tabindex="-1"></a>    <span class="at">--outdir</span> 01_filter_clusters/02_clusters_311-350_noDuplicate</span>
<span id="cb3-13"><a href="phylogenetic-reconstruction.html#cb3-13" tabindex="-1"></a></span>
<span id="cb3-14"><a href="phylogenetic-reconstruction.html#cb3-14" tabindex="-1"></a><span class="co"># retrieve amino acid sequences</span></span>
<span id="cb3-15"><a href="phylogenetic-reconstruction.html#cb3-15" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> 01_filter_clusters/03_clusters_aa</span>
<span id="cb3-16"><a href="phylogenetic-reconstruction.html#cb3-16" tabindex="-1"></a></span>
<span id="cb3-17"><a href="phylogenetic-reconstruction.html#cb3-17" tabindex="-1"></a><span class="co"># format headers</span></span>
<span id="cb3-18"><a href="phylogenetic-reconstruction.html#cb3-18" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/ .*//g&#39;</span> ../all_plusKV.aa.fa</span>
<span id="cb3-19"><a href="phylogenetic-reconstruction.html#cb3-19" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/|.*//g&#39;</span> ../all_plusKV.aa.fa <span class="kw">&amp;</span></span>
<span id="cb3-20"><a href="phylogenetic-reconstruction.html#cb3-20" tabindex="-1"></a></span>
<span id="cb3-21"><a href="phylogenetic-reconstruction.html#cb3-21" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> 02_clusters_311-350_noDuplicates/<span class="pp">*</span><span class="kw">;</span> <span class="cf">do</span> <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/|.*//g&#39;</span> <span class="va">$file</span> <span class="kw">;</span> <span class="cf">done</span></span>
<span id="cb3-22"><a href="phylogenetic-reconstruction.html#cb3-22" tabindex="-1"></a></span>
<span id="cb3-23"><a href="phylogenetic-reconstruction.html#cb3-23" tabindex="-1"></a><span class="ex">python3.5</span> Vikings.CDHITclustersList2ntSeq.py <span class="dt">\</span></span>
<span id="cb3-24"><a href="phylogenetic-reconstruction.html#cb3-24" tabindex="-1"></a>  <span class="at">--fasta</span> all_plusKV.aa.fa <span class="dt">\</span></span>
<span id="cb3-25"><a href="phylogenetic-reconstruction.html#cb3-25" tabindex="-1"></a>  <span class="at">--indir</span> ./01_filter_clusters/02_clusters_311-350_noDuplicates/ <span class="dt">\</span></span>
<span id="cb3-26"><a href="phylogenetic-reconstruction.html#cb3-26" tabindex="-1"></a>  <span class="at">--outdir</span> ./01_filter_clusters/03_clusters_aa/</span></code></pre></div>
<div id="create-animo-acid-and-codon-alignments" class="section level3 hasAnchor" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> Create animo acid and codon alignments<a href="phylogenetic-reconstruction.html#create-animo-acid-and-codon-alignments" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Prepare nucleotidic sequences</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="phylogenetic-reconstruction.html#cb4-1" tabindex="-1"></a><span class="fu">mkdir</span> 02_nt_start/</span>
<span id="cb4-2"><a href="phylogenetic-reconstruction.html#cb4-2" tabindex="-1"></a></span>
<span id="cb4-3"><a href="phylogenetic-reconstruction.html#cb4-3" tabindex="-1"></a><span class="co"># reformat the header to match aa</span></span>
<span id="cb4-4"><a href="phylogenetic-reconstruction.html#cb4-4" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.fa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb4-5"><a href="phylogenetic-reconstruction.html#cb4-5" tabindex="-1"></a>    <span class="va">STRAIN</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> <span class="kw">|</span> <span class="fu">tr</span> <span class="st">&#39;.&#39;</span> <span class="st">&#39; &#39;</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 1 <span class="at">-d</span> <span class="st">&quot; &quot;</span><span class="va">)</span><span class="kw">;</span></span>
<span id="cb4-6"><a href="phylogenetic-reconstruction.html#cb4-6" tabindex="-1"></a>    <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/ .*//g&#39;</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb4-7"><a href="phylogenetic-reconstruction.html#cb4-7" tabindex="-1"></a>    <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/|.*//g&#39;</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb4-8"><a href="phylogenetic-reconstruction.html#cb4-8" tabindex="-1"></a>    <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/&gt;/&gt;_/g&#39;</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb4-9"><a href="phylogenetic-reconstruction.html#cb4-9" tabindex="-1"></a>    <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&quot;s/&gt;/&gt;</span><span class="va">$STRAIN</span><span class="st">/g&quot;</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb4-10"><a href="phylogenetic-reconstruction.html#cb4-10" tabindex="-1"></a><span class="cf">done</span> </span>
<span id="cb4-11"><a href="phylogenetic-reconstruction.html#cb4-11" tabindex="-1"></a></span>
<span id="cb4-12"><a href="phylogenetic-reconstruction.html#cb4-12" tabindex="-1"></a><span class="co"># select nt per cluster</span></span>
<span id="cb4-13"><a href="phylogenetic-reconstruction.html#cb4-13" tabindex="-1"></a><span class="fu">cat</span> <span class="pp">*</span>.fa <span class="op">&gt;</span> ../all_plusKV.nt.fa</span>
<span id="cb4-14"><a href="phylogenetic-reconstruction.html#cb4-14" tabindex="-1"></a><span class="co"># correct worng headers</span></span>
<span id="cb4-15"><a href="phylogenetic-reconstruction.html#cb4-15" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/Skudr_Skudr/Skudr/g&#39;</span> all_plusKV.nt.fa<span class="kw">;</span></span>
<span id="cb4-16"><a href="phylogenetic-reconstruction.html#cb4-16" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/Suvar_Suvar/Suvar/g&#39;</span> all_plusKV.nt.fa<span class="kw">;</span></span>
<span id="cb4-17"><a href="phylogenetic-reconstruction.html#cb4-17" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/Seuba_Seuba/Seuba/g&#39;</span> all_plusKV.nt.fa</span>
<span id="cb4-18"><a href="phylogenetic-reconstruction.html#cb4-18" tabindex="-1"></a><span class="fu">mkdir</span> 03_nt_clusters</span>
<span id="cb4-19"><a href="phylogenetic-reconstruction.html#cb4-19" tabindex="-1"></a></span>
<span id="cb4-20"><a href="phylogenetic-reconstruction.html#cb4-20" tabindex="-1"></a><span class="co"># cluster NT</span></span>
<span id="cb4-21"><a href="phylogenetic-reconstruction.html#cb4-21" tabindex="-1"></a><span class="ex">cd-hit</span> <span class="at">-T</span> 72 <span class="at">-c</span> 0.98 <span class="at">-aL</span> 0.95 <span class="at">-d</span> 0 <span class="at">-i</span> all_plusKV.nt.fa <span class="at">-o</span> all_plusKV.nt.098.fa</span>
<span id="cb4-22"><a href="phylogenetic-reconstruction.html#cb4-22" tabindex="-1"></a></span>
<span id="cb4-23"><a href="phylogenetic-reconstruction.html#cb4-23" tabindex="-1"></a><span class="co"># extract cluster gene lists</span></span>
<span id="cb4-24"><a href="phylogenetic-reconstruction.html#cb4-24" tabindex="-1"></a><span class="ex">python3.5</span> Vikings.filterCDHITclusters.nt.py <span class="at">--clstr</span> all_plusKV.nt.098.fa.clstr</span>
<span id="cb4-25"><a href="phylogenetic-reconstruction.html#cb4-25" tabindex="-1"></a></span>
<span id="cb4-26"><a href="phylogenetic-reconstruction.html#cb4-26" tabindex="-1"></a><span class="co"># select genes</span></span>
<span id="cb4-27"><a href="phylogenetic-reconstruction.html#cb4-27" tabindex="-1"></a><span class="fu">cat</span> gene.lst <span class="dt">\</span></span>
<span id="cb4-28"><a href="phylogenetic-reconstruction.html#cb4-28" tabindex="-1"></a>  <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-n</span> 1 <span class="at">-P</span> 64 <span class="at">-I</span> {} sh <span class="at">-c</span> <span class="dt">\</span></span>
<span id="cb4-29"><a href="phylogenetic-reconstruction.html#cb4-29" tabindex="-1"></a>    <span class="st">&#39;echo {}; \</span></span>
<span id="cb4-30"><a href="phylogenetic-reconstruction.html#cb4-30" tabindex="-1"></a><span class="st">    for file in 00_nt_clusters_lst/*.clstr.*; do </span></span>
<span id="cb4-31"><a href="phylogenetic-reconstruction.html#cb4-31" tabindex="-1"></a><span class="st">      if grep -q {} $file; </span></span>
<span id="cb4-32"><a href="phylogenetic-reconstruction.html#cb4-32" tabindex="-1"></a><span class="st">        then cp $file 00_nt_clusters_lst_select;</span></span>
<span id="cb4-33"><a href="phylogenetic-reconstruction.html#cb4-33" tabindex="-1"></a><span class="st">    fi; done&#39;</span></span>
<span id="cb4-34"><a href="phylogenetic-reconstruction.html#cb4-34" tabindex="-1"></a></span>
<span id="cb4-35"><a href="phylogenetic-reconstruction.html#cb4-35" tabindex="-1"></a><span class="co"># add Hittinger IDs</span></span>
<span id="cb4-36"><a href="phylogenetic-reconstruction.html#cb4-36" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> 99_Hittinger/coding/<span class="pp">*</span>.fsa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb4-37"><a href="phylogenetic-reconstruction.html#cb4-37" tabindex="-1"></a>  <span class="fu">grep</span> <span class="st">&quot;&gt;&quot;</span> <span class="va">$file</span> <span class="dt">\</span></span>
<span id="cb4-38"><a href="phylogenetic-reconstruction.html#cb4-38" tabindex="-1"></a>    <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&#39;s/&gt;//g&#39;</span> <span class="dt">\</span></span>
<span id="cb4-39"><a href="phylogenetic-reconstruction.html#cb4-39" tabindex="-1"></a>    <span class="op">&gt;</span> 99_Hittinger/geneList/<span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .fsa<span class="va">)</span>.lst<span class="kw">;</span></span>
<span id="cb4-40"><a href="phylogenetic-reconstruction.html#cb4-40" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb4-41"><a href="phylogenetic-reconstruction.html#cb4-41" tabindex="-1"></a></span>
<span id="cb4-42"><a href="phylogenetic-reconstruction.html#cb4-42" tabindex="-1"></a><span class="ex">python3.5</span> Vikings.addHittingerID.py <span class="dt">\</span></span>
<span id="cb4-43"><a href="phylogenetic-reconstruction.html#cb4-43" tabindex="-1"></a>  <span class="at">--CDlists</span> 01_nt_clusters_lst_select <span class="dt">\</span></span>
<span id="cb4-44"><a href="phylogenetic-reconstruction.html#cb4-44" tabindex="-1"></a>  <span class="at">--Hit</span> 99_Hittinger/geneList/ <span class="dt">\</span></span>
<span id="cb4-45"><a href="phylogenetic-reconstruction.html#cb4-45" tabindex="-1"></a>  <span class="at">--outdir</span> 02_nt_clusters_lst_select_Hittinger</span>
<span id="cb4-46"><a href="phylogenetic-reconstruction.html#cb4-46" tabindex="-1"></a></span>
<span id="cb4-47"><a href="phylogenetic-reconstruction.html#cb4-47" tabindex="-1"></a><span class="fu">cat</span> all_plusKV.aa.fa 99_Hittinger/coding/<span class="pp">*</span>.aa.mfa <span class="op">&gt;</span> all_plusKV.Hit.aa.fa</span>
<span id="cb4-48"><a href="phylogenetic-reconstruction.html#cb4-48" tabindex="-1"></a><span class="fu">cat</span> all_plusKV.nt.fa 99_Hittinger/coding/<span class="pp">*</span>.fsa <span class="op">&gt;</span> all_plusKV.Hit.nt.fa</span>
<span id="cb4-49"><a href="phylogenetic-reconstruction.html#cb4-49" tabindex="-1"></a></span>
<span id="cb4-50"><a href="phylogenetic-reconstruction.html#cb4-50" tabindex="-1"></a></span>
<span id="cb4-51"><a href="phylogenetic-reconstruction.html#cb4-51" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> 02_nt_clusters_lst_select_Hittinger/<span class="pp">*</span>.Hit<span class="kw">;</span> <span class="cf">do</span> <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/ .*//g&#39;</span> <span class="va">$file</span> <span class="kw">;</span> <span class="cf">done</span></span>
<span id="cb4-52"><a href="phylogenetic-reconstruction.html#cb4-52" tabindex="-1"></a></span>
<span id="cb4-53"><a href="phylogenetic-reconstruction.html#cb4-53" tabindex="-1"></a><span class="co"># select aa for each cluster</span></span>
<span id="cb4-54"><a href="phylogenetic-reconstruction.html#cb4-54" tabindex="-1"></a><span class="ex">python3.5</span> Vikings.CDHITclustersList2ntSeq.py <span class="dt">\</span></span>
<span id="cb4-55"><a href="phylogenetic-reconstruction.html#cb4-55" tabindex="-1"></a>  <span class="at">--fasta</span> all_plusKV.Hit.aa.fa <span class="dt">\</span></span>
<span id="cb4-56"><a href="phylogenetic-reconstruction.html#cb4-56" tabindex="-1"></a>  <span class="at">--indir</span> ./02_nt_clusters_lst_select_Hittinger <span class="dt">\</span></span>
<span id="cb4-57"><a href="phylogenetic-reconstruction.html#cb4-57" tabindex="-1"></a>  <span class="at">--outdir</span> ./03_aa_clusters/</span>
<span id="cb4-58"><a href="phylogenetic-reconstruction.html#cb4-58" tabindex="-1"></a></span>
<span id="cb4-59"><a href="phylogenetic-reconstruction.html#cb4-59" tabindex="-1"></a><span class="co"># align aa clusters</span></span>
<span id="cb4-60"><a href="phylogenetic-reconstruction.html#cb4-60" tabindex="-1"></a><span class="fu">ls</span> ./03_aa_clusters/ <span class="dt">\</span></span>
<span id="cb4-61"><a href="phylogenetic-reconstruction.html#cb4-61" tabindex="-1"></a>  <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-n</span> 8 <span class="at">-P</span> 8 <span class="at">-I</span> {} sh <span class="at">-c</span> <span class="dt">\</span></span>
<span id="cb4-62"><a href="phylogenetic-reconstruction.html#cb4-62" tabindex="-1"></a>  <span class="st">&#39;mafft --localpair --maxiterate 1000 --ep 0.123 --thread 8 ./03_aa_clusters/{} &gt; ./04_aa_clusters_aln/{}.aln.fa&#39;</span></span>
<span id="cb4-63"><a href="phylogenetic-reconstruction.html#cb4-63" tabindex="-1"></a></span>
<span id="cb4-64"><a href="phylogenetic-reconstruction.html#cb4-64" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> ./04_aa_clusters_aln/<span class="pp">*</span>.fa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb4-65"><a href="phylogenetic-reconstruction.html#cb4-65" tabindex="-1"></a>  <span class="fu">perl</span> ~/rubbish/00_Brigida_starting_files/pal2nal.v14/pal2nal.pl <span class="dt">\</span></span>
<span id="cb4-66"><a href="phylogenetic-reconstruction.html#cb4-66" tabindex="-1"></a>    <span class="va">$file</span> 05_nt_clusters/<span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .aln.fa<span class="va">)</span><span class="dt">\</span></span>
<span id="cb4-67"><a href="phylogenetic-reconstruction.html#cb4-67" tabindex="-1"></a>    <span class="at">-output</span> fasta <span class="dt">\</span></span>
<span id="cb4-68"><a href="phylogenetic-reconstruction.html#cb4-68" tabindex="-1"></a>    <span class="op">&gt;</span> 06_nt_clusters_aln/<span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .fa<span class="va">)</span>.fa<span class="kw">;</span></span>
<span id="cb4-69"><a href="phylogenetic-reconstruction.html#cb4-69" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb4-70"><a href="phylogenetic-reconstruction.html#cb4-70" tabindex="-1"></a></span>
<span id="cb4-71"><a href="phylogenetic-reconstruction.html#cb4-71" tabindex="-1"></a><span class="co"># rename headers to keep only strain name</span></span>
<span id="cb4-72"><a href="phylogenetic-reconstruction.html#cb4-72" tabindex="-1"></a><span class="fu">find</span> 05_nt_align_long/<span class="pp">*</span> <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-n</span> 1 <span class="at">-P</span> 64 sed <span class="at">-i</span> <span class="st">&#39;s/_.*//g&#39;</span> </span>
<span id="cb4-73"><a href="phylogenetic-reconstruction.html#cb4-73" tabindex="-1"></a></span>
<span id="cb4-74"><a href="phylogenetic-reconstruction.html#cb4-74" tabindex="-1"></a><span class="fu">mkdir</span> ../07_nt_aln_ready<span class="kw">;</span></span>
<span id="cb4-75"><a href="phylogenetic-reconstruction.html#cb4-75" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> ./<span class="pp">*</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb4-76"><a href="phylogenetic-reconstruction.html#cb4-76" tabindex="-1"></a>  <span class="va">NAME</span><span class="op">=</span><span class="va">$(</span><span class="fu">grep</span> S288C <span class="va">$file</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&#39;s/&gt;S288C_//g&#39;</span><span class="va">)</span><span class="kw">;</span></span>
<span id="cb4-77"><a href="phylogenetic-reconstruction.html#cb4-77" tabindex="-1"></a>  <span class="fu">cp</span> <span class="va">$file</span> ../07_nt_aln_ready/<span class="va">$NAME</span>.nt.aln.fa<span class="kw">;</span></span>
<span id="cb4-78"><a href="phylogenetic-reconstruction.html#cb4-78" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb4-79"><a href="phylogenetic-reconstruction.html#cb4-79" tabindex="-1"></a></span>
<span id="cb4-80"><a href="phylogenetic-reconstruction.html#cb4-80" tabindex="-1"></a><span class="co"># rename headers to keep only strain name</span></span>
<span id="cb4-81"><a href="phylogenetic-reconstruction.html#cb4-81" tabindex="-1"></a><span class="fu">find</span> 07_nt_aln_ready/<span class="pp">*</span> <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-n</span> 1 <span class="at">-P</span> 64 sed <span class="at">-i</span> <span class="st">&#39;s/_.*//g&#39;</span> </span></code></pre></div>
</div>
</div>
<div id="run-phylogenetic-reconstruction" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Run phylogenetic reconstruction<a href="phylogenetic-reconstruction.html#run-phylogenetic-reconstruction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This resulted in 1,390 single-copy genes identified across 151 yeast strains of Gallone et al.2016 and the 44 kveiks strains in this study.
Lets build a Maximum Likelihood tree to reconstruct the phylogenetic relationships between these <em>S. cerevisiae</em> yeast strains!</p>
<div id="create-ml-tree" class="section level3 hasAnchor" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> Create ML tree<a href="phylogenetic-reconstruction.html#create-ml-tree" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="phylogenetic-reconstruction.html#cb5-1" tabindex="-1"></a><span class="co"># concatenate alignmnets</span></span>
<span id="cb5-2"><a href="phylogenetic-reconstruction.html#cb5-2" tabindex="-1"></a><span class="fu">mkdir</span> 08_concatenated_ML</span>
<span id="cb5-3"><a href="phylogenetic-reconstruction.html#cb5-3" tabindex="-1"></a><span class="fu">perl</span> catfasta2phyml.pl <span class="dt">\</span></span>
<span id="cb5-4"><a href="phylogenetic-reconstruction.html#cb5-4" tabindex="-1"></a>  <span class="at">--fasta</span> <span class="dt">\</span></span>
<span id="cb5-5"><a href="phylogenetic-reconstruction.html#cb5-5" tabindex="-1"></a>  <span class="at">--concatenate</span> 07_nt_aln_ready/<span class="pp">*</span>.fa <span class="dt">\</span></span>
<span id="cb5-6"><a href="phylogenetic-reconstruction.html#cb5-6" tabindex="-1"></a>  <span class="op">&gt;</span> 08_concatenated_ML/clstr.all.concat.nt.align.fa</span>
<span id="cb5-7"><a href="phylogenetic-reconstruction.html#cb5-7" tabindex="-1"></a></span>
<span id="cb5-8"><a href="phylogenetic-reconstruction.html#cb5-8" tabindex="-1"></a><span class="co"># ML analysis on concatenated alignment</span></span>
<span id="cb5-9"><a href="phylogenetic-reconstruction.html#cb5-9" tabindex="-1"></a><span class="ex">~/bin/iqtree-1.6.8-Linux/bin/iqtree</span> <span class="dt">\</span></span>
<span id="cb5-10"><a href="phylogenetic-reconstruction.html#cb5-10" tabindex="-1"></a>  <span class="at">-s</span> clstr.all.concat.nt.align.fa <span class="dt">\</span></span>
<span id="cb5-11"><a href="phylogenetic-reconstruction.html#cb5-11" tabindex="-1"></a>  <span class="at">-st</span> CODON <span class="dt">\</span></span>
<span id="cb5-12"><a href="phylogenetic-reconstruction.html#cb5-12" tabindex="-1"></a>  <span class="at">-pre</span> clstr.all.concat.nt.align <span class="dt">\</span></span>
<span id="cb5-13"><a href="phylogenetic-reconstruction.html#cb5-13" tabindex="-1"></a>  <span class="at">-nt</span> 52 <span class="dt">\</span></span>
<span id="cb5-14"><a href="phylogenetic-reconstruction.html#cb5-14" tabindex="-1"></a>  <span class="at">-wbt</span> <span class="dt">\</span></span>
<span id="cb5-15"><a href="phylogenetic-reconstruction.html#cb5-15" tabindex="-1"></a>  <span class="at">-bb</span> 1000 <span class="dt">\</span></span>
<span id="cb5-16"><a href="phylogenetic-reconstruction.html#cb5-16" tabindex="-1"></a>  <span class="at">-alrt</span> 1000 <span class="dt">\</span></span>
<span id="cb5-17"><a href="phylogenetic-reconstruction.html#cb5-17" tabindex="-1"></a>  <span class="at">-m</span> MG+F1X4</span></code></pre></div>
<p>Plot the reconstructed phylogenetic tree.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="phylogenetic-reconstruction.html#cb6-1" tabindex="-1"></a><span class="co"># load tree</span></span>
<span id="cb6-2"><a href="phylogenetic-reconstruction.html#cb6-2" tabindex="-1"></a>best_tree <span class="ot">=</span> <span class="fu">read.tree</span>(<span class="fu">paste</span>(mainDir, <span class="st">&quot;clstr.all.concat.nt.align.short.tre&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb6-3"><a href="phylogenetic-reconstruction.html#cb6-3" tabindex="-1"></a>best_tree <span class="ot">=</span> <span class="fu">as.phylo</span>(best_tree)</span>
<span id="cb6-4"><a href="phylogenetic-reconstruction.html#cb6-4" tabindex="-1"></a>best_tree<span class="sc">$</span>node.label <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(best_tree<span class="sc">$</span>node.label))</span>
<span id="cb6-5"><a href="phylogenetic-reconstruction.html#cb6-5" tabindex="-1"></a>best_tree <span class="ot">=</span> ape<span class="sc">::</span><span class="fu">root</span>(best_tree,</span>
<span id="cb6-6"><a href="phylogenetic-reconstruction.html#cb6-6" tabindex="-1"></a>                      <span class="fu">which</span>(best_tree<span class="sc">$</span>tip.label <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Skud&quot;</span>)),</span>
<span id="cb6-7"><a href="phylogenetic-reconstruction.html#cb6-7" tabindex="-1"></a>                      <span class="at">edgelabel =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-8"><a href="phylogenetic-reconstruction.html#cb6-8" tabindex="-1"></a>best_tree <span class="ot">=</span> <span class="fu">ladderize</span>(best_tree, <span class="at">right =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-9"><a href="phylogenetic-reconstruction.html#cb6-9" tabindex="-1"></a></span>
<span id="cb6-10"><a href="phylogenetic-reconstruction.html#cb6-10" tabindex="-1"></a><span class="co"># load annotations</span></span>
<span id="cb6-11"><a href="phylogenetic-reconstruction.html#cb6-11" tabindex="-1"></a>tree_annot_raw <span class="ot">=</span> <span class="fu">read.table</span>(<span class="fu">paste</span>(mainDir, <span class="st">&quot;genetrees.output.BS.annot.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>),</span>
<span id="cb6-12"><a href="phylogenetic-reconstruction.html#cb6-12" tabindex="-1"></a>                            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb6-13"><a href="phylogenetic-reconstruction.html#cb6-13" tabindex="-1"></a>                            <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb6-14"><a href="phylogenetic-reconstruction.html#cb6-14" tabindex="-1"></a>                            <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb6-15"><a href="phylogenetic-reconstruction.html#cb6-15" tabindex="-1"></a>tree_annot_raw <span class="ot">=</span> <span class="fu">data.frame</span>(tree_annot_raw)</span>
<span id="cb6-16"><a href="phylogenetic-reconstruction.html#cb6-16" tabindex="-1"></a>tree_annot_raw<span class="sc">$</span>Origin <span class="ot">=</span> <span class="fu">factor</span>(tree_annot_raw<span class="sc">$</span>Origin,</span>
<span id="cb6-17"><a href="phylogenetic-reconstruction.html#cb6-17" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;North-West Norway&quot;</span>, <span class="st">&quot;Eastern Norway&quot;</span>, <span class="st">&quot;South-West Norway&quot;</span>,</span>
<span id="cb6-18"><a href="phylogenetic-reconstruction.html#cb6-18" tabindex="-1"></a>                                          <span class="st">&quot;Latvia&quot;</span>, <span class="st">&quot;Lithuania&quot;</span>, <span class="st">&quot;Russia&quot;</span>, <span class="st">&quot; &quot;</span>))</span>
<span id="cb6-19"><a href="phylogenetic-reconstruction.html#cb6-19" tabindex="-1"></a></span>
<span id="cb6-20"><a href="phylogenetic-reconstruction.html#cb6-20" tabindex="-1"></a><span class="co"># plot</span></span>
<span id="cb6-21"><a href="phylogenetic-reconstruction.html#cb6-21" tabindex="-1"></a>p <span class="ot">=</span> <span class="fu">ggtree</span>(best_tree,</span>
<span id="cb6-22"><a href="phylogenetic-reconstruction.html#cb6-22" tabindex="-1"></a>           <span class="at">layout =</span> <span class="st">&quot;fan&quot;</span>,</span>
<span id="cb6-23"><a href="phylogenetic-reconstruction.html#cb6-23" tabindex="-1"></a>           <span class="at">open.angle =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb6-24"><a href="phylogenetic-reconstruction.html#cb6-24" tabindex="-1"></a>  <span class="fu">geom_tiplab2</span>(<span class="at">offset =</span> <span class="fl">0.001</span>,</span>
<span id="cb6-25"><a href="phylogenetic-reconstruction.html#cb6-25" tabindex="-1"></a>               <span class="at">align =</span> <span class="cn">TRUE</span>,</span>
<span id="cb6-26"><a href="phylogenetic-reconstruction.html#cb6-26" tabindex="-1"></a>               <span class="at">size =</span> <span class="fl">2.75</span>) <span class="sc">+</span></span>
<span id="cb6-27"><a href="phylogenetic-reconstruction.html#cb6-27" tabindex="-1"></a>  <span class="fu">geom_nodepoint</span>(<span class="fu">aes</span>(<span class="at">fill =</span> <span class="fu">cut</span>(<span class="fu">as.numeric</span>(label), <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">70</span>, <span class="dv">90</span>, <span class="dv">100</span>))),</span>
<span id="cb6-28"><a href="phylogenetic-reconstruction.html#cb6-28" tabindex="-1"></a>                 <span class="at">shape =</span> <span class="dv">21</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb6-29"><a href="phylogenetic-reconstruction.html#cb6-29" tabindex="-1"></a>  <span class="fu">theme_tree</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.55</span>)) <span class="sc">+</span></span>
<span id="cb6-30"><a href="phylogenetic-reconstruction.html#cb6-30" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;black&quot;</span>),</span>
<span id="cb6-31"><a href="phylogenetic-reconstruction.html#cb6-31" tabindex="-1"></a>                    <span class="at">guide =</span> <span class="st">&quot;legend&quot;</span>,</span>
<span id="cb6-32"><a href="phylogenetic-reconstruction.html#cb6-32" tabindex="-1"></a>                    <span class="at">name =</span> <span class="st">&quot;Bootstrap&quot;</span>,</span>
<span id="cb6-33"><a href="phylogenetic-reconstruction.html#cb6-33" tabindex="-1"></a>                    <span class="at">breaks =</span> <span class="fu">c</span>(<span class="st">&quot;(90,100]&quot;</span>, <span class="st">&quot;(70,90]&quot;</span>, <span class="st">&quot;(0,70]&quot;</span>),</span>
<span id="cb6-34"><a href="phylogenetic-reconstruction.html#cb6-34" tabindex="-1"></a>                    <span class="at">labels =</span> <span class="fu">expression</span>(BP<span class="sc">&gt;=</span><span class="dv">90</span>,<span class="dv">70</span> <span class="sc">&lt;=</span> BP <span class="sc">*</span> <span class="st">&quot; &lt; 90&quot;</span>, BP <span class="sc">&lt;</span> <span class="dv">70</span>))</span>
<span id="cb6-35"><a href="phylogenetic-reconstruction.html#cb6-35" tabindex="-1"></a></span>
<span id="cb6-36"><a href="phylogenetic-reconstruction.html#cb6-36" tabindex="-1"></a>p1 <span class="ot">=</span> p <span class="sc">+</span></span>
<span id="cb6-37"><a href="phylogenetic-reconstruction.html#cb6-37" tabindex="-1"></a>  <span class="fu">new_scale_fill</span>() <span class="sc">+</span></span>
<span id="cb6-38"><a href="phylogenetic-reconstruction.html#cb6-38" tabindex="-1"></a>  <span class="fu">geom_fruit</span>(<span class="at">data =</span> tree_annot_raw,</span>
<span id="cb6-39"><a href="phylogenetic-reconstruction.html#cb6-39" tabindex="-1"></a>             <span class="at">geom =</span> geom_tile,</span>
<span id="cb6-40"><a href="phylogenetic-reconstruction.html#cb6-40" tabindex="-1"></a>             <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">y =</span> Strain, <span class="at">fill =</span> Clade),</span>
<span id="cb6-41"><a href="phylogenetic-reconstruction.html#cb6-41" tabindex="-1"></a>             <span class="at">color =</span> <span class="cn">NA</span>, <span class="at">offset =</span> <span class="fl">0.00975</span>, <span class="at">pwidth =</span> <span class="fl">0.001</span>) <span class="sc">+</span></span>
<span id="cb6-42"><a href="phylogenetic-reconstruction.html#cb6-42" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&#39;#e6194b&#39;</span>, <span class="st">&#39;#3cb44b&#39;</span>, <span class="st">&#39;#ffe119&#39;</span>, <span class="st">&#39;#4363d8&#39;</span>, <span class="st">&#39;#f58231&#39;</span>,</span>
<span id="cb6-43"><a href="phylogenetic-reconstruction.html#cb6-43" tabindex="-1"></a>                               <span class="st">&#39;#911eb4&#39;</span>, <span class="st">&#39;#46f0f0&#39;</span>, <span class="st">&#39;#f032e6&#39;</span>, <span class="st">&#39;#bcf60c&#39;</span>, <span class="st">&#39;#fabebe&#39;</span>,</span>
<span id="cb6-44"><a href="phylogenetic-reconstruction.html#cb6-44" tabindex="-1"></a>                               <span class="st">&#39;#008080&#39;</span>, <span class="st">&#39;#e6beff&#39;</span>, <span class="st">&#39;#9a6324&#39;</span>, <span class="st">&#39;#fffac8&#39;</span>, <span class="st">&#39;#800000&#39;</span>,</span>
<span id="cb6-45"><a href="phylogenetic-reconstruction.html#cb6-45" tabindex="-1"></a>                               <span class="st">&#39;#aaffc3&#39;</span>, <span class="st">&#39;#808000&#39;</span>, <span class="st">&#39;#ffd8b1&#39;</span>, <span class="st">&#39;#000075&#39;</span>, <span class="st">&#39;#808080&#39;</span>,</span>
<span id="cb6-46"><a href="phylogenetic-reconstruction.html#cb6-46" tabindex="-1"></a>                               <span class="st">&#39;#ffffff&#39;</span>, <span class="st">&#39;#000000&#39;</span>)) </span>
<span id="cb6-47"><a href="phylogenetic-reconstruction.html#cb6-47" tabindex="-1"></a>p2 <span class="ot">=</span> p1 <span class="sc">+</span></span>
<span id="cb6-48"><a href="phylogenetic-reconstruction.html#cb6-48" tabindex="-1"></a>  <span class="fu">new_scale_fill</span>() <span class="sc">+</span></span>
<span id="cb6-49"><a href="phylogenetic-reconstruction.html#cb6-49" tabindex="-1"></a>  <span class="fu">geom_fruit</span>(<span class="at">data =</span> tree_annot_raw,</span>
<span id="cb6-50"><a href="phylogenetic-reconstruction.html#cb6-50" tabindex="-1"></a>             <span class="at">geom =</span> geom_tile,</span>
<span id="cb6-51"><a href="phylogenetic-reconstruction.html#cb6-51" tabindex="-1"></a>             <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">y =</span> Strain, <span class="at">fill =</span> Origin),</span>
<span id="cb6-52"><a href="phylogenetic-reconstruction.html#cb6-52" tabindex="-1"></a>             <span class="at">color =</span> <span class="cn">NA</span>, <span class="at">offset =</span> <span class="fl">0.0115</span>, <span class="at">pwidth =</span> <span class="fl">0.001</span>) <span class="sc">+</span></span>
<span id="cb6-53"><a href="phylogenetic-reconstruction.html#cb6-53" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&#39;#03045e&#39;</span>, <span class="st">&#39;#0077b6&#39;</span>, <span class="st">&#39;#00b4d8&#39;</span>, </span>
<span id="cb6-54"><a href="phylogenetic-reconstruction.html#cb6-54" tabindex="-1"></a>                               <span class="st">&#39;#808000&#39;</span>, <span class="st">&#39;#aaffc3&#39;</span>, <span class="st">&#39;#800000&#39;</span>,</span>
<span id="cb6-55"><a href="phylogenetic-reconstruction.html#cb6-55" tabindex="-1"></a>                               <span class="st">&#39;#ffffff&#39;</span>)) <span class="sc">+</span></span>
<span id="cb6-56"><a href="phylogenetic-reconstruction.html#cb6-56" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">1.1</span>, <span class="fl">0.5</span>))</span>
<span id="cb6-57"><a href="phylogenetic-reconstruction.html#cb6-57" tabindex="-1"></a></span>
<span id="cb6-58"><a href="phylogenetic-reconstruction.html#cb6-58" tabindex="-1"></a>p2</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-7"></span>
<img src="_main_files/figure-html/unnamed-chunk-7-1.png" alt="Farmhouse yeasts Maximum Likelihood phynogenetic recontruction" width="672" />
<p class="caption">
Figure 5.1: Farmhouse yeasts Maximum Likelihood phynogenetic recontruction
</p>
</div>
<p>Based on the concatenated ML tree, Farmhouse yeasts seems not monophyletic. However, the well enstablished phylogenetic relationships between industrial <em>S. cerevisiae</em> strains (e.g.: Beer1, Beer2, Wine clades, see Gallone et al.2016, Peter et al.2018) were not reconstructed, suggesting that the presence of highly heterozigous and possibly hybrid (Preiss et al.2018) Farmhouse yeasts may impair the phylogenetic reconstruction.</p>
<p>Lets see if the 1,390 genes we have selected for the phylogenetic reconstruction can confirm the phylogenetic relationships between industrial clades reconstructed by Gallone et al.2016, Peter et al.2018). We therefore remove the farmhouse yeasts sequences from the alignments, recompute the aa and nt alignments and run the supermatrix ML analysis.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="phylogenetic-reconstruction.html#cb7-1" tabindex="-1"></a><span class="co"># load tree</span></span>
<span id="cb7-2"><a href="phylogenetic-reconstruction.html#cb7-2" tabindex="-1"></a>best_tree <span class="ot">=</span> <span class="fu">read.tree</span>(<span class="fu">paste</span>(mainDir, <span class="st">&quot;clstr.all.concat.nt.align.noKV.short.tre&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb7-3"><a href="phylogenetic-reconstruction.html#cb7-3" tabindex="-1"></a>best_tree <span class="ot">=</span> <span class="fu">as.phylo</span>(best_tree)</span>
<span id="cb7-4"><a href="phylogenetic-reconstruction.html#cb7-4" tabindex="-1"></a>best_tree<span class="sc">$</span>node.label <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(best_tree<span class="sc">$</span>node.label))</span>
<span id="cb7-5"><a href="phylogenetic-reconstruction.html#cb7-5" tabindex="-1"></a>best_tree <span class="ot">=</span> ape<span class="sc">::</span><span class="fu">root</span>(best_tree,</span>
<span id="cb7-6"><a href="phylogenetic-reconstruction.html#cb7-6" tabindex="-1"></a>                      <span class="fu">which</span>(best_tree<span class="sc">$</span>tip.label <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Skud&quot;</span>)),</span>
<span id="cb7-7"><a href="phylogenetic-reconstruction.html#cb7-7" tabindex="-1"></a>                      <span class="at">edgelabel =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-8"><a href="phylogenetic-reconstruction.html#cb7-8" tabindex="-1"></a>best_tree <span class="ot">=</span> <span class="fu">ladderize</span>(best_tree, <span class="at">right =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-9"><a href="phylogenetic-reconstruction.html#cb7-9" tabindex="-1"></a></span>
<span id="cb7-10"><a href="phylogenetic-reconstruction.html#cb7-10" tabindex="-1"></a><span class="co"># load annotations</span></span>
<span id="cb7-11"><a href="phylogenetic-reconstruction.html#cb7-11" tabindex="-1"></a>tree_annot_raw <span class="ot">=</span> <span class="fu">read.table</span>(<span class="fu">paste</span>(mainDir, <span class="st">&quot;genetrees.output.BS.annot.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>),</span>
<span id="cb7-12"><a href="phylogenetic-reconstruction.html#cb7-12" tabindex="-1"></a>                            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb7-13"><a href="phylogenetic-reconstruction.html#cb7-13" tabindex="-1"></a>                            <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb7-14"><a href="phylogenetic-reconstruction.html#cb7-14" tabindex="-1"></a>                            <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb7-15"><a href="phylogenetic-reconstruction.html#cb7-15" tabindex="-1"></a>tree_annot_raw <span class="ot">=</span> <span class="fu">data.frame</span>(tree_annot_raw)</span>
<span id="cb7-16"><a href="phylogenetic-reconstruction.html#cb7-16" tabindex="-1"></a>tree_annot_raw<span class="sc">$</span>Origin <span class="ot">=</span> <span class="fu">factor</span>(tree_annot_raw<span class="sc">$</span>Origin,</span>
<span id="cb7-17"><a href="phylogenetic-reconstruction.html#cb7-17" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;North-West Norway&quot;</span>, <span class="st">&quot;Eastern Norway&quot;</span>, <span class="st">&quot;South-West Norway&quot;</span>,</span>
<span id="cb7-18"><a href="phylogenetic-reconstruction.html#cb7-18" tabindex="-1"></a>                                          <span class="st">&quot;Latvia&quot;</span>, <span class="st">&quot;Lithuania&quot;</span>, <span class="st">&quot;Russia&quot;</span>, <span class="st">&quot; &quot;</span>))</span>
<span id="cb7-19"><a href="phylogenetic-reconstruction.html#cb7-19" tabindex="-1"></a></span>
<span id="cb7-20"><a href="phylogenetic-reconstruction.html#cb7-20" tabindex="-1"></a><span class="co"># plot</span></span>
<span id="cb7-21"><a href="phylogenetic-reconstruction.html#cb7-21" tabindex="-1"></a>p <span class="ot">=</span> <span class="fu">ggtree</span>(best_tree,</span>
<span id="cb7-22"><a href="phylogenetic-reconstruction.html#cb7-22" tabindex="-1"></a>           <span class="at">layout =</span> <span class="st">&quot;fan&quot;</span>,</span>
<span id="cb7-23"><a href="phylogenetic-reconstruction.html#cb7-23" tabindex="-1"></a>           <span class="at">open.angle =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb7-24"><a href="phylogenetic-reconstruction.html#cb7-24" tabindex="-1"></a>  <span class="fu">geom_tiplab2</span>(<span class="at">offset =</span> <span class="fl">0.001</span>,</span>
<span id="cb7-25"><a href="phylogenetic-reconstruction.html#cb7-25" tabindex="-1"></a>               <span class="at">align =</span> <span class="cn">TRUE</span>,</span>
<span id="cb7-26"><a href="phylogenetic-reconstruction.html#cb7-26" tabindex="-1"></a>               <span class="at">size =</span> <span class="fl">2.75</span>) <span class="sc">+</span></span>
<span id="cb7-27"><a href="phylogenetic-reconstruction.html#cb7-27" tabindex="-1"></a>  <span class="fu">geom_nodepoint</span>(<span class="fu">aes</span>(<span class="at">fill =</span> <span class="fu">cut</span>(<span class="fu">as.numeric</span>(label), <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">70</span>, <span class="dv">90</span>, <span class="dv">100</span>))),</span>
<span id="cb7-28"><a href="phylogenetic-reconstruction.html#cb7-28" tabindex="-1"></a>                 <span class="at">shape =</span> <span class="dv">21</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb7-29"><a href="phylogenetic-reconstruction.html#cb7-29" tabindex="-1"></a>  <span class="fu">theme_tree</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.55</span>)) <span class="sc">+</span></span>
<span id="cb7-30"><a href="phylogenetic-reconstruction.html#cb7-30" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;black&quot;</span>),</span>
<span id="cb7-31"><a href="phylogenetic-reconstruction.html#cb7-31" tabindex="-1"></a>                    <span class="at">guide =</span> <span class="st">&quot;legend&quot;</span>,</span>
<span id="cb7-32"><a href="phylogenetic-reconstruction.html#cb7-32" tabindex="-1"></a>                    <span class="at">name =</span> <span class="st">&quot;Bootstrap&quot;</span>,</span>
<span id="cb7-33"><a href="phylogenetic-reconstruction.html#cb7-33" tabindex="-1"></a>                    <span class="at">breaks =</span> <span class="fu">c</span>(<span class="st">&quot;(90,100]&quot;</span>, <span class="st">&quot;(70,90]&quot;</span>, <span class="st">&quot;(0,70]&quot;</span>),</span>
<span id="cb7-34"><a href="phylogenetic-reconstruction.html#cb7-34" tabindex="-1"></a>                    <span class="at">labels =</span> <span class="fu">expression</span>(BP<span class="sc">&gt;=</span><span class="dv">90</span>,<span class="dv">70</span> <span class="sc">&lt;=</span> BP <span class="sc">*</span> <span class="st">&quot; &lt; 90&quot;</span>, BP <span class="sc">&lt;</span> <span class="dv">70</span>))</span>
<span id="cb7-35"><a href="phylogenetic-reconstruction.html#cb7-35" tabindex="-1"></a></span>
<span id="cb7-36"><a href="phylogenetic-reconstruction.html#cb7-36" tabindex="-1"></a>p1 <span class="ot">=</span> p <span class="sc">+</span></span>
<span id="cb7-37"><a href="phylogenetic-reconstruction.html#cb7-37" tabindex="-1"></a>  <span class="fu">new_scale_fill</span>() <span class="sc">+</span></span>
<span id="cb7-38"><a href="phylogenetic-reconstruction.html#cb7-38" tabindex="-1"></a>  <span class="fu">geom_fruit</span>(<span class="at">data =</span> tree_annot_raw,</span>
<span id="cb7-39"><a href="phylogenetic-reconstruction.html#cb7-39" tabindex="-1"></a>             <span class="at">geom =</span> geom_tile,</span>
<span id="cb7-40"><a href="phylogenetic-reconstruction.html#cb7-40" tabindex="-1"></a>             <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">y =</span> Strain, <span class="at">fill =</span> Clade),</span>
<span id="cb7-41"><a href="phylogenetic-reconstruction.html#cb7-41" tabindex="-1"></a>             <span class="at">color =</span> <span class="cn">NA</span>, <span class="at">offset =</span> <span class="fl">0.00975</span>, <span class="at">pwidth =</span> <span class="fl">0.001</span>) <span class="sc">+</span></span>
<span id="cb7-42"><a href="phylogenetic-reconstruction.html#cb7-42" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&#39;#e6194b&#39;</span>, <span class="st">&#39;#3cb44b&#39;</span>, <span class="st">&#39;#ffe119&#39;</span>, <span class="st">&#39;#4363d8&#39;</span>, <span class="st">&#39;#f58231&#39;</span>,</span>
<span id="cb7-43"><a href="phylogenetic-reconstruction.html#cb7-43" tabindex="-1"></a>                               <span class="st">&#39;#911eb4&#39;</span>, <span class="st">&#39;#46f0f0&#39;</span>, <span class="st">&#39;#f032e6&#39;</span>, <span class="st">&#39;#bcf60c&#39;</span>, <span class="st">&#39;#fabebe&#39;</span>,</span>
<span id="cb7-44"><a href="phylogenetic-reconstruction.html#cb7-44" tabindex="-1"></a>                               <span class="st">&#39;#008080&#39;</span>, <span class="st">&#39;#e6beff&#39;</span>, <span class="st">&#39;#9a6324&#39;</span>, <span class="st">&#39;#fffac8&#39;</span>, <span class="st">&#39;#800000&#39;</span>,</span>
<span id="cb7-45"><a href="phylogenetic-reconstruction.html#cb7-45" tabindex="-1"></a>                               <span class="st">&#39;#aaffc3&#39;</span>, <span class="st">&#39;#808000&#39;</span>, <span class="st">&#39;#ffd8b1&#39;</span>, <span class="st">&#39;#000075&#39;</span>, <span class="st">&#39;#808080&#39;</span>,</span>
<span id="cb7-46"><a href="phylogenetic-reconstruction.html#cb7-46" tabindex="-1"></a>                               <span class="st">&#39;#ffffff&#39;</span>, <span class="st">&#39;#000000&#39;</span>)) </span>
<span id="cb7-47"><a href="phylogenetic-reconstruction.html#cb7-47" tabindex="-1"></a>p2 <span class="ot">=</span> p1 <span class="sc">+</span></span>
<span id="cb7-48"><a href="phylogenetic-reconstruction.html#cb7-48" tabindex="-1"></a>  <span class="fu">new_scale_fill</span>() <span class="sc">+</span></span>
<span id="cb7-49"><a href="phylogenetic-reconstruction.html#cb7-49" tabindex="-1"></a>  <span class="fu">geom_fruit</span>(<span class="at">data =</span> tree_annot_raw,</span>
<span id="cb7-50"><a href="phylogenetic-reconstruction.html#cb7-50" tabindex="-1"></a>             <span class="at">geom =</span> geom_tile,</span>
<span id="cb7-51"><a href="phylogenetic-reconstruction.html#cb7-51" tabindex="-1"></a>             <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">y =</span> Strain, <span class="at">fill =</span> Origin),</span>
<span id="cb7-52"><a href="phylogenetic-reconstruction.html#cb7-52" tabindex="-1"></a>             <span class="at">color =</span> <span class="cn">NA</span>, <span class="at">offset =</span> <span class="fl">0.0115</span>, <span class="at">pwidth =</span> <span class="fl">0.001</span>) <span class="sc">+</span></span>
<span id="cb7-53"><a href="phylogenetic-reconstruction.html#cb7-53" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&#39;#03045e&#39;</span>, <span class="st">&#39;#0077b6&#39;</span>, <span class="st">&#39;#00b4d8&#39;</span>, </span>
<span id="cb7-54"><a href="phylogenetic-reconstruction.html#cb7-54" tabindex="-1"></a>                               <span class="st">&#39;#808000&#39;</span>, <span class="st">&#39;#aaffc3&#39;</span>, <span class="st">&#39;#800000&#39;</span>,</span>
<span id="cb7-55"><a href="phylogenetic-reconstruction.html#cb7-55" tabindex="-1"></a>                               <span class="st">&#39;#ffffff&#39;</span>)) <span class="sc">+</span></span>
<span id="cb7-56"><a href="phylogenetic-reconstruction.html#cb7-56" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">1.1</span>, <span class="fl">0.5</span>))</span>
<span id="cb7-57"><a href="phylogenetic-reconstruction.html#cb7-57" tabindex="-1"></a></span>
<span id="cb7-58"><a href="phylogenetic-reconstruction.html#cb7-58" tabindex="-1"></a>p2</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-8"></span>
<img src="_main_files/figure-html/unnamed-chunk-8-1.png" alt="Industrial yeasts Maximum Likelihood phynogenetic recontruction" width="672" />
<p class="caption">
Figure 5.2: Industrial yeasts Maximum Likelihood phynogenetic recontruction
</p>
</div>
<p>Yes, we can confirm the well enstablished phylogenetic relationships between already sequenced yeast straind from the industrial clades based on a supermatrix analysis of the concatenated selected 1,390 genes.</p>
</div>
<div id="create-coalescence-based-tree" class="section level3 hasAnchor" number="5.2.2">
<h3><span class="header-section-number">5.2.2</span> Create coalescence-based tree<a href="phylogenetic-reconstruction.html#create-coalescence-based-tree" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>As seen, ML approaches fail to reconstruct an expected phylogenetic tree of <em>S. cerevisiae</em> when Farmhouse yeasts are included. To better handle high levels of heterozygosity and partial lineage sorting, we inferred the species tree by applying coalescence-based phylognetic reconstruction on the single-gene ML trees.</p>
<p>[supermatrix analyses are based on large datasets and can use complex models that can account for the heterogeneity of the substitution process, but cannot account for gene-tree species-tree incongruence, while coalescence-based approaches accommodate incomplete lineage sorting, but rely on topologies inferred from small datasets using simpler models that make them more sensitive to stochastic noise]</p>
<p>Lets reconstruct the single gene ML trees.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="phylogenetic-reconstruction.html#cb8-1" tabindex="-1"></a><span class="co"># run IQtree in parallel for each nt gene alignment</span></span>
<span id="cb8-2"><a href="phylogenetic-reconstruction.html#cb8-2" tabindex="-1"></a><span class="fu">ls</span> ../07_nt_aln_ready/<span class="pp">*</span>.fa <span class="dt">\</span></span>
<span id="cb8-3"><a href="phylogenetic-reconstruction.html#cb8-3" tabindex="-1"></a>  <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-n</span> 8 <span class="at">-P</span> 8 <span class="at">-I</span> {} sh <span class="at">-c</span> <span class="dt">\</span></span>
<span id="cb8-4"><a href="phylogenetic-reconstruction.html#cb8-4" tabindex="-1"></a>    <span class="st">&#39;~/bin/iqtree-1.6.8-Linux/bin/iqtree \</span></span>
<span id="cb8-5"><a href="phylogenetic-reconstruction.html#cb8-5" tabindex="-1"></a><span class="st">       -s {} \</span></span>
<span id="cb8-6"><a href="phylogenetic-reconstruction.html#cb8-6" tabindex="-1"></a><span class="st">       -st CODON \</span></span>
<span id="cb8-7"><a href="phylogenetic-reconstruction.html#cb8-7" tabindex="-1"></a><span class="st">       -pre {} \</span></span>
<span id="cb8-8"><a href="phylogenetic-reconstruction.html#cb8-8" tabindex="-1"></a><span class="st">       -nt 64 \</span></span>
<span id="cb8-9"><a href="phylogenetic-reconstruction.html#cb8-9" tabindex="-1"></a><span class="st">       -wbt \</span></span>
<span id="cb8-10"><a href="phylogenetic-reconstruction.html#cb8-10" tabindex="-1"></a><span class="st">       -bb 1000 \</span></span>
<span id="cb8-11"><a href="phylogenetic-reconstruction.html#cb8-11" tabindex="-1"></a><span class="st">       -alrt 1000 \</span></span>
<span id="cb8-12"><a href="phylogenetic-reconstruction.html#cb8-12" tabindex="-1"></a><span class="st">       -m MG+F1X4&#39;</span></span></code></pre></div>
<p>Lets run a coalescence based approach on the reonstructed single-gene ML trees plus the 1,000 ultrafast Boostrap trees generated for each tree by IQtree (~1,400,000 single-gene tree). The coalescensce-based phylogeny was inferred 100 times allowing gene resampling to obtain Bootstrap estimated support values for each node.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="phylogenetic-reconstruction.html#cb9-1" tabindex="-1"></a><span class="co"># colaescence-based species tree</span></span>
<span id="cb9-2"><a href="phylogenetic-reconstruction.html#cb9-2" tabindex="-1"></a><span class="ex">java</span> <span class="at">-Xmx30000M</span> <span class="at">-jar</span> ~/bin/Astral/astral.5.6.3.jar <span class="dt">\</span></span>
<span id="cb9-3"><a href="phylogenetic-reconstruction.html#cb9-3" tabindex="-1"></a>  <span class="at">--bootstraps</span> genetrees.input.BS <span class="dt">\</span></span>
<span id="cb9-4"><a href="phylogenetic-reconstruction.html#cb9-4" tabindex="-1"></a>  <span class="at">--gene-resampling</span> <span class="dt">\</span></span>
<span id="cb9-5"><a href="phylogenetic-reconstruction.html#cb9-5" tabindex="-1"></a>  <span class="at">-r</span> 100 <span class="dt">\</span></span>
<span id="cb9-6"><a href="phylogenetic-reconstruction.html#cb9-6" tabindex="-1"></a>  <span class="at">--input</span> genetrees.input.tre <span class="dt">\</span></span>
<span id="cb9-7"><a href="phylogenetic-reconstruction.html#cb9-7" tabindex="-1"></a>  <span class="at">--output</span> genetrees.output.BS.tre <span class="dt">\</span></span>
<span id="cb9-8"><a href="phylogenetic-reconstruction.html#cb9-8" tabindex="-1"></a>  <span class="dv">2</span><span class="op">&gt;</span> genetrees.output.BS.log</span></code></pre></div>
<p>Plot the reconstructed species tree (based on coalescence).</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="phylogenetic-reconstruction.html#cb10-1" tabindex="-1"></a><span class="co"># load tree</span></span>
<span id="cb10-2"><a href="phylogenetic-reconstruction.html#cb10-2" tabindex="-1"></a>best_tree <span class="ot">=</span> <span class="fu">read.tree</span>(<span class="fu">paste</span>(mainDir, <span class="st">&quot;genetrees.output.BS.best.tre&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb10-3"><a href="phylogenetic-reconstruction.html#cb10-3" tabindex="-1"></a>best_tree <span class="ot">=</span> <span class="fu">as.phylo</span>(best_tree)</span>
<span id="cb10-4"><a href="phylogenetic-reconstruction.html#cb10-4" tabindex="-1"></a>best_tree<span class="sc">$</span>node.label <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(best_tree<span class="sc">$</span>node.label))</span>
<span id="cb10-5"><a href="phylogenetic-reconstruction.html#cb10-5" tabindex="-1"></a>best_tree <span class="ot">=</span> ape<span class="sc">::</span><span class="fu">root</span>(best_tree,</span>
<span id="cb10-6"><a href="phylogenetic-reconstruction.html#cb10-6" tabindex="-1"></a>                      <span class="fu">which</span>(best_tree<span class="sc">$</span>tip.label <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Skud&quot;</span>)),</span>
<span id="cb10-7"><a href="phylogenetic-reconstruction.html#cb10-7" tabindex="-1"></a>                      <span class="at">edgelabel =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-8"><a href="phylogenetic-reconstruction.html#cb10-8" tabindex="-1"></a>best_tree <span class="ot">=</span> <span class="fu">ladderize</span>(best_tree, <span class="at">right =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-9"><a href="phylogenetic-reconstruction.html#cb10-9" tabindex="-1"></a></span>
<span id="cb10-10"><a href="phylogenetic-reconstruction.html#cb10-10" tabindex="-1"></a><span class="co"># load annotations</span></span>
<span id="cb10-11"><a href="phylogenetic-reconstruction.html#cb10-11" tabindex="-1"></a>tree_annot_raw <span class="ot">=</span> <span class="fu">read.table</span>(<span class="fu">paste</span>(mainDir, <span class="st">&quot;genetrees.output.BS.annot.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>),</span>
<span id="cb10-12"><a href="phylogenetic-reconstruction.html#cb10-12" tabindex="-1"></a>                            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb10-13"><a href="phylogenetic-reconstruction.html#cb10-13" tabindex="-1"></a>                            <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-14"><a href="phylogenetic-reconstruction.html#cb10-14" tabindex="-1"></a>                            <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb10-15"><a href="phylogenetic-reconstruction.html#cb10-15" tabindex="-1"></a>tree_annot_raw <span class="ot">=</span> <span class="fu">data.frame</span>(tree_annot_raw)</span>
<span id="cb10-16"><a href="phylogenetic-reconstruction.html#cb10-16" tabindex="-1"></a>tree_annot_raw<span class="sc">$</span>Origin <span class="ot">=</span> <span class="fu">factor</span>(tree_annot_raw<span class="sc">$</span>Origin,</span>
<span id="cb10-17"><a href="phylogenetic-reconstruction.html#cb10-17" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;North-West Norway&quot;</span>, <span class="st">&quot;Eastern Norway&quot;</span>, <span class="st">&quot;South-West Norway&quot;</span>,</span>
<span id="cb10-18"><a href="phylogenetic-reconstruction.html#cb10-18" tabindex="-1"></a>                                          <span class="st">&quot;Latvia&quot;</span>, <span class="st">&quot;Lithuania&quot;</span>, <span class="st">&quot;Russia&quot;</span>, <span class="st">&quot; &quot;</span>))</span>
<span id="cb10-19"><a href="phylogenetic-reconstruction.html#cb10-19" tabindex="-1"></a></span>
<span id="cb10-20"><a href="phylogenetic-reconstruction.html#cb10-20" tabindex="-1"></a><span class="co"># plot</span></span>
<span id="cb10-21"><a href="phylogenetic-reconstruction.html#cb10-21" tabindex="-1"></a>p <span class="ot">=</span> <span class="fu">ggtree</span>(best_tree,</span>
<span id="cb10-22"><a href="phylogenetic-reconstruction.html#cb10-22" tabindex="-1"></a>           <span class="at">layout =</span> <span class="st">&quot;fan&quot;</span>,</span>
<span id="cb10-23"><a href="phylogenetic-reconstruction.html#cb10-23" tabindex="-1"></a>           <span class="at">open.angle =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb10-24"><a href="phylogenetic-reconstruction.html#cb10-24" tabindex="-1"></a>  <span class="fu">geom_tiplab2</span>(<span class="at">offset =</span> <span class="dv">3</span>,</span>
<span id="cb10-25"><a href="phylogenetic-reconstruction.html#cb10-25" tabindex="-1"></a>               <span class="at">size =</span> <span class="fl">2.75</span>) <span class="sc">+</span></span>
<span id="cb10-26"><a href="phylogenetic-reconstruction.html#cb10-26" tabindex="-1"></a>  <span class="fu">geom_nodepoint</span>(<span class="fu">aes</span>(<span class="at">fill =</span> <span class="fu">cut</span>(<span class="fu">as.numeric</span>(label), <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">70</span>, <span class="dv">90</span>, <span class="dv">100</span>))),</span>
<span id="cb10-27"><a href="phylogenetic-reconstruction.html#cb10-27" tabindex="-1"></a>                 <span class="at">shape =</span> <span class="dv">21</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb10-28"><a href="phylogenetic-reconstruction.html#cb10-28" tabindex="-1"></a>  <span class="fu">theme_tree</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.55</span>)) <span class="sc">+</span></span>
<span id="cb10-29"><a href="phylogenetic-reconstruction.html#cb10-29" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;black&quot;</span>),</span>
<span id="cb10-30"><a href="phylogenetic-reconstruction.html#cb10-30" tabindex="-1"></a>                    <span class="at">guide =</span> <span class="st">&quot;legend&quot;</span>,</span>
<span id="cb10-31"><a href="phylogenetic-reconstruction.html#cb10-31" tabindex="-1"></a>                    <span class="at">name =</span> <span class="st">&quot;Bootstrap&quot;</span>,</span>
<span id="cb10-32"><a href="phylogenetic-reconstruction.html#cb10-32" tabindex="-1"></a>                    <span class="at">breaks =</span> <span class="fu">c</span>(<span class="st">&quot;(90,100]&quot;</span>, <span class="st">&quot;(70,90]&quot;</span>, <span class="st">&quot;(0,70]&quot;</span>),</span>
<span id="cb10-33"><a href="phylogenetic-reconstruction.html#cb10-33" tabindex="-1"></a>                    <span class="at">labels =</span> <span class="fu">expression</span>(BP<span class="sc">&gt;=</span><span class="dv">90</span>,<span class="dv">70</span> <span class="sc">&lt;=</span> BP <span class="sc">*</span> <span class="st">&quot; &lt; 90&quot;</span>, BP <span class="sc">&lt;</span> <span class="dv">70</span>))</span>
<span id="cb10-34"><a href="phylogenetic-reconstruction.html#cb10-34" tabindex="-1"></a>p <span class="ot">=</span> <span class="fu">flip</span>(p, <span class="dv">250</span>, <span class="dv">317</span>)</span>
<span id="cb10-35"><a href="phylogenetic-reconstruction.html#cb10-35" tabindex="-1"></a></span>
<span id="cb10-36"><a href="phylogenetic-reconstruction.html#cb10-36" tabindex="-1"></a>p1 <span class="ot">=</span> p <span class="sc">+</span></span>
<span id="cb10-37"><a href="phylogenetic-reconstruction.html#cb10-37" tabindex="-1"></a>  <span class="fu">new_scale_fill</span>() <span class="sc">+</span></span>
<span id="cb10-38"><a href="phylogenetic-reconstruction.html#cb10-38" tabindex="-1"></a>  <span class="fu">geom_fruit</span>(<span class="at">data =</span> tree_annot_raw,</span>
<span id="cb10-39"><a href="phylogenetic-reconstruction.html#cb10-39" tabindex="-1"></a>             <span class="at">geom =</span> geom_tile,</span>
<span id="cb10-40"><a href="phylogenetic-reconstruction.html#cb10-40" tabindex="-1"></a>             <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">y =</span> Strain, <span class="at">fill =</span> Clade),</span>
<span id="cb10-41"><a href="phylogenetic-reconstruction.html#cb10-41" tabindex="-1"></a>             <span class="at">color =</span> <span class="cn">NA</span>, <span class="at">offset =</span> <span class="fl">0.04</span>, <span class="at">size =</span> <span class="fl">0.02</span>) <span class="sc">+</span></span>
<span id="cb10-42"><a href="phylogenetic-reconstruction.html#cb10-42" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&#39;#e6194b&#39;</span>, <span class="st">&#39;#3cb44b&#39;</span>, <span class="st">&#39;#ffe119&#39;</span>, <span class="st">&#39;#4363d8&#39;</span>, <span class="st">&#39;#f58231&#39;</span>,</span>
<span id="cb10-43"><a href="phylogenetic-reconstruction.html#cb10-43" tabindex="-1"></a>                               <span class="st">&#39;#911eb4&#39;</span>, <span class="st">&#39;#46f0f0&#39;</span>, <span class="st">&#39;#f032e6&#39;</span>, <span class="st">&#39;#bcf60c&#39;</span>, <span class="st">&#39;#fabebe&#39;</span>,</span>
<span id="cb10-44"><a href="phylogenetic-reconstruction.html#cb10-44" tabindex="-1"></a>                               <span class="st">&#39;#008080&#39;</span>, <span class="st">&#39;#e6beff&#39;</span>, <span class="st">&#39;#9a6324&#39;</span>, <span class="st">&#39;#fffac8&#39;</span>, <span class="st">&#39;#800000&#39;</span>,</span>
<span id="cb10-45"><a href="phylogenetic-reconstruction.html#cb10-45" tabindex="-1"></a>                               <span class="st">&#39;#aaffc3&#39;</span>, <span class="st">&#39;#808000&#39;</span>, <span class="st">&#39;#ffd8b1&#39;</span>, <span class="st">&#39;#000075&#39;</span>, <span class="st">&#39;#808080&#39;</span>,</span>
<span id="cb10-46"><a href="phylogenetic-reconstruction.html#cb10-46" tabindex="-1"></a>                               <span class="st">&#39;#ffffff&#39;</span>, <span class="st">&#39;#000000&#39;</span>)) </span>
<span id="cb10-47"><a href="phylogenetic-reconstruction.html#cb10-47" tabindex="-1"></a>p2 <span class="ot">=</span> p1 <span class="sc">+</span></span>
<span id="cb10-48"><a href="phylogenetic-reconstruction.html#cb10-48" tabindex="-1"></a>  <span class="fu">new_scale_fill</span>() <span class="sc">+</span></span>
<span id="cb10-49"><a href="phylogenetic-reconstruction.html#cb10-49" tabindex="-1"></a>  <span class="fu">geom_fruit</span>(<span class="at">data =</span> tree_annot_raw,</span>
<span id="cb10-50"><a href="phylogenetic-reconstruction.html#cb10-50" tabindex="-1"></a>             <span class="at">geom =</span> geom_tile,</span>
<span id="cb10-51"><a href="phylogenetic-reconstruction.html#cb10-51" tabindex="-1"></a>             <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">y =</span> Strain, <span class="at">fill =</span> Origin),</span>
<span id="cb10-52"><a href="phylogenetic-reconstruction.html#cb10-52" tabindex="-1"></a>             <span class="at">color =</span> <span class="cn">NA</span>, <span class="at">offset =</span> <span class="fl">0.06</span>, <span class="at">size =</span> <span class="fl">0.02</span>) <span class="sc">+</span></span>
<span id="cb10-53"><a href="phylogenetic-reconstruction.html#cb10-53" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&#39;#03045e&#39;</span>, <span class="st">&#39;#0077b6&#39;</span>, <span class="st">&#39;#00b4d8&#39;</span>, </span>
<span id="cb10-54"><a href="phylogenetic-reconstruction.html#cb10-54" tabindex="-1"></a>                               <span class="st">&#39;#808000&#39;</span>, <span class="st">&#39;#aaffc3&#39;</span>, <span class="st">&#39;#800000&#39;</span>,</span>
<span id="cb10-55"><a href="phylogenetic-reconstruction.html#cb10-55" tabindex="-1"></a>                               <span class="st">&#39;#ffffff&#39;</span>)) <span class="sc">+</span></span>
<span id="cb10-56"><a href="phylogenetic-reconstruction.html#cb10-56" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">1.1</span>, <span class="fl">0.5</span>))</span>
<span id="cb10-57"><a href="phylogenetic-reconstruction.html#cb10-57" tabindex="-1"></a></span>
<span id="cb10-58"><a href="phylogenetic-reconstruction.html#cb10-58" tabindex="-1"></a>p2</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-11"></span>
<img src="_main_files/figure-html/unnamed-chunk-11-1.png" alt="Farmhouse yeasts coalescence-based phynogenetic recontruction" width="672" />
<p class="caption">
Figure 5.3: Farmhouse yeasts coalescence-based phynogenetic recontruction
</p>
</div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="horizontal-gene-transfer.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="variant-calling.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/05-Phylogenetic-Reconstruction.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
