<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Variant Calling | Vikings Project</title>
  <meta name="description" content="Exploring yeast phenotypic diversity in traditional European farmhouse ale cultures as a source of novel industrial traits" />
  <meta name="generator" content="bookdown 0.32 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Variant Calling | Vikings Project" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Exploring yeast phenotypic diversity in traditional European farmhouse ale cultures as a source of novel industrial traits" />
  <meta name="github-repo" content="19ADC99/VikingsProject" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Variant Calling | Vikings Project" />
  
  <meta name="twitter:description" content="Exploring yeast phenotypic diversity in traditional European farmhouse ale cultures as a source of novel industrial traits" />
  

<meta name="author" content="Andrea Del Cortona" />


<meta name="date" content="2023-03-21" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="phylogenetic-reconstruction.html"/>
<link rel="next" href="farmhouse-yeasts-culturess-heterogeneity.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Vikings Project</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="1" data-path="qc-and-genome-composition-assessment.html"><a href="qc-and-genome-composition-assessment.html"><i class="fa fa-check"></i><b>1</b> QC and Genome Composition assessment</a>
<ul>
<li class="chapter" data-level="1.1" data-path="qc-and-genome-composition-assessment.html"><a href="qc-and-genome-composition-assessment.html#reads-qc-and-filtering"><i class="fa fa-check"></i><b>1.1</b> Reads QC and Filtering</a></li>
<li class="chapter" data-level="1.2" data-path="qc-and-genome-composition-assessment.html"><a href="qc-and-genome-composition-assessment.html#genome-composition"><i class="fa fa-check"></i><b>1.2</b> Genome Composition</a></li>
<li class="chapter" data-level="1.3" data-path="qc-and-genome-composition-assessment.html"><a href="qc-and-genome-composition-assessment.html#ploydies-estimation"><i class="fa fa-check"></i><b>1.3</b> Ploydies estimation</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="copy-number-variants.html"><a href="copy-number-variants.html"><i class="fa fa-check"></i><b>2</b> Copy Number Variants</a>
<ul>
<li class="chapter" data-level="2.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#call-cnvs"><i class="fa fa-check"></i><b>2.1</b> Call CNVs</a></li>
<li class="chapter" data-level="2.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#identify-common-cnvs-in-farmhouse-yeasts"><i class="fa fa-check"></i><b>2.2</b> Identify common CNVs in Farmhouse yeasts</a></li>
<li class="chapter" data-level="2.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv-functional-enrichment"><i class="fa fa-check"></i><b>2.3</b> CNV Functional Enrichment</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---i1-160000"><i class="fa fa-check"></i><b>2.3.1</b> CNV - I:1-160,000</a>
<ul>
<li class="chapter" data-level="2.3.1.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms"><i class="fa fa-check"></i><b>2.3.1.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.1.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms"><i class="fa fa-check"></i><b>2.3.1.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.1.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms"><i class="fa fa-check"></i><b>2.3.1.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.1.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways"><i class="fa fa-check"></i><b>2.3.1.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---i160501-230500"><i class="fa fa-check"></i><b>2.3.2</b> CNV - I:160,501-230,500</a>
<ul>
<li class="chapter" data-level="2.3.2.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-1"><i class="fa fa-check"></i><b>2.3.2.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.2.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-1"><i class="fa fa-check"></i><b>2.3.2.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.2.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-1"><i class="fa fa-check"></i><b>2.3.2.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.2.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-1"><i class="fa fa-check"></i><b>2.3.2.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---iii1001-317000"><i class="fa fa-check"></i><b>2.3.3</b> CNV - III:1,001-317,000</a>
<ul>
<li class="chapter" data-level="2.3.3.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-2"><i class="fa fa-check"></i><b>2.3.3.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.3.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-2"><i class="fa fa-check"></i><b>2.3.3.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.3.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-2"><i class="fa fa-check"></i><b>2.3.3.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.3.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-2"><i class="fa fa-check"></i><b>2.3.3.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---iv1-18500"><i class="fa fa-check"></i><b>2.3.4</b> CNV - IV:1-18,500</a>
<ul>
<li class="chapter" data-level="2.3.4.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-3"><i class="fa fa-check"></i><b>2.3.4.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.4.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-3"><i class="fa fa-check"></i><b>2.3.4.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.4.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-3"><i class="fa fa-check"></i><b>2.3.4.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.4.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-3"><i class="fa fa-check"></i><b>2.3.4.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.5" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---iv527001-540000"><i class="fa fa-check"></i><b>2.3.5</b> CNV - IV:527,001-540,000</a>
<ul>
<li class="chapter" data-level="2.3.5.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-4"><i class="fa fa-check"></i><b>2.3.5.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.5.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-4"><i class="fa fa-check"></i><b>2.3.5.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.5.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-4"><i class="fa fa-check"></i><b>2.3.5.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.5.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-4"><i class="fa fa-check"></i><b>2.3.5.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.6" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---vi4001-270500"><i class="fa fa-check"></i><b>2.3.6</b> CNV - VI:4,001-270,500</a>
<ul>
<li class="chapter" data-level="2.3.6.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-5"><i class="fa fa-check"></i><b>2.3.6.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.6.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-5"><i class="fa fa-check"></i><b>2.3.6.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.6.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-5"><i class="fa fa-check"></i><b>2.3.6.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.6.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-5"><i class="fa fa-check"></i><b>2.3.6.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.7" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---viii85501-556000"><i class="fa fa-check"></i><b>2.3.7</b> CNV - VIII:85,501-556,000</a>
<ul>
<li class="chapter" data-level="2.3.7.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-6"><i class="fa fa-check"></i><b>2.3.7.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.7.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-6"><i class="fa fa-check"></i><b>2.3.7.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.7.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-6"><i class="fa fa-check"></i><b>2.3.7.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.7.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-6"><i class="fa fa-check"></i><b>2.3.7.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.8" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---xi645001-667000"><i class="fa fa-check"></i><b>2.3.8</b> CNV - XI:645,001-667,000</a>
<ul>
<li class="chapter" data-level="2.3.8.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-7"><i class="fa fa-check"></i><b>2.3.8.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.8.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-7"><i class="fa fa-check"></i><b>2.3.8.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.8.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-7"><i class="fa fa-check"></i><b>2.3.8.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.8.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-7"><i class="fa fa-check"></i><b>2.3.8.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.9" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---xii451001-490000"><i class="fa fa-check"></i><b>2.3.9</b> CNV - XII:451,001-490,000</a>
<ul>
<li class="chapter" data-level="2.3.9.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-8"><i class="fa fa-check"></i><b>2.3.9.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.9.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-8"><i class="fa fa-check"></i><b>2.3.9.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.9.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-8"><i class="fa fa-check"></i><b>2.3.9.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.9.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-8"><i class="fa fa-check"></i><b>2.3.9.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.10" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---xv1-38000"><i class="fa fa-check"></i><b>2.3.10</b> CNV - XV:1-38,000</a>
<ul>
<li class="chapter" data-level="2.3.10.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-9"><i class="fa fa-check"></i><b>2.3.10.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.10.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-9"><i class="fa fa-check"></i><b>2.3.10.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.10.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-9"><i class="fa fa-check"></i><b>2.3.10.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.10.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-9"><i class="fa fa-check"></i><b>2.3.10.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.11" data-path="copy-number-variants.html"><a href="copy-number-variants.html#summary-of-enriched-or-depleted-functions"><i class="fa fa-check"></i><b>2.3.11</b> Summary of enriched or depleted functions</a>
<ul>
<li class="chapter" data-level="2.3.11.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#enriched-go-terms"><i class="fa fa-check"></i><b>2.3.11.1</b> Enriched GO terms</a></li>
<li class="chapter" data-level="2.3.11.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#enriched-kegg-pathways"><i class="fa fa-check"></i><b>2.3.11.2</b> Enriched KEGG pathways</a></li>
<li class="chapter" data-level="2.3.11.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#enriched-reactome-pathways"><i class="fa fa-check"></i><b>2.3.11.3</b> Enriched Reactome pathways</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="genome-assembly-and-annotation.html"><a href="genome-assembly-and-annotation.html"><i class="fa fa-check"></i><b>3</b> Genome Assembly and Annotation</a>
<ul>
<li class="chapter" data-level="3.1" data-path="genome-assembly-and-annotation.html"><a href="genome-assembly-and-annotation.html#genome-assembly"><i class="fa fa-check"></i><b>3.1</b> Genome Assembly</a></li>
<li class="chapter" data-level="3.2" data-path="genome-assembly-and-annotation.html"><a href="genome-assembly-and-annotation.html#genome-annotation"><i class="fa fa-check"></i><b>3.2</b> Genome Annotation</a></li>
<li class="chapter" data-level="3.3" data-path="genome-assembly-and-annotation.html"><a href="genome-assembly-and-annotation.html#identification-of-farmhouse-specific-gene-families"><i class="fa fa-check"></i><b>3.3</b> Identification of Farmhouse-specific Gene Families</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html"><i class="fa fa-check"></i><b>4</b> Horizontal Gene Transfer</a>
<ul>
<li class="chapter" data-level="4.1" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#taxonomic-annotation-of-predicted-genes"><i class="fa fa-check"></i><b>4.1</b> Taxonomic annotation of predicted genes</a></li>
<li class="chapter" data-level="4.2" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#evidence-for-bacterial-hgt"><i class="fa fa-check"></i><b>4.2</b> Evidence for bacterial HGT</a></li>
<li class="chapter" data-level="4.3" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#evidence-for-hgt-from-ascomycota"><i class="fa fa-check"></i><b>4.3</b> Evidence for HGT from Ascomycota</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#zygosaccharomyces-parabailli-transferred-operon"><i class="fa fa-check"></i><b>4.3.1</b> <em>Zygosaccharomyces parabailli</em> transferred operon</a>
<ul>
<li class="chapter" data-level="4.3.1.1" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#overrepresented-go-bp-terms"><i class="fa fa-check"></i><b>4.3.1.1</b> Overrepresented GO BP terms</a></li>
<li class="chapter" data-level="4.3.1.2" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#overrepresented-go-cc-terms"><i class="fa fa-check"></i><b>4.3.1.2</b> Overrepresented GO CC terms</a></li>
<li class="chapter" data-level="4.3.1.3" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#overrepresented-go-mf-terms"><i class="fa fa-check"></i><b>4.3.1.3</b> Overrepresented GO MF terms</a></li>
<li class="chapter" data-level="4.3.1.4" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#overrepresented-reactome-pathways"><i class="fa fa-check"></i><b>4.3.1.4</b> Overrepresented Reactome Pathways</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="phylogenetic-reconstruction.html"><a href="phylogenetic-reconstruction.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic reconstruction</a>
<ul>
<li class="chapter" data-level="5.1" data-path="phylogenetic-reconstruction.html"><a href="phylogenetic-reconstruction.html#identify-quasi-single-copy-genes"><i class="fa fa-check"></i><b>5.1</b> Identify quasi single-copy genes</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="phylogenetic-reconstruction.html"><a href="phylogenetic-reconstruction.html#create-animo-acid-and-codon-alignments"><i class="fa fa-check"></i><b>5.1.1</b> Create animo acid and codon alignments</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="phylogenetic-reconstruction.html"><a href="phylogenetic-reconstruction.html#run-phylogenetic-reconstruction"><i class="fa fa-check"></i><b>5.2</b> Run phylogenetic reconstruction</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="variant-calling.html"><a href="variant-calling.html"><i class="fa fa-check"></i><b>6</b> Variant Calling</a>
<ul>
<li class="chapter" data-level="6.1" data-path="variant-calling.html"><a href="variant-calling.html#variant-calling-1"><i class="fa fa-check"></i><b>6.1</b> Variant Calling</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="variant-calling.html"><a href="variant-calling.html#reads-alignment"><i class="fa fa-check"></i><b>6.1.1</b> Reads alignment</a></li>
<li class="chapter" data-level="6.1.2" data-path="variant-calling.html"><a href="variant-calling.html#variant-calling-2"><i class="fa fa-check"></i><b>6.1.2</b> Variant Calling</a></li>
<li class="chapter" data-level="6.1.3" data-path="variant-calling.html"><a href="variant-calling.html#variant-filtering-and-convert-to-plink-bed-input"><i class="fa fa-check"></i><b>6.1.3</b> Variant filtering and convert to plink bed input</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="variant-calling.html"><a href="variant-calling.html#heterozigous-positions"><i class="fa fa-check"></i><b>6.2</b> Heterozigous positions</a></li>
<li class="chapter" data-level="6.3" data-path="variant-calling.html"><a href="variant-calling.html#loh"><i class="fa fa-check"></i><b>6.3</b> LOH</a></li>
<li class="chapter" data-level="6.4" data-path="variant-calling.html"><a href="variant-calling.html#population-structure"><i class="fa fa-check"></i><b>6.4</b> Population Structure</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="variant-calling.html"><a href="variant-calling.html#farmhouse-yeasts-only"><i class="fa fa-check"></i><b>6.4.1</b> Farmhouse yeasts only</a></li>
<li class="chapter" data-level="6.4.2" data-path="variant-calling.html"><a href="variant-calling.html#farmhouse-yeasts-gallone-2019"><i class="fa fa-check"></i><b>6.4.2</b> Farmhouse yeasts + Gallone 2019</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="farmhouse-yeasts-culturess-heterogeneity.html"><a href="farmhouse-yeasts-culturess-heterogeneity.html"><i class="fa fa-check"></i><b>7</b> Farmhouse Yeasts Culturess Heterogeneity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="farmhouse-yeasts-culturess-heterogeneity.html"><a href="farmhouse-yeasts-culturess-heterogeneity.html#vaaa"><i class="fa fa-check"></i><b>7.1</b> Vaaa</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Vikings Project</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="variant-calling" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Variant Calling<a href="variant-calling.html#variant-calling" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>We will now analyse the DNA variants between the sequenced farmhouse yeasts and the reference <em>S. cerevisiae</em> lab strain S288C. Genetic variants will be combined with information from the 151 yeast strains sequenced in Gallone et al 2019 used to recostruct (kveiks) population structures, levels of heterozygosities, puatative regions with Loss Of Heterozygosity (LOH) and the eventaul genes affected by LOH.</p>
<div id="variant-calling-1" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Variant Calling<a href="variant-calling.html#variant-calling-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="reads-alignment" class="section level3 hasAnchor" number="6.1.1">
<h3><span class="header-section-number">6.1.1</span> Reads alignment<a href="variant-calling.html#reads-alignment" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Here the steps are the standard I always do:</p>
<ul>
<li>align reads to reference genome with bwa mem;</li>
<li>sort the bam file by read position;</li>
<li>mark reads duplicates with what used to be picard tools;</li>
<li>assign read groups, so that each sequencing library is assigned to the right sample (necessary for when I merge the vcf files)</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="variant-calling.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">## align reads to reference genome S288C</span></span>
<span id="cb1-2"><a href="variant-calling.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co"># align reads</span></span>
<span id="cb1-3"><a href="variant-calling.html#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="va">DB</span><span class="op">=</span>~/00_Brigida_starting_files/variants/Saccharomyces_cerevisiae.EF4.73.dna.chromosome.all.fa</span>
<span id="cb1-4"><a href="variant-calling.html#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="variant-calling.html#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb1-6"><a href="variant-calling.html#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="ex">bwa</span> mem <span class="at">-t</span> 72 <span class="dt">\</span></span>
<span id="cb1-7"><a href="variant-calling.html#cb1-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">-K</span> 100000000 <span class="va">$DB</span> <span class="dt">\</span></span>
<span id="cb1-8"><a href="variant-calling.html#cb1-8" aria-hidden="true" tabindex="-1"></a>    ../00_trim_reads/<span class="va">$line</span>.R1.tr.fq.gz <span class="dt">\</span></span>
<span id="cb1-9"><a href="variant-calling.html#cb1-9" aria-hidden="true" tabindex="-1"></a>    ../00_trim_reads/<span class="va">$line</span>.R2.tr.fq.gz <span class="dt">\</span></span>
<span id="cb1-10"><a href="variant-calling.html#cb1-10" aria-hidden="true" tabindex="-1"></a>    <span class="op">&gt;</span> <span class="va">$line</span>.S288C.align.sam<span class="kw">;</span></span>
<span id="cb1-11"><a href="variant-calling.html#cb1-11" aria-hidden="true" tabindex="-1"></a>  <span class="ex">samtools</span> view <span class="at">-@</span> 72 <span class="at">-Sb</span> <span class="dt">\</span></span>
<span id="cb1-12"><a href="variant-calling.html#cb1-12" aria-hidden="true" tabindex="-1"></a>    <span class="va">$line</span>.S288C.align.sam <span class="op">&gt;</span> <span class="va">$line</span>.S288C.align.bam<span class="kw">;</span></span>
<span id="cb1-13"><a href="variant-calling.html#cb1-13" aria-hidden="true" tabindex="-1"></a>  <span class="ex">samtools</span> sort <span class="at">-@</span> 72 <span class="dt">\</span></span>
<span id="cb1-14"><a href="variant-calling.html#cb1-14" aria-hidden="true" tabindex="-1"></a>    <span class="va">$line</span>.S288C.align.bam <span class="va">$line</span>.S288C.align.sort<span class="kw">;</span></span>
<span id="cb1-15"><a href="variant-calling.html#cb1-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span> <span class="va">$line</span>.S288C.align.sam <span class="va">$line</span>.S288C.align.bam<span class="kw">;</span></span>
<span id="cb1-16"><a href="variant-calling.html#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span> <span class="op">&lt;</span> ../sample.lst</span>
<span id="cb1-17"><a href="variant-calling.html#cb1-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-18"><a href="variant-calling.html#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="co">## mark duplicates</span></span>
<span id="cb1-19"><a href="variant-calling.html#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.bam<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb1-20"><a href="variant-calling.html#cb1-20" aria-hidden="true" tabindex="-1"></a>  <span class="ex">~/bin/gatk-4.0.11.0/gatk</span> MarkDuplicates <span class="dt">\</span></span>
<span id="cb1-21"><a href="variant-calling.html#cb1-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">--INPUT</span><span class="op">=</span><span class="va">$file</span> <span class="dt">\</span></span>
<span id="cb1-22"><a href="variant-calling.html#cb1-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">--OUTPUT</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .bam<span class="va">)</span>.md.bam <span class="dt">\</span></span>
<span id="cb1-23"><a href="variant-calling.html#cb1-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">--METRICS_FILE</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .bam<span class="va">)</span>.md.log <span class="dt">\</span></span>
<span id="cb1-24"><a href="variant-calling.html#cb1-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">--CREATE_INDEX</span><span class="op">=</span>true<span class="kw">;</span></span>
<span id="cb1-25"><a href="variant-calling.html#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb1-26"><a href="variant-calling.html#cb1-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-27"><a href="variant-calling.html#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="co"># assign read group</span></span>
<span id="cb1-28"><a href="variant-calling.html#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb1-29"><a href="variant-calling.html#cb1-29" aria-hidden="true" tabindex="-1"></a>  <span class="ex">~/bin/gatk-4.0.11.0/gatk</span> AddOrReplaceReadGroups <span class="dt">\</span></span>
<span id="cb1-30"><a href="variant-calling.html#cb1-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">--INPUT</span> <span class="va">$line</span>.S288C.align.sort.md.bam <span class="dt">\</span></span>
<span id="cb1-31"><a href="variant-calling.html#cb1-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">--OUTPUT</span> <span class="va">$line</span>.S288C.align.sort.md.r.bam <span class="dt">\</span></span>
<span id="cb1-32"><a href="variant-calling.html#cb1-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">--RGID</span> <span class="va">$line</span> <span class="at">--RGPL</span> <span class="va">$line</span> <span class="at">--RGLB</span> <span class="va">$line</span> <span class="dt">\</span></span>
<span id="cb1-33"><a href="variant-calling.html#cb1-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">--RGPU</span> <span class="va">$line</span> <span class="at">--RGSM</span> <span class="va">$line</span><span class="kw">;</span></span>
<span id="cb1-34"><a href="variant-calling.html#cb1-34" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span> <span class="op">&lt;</span> ../sample.lst</span></code></pre></div>
</div>
<div id="variant-calling-2" class="section level3 hasAnchor" number="6.1.2">
<h3><span class="header-section-number">6.1.2</span> Variant Calling<a href="variant-calling.html#variant-calling-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>I run HaplotypeCaller in gVCF mode, sample ploidy 2 and calling all the kveiks positions at the same time to reduce the number of false positives.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="variant-calling.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">## run haplotype caller</span></span>
<span id="cb2-2"><a href="variant-calling.html#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="va">DB</span><span class="op">=</span>~/00_Brigida_starting_files/variants/Saccharomyces_cerevisiae.EF4.73.dna.chromosome.all.fa</span>
<span id="cb2-3"><a href="variant-calling.html#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="variant-calling.html#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> chr1 chr2 chr3 chr4 chr5 chr6 chr7 chr8 chr9 chr10 chr11 chr12 chr13 chr14 chr15 chr16<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb2-5"><a href="variant-calling.html#cb2-5" aria-hidden="true" tabindex="-1"></a>  <span class="ex">java</span> <span class="at">-Xmx8g</span> <span class="dt">\</span></span>
<span id="cb2-6"><a href="variant-calling.html#cb2-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">-jar</span> ~/bin/GenomeAnalysisTK-3.3-0-g37228af/GenomeAnalysisTK.jar <span class="dt">\</span></span>
<span id="cb2-7"><a href="variant-calling.html#cb2-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">--logging_level</span> INFO <span class="at">--num_threads</span> 4 <span class="dt">\</span></span>
<span id="cb2-8"><a href="variant-calling.html#cb2-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">--analysis_type</span> UnifiedGenotyper <span class="dt">\</span></span>
<span id="cb2-9"><a href="variant-calling.html#cb2-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">--reference_sequence</span> <span class="va">$DB</span> <span class="dt">\</span></span>
<span id="cb2-10"><a href="variant-calling.html#cb2-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">-I</span> bam_files.list <span class="at">-o</span> S288C.variants.<span class="va">$i</span>.gvcf <span class="dt">\</span></span>
<span id="cb2-11"><a href="variant-calling.html#cb2-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">--intervals</span> Saccharomyces_cerevisiae.EF4.73.dna.chromosome.<span class="va">$i</span>.interval_list <span class="dt">\</span></span>
<span id="cb2-12"><a href="variant-calling.html#cb2-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">--standard_min_confidence_threshold_for_calling</span> 20 <span class="dt">\</span></span>
<span id="cb2-13"><a href="variant-calling.html#cb2-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">-stand_emit_conf</span> 30.0 <span class="at">--min_base_quality_score</span> 20 <span class="dt">\</span></span>
<span id="cb2-14"><a href="variant-calling.html#cb2-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">-glm</span> BOTH <span class="at">--sample_ploidy</span> 2 <span class="dt">\</span></span>
<span id="cb2-15"><a href="variant-calling.html#cb2-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">--validation_strictness</span> SILENT <span class="at">--output_mode</span> EMIT_ALL_SITES <span class="kw">&amp;</span></span>
<span id="cb2-16"><a href="variant-calling.html#cb2-16" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb2-17"><a href="variant-calling.html#cb2-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-18"><a href="variant-calling.html#cb2-18" aria-hidden="true" tabindex="-1"></a><span class="co"># merge kveiks with with brigida&#39;s chromosomes</span></span>
<span id="cb2-19"><a href="variant-calling.html#cb2-19" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> chr1 chr2 chr3 chr4 chr5 chr6 chr7 chr8 chr9 chr10 chr11 chr12 chr13 chr14 chr15 chr16<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb2-20"><a href="variant-calling.html#cb2-20" aria-hidden="true" tabindex="-1"></a>  <span class="ex">bcftools</span> merge <span class="at">--threads</span> 4 <span class="dt">\</span></span>
<span id="cb2-21"><a href="variant-calling.html#cb2-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">--output-type</span> z <span class="at">--output</span> all_plusKV.<span class="va">$i</span>.vcf.gz <span class="dt">\</span></span>
<span id="cb2-22"><a href="variant-calling.html#cb2-22" aria-hidden="true" tabindex="-1"></a>    kveiks.S288C.variants.<span class="va">$i</span>.gvcf.gz ../03_Bri_chr/Scer.<span class="va">$i</span>.raw.vcf.gz <span class="kw">&amp;</span></span>
<span id="cb2-23"><a href="variant-calling.html#cb2-23" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb2-24"><a href="variant-calling.html#cb2-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-25"><a href="variant-calling.html#cb2-25" aria-hidden="true" tabindex="-1"></a><span class="co"># merge all chromosomes together</span></span>
<span id="cb2-26"><a href="variant-calling.html#cb2-26" aria-hidden="true" tabindex="-1"></a><span class="ex">bcftools</span> concat all_plusKV.chr1.vcf.gz all_plusKV.chr2.vcf.gz <span class="dt">\</span></span>
<span id="cb2-27"><a href="variant-calling.html#cb2-27" aria-hidden="true" tabindex="-1"></a>  all_plusKV.chr3.vcf.gz all_plusKV.chr4.vcf.gz all_plusKV.chr5.vcf.gz <span class="dt">\</span></span>
<span id="cb2-28"><a href="variant-calling.html#cb2-28" aria-hidden="true" tabindex="-1"></a>  all_plusKV.chr6.vcf.gz all_plusKV.chr7.vcf.gz all_plusKV.chr8.vcf.gz <span class="dt">\</span></span>
<span id="cb2-29"><a href="variant-calling.html#cb2-29" aria-hidden="true" tabindex="-1"></a>  all_plusKV.chr9.vcf.gz all_plusKV.chr10.vcf.gz all_plusKV.chr11.vcf.gz <span class="dt">\</span></span>
<span id="cb2-30"><a href="variant-calling.html#cb2-30" aria-hidden="true" tabindex="-1"></a>  all_plusKV.chr12.vcf.gz all_plusKV.chr13.vcf.gz all_plusKV.chr14.vcf.gz <span class="dt">\</span></span>
<span id="cb2-31"><a href="variant-calling.html#cb2-31" aria-hidden="true" tabindex="-1"></a>  all_plusKV.chr15.vcf.gz all_plusKV.chr16.vcf.gz <span class="dt">\</span></span>
<span id="cb2-32"><a href="variant-calling.html#cb2-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">--output-type</span> z <span class="at">--output</span> all_plusKV.vcf.gz <span class="at">--threads</span> 64</span></code></pre></div>
</div>
<div id="variant-filtering-and-convert-to-plink-bed-input" class="section level3 hasAnchor" number="6.1.3">
<h3><span class="header-section-number">6.1.3</span> Variant filtering and convert to plink bed input<a href="variant-calling.html#variant-filtering-and-convert-to-plink-bed-input" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Here I filter for SNPs that are informative for Population Structure analysis
What I do:</p>
<ul>
<li>apply hard filters to flag low quality positions;</li>
<li>select biallelic and high frequency SNPs, SNPs in linkage disequilibrium.</li>
</ul>
<p>Here I select biallelic SNPs and filter out minor allele frequency. Then I merge my vcf file with the
vcf files from your dataset. After, I just invoke plink for generate the input files for FastStructure</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="variant-calling.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># hard filter low quality variants</span></span>
<span id="cb3-2"><a href="variant-calling.html#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="ex">java</span> <span class="at">-jar</span> ~/bin/GenomeAnalysisTK-3.3-0-g37228af/GenomeAnalysisTK.jar <span class="dt">\</span></span>
<span id="cb3-3"><a href="variant-calling.html#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--analysis_type</span> VariantFiltration <span class="dt">\</span></span>
<span id="cb3-4"><a href="variant-calling.html#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--variant</span> all_plusKV.vcf.gz <span class="at">--out</span> all_plusKV.fltr.vcf <span class="dt">\</span></span>
<span id="cb3-5"><a href="variant-calling.html#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--reference_sequence</span> <span class="va">$DB</span> <span class="dt">\</span></span>
<span id="cb3-6"><a href="variant-calling.html#cb3-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">--filterExpression</span> <span class="st">&quot;QD &lt; 2.0&quot;</span> <span class="at">--filterName</span> <span class="st">&quot;QD_Low&quot;</span> <span class="dt">\</span></span>
<span id="cb3-7"><a href="variant-calling.html#cb3-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">--filterExpression</span> <span class="st">&quot;MQ &lt; 40.0&quot;</span> <span class="at">--filterName</span> <span class="st">&quot;MQ_Low&quot;</span> <span class="dt">\</span></span>
<span id="cb3-8"><a href="variant-calling.html#cb3-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">--filterExpression</span> <span class="st">&quot;GQ &lt; 30&quot;</span> <span class="at">--filterName</span> <span class="st">&quot;GQ_Low&quot;</span> <span class="dt">\</span></span>
<span id="cb3-9"><a href="variant-calling.html#cb3-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">--filterExpression</span> <span class="st">&quot;AD &lt; 5&quot;</span> <span class="at">--filterName</span> <span class="st">&quot;AD_Low&quot;</span></span>
<span id="cb3-10"><a href="variant-calling.html#cb3-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-11"><a href="variant-calling.html#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="co"># select biallelic loci, filter for MAF and linkage disequilibrium</span></span>
<span id="cb3-12"><a href="variant-calling.html#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="ex">~/bin/plink_linux_x86_64_20181202/plink</span> <span class="dt">\</span></span>
<span id="cb3-13"><a href="variant-calling.html#cb3-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">--make-bed</span> <span class="at">--double-id</span> <span class="at">--allow-extra-chr</span> <span class="dt">\</span></span>
<span id="cb3-14"><a href="variant-calling.html#cb3-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">--indep-pairwise</span> 50 5 0.5 <span class="at">--maf</span> 0.05 <span class="dt">\</span></span>
<span id="cb3-15"><a href="variant-calling.html#cb3-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">--biallelic-only</span> <span class="at">--geno</span> 0.25 <span class="dt">\</span></span>
<span id="cb3-16"><a href="variant-calling.html#cb3-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">--vcf</span> all_plusKV.fltr.vcf.gz <span class="at">--out</span> all_plusKV.fltr.bi</span></code></pre></div>
</div>
</div>
<div id="heterozigous-positions" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Heterozigous positions<a href="variant-calling.html#heterozigous-positions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Farmhouse yeasts contains a high degree of Heterozygosity, comparable to yeast strains from the “mixed” Clade.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="variant-calling.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># import dataset</span></span>
<span id="cb4-2"><a href="variant-calling.html#cb4-2" aria-hidden="true" tabindex="-1"></a>kveik <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="fu">paste</span>(mainDir, <span class="st">&quot;all_plusKV.heterozygous.positions.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-3"><a href="variant-calling.html#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="variant-calling.html#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co"># sort </span></span>
<span id="cb4-5"><a href="variant-calling.html#cb4-5" aria-hidden="true" tabindex="-1"></a>kveik<span class="sc">$</span>Pop <span class="ot">=</span> <span class="fu">factor</span>(</span>
<span id="cb4-6"><a href="variant-calling.html#cb4-6" aria-hidden="true" tabindex="-1"></a>  kveik<span class="sc">$</span>Pop,</span>
<span id="cb4-7"><a href="variant-calling.html#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;Farmhouse&quot;</span>, <span class="st">&quot;Asian&quot;</span>, <span class="st">&quot;Beer1&quot;</span>, <span class="st">&quot;Beer2&quot;</span>, <span class="st">&quot;Bread&quot;</span>,</span>
<span id="cb4-8"><a href="variant-calling.html#cb4-8" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;Mixed&quot;</span>, <span class="st">&quot;Mosaic&quot;</span>, <span class="st">&quot;Sake&quot;</span>, <span class="st">&quot;Wild&quot;</span>, <span class="st">&quot;Wine&quot;</span>)</span>
<span id="cb4-9"><a href="variant-calling.html#cb4-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-10"><a href="variant-calling.html#cb4-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-11"><a href="variant-calling.html#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="co"># prepare count labels</span></span>
<span id="cb4-12"><a href="variant-calling.html#cb4-12" aria-hidden="true" tabindex="-1"></a>stat_box_data <span class="ot">&lt;-</span> <span class="cf">function</span>(y) {</span>
<span id="cb4-13"><a href="variant-calling.html#cb4-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>( </span>
<span id="cb4-14"><a href="variant-calling.html#cb4-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(</span>
<span id="cb4-15"><a href="variant-calling.html#cb4-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="dv">6</span>,</span>
<span id="cb4-16"><a href="variant-calling.html#cb4-16" aria-hidden="true" tabindex="-1"></a>      <span class="co">#sign =tests1[which(tests1$group2 == y),]$p.signif,</span></span>
<span id="cb4-17"><a href="variant-calling.html#cb4-17" aria-hidden="true" tabindex="-1"></a>      <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">&#39;n =&#39;</span>, <span class="fu">length</span>(y))</span>
<span id="cb4-18"><a href="variant-calling.html#cb4-18" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb4-19"><a href="variant-calling.html#cb4-19" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb4-20"><a href="variant-calling.html#cb4-20" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-21"><a href="variant-calling.html#cb4-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-22"><a href="variant-calling.html#cb4-22" aria-hidden="true" tabindex="-1"></a><span class="co"># plot</span></span>
<span id="cb4-23"><a href="variant-calling.html#cb4-23" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(kveik, <span class="fu">aes</span>(<span class="at">x =</span> Pop, <span class="at">y =</span> HET_CT)) <span class="sc">+</span></span>
<span id="cb4-24"><a href="variant-calling.html#cb4-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.25</span>),</span>
<span id="cb4-25"><a href="variant-calling.html#cb4-25" aria-hidden="true" tabindex="-1"></a>             <span class="at">shape =</span> <span class="dv">21</span>, <span class="at">fill =</span> <span class="st">&quot;grey50&quot;</span>, </span>
<span id="cb4-26"><a href="variant-calling.html#cb4-26" aria-hidden="true" tabindex="-1"></a>             <span class="at">alpha =</span> <span class="fl">0.25</span>) <span class="sc">+</span></span>
<span id="cb4-27"><a href="variant-calling.html#cb4-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb4-28"><a href="variant-calling.html#cb4-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb4-29"><a href="variant-calling.html#cb4-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_summary</span>(<span class="at">fun.data =</span> stat_box_data, <span class="at">geom =</span> <span class="st">&quot;text&quot;</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">vjust =</span> <span class="fl">0.9</span>) <span class="sc">+</span></span>
<span id="cb4-30"><a href="variant-calling.html#cb4-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">method =</span> <span class="st">&quot;anova&quot;</span>, <span class="at">label.y =</span> <span class="fl">0.01</span>) <span class="sc">+</span></span>
<span id="cb4-31"><a href="variant-calling.html#cb4-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">label =</span> <span class="st">&quot;p.signif&quot;</span>, <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>, <span class="at">ref.group =</span> <span class="st">&quot;Farmhouse&quot;</span>) <span class="sc">+</span></span>
<span id="cb4-32"><a href="variant-calling.html#cb4-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="at">labels =</span> scales<span class="sc">::</span>comma, <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1300000</span>)) <span class="sc">+</span></span>
<span id="cb4-33"><a href="variant-calling.html#cb4-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Heterozygosity in yeast populations&quot;</span>,</span>
<span id="cb4-34"><a href="variant-calling.html#cb4-34" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;Populations&quot;</span>,</span>
<span id="cb4-35"><a href="variant-calling.html#cb4-35" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;log10 heterozygous positions&quot;</span>) <span class="sc">+</span></span>
<span id="cb4-36"><a href="variant-calling.html#cb4-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">24</span>),</span>
<span id="cb4-37"><a href="variant-calling.html#cb4-37" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.95</span>, <span class="at">angle =</span> <span class="dv">30</span>, <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb4-38"><a href="variant-calling.html#cb4-38" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb4-39"><a href="variant-calling.html#cb4-39" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb4-40"><a href="variant-calling.html#cb4-40" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb4-41"><a href="variant-calling.html#cb4-41" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="at">fill =</span> <span class="cn">NA</span>),</span>
<span id="cb4-42"><a href="variant-calling.html#cb4-42" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-43"><a href="variant-calling.html#cb4-43" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-44"><a href="variant-calling.html#cb4-44" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major.y =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">&quot;grey75&quot;</span>),</span>
<span id="cb4-45"><a href="variant-calling.html#cb4-45" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor.y =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-5-1"></span>
<img src="_main_files/figure-html/unnamed-chunk-5-1.png" alt="Heterozygosity of yeasts from different clades" width="672" />
<p class="caption">
Figure 6.1: Heterozygosity of yeasts from different clades
</p>
</div>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="variant-calling.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot</span></span>
<span id="cb5-2"><a href="variant-calling.html#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(kveik, <span class="fu">aes</span>(<span class="at">x =</span> Pop, <span class="at">y =</span> HET_CT)) <span class="sc">+</span></span>
<span id="cb5-3"><a href="variant-calling.html#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="st">&quot;-&quot;</span>, <span class="at">size =</span> <span class="fl">3.5</span>) <span class="sc">+</span></span>
<span id="cb5-4"><a href="variant-calling.html#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">fill =</span> <span class="st">&quot;grey95&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.75</span>, <span class="at">width =</span> <span class="fl">0.25</span>,</span>
<span id="cb5-5"><a href="variant-calling.html#cb5-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">position =</span> <span class="fu">position_nudge</span>(<span class="at">x =</span> <span class="sc">-</span><span class="fl">0.2</span>),</span>
<span id="cb5-6"><a href="variant-calling.html#cb5-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">outlier.colour =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb5-7"><a href="variant-calling.html#cb5-7" aria-hidden="true" tabindex="-1"></a>  ggdist<span class="sc">::</span><span class="fu">stat_halfeye</span>(<span class="at">fill =</span> <span class="st">&quot;grey95&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.75</span>, <span class="at">slab_color =</span> <span class="st">&quot;grey45&quot;</span>, <span class="at">slab_size =</span> <span class="fl">0.25</span>, </span>
<span id="cb5-8"><a href="variant-calling.html#cb5-8" aria-hidden="true" tabindex="-1"></a>                       <span class="at">adjust =</span> <span class="dv">5</span>, <span class="at">justification =</span> <span class="sc">-</span><span class="fl">0.05</span>, <span class="at">.width =</span> <span class="dv">0</span>, <span class="at">point_colour =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb5-9"><a href="variant-calling.html#cb5-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb5-10"><a href="variant-calling.html#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_summary</span>(<span class="at">fun.data =</span> stat_box_data, <span class="at">geom =</span> <span class="st">&quot;text&quot;</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">vjust =</span> <span class="fl">0.9</span>) <span class="sc">+</span></span>
<span id="cb5-11"><a href="variant-calling.html#cb5-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">method =</span> <span class="st">&quot;anova&quot;</span>, <span class="at">label.y =</span> <span class="fl">0.01</span>) <span class="sc">+</span></span>
<span id="cb5-12"><a href="variant-calling.html#cb5-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">label =</span> <span class="st">&quot;p.signif&quot;</span>, <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>, <span class="at">ref.group =</span> <span class="st">&quot;Farmhouse&quot;</span>) <span class="sc">+</span></span>
<span id="cb5-13"><a href="variant-calling.html#cb5-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="at">labels =</span> scales<span class="sc">::</span>comma, <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1300000</span>)) <span class="sc">+</span></span>
<span id="cb5-14"><a href="variant-calling.html#cb5-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Heterozygosity in yeast populations&quot;</span>,</span>
<span id="cb5-15"><a href="variant-calling.html#cb5-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;Populations&quot;</span>,</span>
<span id="cb5-16"><a href="variant-calling.html#cb5-16" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;log10 heterozygous positions&quot;</span>) <span class="sc">+</span></span>
<span id="cb5-17"><a href="variant-calling.html#cb5-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">24</span>),</span>
<span id="cb5-18"><a href="variant-calling.html#cb5-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.95</span>, <span class="at">angle =</span> <span class="dv">30</span>, <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb5-19"><a href="variant-calling.html#cb5-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb5-20"><a href="variant-calling.html#cb5-20" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb5-21"><a href="variant-calling.html#cb5-21" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb5-22"><a href="variant-calling.html#cb5-22" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="at">fill =</span> <span class="cn">NA</span>),</span>
<span id="cb5-23"><a href="variant-calling.html#cb5-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb5-24"><a href="variant-calling.html#cb5-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb5-25"><a href="variant-calling.html#cb5-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major.y =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">&quot;grey75&quot;</span>),</span>
<span id="cb5-26"><a href="variant-calling.html#cb5-26" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor.y =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-5-2"></span>
<img src="_main_files/figure-html/unnamed-chunk-5-2.png" alt="Heterozygosity of yeasts from different clades" width="672" />
<p class="caption">
Figure 6.2: Heterozygosity of yeasts from different clades
</p>
</div>
</div>
<div id="loh" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> LOH<a href="variant-calling.html#loh" class="anchor-section" aria-label="Anchor link to header"></a></h2>
</div>
<div id="population-structure" class="section level2 hasAnchor" number="6.4">
<h2><span class="header-section-number">6.4</span> Population Structure<a href="variant-calling.html#population-structure" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Here I simply run FastStructure for a range of k, and then identify the best kmer. I have run FastStructure as well giving logarithmic priors, but the results where comparable</p>
<div id="farmhouse-yeasts-only" class="section level3 hasAnchor" number="6.4.1">
<h3><span class="header-section-number">6.4.1</span> Farmhouse yeasts only<a href="variant-calling.html#farmhouse-yeasts-only" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="variant-calling.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># merge all chromosomes together</span></span>
<span id="cb6-2"><a href="variant-calling.html#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="ex">~/bin/bcftools-1.9/bcftools</span> concat <span class="dt">\</span></span>
<span id="cb6-3"><a href="variant-calling.html#cb6-3" aria-hidden="true" tabindex="-1"></a>  kveiks.S288C.variants.chr1.gvcf.gz kveiks.S288C.variants.chr2.gvcf.gz <span class="dt">\</span></span>
<span id="cb6-4"><a href="variant-calling.html#cb6-4" aria-hidden="true" tabindex="-1"></a>  kveiks.S288C.variants.chr3.gvcf.gz kveiks.S288C.variants.chr4.gvcf.gz <span class="dt">\</span></span>
<span id="cb6-5"><a href="variant-calling.html#cb6-5" aria-hidden="true" tabindex="-1"></a>  kveiks.S288C.variants.chr5.gvcf.gz kveiks.S288C.variants.chr6.gvcf.gz <span class="dt">\</span></span>
<span id="cb6-6"><a href="variant-calling.html#cb6-6" aria-hidden="true" tabindex="-1"></a>  kveiks.S288C.variants.chr7.gvcf.gz kveiks.S288C.variants.chr8.gvcf.gz <span class="dt">\</span></span>
<span id="cb6-7"><a href="variant-calling.html#cb6-7" aria-hidden="true" tabindex="-1"></a>  kveiks.S288C.variants.chr9.gvcf.gz kveiks.S288C.variants.chr10.gvcf.gz <span class="dt">\</span></span>
<span id="cb6-8"><a href="variant-calling.html#cb6-8" aria-hidden="true" tabindex="-1"></a>  kveiks.S288C.variants.chr11.gvcf.gz kveiks.S288C.variants.chr12.gvcf.gz <span class="dt">\</span></span>
<span id="cb6-9"><a href="variant-calling.html#cb6-9" aria-hidden="true" tabindex="-1"></a>  kveiks.S288C.variants.chr13.gvcf.gz kveiks.S288C.variants.chr14.gvcf.gz <span class="dt">\</span></span>
<span id="cb6-10"><a href="variant-calling.html#cb6-10" aria-hidden="true" tabindex="-1"></a>  kveiks.S288C.variants.chr15.gvcf.gz kveiks.S288C.variants.chr16.gvcf.gz <span class="dt">\</span></span>
<span id="cb6-11"><a href="variant-calling.html#cb6-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">--output-type</span> z <span class="at">--output</span> kveiks.S288C.variants.vcf.gz <span class="at">--threads</span> 16</span>
<span id="cb6-12"><a href="variant-calling.html#cb6-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-13"><a href="variant-calling.html#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="co"># hard filter low quality variants</span></span>
<span id="cb6-14"><a href="variant-calling.html#cb6-14" aria-hidden="true" tabindex="-1"></a><span class="ex">java</span> <span class="at">-jar</span> ~/bin/GenomeAnalysisTK-3.3-0-g37228af/GenomeAnalysisTK.jar <span class="dt">\</span></span>
<span id="cb6-15"><a href="variant-calling.html#cb6-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">--analysis_type</span> VariantFiltration <span class="dt">\</span></span>
<span id="cb6-16"><a href="variant-calling.html#cb6-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">--variant</span> kveiks.S288C.variants.vcf.gz <span class="dt">\</span></span>
<span id="cb6-17"><a href="variant-calling.html#cb6-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">--out</span> kveiks.S288C.variants.fltr.vcf <span class="dt">\</span></span>
<span id="cb6-18"><a href="variant-calling.html#cb6-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">--reference_sequence</span> <span class="va">$DB</span> <span class="dt">\</span></span>
<span id="cb6-19"><a href="variant-calling.html#cb6-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">--filterExpression</span> <span class="st">&quot;QD &lt; 2.0&quot;</span> <span class="at">--filterName</span> <span class="st">&quot;QD_Low&quot;</span> <span class="dt">\</span></span>
<span id="cb6-20"><a href="variant-calling.html#cb6-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">--filterExpression</span> <span class="st">&quot;MQ &lt; 40.0&quot;</span> <span class="at">--filterName</span> <span class="st">&quot;MQ_Low&quot;</span> <span class="dt">\</span></span>
<span id="cb6-21"><a href="variant-calling.html#cb6-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">--filterExpression</span> <span class="st">&quot;GQ &lt; 30&quot;</span> <span class="at">--filterName</span> <span class="st">&quot;GQ_Low&quot;</span> <span class="dt">\</span></span>
<span id="cb6-22"><a href="variant-calling.html#cb6-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">--filterExpression</span> <span class="st">&quot;AD &lt; 5&quot;</span> <span class="at">--filterName</span> <span class="st">&quot;AD_Low&quot;</span></span>
<span id="cb6-23"><a href="variant-calling.html#cb6-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-24"><a href="variant-calling.html#cb6-24" aria-hidden="true" tabindex="-1"></a><span class="co"># select biallelic loci, filter for MAF and linkage disequilibrium</span></span>
<span id="cb6-25"><a href="variant-calling.html#cb6-25" aria-hidden="true" tabindex="-1"></a><span class="ex">~/bin/plink_linux_x86_64_20181202/plink</span> <span class="dt">\</span></span>
<span id="cb6-26"><a href="variant-calling.html#cb6-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">--make-bed</span> <span class="at">--double-id</span> <span class="at">--allow-extra-chr</span> <span class="dt">\</span></span>
<span id="cb6-27"><a href="variant-calling.html#cb6-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">--indep-pairwise</span> 50 5 0.5 <span class="at">--maf</span> 0.05 <span class="dt">\</span></span>
<span id="cb6-28"><a href="variant-calling.html#cb6-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">--biallelic-only</span> <span class="at">--geno</span> 0.25 <span class="dt">\</span></span>
<span id="cb6-29"><a href="variant-calling.html#cb6-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">--vcf</span> kveiks.S288C.variants.fltr.vcf.gz <span class="dt">\</span></span>
<span id="cb6-30"><a href="variant-calling.html#cb6-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">--out</span> kveiks.S288C.variants.fltr.bi</span>
<span id="cb6-31"><a href="variant-calling.html#cb6-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-32"><a href="variant-calling.html#cb6-32" aria-hidden="true" tabindex="-1"></a><span class="co"># run fastStructure simple</span></span>
<span id="cb6-33"><a href="variant-calling.html#cb6-33" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> k <span class="kw">in</span> 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb6-34"><a href="variant-calling.html#cb6-34" aria-hidden="true" tabindex="-1"></a>  <span class="ex">python2.7</span> ~/bin/faststructure/fastStructure-1.0/structure.py <span class="dt">\</span></span>
<span id="cb6-35"><a href="variant-calling.html#cb6-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">-K</span> <span class="st">&quot;</span><span class="va">${k}</span><span class="st">&quot;</span> <span class="at">--input</span><span class="op">=</span>kveiks.S288C.variants.fltr.bi <span class="dt">\</span></span>
<span id="cb6-36"><a href="variant-calling.html#cb6-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">--output</span><span class="op">=</span>kveiks.S288C.variants.fltr.bi.structure <span class="dt">\</span></span>
<span id="cb6-37"><a href="variant-calling.html#cb6-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">--prior</span><span class="op">=</span>simple <span class="kw">&amp;</span></span>
<span id="cb6-38"><a href="variant-calling.html#cb6-38" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb6-39"><a href="variant-calling.html#cb6-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-40"><a href="variant-calling.html#cb6-40" aria-hidden="true" tabindex="-1"></a><span class="co"># choose the best K-mer</span></span>
<span id="cb6-41"><a href="variant-calling.html#cb6-41" aria-hidden="true" tabindex="-1"></a><span class="ex">python2.7</span> ~/bin/faststructure/fastStructure-1.0/chooseK.py <span class="dt">\</span></span>
<span id="cb6-42"><a href="variant-calling.html#cb6-42" aria-hidden="true" tabindex="-1"></a>  <span class="at">--input</span><span class="op">=</span>all_plusKV.fltr.bi.structure</span></code></pre></div>
<p>==&gt; Model complexity that maximizes marginal likelihood = 4
==&gt; Model components used to explain structure in data = 5</p>
</div>
<div id="farmhouse-yeasts-gallone-2019" class="section level3 hasAnchor" number="6.4.2">
<h3><span class="header-section-number">6.4.2</span> Farmhouse yeasts + Gallone 2019<a href="variant-calling.html#farmhouse-yeasts-gallone-2019" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="variant-calling.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run fastStructure simple</span></span>
<span id="cb7-2"><a href="variant-calling.html#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> k <span class="kw">in</span> 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb7-3"><a href="variant-calling.html#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="ex">python2.7</span> ~/bin/faststructure/fastStructure-1.0/structure.py <span class="dt">\</span></span>
<span id="cb7-4"><a href="variant-calling.html#cb7-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">-K</span> <span class="st">&quot;</span><span class="va">${k}</span><span class="st">&quot;</span> <span class="at">--input</span><span class="op">=</span>all_plusKV.fltr.bi <span class="dt">\</span></span>
<span id="cb7-5"><a href="variant-calling.html#cb7-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">--output</span><span class="op">=</span>all_plusKV.fltr.bi.structure <span class="dt">\</span></span>
<span id="cb7-6"><a href="variant-calling.html#cb7-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">--prior</span><span class="op">=</span>simple <span class="kw">&amp;</span></span>
<span id="cb7-7"><a href="variant-calling.html#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb7-8"><a href="variant-calling.html#cb7-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-9"><a href="variant-calling.html#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co"># choose the best K-mer</span></span>
<span id="cb7-10"><a href="variant-calling.html#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="ex">python2.7</span> ~/bin/faststructure/fastStructure-1.0/chooseK.py <span class="dt">\</span></span>
<span id="cb7-11"><a href="variant-calling.html#cb7-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">--input</span><span class="op">=</span>all_plusKV.fltr.bi.structure</span></code></pre></div>
<p>==&gt; Model complexity that maximizes marginal likelihood = 20
==&gt; Model components used to explain structure in data = 15</p>
<p><strong>CODE BELOW TO BE DOUBLECHECKED</strong></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="variant-calling.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># print distruct plots</span></span>
<span id="cb8-2"><a href="variant-calling.html#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> k <span class="kw">in</span> 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20<span class="kw">;</span> <span class="cf">do</span> <span class="ex">python2.7</span></span>
<span id="cb8-3"><a href="variant-calling.html#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="ex">~/bin/faststructure/fastStructure-1.0/distruct.py</span></span>
<span id="cb8-4"><a href="variant-calling.html#cb8-4" aria-hidden="true" tabindex="-1"></a> <span class="ex">-K</span> <span class="st">&quot;</span><span class="va">${k}</span><span class="st">&quot;</span> <span class="at">--</span></span>
<span id="cb8-5"><a href="variant-calling.html#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="va">input</span><span class="op">=</span>all_plusKV.fltr.bi.structure <span class="ex">--</span></span>
<span id="cb8-6"><a href="variant-calling.html#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="va">output</span><span class="op">=</span>all_plusKV.fltr.bi.structure.<span class="st">&quot;</span><span class="va">${k}</span><span class="st">&quot;</span>.svg <span class="ex">--popfile=all_plusKV.poplabels</span><span class="kw">;</span></span>
<span id="cb8-7"><a href="variant-calling.html#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb8-8"><a href="variant-calling.html#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="ex">The</span> population labels file all_plusKV.poplabels looks like:</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="phylogenetic-reconstruction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="farmhouse-yeasts-culturess-heterogeneity.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/06-Variant-Calling-and-Population-Structure.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
