<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Horizontal Gene Transfer | Vikings Project</title>
  <meta name="description" content="Exploring yeast phenotypic diversity in traditional European farmhouse ale cultures as a source of novel industrial traits" />
  <meta name="generator" content="bookdown 0.32 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Horizontal Gene Transfer | Vikings Project" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Exploring yeast phenotypic diversity in traditional European farmhouse ale cultures as a source of novel industrial traits" />
  <meta name="github-repo" content="19ADC99/VikingsProject" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Horizontal Gene Transfer | Vikings Project" />
  
  <meta name="twitter:description" content="Exploring yeast phenotypic diversity in traditional European farmhouse ale cultures as a source of novel industrial traits" />
  

<meta name="author" content="Andrea Del Cortona" />


<meta name="date" content="2023-03-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="genome-assembly-and-annotation.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.24/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-fixedcolumns-1.11.3/css/fixedColumns.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-fixedcolumns-1.11.3/js/dataTables.fixedColumns.min.js"></script>
<link href="libs/dt-ext-fixedheader-1.11.3/css/fixedHeader.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-fixedheader-1.11.3/js/dataTables.fixedHeader.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Vikings Project</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="1" data-path="qc-and-genome-composition-assessment.html"><a href="qc-and-genome-composition-assessment.html"><i class="fa fa-check"></i><b>1</b> QC and Genome Composition assessment</a>
<ul>
<li class="chapter" data-level="1.1" data-path="qc-and-genome-composition-assessment.html"><a href="qc-and-genome-composition-assessment.html#reads-qc-and-filtering"><i class="fa fa-check"></i><b>1.1</b> Reads QC and Filtering</a></li>
<li class="chapter" data-level="1.2" data-path="qc-and-genome-composition-assessment.html"><a href="qc-and-genome-composition-assessment.html#genome-composition"><i class="fa fa-check"></i><b>1.2</b> Genome Composition</a></li>
<li class="chapter" data-level="1.3" data-path="qc-and-genome-composition-assessment.html"><a href="qc-and-genome-composition-assessment.html#ploydies-estimation"><i class="fa fa-check"></i><b>1.3</b> Ploydies estimation</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="copy-number-variants.html"><a href="copy-number-variants.html"><i class="fa fa-check"></i><b>2</b> Copy Number Variants</a>
<ul>
<li class="chapter" data-level="2.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#call-cnvs"><i class="fa fa-check"></i><b>2.1</b> Call CNVs</a></li>
<li class="chapter" data-level="2.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#identify-common-cnvs-in-farmhouse-yeasts"><i class="fa fa-check"></i><b>2.2</b> Identify common CNVs in Farmhouse yeasts</a></li>
<li class="chapter" data-level="2.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv-functional-enrichment"><i class="fa fa-check"></i><b>2.3</b> CNV Functional Enrichment</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---i1-160000"><i class="fa fa-check"></i><b>2.3.1</b> CNV - I:1-160,000</a>
<ul>
<li class="chapter" data-level="2.3.1.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms"><i class="fa fa-check"></i><b>2.3.1.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.1.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms"><i class="fa fa-check"></i><b>2.3.1.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.1.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms"><i class="fa fa-check"></i><b>2.3.1.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.1.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways"><i class="fa fa-check"></i><b>2.3.1.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---i160501-230500"><i class="fa fa-check"></i><b>2.3.2</b> CNV - I:160,501-230,500</a>
<ul>
<li class="chapter" data-level="2.3.2.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-1"><i class="fa fa-check"></i><b>2.3.2.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.2.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-1"><i class="fa fa-check"></i><b>2.3.2.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.2.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-1"><i class="fa fa-check"></i><b>2.3.2.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.2.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-1"><i class="fa fa-check"></i><b>2.3.2.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---iii1001-317000"><i class="fa fa-check"></i><b>2.3.3</b> CNV - III:1,001-317,000</a>
<ul>
<li class="chapter" data-level="2.3.3.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-2"><i class="fa fa-check"></i><b>2.3.3.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.3.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-2"><i class="fa fa-check"></i><b>2.3.3.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.3.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-2"><i class="fa fa-check"></i><b>2.3.3.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.3.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-2"><i class="fa fa-check"></i><b>2.3.3.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---iv1-18500"><i class="fa fa-check"></i><b>2.3.4</b> CNV - IV:1-18,500</a>
<ul>
<li class="chapter" data-level="2.3.4.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-3"><i class="fa fa-check"></i><b>2.3.4.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.4.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-3"><i class="fa fa-check"></i><b>2.3.4.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.4.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-3"><i class="fa fa-check"></i><b>2.3.4.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.4.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-3"><i class="fa fa-check"></i><b>2.3.4.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.5" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---iv527001-540000"><i class="fa fa-check"></i><b>2.3.5</b> CNV - IV:527,001-540,000</a>
<ul>
<li class="chapter" data-level="2.3.5.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-4"><i class="fa fa-check"></i><b>2.3.5.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.5.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-4"><i class="fa fa-check"></i><b>2.3.5.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.5.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-4"><i class="fa fa-check"></i><b>2.3.5.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.5.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-4"><i class="fa fa-check"></i><b>2.3.5.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.6" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---vi4001-270500"><i class="fa fa-check"></i><b>2.3.6</b> CNV - VI:4,001-270,500</a>
<ul>
<li class="chapter" data-level="2.3.6.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-5"><i class="fa fa-check"></i><b>2.3.6.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.6.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-5"><i class="fa fa-check"></i><b>2.3.6.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.6.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-5"><i class="fa fa-check"></i><b>2.3.6.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.6.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-5"><i class="fa fa-check"></i><b>2.3.6.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.7" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---viii85501-556000"><i class="fa fa-check"></i><b>2.3.7</b> CNV - VIII:85,501-556,000</a>
<ul>
<li class="chapter" data-level="2.3.7.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-6"><i class="fa fa-check"></i><b>2.3.7.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.7.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-6"><i class="fa fa-check"></i><b>2.3.7.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.7.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-6"><i class="fa fa-check"></i><b>2.3.7.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.7.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-6"><i class="fa fa-check"></i><b>2.3.7.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.8" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---xi645001-667000"><i class="fa fa-check"></i><b>2.3.8</b> CNV - XI:645,001-667,000</a>
<ul>
<li class="chapter" data-level="2.3.8.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-7"><i class="fa fa-check"></i><b>2.3.8.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.8.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-7"><i class="fa fa-check"></i><b>2.3.8.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.8.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-7"><i class="fa fa-check"></i><b>2.3.8.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.8.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-7"><i class="fa fa-check"></i><b>2.3.8.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.9" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---xii451001-490000"><i class="fa fa-check"></i><b>2.3.9</b> CNV - XII:451,001-490,000</a>
<ul>
<li class="chapter" data-level="2.3.9.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-8"><i class="fa fa-check"></i><b>2.3.9.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.9.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-8"><i class="fa fa-check"></i><b>2.3.9.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.9.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-8"><i class="fa fa-check"></i><b>2.3.9.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.9.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-8"><i class="fa fa-check"></i><b>2.3.9.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.10" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---xv1-38000"><i class="fa fa-check"></i><b>2.3.10</b> CNV - XV:1-38,000</a>
<ul>
<li class="chapter" data-level="2.3.10.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-9"><i class="fa fa-check"></i><b>2.3.10.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.10.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-9"><i class="fa fa-check"></i><b>2.3.10.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.10.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-9"><i class="fa fa-check"></i><b>2.3.10.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.10.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-9"><i class="fa fa-check"></i><b>2.3.10.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.11" data-path="copy-number-variants.html"><a href="copy-number-variants.html#summary-of-enriched-or-depleted-functions"><i class="fa fa-check"></i><b>2.3.11</b> Summary of enriched or depleted functions</a>
<ul>
<li class="chapter" data-level="2.3.11.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#enriched-go-terms"><i class="fa fa-check"></i><b>2.3.11.1</b> Enriched GO terms</a></li>
<li class="chapter" data-level="2.3.11.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#enriched-kegg-pathways"><i class="fa fa-check"></i><b>2.3.11.2</b> Enriched KEGG pathways</a></li>
<li class="chapter" data-level="2.3.11.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#enriched-reactome-pathways"><i class="fa fa-check"></i><b>2.3.11.3</b> Enriched Reactome pathways</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="genome-assembly-and-annotation.html"><a href="genome-assembly-and-annotation.html"><i class="fa fa-check"></i><b>3</b> Genome Assembly and Annotation</a>
<ul>
<li class="chapter" data-level="3.1" data-path="genome-assembly-and-annotation.html"><a href="genome-assembly-and-annotation.html#genome-assembly"><i class="fa fa-check"></i><b>3.1</b> Genome Assembly</a></li>
<li class="chapter" data-level="3.2" data-path="genome-assembly-and-annotation.html"><a href="genome-assembly-and-annotation.html#genome-annotation"><i class="fa fa-check"></i><b>3.2</b> Genome Annotation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html"><i class="fa fa-check"></i><b>4</b> Horizontal Gene Transfer</a>
<ul>
<li class="chapter" data-level="4.1" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#taxonomic-annotation-of-predicted-genes"><i class="fa fa-check"></i><b>4.1</b> Taxonomic annotation of predicted genes</a></li>
<li class="chapter" data-level="4.2" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#evidence-for-bacterial-hgt"><i class="fa fa-check"></i><b>4.2</b> Evidence for bacterial HGT</a></li>
<li class="chapter" data-level="4.3" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#evidence-for-hgt-from-ascomycota"><i class="fa fa-check"></i><b>4.3</b> Evidence for HGT from Ascomycota</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Vikings Project</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="horizontal-gene-transfer" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> Horizontal Gene Transfer<a href="horizontal-gene-transfer.html#horizontal-gene-transfer" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>To identify genuine Bacterial (or Fungal) Horizontal Gene Transfer, we apply the following protocol:</p>
<ul>
<li>sequence similarity search against non-redundant proteins database, including taxonomic annotations</li>
<li>identify protein coding genes with best hits (top 5) to Bacterial proteins</li>
<li>select assembled contigs where putative bacterial genes are</li>
<li>discard short contigs with only one bacterial hit (Noise)</li>
<li>if putative bacterial gene flanked by Eukaryotic genes, manual sequence similarity search to confirm a <em>bona fide</em> hit</li>
</ul>
<p>To have internal controls in this (and in the following analyses), we will add the ones of 18 <em>S. cerevisiae</em> industrial strains as well.</p>
<div id="taxonomic-annotation-of-predicted-genes" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Taxonomic annotation of predicted genes<a href="horizontal-gene-transfer.html#taxonomic-annotation-of-predicted-genes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For each predicted protein-coding gene, we can make a sequence similarity search against the non-redundant protein database at NCBI (nr). To assign the taxonomy id of the nr BLAST hit, we need the <code>prot.accession2taxid.gz</code> and <code>nodes.dmp</code> files provided by NCBI taxonomy. This can help us identify horizontal gene transfer and traces of contamination in the library prep of yeast samples.</p>
<p>For each sample, first we do a blast search using DIAMOND, and then we associate to the protein_id of the top 5 best hits to the corresponding taxonomy the python library ete and customs scripts. BLAST is run against nr, excluding <em>Saccharomyces cerevisiae</em>, so that we can judge if the protein was present in other yeasts, or if it is indeed of bacterial origin.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="horizontal-gene-transfer.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># BLAST search</span></span>
<span id="cb1-2"><a href="horizontal-gene-transfer.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.all.maker.proteins.fasta<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb1-3"><a href="horizontal-gene-transfer.html#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="ex">~/bin/DIAMOND/diamond</span> blastp <span class="dt">\</span></span>
<span id="cb1-4"><a href="horizontal-gene-transfer.html#cb1-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">--query</span> <span class="va">$file</span> <span class="dt">\</span></span>
<span id="cb1-5"><a href="horizontal-gene-transfer.html#cb1-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">--db</span> ~/taxonomy/nr <span class="dt">\</span></span>
<span id="cb1-6"><a href="horizontal-gene-transfer.html#cb1-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">--taxonmap</span> ~/taxonomy/prot.accession2taxid.gz <span class="dt">\</span></span>
<span id="cb1-7"><a href="horizontal-gene-transfer.html#cb1-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">--taxonnodes</span> ~/taxonomy/nodes.dmp <span class="dt">\</span></span>
<span id="cb1-8"><a href="horizontal-gene-transfer.html#cb1-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">--threads</span> 70 <span class="dt">\</span></span>
<span id="cb1-9"><a href="horizontal-gene-transfer.html#cb1-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">--sensitive</span> <span class="dt">\</span></span>
<span id="cb1-10"><a href="horizontal-gene-transfer.html#cb1-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">--max-target-seqs</span> 5 <span class="dt">\</span></span>
<span id="cb1-11"><a href="horizontal-gene-transfer.html#cb1-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">--outfmt</span> 6 qseqid sseqid pident length mismatch gapopen qstart qend</span>
<span id="cb1-12"><a href="horizontal-gene-transfer.html#cb1-12" aria-hidden="true" tabindex="-1"></a>    <span class="ex">sstart</span> send evalue bitscore staxids <span class="op">&gt;</span> <span class="va">$file</span>.diamond</span>
<span id="cb1-13"><a href="horizontal-gene-transfer.html#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb1-14"><a href="horizontal-gene-transfer.html#cb1-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-15"><a href="horizontal-gene-transfer.html#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="co"># BLAST to taxid</span></span>
<span id="cb1-16"><a href="horizontal-gene-transfer.html#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.diamond<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb1-17"><a href="horizontal-gene-transfer.html#cb1-17" aria-hidden="true" tabindex="-1"></a>  <span class="ex">~/anaconda_ete/bin/python3.6</span> Vikings.tax.topath.py <span class="at">--input</span> <span class="va">$file</span> <span class="op">&gt;</span> <span class="va">$file</span>.tax</span>
<span id="cb1-18"><a href="horizontal-gene-transfer.html#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb1-19"><a href="horizontal-gene-transfer.html#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.tax <span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb1-20"><a href="horizontal-gene-transfer.html#cb1-20" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span> <span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb1-21"><a href="horizontal-gene-transfer.html#cb1-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grep</span> <span class="va">$line</span> <span class="va">$file</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 1,5<span class="kw">;</span></span>
<span id="cb1-22"><a href="horizontal-gene-transfer.html#cb1-22" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span><span class="kw">;</span></span>
<span id="cb1-23"><a href="horizontal-gene-transfer.html#cb1-23" aria-hidden="true" tabindex="-1"></a>  <span class="cf">done</span> <span class="op">&lt;</span> <span class="op">&lt;(</span><span class="fu">grep</span> Bacteria <span class="va">$file</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 1 <span class="kw">|</span> <span class="fu">sort</span> <span class="at">-u</span><span class="op">)</span> <span class="op">&gt;</span> <span class="va">$file</span>.bact</span>
<span id="cb1-24"><a href="horizontal-gene-transfer.html#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>Now we have the full taxonomic annotation of the five best hits for each of the protein coding genes we annotated on the kveiks.</p>
</div>
<div id="evidence-for-bacterial-hgt" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Evidence for bacterial HGT<a href="horizontal-gene-transfer.html#evidence-for-bacterial-hgt" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="horizontal-gene-transfer.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># filter annot.gff files</span></span>
<span id="cb2-2"><a href="horizontal-gene-transfer.html#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.gff <span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb2-3"><a href="horizontal-gene-transfer.html#cb2-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grep</span> CDS <span class="va">$file</span> <span class="op">&gt;</span> <span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .gff<span class="va">)</span>.CDSonly.gff<span class="kw">;</span></span>
<span id="cb2-4"><a href="horizontal-gene-transfer.html#cb2-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gzip</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb2-5"><a href="horizontal-gene-transfer.html#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb2-6"><a href="horizontal-gene-transfer.html#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="horizontal-gene-transfer.html#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co"># identify putative bacterial genes and the assembled contigs harbouring them</span></span>
<span id="cb2-8"><a href="horizontal-gene-transfer.html#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.tax<span class="kw">;</span> <span class="cf">do</span> <span class="fu">cut</span> <span class="at">-f</span> 1,5 <span class="va">$file</span> <span class="op">&gt;</span> <span class="va">$file</span>.all<span class="kw">;</span> <span class="cf">done</span></span>
<span id="cb2-9"><a href="horizontal-gene-transfer.html#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.all<span class="kw">;</span> <span class="cf">do</span> <span class="fu">grep</span> Bact <span class="va">$file</span> <span class="op">&gt;</span> <span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .all<span class="va">)</span>.bact<span class="kw">;</span> <span class="cf">done</span></span>
<span id="cb2-10"><a href="horizontal-gene-transfer.html#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.bact<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb2-11"><a href="horizontal-gene-transfer.html#cb2-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb2-12"><a href="horizontal-gene-transfer.html#cb2-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="kw">[[</span> <span class="va">$line</span> <span class="ot">=</span> <span class="st">&quot;&quot;</span> <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb2-13"><a href="horizontal-gene-transfer.html#cb2-13" aria-hidden="true" tabindex="-1"></a>      <span class="cf">continue</span><span class="kw">;</span></span>
<span id="cb2-14"><a href="horizontal-gene-transfer.html#cb2-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span></span>
<span id="cb2-15"><a href="horizontal-gene-transfer.html#cb2-15" aria-hidden="true" tabindex="-1"></a>      <span class="va">GENE</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$line</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-d</span> <span class="st">&quot; &quot;</span> <span class="at">-f</span> 1 <span class="va">)</span><span class="kw">;</span></span>
<span id="cb2-16"><a href="horizontal-gene-transfer.html#cb2-16" aria-hidden="true" tabindex="-1"></a>      <span class="va">CHR</span><span class="op">=</span><span class="va">$(</span><span class="fu">grep</span> <span class="st">&quot;</span><span class="va">${GENE}</span><span class="st">&quot;</span> <span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .maker.proteins.fasta.diamond.tax.bact<span class="va">)</span>.CDSonly.gff <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 1 <span class="va">)</span><span class="kw">;</span></span>
<span id="cb2-17"><a href="horizontal-gene-transfer.html#cb2-17" aria-hidden="true" tabindex="-1"></a>      <span class="fu">grep</span> <span class="st">&quot;</span><span class="va">${CHR}</span><span class="st">&quot;</span> <span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .bact<span class="va">)</span>.all <span class="dt">\</span></span>
<span id="cb2-18"><a href="horizontal-gene-transfer.html#cb2-18" aria-hidden="true" tabindex="-1"></a>        <span class="kw">|</span> <span class="fu">uniq</span> <span class="dt">\</span></span>
<span id="cb2-19"><a href="horizontal-gene-transfer.html#cb2-19" aria-hidden="true" tabindex="-1"></a>        <span class="kw">|</span> <span class="fu">grep</span> <span class="at">-C5</span> <span class="st">&quot;</span><span class="va">${GENE}</span><span class="st">&quot;</span> <span class="dt">\</span></span>
<span id="cb2-20"><a href="horizontal-gene-transfer.html#cb2-20" aria-hidden="true" tabindex="-1"></a>        <span class="kw">|</span> <span class="fu">grep</span> <span class="at">-v</span> <span class="st">&quot;</span><span class="va">${GENE}</span><span class="st">&quot;</span> <span class="dt">\</span></span>
<span id="cb2-21"><a href="horizontal-gene-transfer.html#cb2-21" aria-hidden="true" tabindex="-1"></a>        <span class="op">&gt;&gt;</span> <span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .maker.proteins.fasta.diamond.tax.bact<span class="va">)</span>.<span class="st">&quot;</span><span class="va">${GENE}</span><span class="st">&quot;</span>.tmp<span class="kw">;</span></span>
<span id="cb2-22"><a href="horizontal-gene-transfer.html#cb2-22" aria-hidden="true" tabindex="-1"></a>    <span class="cf">fi</span><span class="kw">;</span></span>
<span id="cb2-23"><a href="horizontal-gene-transfer.html#cb2-23" aria-hidden="true" tabindex="-1"></a>  <span class="cf">done</span> <span class="op">&lt;</span> <span class="op">&lt;(</span><span class="fu">cut</span> <span class="at">-f</span> 1 <span class="va">$file</span> <span class="kw">|</span> <span class="fu">sort</span> <span class="at">-u</span><span class="op">)</span><span class="kw">;</span></span>
<span id="cb2-24"><a href="horizontal-gene-transfer.html#cb2-24" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb2-25"><a href="horizontal-gene-transfer.html#cb2-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-26"><a href="horizontal-gene-transfer.html#cb2-26" aria-hidden="true" tabindex="-1"></a><span class="co"># pull the putative bacterial contigs</span></span>
<span id="cb2-27"><a href="horizontal-gene-transfer.html#cb2-27" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.bact<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb2-28"><a href="horizontal-gene-transfer.html#cb2-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cut</span> <span class="at">-f</span> 1 <span class="va">$file</span> <span class="kw">|</span> <span class="fu">sort</span> <span class="at">-u</span> <span class="op">&gt;</span> <span class="va">$file</span>.lst<span class="kw">;</span></span>
<span id="cb2-29"><a href="horizontal-gene-transfer.html#cb2-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">perl</span> ~/scripts/SelectList_Fasta.pl <span class="dt">\</span></span>
<span id="cb2-30"><a href="horizontal-gene-transfer.html#cb2-30" aria-hidden="true" tabindex="-1"></a>    <span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .diamond.tax.bact<span class="va">)</span> <span class="dt">\</span></span>
<span id="cb2-31"><a href="horizontal-gene-transfer.html#cb2-31" aria-hidden="true" tabindex="-1"></a>    <span class="va">$file</span>.lst <span class="dt">\</span></span>
<span id="cb2-32"><a href="horizontal-gene-transfer.html#cb2-32" aria-hidden="true" tabindex="-1"></a>    <span class="op">&gt;</span> <span class="va">$file</span>.fa<span class="kw">;</span></span>
<span id="cb2-33"><a href="horizontal-gene-transfer.html#cb2-33" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb2-34"><a href="horizontal-gene-transfer.html#cb2-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-35"><a href="horizontal-gene-transfer.html#cb2-35" aria-hidden="true" tabindex="-1"></a><span class="co"># check if putative bacterial transcripts are flanked by Euk genes</span></span>
<span id="cb2-36"><a href="horizontal-gene-transfer.html#cb2-36" aria-hidden="true" tabindex="-1"></a><span class="ex">python3.6</span> Vikings.check_eukbact_contigs.1.py <span class="at">--input</span> ../samples.lst</span>
<span id="cb2-37"><a href="horizontal-gene-transfer.html#cb2-37" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>HGT.table<span class="kw">;</span> <span class="cf">do</span> <span class="fu">uniq</span> <span class="va">$file</span> <span class="op">&gt;</span> temp<span class="kw">;</span> <span class="fu">mv</span> temp <span class="va">$file</span> <span class="kw">;</span> <span class="cf">done</span></span>
<span id="cb2-38"><a href="horizontal-gene-transfer.html#cb2-38" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>HGT.table<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb2-39"><a href="horizontal-gene-transfer.html#cb2-39" aria-hidden="true" tabindex="-1"></a>  <span class="ex">python3</span> ../Vikings.Bactmatch.py <span class="at">--input</span> <span class="va">$file</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 2 <span class="op">&gt;</span> temp<span class="kw">;</span></span>
<span id="cb2-40"><a href="horizontal-gene-transfer.html#cb2-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cp</span> temp temp2<span class="kw">;</span></span>
<span id="cb2-41"><a href="horizontal-gene-transfer.html#cb2-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">paste</span> temp temp2 <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&#39;s/\t/\|\&gt;/g&#39;</span> <span class="op">&gt;</span> <span class="va">$file</span>.lst<span class="kw">;</span></span>
<span id="cb2-42"><a href="horizontal-gene-transfer.html#cb2-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span> temp temp2<span class="kw">;</span></span>
<span id="cb2-43"><a href="horizontal-gene-transfer.html#cb2-43" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb2-44"><a href="horizontal-gene-transfer.html#cb2-44" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>HGT.table.lst<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb2-45"><a href="horizontal-gene-transfer.html#cb2-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">perl</span> ~/scripts/SelectList_Fasta.pl <span class="dt">\</span></span>
<span id="cb2-46"><a href="horizontal-gene-transfer.html#cb2-46" aria-hidden="true" tabindex="-1"></a>    <span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .HGT.table.lst<span class="va">)</span>.bact.fa <span class="dt">\</span></span>
<span id="cb2-47"><a href="horizontal-gene-transfer.html#cb2-47" aria-hidden="true" tabindex="-1"></a>    <span class="va">$file</span> <span class="dt">\</span></span>
<span id="cb2-48"><a href="horizontal-gene-transfer.html#cb2-48" aria-hidden="true" tabindex="-1"></a>    <span class="op">&gt;</span> <span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .lst<span class="va">)</span>.fa<span class="kw">;</span></span>
<span id="cb2-49"><a href="horizontal-gene-transfer.html#cb2-49" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p><strong>THERE ARE NO EVIDENCES FOR SIGNIFICANT BACTERIAL HORIZONTAL GENE TRANSFER.</strong></p>
</div>
<div id="evidence-for-hgt-from-ascomycota" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Evidence for HGT from Ascomycota<a href="horizontal-gene-transfer.html#evidence-for-hgt-from-ascomycota" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We can check the presence of fungal non-Saccharomyces genes and operons starting from the DIAMOND blast results we have.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="horizontal-gene-transfer.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select non Saccharomyces genes</span></span>
<span id="cb3-2"><a href="horizontal-gene-transfer.html#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="co"># filter for genes with no top hits to Saccharomyces</span></span>
<span id="cb3-3"><a href="horizontal-gene-transfer.html#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>tax<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb3-4"><a href="horizontal-gene-transfer.html#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grep</span> Ascomycota <span class="va">$file</span> <span class="kw">|</span> <span class="fu">grep</span> <span class="at">-v</span> Saccharomyces <span class="op">&gt;</span> <span class="va">$file</span>.Asco<span class="kw">;</span></span>
<span id="cb3-5"><a href="horizontal-gene-transfer.html#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb3-6"><a href="horizontal-gene-transfer.html#cb3-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grep</span> <span class="va">$line</span> <span class="va">$file</span> <span class="op">&gt;</span> <span class="va">$file</span>.Asco.all<span class="kw">;</span></span>
<span id="cb3-7"><a href="horizontal-gene-transfer.html#cb3-7" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="kw">[[</span> <span class="va">$(</span><span class="fu">grep</span> <span class="va">$line</span> <span class="va">$file</span> <span class="kw">|</span> <span class="fu">grep</span> Saccharomyces<span class="va">)</span> <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb3-8"><a href="horizontal-gene-transfer.html#cb3-8" aria-hidden="true" tabindex="-1"></a>      <span class="cf">continue</span><span class="kw">;</span></span>
<span id="cb3-9"><a href="horizontal-gene-transfer.html#cb3-9" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span></span>
<span id="cb3-10"><a href="horizontal-gene-transfer.html#cb3-10" aria-hidden="true" tabindex="-1"></a>      <span class="bu">echo</span> <span class="va">$line</span><span class="kw">;</span></span>
<span id="cb3-11"><a href="horizontal-gene-transfer.html#cb3-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">fi</span><span class="kw">;</span></span>
<span id="cb3-12"><a href="horizontal-gene-transfer.html#cb3-12" aria-hidden="true" tabindex="-1"></a>  <span class="cf">done</span> <span class="op">&lt;</span> <span class="op">&lt;(</span><span class="fu">cut</span> <span class="at">-f</span> 1 <span class="va">$file</span>.Asco <span class="kw">|</span> <span class="fu">sort</span> <span class="at">-u</span><span class="op">)</span> <span class="op">&gt;</span> <span class="va">$file</span>.Asco.candidates<span class="kw">;</span></span>
<span id="cb3-13"><a href="horizontal-gene-transfer.html#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb3-14"><a href="horizontal-gene-transfer.html#cb3-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-15"><a href="horizontal-gene-transfer.html#cb3-15" aria-hidden="true" tabindex="-1"></a><span class="co"># retrieve fasta sequences</span></span>
<span id="cb3-16"><a href="horizontal-gene-transfer.html#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>candidates<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb3-17"><a href="horizontal-gene-transfer.html#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="fu">perl</span> ~/scripts/SelectList_Fasta.pl <span class="dt">\</span></span>
<span id="cb3-18"><a href="horizontal-gene-transfer.html#cb3-18" aria-hidden="true" tabindex="-1"></a>  ../../11_domestication/00_prot_DB/<span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span></span>
<span id="cb3-19"><a href="horizontal-gene-transfer.html#cb3-19" aria-hidden="true" tabindex="-1"></a>  <span class="ex">.diamond.tax.Asco.candidates</span><span class="va">)</span> <span class="dt">\</span></span>
<span id="cb3-20"><a href="horizontal-gene-transfer.html#cb3-20" aria-hidden="true" tabindex="-1"></a>  <span class="va">$file</span> <span class="op">&gt;</span> <span class="va">$file</span>.fa<span class="kw">;</span></span>
<span id="cb3-21"><a href="horizontal-gene-transfer.html#cb3-21" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb3-22"><a href="horizontal-gene-transfer.html#cb3-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-23"><a href="horizontal-gene-transfer.html#cb3-23" aria-hidden="true" tabindex="-1"></a><span class="co"># generate table stats</span></span>
<span id="cb3-24"><a href="horizontal-gene-transfer.html#cb3-24" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> 2 3<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb3-25"><a href="horizontal-gene-transfer.html#cb3-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">wc</span> <span class="at">-l</span> <span class="pp">*</span>candidates <span class="dt">\</span></span>
<span id="cb3-26"><a href="horizontal-gene-transfer.html#cb3-26" aria-hidden="true" tabindex="-1"></a>    <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/.SPAdes.*//g&quot;</span> <span class="dt">\</span></span>
<span id="cb3-27"><a href="horizontal-gene-transfer.html#cb3-27" aria-hidden="true" tabindex="-1"></a>    <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/.aa.*//g&quot;</span> <span class="dt">\</span></span>
<span id="cb3-28"><a href="horizontal-gene-transfer.html#cb3-28" aria-hidden="true" tabindex="-1"></a>    <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/.contigs.*//g&quot;</span> <span class="dt">\</span></span>
<span id="cb3-29"><a href="horizontal-gene-transfer.html#cb3-29" aria-hidden="true" tabindex="-1"></a>    <span class="kw">|</span> <span class="fu">grep</span> <span class="at">-v</span> total <span class="dt">\</span></span>
<span id="cb3-30"><a href="horizontal-gene-transfer.html#cb3-30" aria-hidden="true" tabindex="-1"></a>    <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/  / /g&quot;</span> <span class="dt">\</span></span>
<span id="cb3-31"><a href="horizontal-gene-transfer.html#cb3-31" aria-hidden="true" tabindex="-1"></a>    <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/  / /g&quot;</span> <span class="dt">\</span></span>
<span id="cb3-32"><a href="horizontal-gene-transfer.html#cb3-32" aria-hidden="true" tabindex="-1"></a>    <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> <span class="va">$i</span> <span class="at">-d</span> <span class="st">&quot; &quot;</span> <span class="dt">\</span></span>
<span id="cb3-33"><a href="horizontal-gene-transfer.html#cb3-33" aria-hidden="true" tabindex="-1"></a>    <span class="op">&gt;</span> temp.<span class="va">$i</span><span class="kw">;</span></span>
<span id="cb3-34"><a href="horizontal-gene-transfer.html#cb3-34" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span><span class="kw">;</span></span>
<span id="cb3-35"><a href="horizontal-gene-transfer.html#cb3-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-36"><a href="horizontal-gene-transfer.html#cb3-36" aria-hidden="true" tabindex="-1"></a><span class="fu">paste</span> temp.3 temp.2 <span class="op">&gt;</span> Vikings.Asco.HGT.stats<span class="kw">;</span></span>
<span id="cb3-37"><a href="horizontal-gene-transfer.html#cb3-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-38"><a href="horizontal-gene-transfer.html#cb3-38" aria-hidden="true" tabindex="-1"></a><span class="co"># clean up</span></span>
<span id="cb3-39"><a href="horizontal-gene-transfer.html#cb3-39" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> temp.2 temp.3</span></code></pre></div>
<div id="htmlwidget-ca0766b0cae204d90b70" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ca0766b0cae204d90b70">{"x":{"filter":"none","vertical":false,"extensions":["FixedColumns","FixedHeader"],"caption":"<caption>Farmhouse yeasts Ascomycota HGT genes<\/caption>","data":[["14R30","14R6","16R23","16R37","17P5","19R18","1R16","21P1","21R38","27R17","28P1","28P6","28R21","28R33","28R8","2R23","38R16","39R20","3R11","40R14","40R1","40R20","41R10","41R15","42R20","42R31","44R32","44R7","45P5","45R11","46R12","46R37","6R15","7R7","8R19","9R40","Granvin1","Hornindal1","Hornindal2","k7R25","Laerdal2","Muri","SortdalEbbe1","Voss1","X1002","X1014","X1018","X1030","X1129","X5508","X1005","X1007","X1118","X1131","X1148","X7078-B","X1046","X4009","X4037","X4040","X4042","X4051"],["76","74","4","22","92","90","92","106","4","73","4","6","4","4","5","90","56","8","4","5","4","96","4","6","5","13","90","76","14","13","16","11","109","81","57","58","4","5","4","21","5","15","6","5","3","3","NA","4","4","3","NA","3","3","NA","3","NA","6","NA","3","NA","NA","4"],["Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Farmhouse","Beer1","Beer1","Beer1","Beer1","Beer1","Beer1","Beer2","Beer2","Beer2","Beer2","Beer2","Beer2","Wine","Wine","Wine","Wine","Wine","Wine"],["North-West Norway","North-West Norway","North-West Norway","North-West Norway","North-West Norway","North-West Norway","North-West Norway","North-West Norway","North-West Norway","North-West Norway","North-West Norway","North-West Norway","North-West Norway","North-West Norway","South-West Norway","South-West Norway","South-West Norway","South-West Norway","South-West Norway","South-West Norway","South-West Norway","South-West Norway","South-West Norway","South-West Norway","Central-Eastern Norway","Central-Eastern Norway","Central-Eastern Norway","Central-Eastern Norway","Central-Eastern Norway","Central-Eastern Norway","Latvia","Latvia","Latvia","Latvia","Lithuania","Lithuania","Lithuania","Lithuania","Russia","Russia","Russia","Russia","Other","Other","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Strains<\/th>\n      <th>Putative Ascomicota HGT<\/th>\n      <th>Industrial niche<\/th>\n      <th>Geographical origin<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"paging":true,"fixedHeader":false,"columnDefs":[],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Kveik strains seems to have a significant higher number of Ascomycetes genes than the other industrial strains analysed. Are there additional proteins shared between all the kveiks (or by some kveiks with same geographical origin?). We can group these proteins in gene families and see if we see a common pattern. If not, we can BLAST back the proteins and see which specific genes have been acquired.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="horizontal-gene-transfer.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># group candidates in gene families</span></span>
<span id="cb4-2"><a href="horizontal-gene-transfer.html#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="ex">~/bin/OrthoFinder-2.3.3/orthofinder</span> <span class="at">-f</span> ./ <span class="dt">\</span></span>
<span id="cb4-3"><a href="horizontal-gene-transfer.html#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">-t</span> 20 <span class="dt">\</span></span>
<span id="cb4-4"><a href="horizontal-gene-transfer.html#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">-M</span> msa <span class="dt">\</span></span>
<span id="cb4-5"><a href="horizontal-gene-transfer.html#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">-S</span> diamond <span class="dt">\</span></span>
<span id="cb4-6"><a href="horizontal-gene-transfer.html#cb4-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">-A</span> mafft <span class="dt">\</span></span>
<span id="cb4-7"><a href="horizontal-gene-transfer.html#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">-T</span> iqtree <span class="dt">\</span></span>
<span id="cb4-8"><a href="horizontal-gene-transfer.html#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">-I</span> 2</span></code></pre></div>
<p>From OrthoFinder output we can identify 49 orthogroups containing four genes or more. We visualize them as heatmap, anc we can see that kveiks strains have a higher number of common additional orthogroups that are absent from S288C and industrial strains. Interestingly, the distribution pattern does not seem to overlap with the geographical isolation of the kveik culture.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="horizontal-gene-transfer.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># import table</span></span>
<span id="cb5-2"><a href="horizontal-gene-transfer.html#cb5-2" aria-hidden="true" tabindex="-1"></a>heatfile <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="fu">paste</span>(mainDir, <span class="st">&quot;Vikings.Asco.HGT.heatmap.tab&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-3"><a href="horizontal-gene-transfer.html#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co"># relevel</span></span>
<span id="cb5-4"><a href="horizontal-gene-transfer.html#cb5-4" aria-hidden="true" tabindex="-1"></a>heatfile<span class="sc">$</span>V2 <span class="ot">=</span> <span class="fu">factor</span>(heatfile<span class="sc">$</span>V2,</span>
<span id="cb5-5"><a href="horizontal-gene-transfer.html#cb5-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;X1002&quot;</span>, <span class="st">&quot;X1014&quot;</span>, <span class="st">&quot;X1018&quot;</span>, <span class="st">&quot;X1030&quot;</span>, <span class="st">&quot;X1129&quot;</span>, <span class="st">&quot;X5508&quot;</span>,</span>
<span id="cb5-6"><a href="horizontal-gene-transfer.html#cb5-6" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot;X1005&quot;</span>, <span class="st">&quot;X1007&quot;</span>, <span class="st">&quot;X1118&quot;</span>, <span class="st">&quot;X1131&quot;</span>, <span class="st">&quot;X1148&quot;</span>, <span class="st">&quot;X7078-B&quot;</span>,</span>
<span id="cb5-7"><a href="horizontal-gene-transfer.html#cb5-7" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot;X1046&quot;</span>, <span class="st">&quot;X4009&quot;</span>, <span class="st">&quot;X4037&quot;</span>, <span class="st">&quot;X4040&quot;</span>, <span class="st">&quot;X4042&quot;</span>, <span class="st">&quot;X4051&quot;</span>,</span>
<span id="cb5-8"><a href="horizontal-gene-transfer.html#cb5-8" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot;Voss1&quot;</span>,</span>
<span id="cb5-9"><a href="horizontal-gene-transfer.html#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;SortdalEbbe1&quot;</span>, <span class="st">&quot;Muri&quot;</span>, <span class="st">&quot;Laerdal2&quot;</span>, <span class="st">&quot;k7R25&quot;</span>,</span>
<span id="cb5-10"><a href="horizontal-gene-transfer.html#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Hornindal2&quot;</span>, <span class="st">&quot;Hornindal1&quot;</span>, <span class="st">&quot;Granvin1&quot;</span>, <span class="st">&quot;9R40&quot;</span>, <span class="st">&quot;8R19&quot;</span>, <span class="st">&quot;7R7&quot;</span>,</span>
<span id="cb5-11"><a href="horizontal-gene-transfer.html#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;6R15&quot;</span>, <span class="st">&quot;46R37&quot;</span>, <span class="st">&quot;46R12&quot;</span>, <span class="st">&quot;45R11&quot;</span>, <span class="st">&quot;45P5&quot;</span>,</span>
<span id="cb5-12"><a href="horizontal-gene-transfer.html#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;44R7&quot;</span>, <span class="st">&quot;44R32&quot;</span>, <span class="st">&quot;42R31&quot;</span>, <span class="st">&quot;42R20&quot;</span>, <span class="st">&quot;41R15&quot;</span>, <span class="st">&quot;41R10&quot;</span>, <span class="st">&quot;40R20&quot;</span>,</span>
<span id="cb5-13"><a href="horizontal-gene-transfer.html#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;40R1&quot;</span>, <span class="st">&quot;40R14&quot;</span>, <span class="st">&quot;3R11&quot;</span>, <span class="st">&quot;39R20&quot;</span>, <span class="st">&quot;38R16&quot;</span>,</span>
<span id="cb5-14"><a href="horizontal-gene-transfer.html#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;2R23&quot;</span>, <span class="st">&quot;28R8&quot;</span>, <span class="st">&quot;28R33&quot;</span>, <span class="st">&quot;28R21&quot;</span>, <span class="st">&quot;28P6&quot;</span>, <span class="st">&quot;28P1&quot;</span>, <span class="st">&quot;27R17&quot;</span>,</span>
<span id="cb5-15"><a href="horizontal-gene-transfer.html#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;21R38&quot;</span>, <span class="st">&quot;21P1&quot;</span>, <span class="st">&quot;1R16&quot;</span>, <span class="st">&quot;19R18&quot;</span>, <span class="st">&quot;17P5&quot;</span>,</span>
<span id="cb5-16"><a href="horizontal-gene-transfer.html#cb5-16" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;16R37&quot;</span>, <span class="st">&quot;16R23&quot;</span>, <span class="st">&quot;14R6&quot;</span>, <span class="st">&quot;14R30&quot;</span>))</span>
<span id="cb5-17"><a href="horizontal-gene-transfer.html#cb5-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-18"><a href="horizontal-gene-transfer.html#cb5-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-19"><a href="horizontal-gene-transfer.html#cb5-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-20"><a href="horizontal-gene-transfer.html#cb5-20" aria-hidden="true" tabindex="-1"></a><span class="co"># prepare heatmap</span></span>
<span id="cb5-21"><a href="horizontal-gene-transfer.html#cb5-21" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(heatfile) <span class="sc">+</span></span>
<span id="cb5-22"><a href="horizontal-gene-transfer.html#cb5-22" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">x =</span> V1, <span class="at">y =</span> V2, <span class="at">fill =</span> V3)) <span class="sc">+</span></span>
<span id="cb5-23"><a href="horizontal-gene-transfer.html#cb5-23" aria-hidden="true" tabindex="-1"></a><span class="fu">scale_fill_gradientn</span>(<span class="at">na.value =</span> <span class="st">&quot;white&quot;</span>, <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">2</span>),</span>
<span id="cb5-24"><a href="horizontal-gene-transfer.html#cb5-24" aria-hidden="true" tabindex="-1"></a><span class="at">colours =</span> <span class="fu">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;cyan&quot;</span>, <span class="st">&quot;steelblue&quot;</span>),</span>
<span id="cb5-25"><a href="horizontal-gene-transfer.html#cb5-25" aria-hidden="true" tabindex="-1"></a><span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb5-26"><a href="horizontal-gene-transfer.html#cb5-26" aria-hidden="true" tabindex="-1"></a><span class="fu">coord_cartesian</span>(<span class="at">expand =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb5-27"><a href="horizontal-gene-transfer.html#cb5-27" aria-hidden="true" tabindex="-1"></a><span class="fu">labs</span>(<span class="at">fill =</span> <span class="st">&quot;Gene counts&quot;</span>,</span>
<span id="cb5-28"><a href="horizontal-gene-transfer.html#cb5-28" aria-hidden="true" tabindex="-1"></a><span class="at">y =</span> <span class="st">&quot;Strain&quot;</span>,</span>
<span id="cb5-29"><a href="horizontal-gene-transfer.html#cb5-29" aria-hidden="true" tabindex="-1"></a><span class="at">x =</span> <span class="st">&quot;Orthologous groups&quot;</span>) <span class="sc">+</span></span>
<span id="cb5-30"><a href="horizontal-gene-transfer.html#cb5-30" aria-hidden="true" tabindex="-1"></a><span class="fu">theme</span>(<span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb5-31"><a href="horizontal-gene-transfer.html#cb5-31" aria-hidden="true" tabindex="-1"></a><span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="fl">0.95</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb5-32"><a href="horizontal-gene-transfer.html#cb5-32" aria-hidden="true" tabindex="-1"></a><span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb5-33"><a href="horizontal-gene-transfer.html#cb5-33" aria-hidden="true" tabindex="-1"></a><span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb5-34"><a href="horizontal-gene-transfer.html#cb5-34" aria-hidden="true" tabindex="-1"></a><span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">size =</span> <span class="fl">0.75</span>)) <span class="sc">+</span></span>
<span id="cb5-35"><a href="horizontal-gene-transfer.html#cb5-35" aria-hidden="true" tabindex="-1"></a><span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_colourbar</span>(<span class="at">ticks.colour =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb5-36"><a href="horizontal-gene-transfer.html#cb5-36" aria-hidden="true" tabindex="-1"></a><span class="at">frame.colour =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb5-37"><a href="horizontal-gene-transfer.html#cb5-37" aria-hidden="true" tabindex="-1"></a><span class="at">frame.linewidth =</span> <span class="dv">1</span>))</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-7"></span>
<img src="main_files/figure-html/unnamed-chunk-7-1.png" alt="Ascomycota HTG orthogroup candidates" width="672" />
<p class="caption">
Figure 4.1: Ascomycota HTG orthogroup candidates
</p>
</div>

</div>
</div>






            </section>

          </div>
        </div>
      </div>
<a href="genome-assembly-and-annotation.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/04-HorizontalGeneTransfer.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
