<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>3&nbsp; Variant Calling – Farmhouse ale cultures preserve brewing’s lost yeast diversity</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./p01-04-copy-number-variants.html" rel="next">
<link href="./p01-02-genome-assembly-and-annotation.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<style>
#quarto-sidebar .sidebar-item-text {
  font-weight: bold;
}
#quarto-sidebar .chapter-number {
  font-weight: normal
}
#quarto-sidebar .chapter-title {
  font-weight: normal
}
#.nav-tabs .nav-link.active {
  background-color: rgba(239, 236, 255, 0.5);
}
</style>


</head>

<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Farmhouse ale cultures preserve brewing’s lost yeast diversity</span>
    </a>
  </div>
        <div class="quarto-navbar-tools tools-end">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./p01-01-qc-and-genome-composition.html">Part 1: Bioinformatics analyses</a></li><li class="breadcrumb-item"><a href="./p01-03-variant-calling.html"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Variant Calling</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Abstract</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Part 1: Bioinformatics analyses</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./p01-01-qc-and-genome-composition.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">QC and Genome Composition assessment</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./p01-02-genome-assembly-and-annotation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Genome Assembly and Annotation</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./p01-03-variant-calling.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Variant Calling</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./p01-04-copy-number-variants.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Copy Number Variants</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./p01-05-phylogeny-and-population-structure.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Phylogeny and Population Structure</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./p01-06-phenotyping.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">High-throughput phenotyping</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Part 2: Publications figures</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./p02-01-Figure1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Figure 1</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./p02-02-Figure2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Figure 2</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./p02-03-Figure3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Figure 3</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./p02-04-Figure4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Figure 4</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./p02-05-Figure5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Figure 5</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./p02-06-Figure6.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Figure 6</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./p02-07-FigureS1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Figure S1</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./p02-08-FigureS2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Figure S2</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./p02-09-FigureS3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Figure 5</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./p02-10-FigureS4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">Figure S4</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./p02-11-FigureS5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">17</span>&nbsp; <span class="chapter-title">Figure 5</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./p02-12-FigureS6.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">18</span>&nbsp; <span class="chapter-title">Figure 5</span></span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#on-this-page" id="toc-on-this-page" class="nav-link active" data-scroll-target="#on-this-page"><span class="header-section-number">3.1</span> On this page</a></li>
  <li><a href="#variant-calling" id="toc-variant-calling" class="nav-link" data-scroll-target="#variant-calling"><span class="header-section-number">3.2</span> Variant Calling</a>
  <ul>
  <li><a href="#reads-alignment" id="toc-reads-alignment" class="nav-link" data-scroll-target="#reads-alignment"><span class="header-section-number">3.2.1</span> Reads alignment</a></li>
  <li><a href="#variant-calling-1" id="toc-variant-calling-1" class="nav-link" data-scroll-target="#variant-calling-1"><span class="header-section-number">3.2.2</span> Variant Calling</a></li>
  <li><a href="#variant-filtering-and-convert-to-plink-bed-input" id="toc-variant-filtering-and-convert-to-plink-bed-input" class="nav-link" data-scroll-target="#variant-filtering-and-convert-to-plink-bed-input"><span class="header-section-number">3.2.3</span> Variant filtering and convert to plink bed input</a></li>
  </ul></li>
  <li><a href="#farmhouse-yeasts-heterozygosity" id="toc-farmhouse-yeasts-heterozygosity" class="nav-link" data-scroll-target="#farmhouse-yeasts-heterozygosity"><span class="header-section-number">3.3</span> Farmhouse yeasts heterozygosity</a></li>
  <li><a href="#loss-of-heterozygosity-loh" id="toc-loss-of-heterozygosity-loh" class="nav-link" data-scroll-target="#loss-of-heterozygosity-loh"><span class="header-section-number">3.4</span> Loss Of Heterozygosity (LOH)</a></li>
  <li><a href="#search-for-signature-of-culture-28-hybridization" id="toc-search-for-signature-of-culture-28-hybridization" class="nav-link" data-scroll-target="#search-for-signature-of-culture-28-hybridization"><span class="header-section-number">3.5</span> Search for signature of Culture #28 hybridization</a></li>
  <li><a href="#genome-domestication-signature" id="toc-genome-domestication-signature" class="nav-link" data-scroll-target="#genome-domestication-signature"><span class="header-section-number">3.6</span> Genome Domestication Signature</a>
  <ul>
  <li><a href="#identifying-genes-missing-in-s288c" id="toc-identifying-genes-missing-in-s288c" class="nav-link" data-scroll-target="#identifying-genes-missing-in-s288c"><span class="header-section-number">3.6.1</span> Identifying genes missing in S288C</a></li>
  <li><a href="#annotate-variants" id="toc-annotate-variants" class="nav-link" data-scroll-target="#annotate-variants"><span class="header-section-number">3.6.2</span> Annotate variants</a></li>
  <li><a href="#parse-variants" id="toc-parse-variants" class="nav-link" data-scroll-target="#parse-variants"><span class="header-section-number">3.6.3</span> Parse variants</a></li>
  <li><a href="#identify-disrupted-genes" id="toc-identify-disrupted-genes" class="nav-link" data-scroll-target="#identify-disrupted-genes"><span class="header-section-number">3.6.4</span> Identify disrupted genes</a></li>
  <li><a href="#phylogenetic-relationships" id="toc-phylogenetic-relationships" class="nav-link" data-scroll-target="#phylogenetic-relationships"><span class="header-section-number">3.6.5</span> Phylogenetic relationships</a></li>
  </ul></li>
  <li><a href="#sec-lesson-learnt-feature-selection" id="toc-sec-lesson-learnt-feature-selection" class="nav-link" data-scroll-target="#sec-lesson-learnt-feature-selection"><span class="header-section-number">3.7</span> Lessons Learnt</a></li>
  <li><a href="#session-information" id="toc-session-information" class="nav-link" data-scroll-target="#session-information"><span class="header-section-number">3.8</span> Session Information</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./p01-01-qc-and-genome-composition.html">Part 1: Bioinformatics analyses</a></li><li class="breadcrumb-item"><a href="./p01-03-variant-calling.html"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Variant Calling</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Variant Calling</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<div class="cell">
<style type="text/css">

pre {
  max-height: 300px;
}

div.sourceCode {
  overflow: scroll;
}

.bash-code {
  font-style: italic;
}


</style>
</div>
<section id="on-this-page" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="on-this-page"><span class="header-section-number">3.1</span> On this page</h2>
<p>Biological insights and take-home messages are at the bottom of the page at section Lesson Learnt: <a href="#sec-lesson-learnt-feature-selection" class="quarto-xref"><span class="quarto-unresolved-ref">sec-lesson-learnt-feature-selection</span></a>.</p>
<ul>
<li>Here</li>
</ul>
</section>
<section id="variant-calling" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="variant-calling"><span class="header-section-number">3.2</span> Variant Calling</h2>
<p>We will now analyse the DNA variants between the sequenced farmhouse yeasts and the reference <em>S. cerevisiae</em> lab strain S288C. Genetic variants will be combined with information from the 151 yeast strains sequenced in Gallone et al 2019 used to reconstruct (kveiks) population structures, levels of heterozygosities, putative regions with Loss Of Heterozygosity (LOH) and the eventual genes affected by LOH.</p>
<section id="reads-alignment" class="level3" data-number="3.2.1">
<h3 data-number="3.2.1" class="anchored" data-anchor-id="reads-alignment"><span class="header-section-number">3.2.1</span> Reads alignment</h3>
<p>Here the steps are the standard I always do:</p>
<ul>
<li>align reads to reference genome with bwa mem;</li>
<li>sort the bam file by read position;</li>
<li>mark reads duplicates with what used to be picard tools;</li>
<li>assign read groups, so that each sequencing library is assigned to the right sample (necessary for when I merge the vcf files)</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1"></a><span class="co">## align reads to reference genome S288C</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="co"># align reads</span></span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="va">DB</span><span class="op">=</span>~/00_Brigida_starting_files/variants/Saccharomyces_cerevisiae.EF4.73.dna.chromosome.all.fa</span>
<span id="cb1-4"><a href="#cb1-4"></a></span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb1-6"><a href="#cb1-6"></a></span>
<span id="cb1-7"><a href="#cb1-7"></a>  <span class="ex">bwa</span> mem <span class="at">-t</span> 72 <span class="dt">\</span></span>
<span id="cb1-8"><a href="#cb1-8"></a>    <span class="at">-K</span> 100000000 <span class="dt">\</span></span>
<span id="cb1-9"><a href="#cb1-9"></a>    <span class="va">$DB</span> <span class="dt">\</span></span>
<span id="cb1-10"><a href="#cb1-10"></a>    ../00_trim_reads/<span class="va">$line</span>.R1.tr.fq.gz <span class="dt">\</span></span>
<span id="cb1-11"><a href="#cb1-11"></a>    ../00_trim_reads/<span class="va">$line</span>.R2.tr.fq.gz <span class="dt">\</span></span>
<span id="cb1-12"><a href="#cb1-12"></a>    <span class="op">&gt;</span> <span class="va">$line</span>.S288C.align.sam<span class="kw">;</span></span>
<span id="cb1-13"><a href="#cb1-13"></a>    </span>
<span id="cb1-14"><a href="#cb1-14"></a>  <span class="ex">samtools</span> view <span class="dt">\</span></span>
<span id="cb1-15"><a href="#cb1-15"></a>    <span class="at">-@</span> 72 <span class="dt">\</span></span>
<span id="cb1-16"><a href="#cb1-16"></a>    <span class="at">-Sb</span> <span class="dt">\</span></span>
<span id="cb1-17"><a href="#cb1-17"></a>    <span class="va">$line</span>.S288C.align.sam <span class="dt">\</span></span>
<span id="cb1-18"><a href="#cb1-18"></a>    <span class="op">&gt;</span> <span class="va">$line</span>.S288C.align.bam<span class="kw">;</span></span>
<span id="cb1-19"><a href="#cb1-19"></a>  </span>
<span id="cb1-20"><a href="#cb1-20"></a>  <span class="ex">samtools</span> sort <span class="dt">\</span></span>
<span id="cb1-21"><a href="#cb1-21"></a>    <span class="at">-@</span> 72 <span class="dt">\</span></span>
<span id="cb1-22"><a href="#cb1-22"></a>    <span class="va">$line</span>.S288C.align.bam <span class="va">$line</span>.S288C.align.sort<span class="kw">;</span></span>
<span id="cb1-23"><a href="#cb1-23"></a>  </span>
<span id="cb1-24"><a href="#cb1-24"></a>  <span class="fu">rm</span> <span class="va">$line</span>.S288C.align.sam <span class="va">$line</span>.S288C.align.bam<span class="kw">;</span></span>
<span id="cb1-25"><a href="#cb1-25"></a></span>
<span id="cb1-26"><a href="#cb1-26"></a><span class="cf">done</span> <span class="op">&lt;</span> ../sample.lst</span>
<span id="cb1-27"><a href="#cb1-27"></a></span>
<span id="cb1-28"><a href="#cb1-28"></a><span class="co">## mark duplicates</span></span>
<span id="cb1-29"><a href="#cb1-29"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.bam<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb1-30"><a href="#cb1-30"></a>  <span class="ex">~/bin/gatk-4.0.11.0/gatk</span> MarkDuplicates <span class="dt">\</span></span>
<span id="cb1-31"><a href="#cb1-31"></a>    <span class="at">--INPUT</span><span class="op">=</span><span class="va">$file</span> <span class="dt">\</span></span>
<span id="cb1-32"><a href="#cb1-32"></a>    <span class="at">--OUTPUT</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .bam<span class="va">)</span>.md.bam <span class="dt">\</span></span>
<span id="cb1-33"><a href="#cb1-33"></a>    <span class="at">--METRICS_FILE</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .bam<span class="va">)</span>.md.log <span class="dt">\</span></span>
<span id="cb1-34"><a href="#cb1-34"></a>    <span class="at">--CREATE_INDEX</span><span class="op">=</span>true<span class="kw">;</span></span>
<span id="cb1-35"><a href="#cb1-35"></a><span class="cf">done</span></span>
<span id="cb1-36"><a href="#cb1-36"></a></span>
<span id="cb1-37"><a href="#cb1-37"></a><span class="co"># assign read group</span></span>
<span id="cb1-38"><a href="#cb1-38"></a><span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb1-39"><a href="#cb1-39"></a>  <span class="ex">~/bin/gatk-4.0.11.0/gatk</span> AddOrReplaceReadGroups <span class="dt">\</span></span>
<span id="cb1-40"><a href="#cb1-40"></a>    <span class="at">--INPUT</span> <span class="va">$line</span>.S288C.align.sort.md.bam <span class="dt">\</span></span>
<span id="cb1-41"><a href="#cb1-41"></a>    <span class="at">--OUTPUT</span> <span class="va">$line</span>.S288C.align.sort.md.r.bam <span class="dt">\</span></span>
<span id="cb1-42"><a href="#cb1-42"></a>    <span class="at">--RGID</span> <span class="va">$line</span> <span class="dt">\</span></span>
<span id="cb1-43"><a href="#cb1-43"></a>    <span class="at">--RGPL</span> <span class="va">$line</span> <span class="dt">\</span></span>
<span id="cb1-44"><a href="#cb1-44"></a>    <span class="at">--RGLB</span> <span class="va">$line</span> <span class="dt">\</span></span>
<span id="cb1-45"><a href="#cb1-45"></a>    <span class="at">--RGPU</span> <span class="va">$line</span> <span class="dt">\</span></span>
<span id="cb1-46"><a href="#cb1-46"></a>    <span class="at">--RGSM</span> <span class="va">$line</span><span class="kw">;</span></span>
<span id="cb1-47"><a href="#cb1-47"></a><span class="cf">done</span> <span class="op">&lt;</span> ../sample.lst</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="variant-calling-1" class="level3" data-number="3.2.2">
<h3 data-number="3.2.2" class="anchored" data-anchor-id="variant-calling-1"><span class="header-section-number">3.2.2</span> Variant Calling</h3>
<p>I run HaplotypeCaller in gVCF mode, sample ploidy 2 and calling all the kveiks positions at the same time to reduce the number of false positives.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1"></a><span class="co">## run haplotype caller</span></span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="va">DB</span><span class="op">=</span>~/00_Brigida_starting_files/variants/Saccharomyces_cerevisiae.EF4.73.dna.chromosome.all.fa</span>
<span id="cb2-3"><a href="#cb2-3"></a></span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="cf">for</span> i <span class="kw">in</span> chr1 chr2 chr3 chr4 chr5 chr6 chr7 chr8 chr9 chr10 chr11 chr12 chr13 chr14 chr15 chr16<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb2-5"><a href="#cb2-5"></a>  <span class="ex">java</span> <span class="at">-Xmx8g</span> <span class="dt">\</span></span>
<span id="cb2-6"><a href="#cb2-6"></a>    <span class="at">-jar</span> ~/bin/GenomeAnalysisTK-3.3-0-g37228af/GenomeAnalysisTK.jar <span class="dt">\</span></span>
<span id="cb2-7"><a href="#cb2-7"></a>    <span class="at">--logging_level</span> INFO <span class="dt">\</span></span>
<span id="cb2-8"><a href="#cb2-8"></a>    <span class="at">--num_threads</span> 4 <span class="dt">\</span></span>
<span id="cb2-9"><a href="#cb2-9"></a>    <span class="at">--analysis_type</span> UnifiedGenotyper <span class="dt">\</span></span>
<span id="cb2-10"><a href="#cb2-10"></a>    <span class="at">--reference_sequence</span> <span class="va">$DB</span> <span class="dt">\</span></span>
<span id="cb2-11"><a href="#cb2-11"></a>    <span class="at">-I</span> bam_files.list <span class="dt">\</span></span>
<span id="cb2-12"><a href="#cb2-12"></a>    <span class="at">-o</span> S288C.variants.<span class="va">$i</span>.gvcf <span class="dt">\</span></span>
<span id="cb2-13"><a href="#cb2-13"></a>    <span class="at">--intervals</span> Saccharomyces_cerevisiae.EF4.73.dna.chromosome.<span class="va">$i</span>.interval_list <span class="dt">\</span></span>
<span id="cb2-14"><a href="#cb2-14"></a>    <span class="at">--standard_min_confidence_threshold_for_calling</span> 20 <span class="dt">\</span></span>
<span id="cb2-15"><a href="#cb2-15"></a>    <span class="at">-stand_emit_conf</span> 30.0 <span class="dt">\</span></span>
<span id="cb2-16"><a href="#cb2-16"></a>    <span class="at">--min_base_quality_score</span> 20 <span class="dt">\</span></span>
<span id="cb2-17"><a href="#cb2-17"></a>    <span class="at">-glm</span> BOTH <span class="dt">\</span></span>
<span id="cb2-18"><a href="#cb2-18"></a>    <span class="at">--sample_ploidy</span> 2 <span class="dt">\</span></span>
<span id="cb2-19"><a href="#cb2-19"></a>    <span class="at">--validation_strictness</span> SILENT <span class="dt">\</span></span>
<span id="cb2-20"><a href="#cb2-20"></a>    <span class="at">--output_mode</span> EMIT_ALL_SITES <span class="kw">&amp;</span></span>
<span id="cb2-21"><a href="#cb2-21"></a><span class="cf">done</span></span>
<span id="cb2-22"><a href="#cb2-22"></a></span>
<span id="cb2-23"><a href="#cb2-23"></a><span class="co"># merge kveiks with with brigida's chromosomes</span></span>
<span id="cb2-24"><a href="#cb2-24"></a><span class="cf">for</span> i <span class="kw">in</span> chr1 chr2 chr3 chr4 chr5 chr6 chr7 chr8 chr9 chr10 chr11 chr12 chr13 chr14 chr15 chr16<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb2-25"><a href="#cb2-25"></a>  <span class="ex">bcftools</span> merge <span class="dt">\</span></span>
<span id="cb2-26"><a href="#cb2-26"></a>    <span class="at">--threads</span> 4 <span class="dt">\</span></span>
<span id="cb2-27"><a href="#cb2-27"></a>    <span class="at">--output-type</span> z <span class="dt">\</span></span>
<span id="cb2-28"><a href="#cb2-28"></a>    <span class="at">--output</span> all_plusKV.<span class="va">$i</span>.vcf.gz <span class="dt">\</span></span>
<span id="cb2-29"><a href="#cb2-29"></a>    kveiks.S288C.variants.<span class="va">$i</span>.gvcf.gz <span class="dt">\</span></span>
<span id="cb2-30"><a href="#cb2-30"></a>    ../03_Bri_chr/Scer.<span class="va">$i</span>.raw.vcf.gz <span class="kw">&amp;</span></span>
<span id="cb2-31"><a href="#cb2-31"></a><span class="cf">done</span></span>
<span id="cb2-32"><a href="#cb2-32"></a></span>
<span id="cb2-33"><a href="#cb2-33"></a><span class="co"># merge all chromosomes together</span></span>
<span id="cb2-34"><a href="#cb2-34"></a><span class="ex">bcftools</span> concat <span class="dt">\</span></span>
<span id="cb2-35"><a href="#cb2-35"></a>  all_plusKV.chr1.vcf.gz <span class="dt">\</span></span>
<span id="cb2-36"><a href="#cb2-36"></a>  all_plusKV.chr2.vcf.gz <span class="dt">\</span></span>
<span id="cb2-37"><a href="#cb2-37"></a>  all_plusKV.chr3.vcf.gz <span class="dt">\</span></span>
<span id="cb2-38"><a href="#cb2-38"></a>  all_plusKV.chr4.vcf.gz <span class="dt">\</span></span>
<span id="cb2-39"><a href="#cb2-39"></a>  all_plusKV.chr5.vcf.gz <span class="dt">\</span></span>
<span id="cb2-40"><a href="#cb2-40"></a>  all_plusKV.chr6.vcf.gz <span class="dt">\</span></span>
<span id="cb2-41"><a href="#cb2-41"></a>  all_plusKV.chr7.vcf.gz <span class="dt">\</span></span>
<span id="cb2-42"><a href="#cb2-42"></a>  all_plusKV.chr8.vcf.gz <span class="dt">\</span></span>
<span id="cb2-43"><a href="#cb2-43"></a>  all_plusKV.chr9.vcf.gz <span class="dt">\</span></span>
<span id="cb2-44"><a href="#cb2-44"></a>  all_plusKV.chr10.vcf.gz <span class="dt">\</span></span>
<span id="cb2-45"><a href="#cb2-45"></a>  all_plusKV.chr11.vcf.gz <span class="dt">\</span></span>
<span id="cb2-46"><a href="#cb2-46"></a>  all_plusKV.chr12.vcf.gz <span class="dt">\</span></span>
<span id="cb2-47"><a href="#cb2-47"></a>  all_plusKV.chr13.vcf.gz <span class="dt">\</span></span>
<span id="cb2-48"><a href="#cb2-48"></a>  all_plusKV.chr14.vcf.gz <span class="dt">\</span></span>
<span id="cb2-49"><a href="#cb2-49"></a>  all_plusKV.chr15.vcf.gz <span class="dt">\</span></span>
<span id="cb2-50"><a href="#cb2-50"></a>  all_plusKV.chr16.vcf.gz <span class="dt">\</span></span>
<span id="cb2-51"><a href="#cb2-51"></a>  <span class="at">--output-type</span> z <span class="dt">\</span></span>
<span id="cb2-52"><a href="#cb2-52"></a>  <span class="at">--output</span> all_plusKV.vcf.gz <span class="dt">\</span></span>
<span id="cb2-53"><a href="#cb2-53"></a>  <span class="at">--threads</span> 64</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="variant-filtering-and-convert-to-plink-bed-input" class="level3" data-number="3.2.3">
<h3 data-number="3.2.3" class="anchored" data-anchor-id="variant-filtering-and-convert-to-plink-bed-input"><span class="header-section-number">3.2.3</span> Variant filtering and convert to plink bed input</h3>
<p>Here I filter for SNPs that are informative for Population Structure analysis. What I do:</p>
<ul>
<li>apply hard filters to flag low quality positions;</li>
<li>select biallelic and high frequency SNPs, SNPs in linkage disequilibrium.</li>
</ul>
<p>Here I select biallelic SNPs and filter out minor allele frequency. Then I merge my vcf file with the vcf files from your dataset. After, I just invoke plink for generate the input files for FastStructure</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1"></a><span class="co"># hard filter low quality variants</span></span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="ex">java</span> <span class="at">-jar</span> ~/bin/GenomeAnalysisTK-3.3-0-g37228af/GenomeAnalysisTK.jar <span class="dt">\</span></span>
<span id="cb3-3"><a href="#cb3-3"></a>  <span class="at">--analysis_type</span> VariantFiltration <span class="dt">\</span></span>
<span id="cb3-4"><a href="#cb3-4"></a>  <span class="at">--variant</span> all_plusKV.vcf.gz <span class="dt">\</span></span>
<span id="cb3-5"><a href="#cb3-5"></a>  <span class="at">--out</span> all_plusKV.fltr.vcf <span class="dt">\</span></span>
<span id="cb3-6"><a href="#cb3-6"></a>  <span class="at">--reference_sequence</span> <span class="va">$DB</span> <span class="dt">\</span></span>
<span id="cb3-7"><a href="#cb3-7"></a>  <span class="at">--filterExpression</span> <span class="st">"QD &lt; 2.0"</span> <span class="at">--filterName</span> <span class="st">"QD_Low"</span> <span class="dt">\</span></span>
<span id="cb3-8"><a href="#cb3-8"></a>  <span class="at">--filterExpression</span> <span class="st">"MQ &lt; 40.0"</span> <span class="at">--filterName</span> <span class="st">"MQ_Low"</span> <span class="dt">\</span></span>
<span id="cb3-9"><a href="#cb3-9"></a>  <span class="at">--filterExpression</span> <span class="st">"GQ &lt; 30"</span> <span class="at">--filterName</span> <span class="st">"GQ_Low"</span> <span class="dt">\</span></span>
<span id="cb3-10"><a href="#cb3-10"></a>  <span class="at">--filterExpression</span> <span class="st">"AD &lt; 5"</span> <span class="at">--filterName</span> <span class="st">"AD_Low"</span></span>
<span id="cb3-11"><a href="#cb3-11"></a></span>
<span id="cb3-12"><a href="#cb3-12"></a><span class="co"># select biallelic loci, filter for MAF and linkage disequilibrium</span></span>
<span id="cb3-13"><a href="#cb3-13"></a><span class="ex">~/bin/plink_linux_x86_64_20181202/plink</span> <span class="dt">\</span></span>
<span id="cb3-14"><a href="#cb3-14"></a>  <span class="at">--make-bed</span> <span class="dt">\</span></span>
<span id="cb3-15"><a href="#cb3-15"></a>  <span class="at">--double-id</span> <span class="dt">\</span></span>
<span id="cb3-16"><a href="#cb3-16"></a>  <span class="at">--allow-extra-chr</span> <span class="dt">\</span></span>
<span id="cb3-17"><a href="#cb3-17"></a>  <span class="at">--indep-pairwise</span> 50 5 0.5 <span class="dt">\</span></span>
<span id="cb3-18"><a href="#cb3-18"></a>  <span class="at">--maf</span> 0.05 <span class="dt">\</span></span>
<span id="cb3-19"><a href="#cb3-19"></a>  <span class="at">--biallelic-only</span> <span class="dt">\</span></span>
<span id="cb3-20"><a href="#cb3-20"></a>  <span class="at">--geno</span> 0.25 <span class="dt">\</span></span>
<span id="cb3-21"><a href="#cb3-21"></a>  <span class="at">--vcf</span> all_plusKV.fltr.vcf.gz <span class="dt">\</span></span>
<span id="cb3-22"><a href="#cb3-22"></a>  <span class="at">--out</span> all_plusKV.fltr.bi</span>
<span id="cb3-23"><a href="#cb3-23"></a>  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="farmhouse-yeasts-heterozygosity" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="farmhouse-yeasts-heterozygosity"><span class="header-section-number">3.3</span> Farmhouse yeasts heterozygosity</h2>
<p>Farmhouse yeasts contains a high degree of Heterozygosity, comparable to yeast strains from the “mixed” Clade.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co"># import dataset</span></span>
<span id="cb4-2"><a href="#cb4-2"></a>kveik_het <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="st">"./data/p01-03/all_plusKV.heterozygous.positions.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-3"><a href="#cb4-3"></a></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="co"># sort </span></span>
<span id="cb4-5"><a href="#cb4-5"></a>kveik_het<span class="sc">$</span>Pop <span class="ot">=</span> <span class="fu">factor</span>(kveik_het<span class="sc">$</span>Pop,</span>
<span id="cb4-6"><a href="#cb4-6"></a>                       <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Farmhouse"</span>, <span class="st">"Asian"</span>, <span class="st">"Beer1"</span>, <span class="st">"Beer2"</span>, <span class="st">"Bread"</span>,</span>
<span id="cb4-7"><a href="#cb4-7"></a>                                  <span class="st">"Mixed"</span>, <span class="st">"Mosaic"</span>, <span class="st">"Sake"</span>, <span class="st">"Wild"</span>, <span class="st">"Wine"</span>))</span>
<span id="cb4-8"><a href="#cb4-8"></a>kveik_het<span class="sc">$</span>Pop <span class="ot">=</span> <span class="fu">relevel</span>(kveik_het<span class="sc">$</span>Pop, <span class="st">"Farmhouse"</span>)</span>
<span id="cb4-9"><a href="#cb4-9"></a>kveik_het<span class="sc">$</span>Geographic.origin <span class="ot">=</span> <span class="fu">ifelse</span>(</span>
<span id="cb4-10"><a href="#cb4-10"></a>  kveik_het<span class="sc">$</span>Geographic.origin <span class="sc">==</span> <span class="st">"Central-E Norway"</span>,</span>
<span id="cb4-11"><a href="#cb4-11"></a>  <span class="st">"C-E Norway"</span>,</span>
<span id="cb4-12"><a href="#cb4-12"></a>  kveik_het<span class="sc">$</span>Geographic.origin</span>
<span id="cb4-13"><a href="#cb4-13"></a>)</span>
<span id="cb4-14"><a href="#cb4-14"></a></span>
<span id="cb4-15"><a href="#cb4-15"></a><span class="fu">colnames</span>(kveik_het) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"ID"</span>, <span class="st">"Pop"</span>, <span class="st">"HET_CT"</span>, <span class="st">"Subpopulation"</span>, <span class="st">"Geographic.origin"</span>)</span>
<span id="cb4-16"><a href="#cb4-16"></a></span>
<span id="cb4-17"><a href="#cb4-17"></a><span class="co"># mapping swap_yeasts</span></span>
<span id="cb4-18"><a href="#cb4-18"></a>to_swap <span class="ot">=</span> <span class="fu">data.frame</span>(</span>
<span id="cb4-19"><a href="#cb4-19"></a>  <span class="at">old_name =</span> <span class="fu">c</span>(<span class="st">"21P1"</span>, <span class="st">"17P5"</span>, <span class="st">"45P5"</span>, <span class="st">"28P1"</span>, <span class="st">"28P6"</span>),</span>
<span id="cb4-20"><a href="#cb4-20"></a>  <span class="at">new_name =</span> <span class="fu">c</span>(<span class="st">"21R40"</span>, <span class="st">"17R20"</span>, <span class="st">"45R38"</span>, <span class="st">"28R31"</span>, <span class="st">"28R1"</span>)</span>
<span id="cb4-21"><a href="#cb4-21"></a>)</span>
<span id="cb4-22"><a href="#cb4-22"></a></span>
<span id="cb4-23"><a href="#cb4-23"></a><span class="co"># import final clade list</span></span>
<span id="cb4-24"><a href="#cb4-24"></a>final_clades <span class="ot">=</span> <span class="fu">read.table</span>(</span>
<span id="cb4-25"><a href="#cb4-25"></a>  <span class="st">"./data/p02-04/final_clades_for_pub.txt"</span>,</span>
<span id="cb4-26"><a href="#cb4-26"></a>  <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb4-27"><a href="#cb4-27"></a>  <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-28"><a href="#cb4-28"></a>  <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb4-29"><a href="#cb4-29"></a>)</span>
<span id="cb4-30"><a href="#cb4-30"></a></span>
<span id="cb4-31"><a href="#cb4-31"></a><span class="co"># load annotations</span></span>
<span id="cb4-32"><a href="#cb4-32"></a>tree_annot_raw <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">"./data/p01-03/genetrees.output.BS.annot-PETER.txt"</span>,</span>
<span id="cb4-33"><a href="#cb4-33"></a>                            <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb4-34"><a href="#cb4-34"></a>                            <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-35"><a href="#cb4-35"></a>                            <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb4-36"><a href="#cb4-36"></a>tree_annot_raw <span class="ot">=</span> <span class="fu">data.frame</span>(tree_annot_raw)</span>
<span id="cb4-37"><a href="#cb4-37"></a></span>
<span id="cb4-38"><a href="#cb4-38"></a><span class="cf">for</span>(k <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(to_swap)){</span>
<span id="cb4-39"><a href="#cb4-39"></a>  tree_annot_raw<span class="sc">$</span>Strain <span class="ot">=</span> stringr<span class="sc">::</span><span class="fu">str_replace</span>(</span>
<span id="cb4-40"><a href="#cb4-40"></a>    tree_annot_raw<span class="sc">$</span>Strain,</span>
<span id="cb4-41"><a href="#cb4-41"></a>    to_swap[k, <span class="st">"old_name"</span>],</span>
<span id="cb4-42"><a href="#cb4-42"></a>    to_swap[k, <span class="st">"new_name"</span>]</span>
<span id="cb4-43"><a href="#cb4-43"></a>  )</span>
<span id="cb4-44"><a href="#cb4-44"></a>  tree_annot_raw<span class="sc">$</span>Name <span class="ot">=</span> stringr<span class="sc">::</span><span class="fu">str_replace</span>(</span>
<span id="cb4-45"><a href="#cb4-45"></a>    tree_annot_raw<span class="sc">$</span>Name,</span>
<span id="cb4-46"><a href="#cb4-46"></a>    to_swap[k, <span class="st">"old_name"</span>],</span>
<span id="cb4-47"><a href="#cb4-47"></a>    to_swap[k, <span class="st">"new_name"</span>]</span>
<span id="cb4-48"><a href="#cb4-48"></a>  )</span>
<span id="cb4-49"><a href="#cb4-49"></a>}</span>
<span id="cb4-50"><a href="#cb4-50"></a></span>
<span id="cb4-51"><a href="#cb4-51"></a><span class="co"># merge with last clade grouping from 20250630</span></span>
<span id="cb4-52"><a href="#cb4-52"></a>tree_annot <span class="ot">=</span> tree_annot_raw <span class="sc">%&gt;%</span></span>
<span id="cb4-53"><a href="#cb4-53"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(., final_clades, <span class="at">by =</span> <span class="st">"Strain"</span>)</span>
<span id="cb4-54"><a href="#cb4-54"></a>tree_annot<span class="sc">$</span>Clade <span class="ot">=</span> <span class="fu">ifelse</span>(tree_annot<span class="sc">$</span>niche1 <span class="sc">==</span> <span class="st">"farmhouse"</span>, <span class="st">"Farmhouse"</span>, tree_annot<span class="sc">$</span>Clade)</span>
<span id="cb4-55"><a href="#cb4-55"></a>tree_annot<span class="sc">$</span>Clade <span class="ot">=</span> <span class="fu">ifelse</span>(</span>
<span id="cb4-56"><a href="#cb4-56"></a>  <span class="fu">is.na</span>(tree_annot<span class="sc">$</span>Clade) <span class="sc">&amp;</span> tree_annot<span class="sc">$</span>group <span class="sc">==</span> <span class="st">'beer2'</span>,</span>
<span id="cb4-57"><a href="#cb4-57"></a>  <span class="st">"Beer2"</span>, tree_annot<span class="sc">$</span>Clade</span>
<span id="cb4-58"><a href="#cb4-58"></a>)</span>
<span id="cb4-59"><a href="#cb4-59"></a>tree_annot<span class="sc">$</span>Clade <span class="ot">=</span> <span class="fu">ifelse</span>(</span>
<span id="cb4-60"><a href="#cb4-60"></a>  <span class="fu">is.na</span>(tree_annot<span class="sc">$</span>Clade) <span class="sc">&amp;</span> tree_annot<span class="sc">$</span>group <span class="sc">==</span> <span class="st">'wine'</span>,</span>
<span id="cb4-61"><a href="#cb4-61"></a>  <span class="st">"Wine"</span>, tree_annot<span class="sc">$</span>Clade</span>
<span id="cb4-62"><a href="#cb4-62"></a>)</span>
<span id="cb4-63"><a href="#cb4-63"></a>tree_annot<span class="sc">$</span>Clade <span class="ot">=</span> <span class="fu">ifelse</span>(</span>
<span id="cb4-64"><a href="#cb4-64"></a>  <span class="fu">is.na</span>(tree_annot<span class="sc">$</span>Clade) <span class="sc">&amp;</span> tree_annot<span class="sc">$</span>group <span class="sc">==</span> <span class="st">'asian'</span>,</span>
<span id="cb4-65"><a href="#cb4-65"></a>  <span class="st">"Asia"</span>, tree_annot<span class="sc">$</span>Clade</span>
<span id="cb4-66"><a href="#cb4-66"></a>)</span>
<span id="cb4-67"><a href="#cb4-67"></a>tree_annot<span class="sc">$</span>Clade <span class="ot">=</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(tree_annot<span class="sc">$</span>Clade), <span class="st">"Other"</span>, tree_annot<span class="sc">$</span>Clade)</span>
<span id="cb4-68"><a href="#cb4-68"></a>tree_annot <span class="ot">=</span> tree_annot[, <span class="sc">-</span><span class="fu">which</span>(<span class="fu">colnames</span>(tree_annot) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"niche1"</span>, <span class="st">"niche2"</span>, <span class="st">"niche3"</span>, <span class="st">"group"</span>, <span class="st">"seq_name"</span>, <span class="st">"Mosaic"</span>))]</span>
<span id="cb4-69"><a href="#cb4-69"></a><span class="fu">colnames</span>(tree_annot) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Strain"</span>, <span class="st">"ID"</span>, <span class="st">"Clade"</span>)</span>
<span id="cb4-70"><a href="#cb4-70"></a>tree_annot<span class="sc">$</span>ID <span class="ot">=</span> stringr<span class="sc">::</span><span class="fu">str_replace_all</span>(tree_annot<span class="sc">$</span>ID, <span class="st">"x"</span>, <span class="st">"X"</span>)</span>
<span id="cb4-71"><a href="#cb4-71"></a></span>
<span id="cb4-72"><a href="#cb4-72"></a></span>
<span id="cb4-73"><a href="#cb4-73"></a><span class="co"># merge</span></span>
<span id="cb4-74"><a href="#cb4-74"></a>kveik_het <span class="ot">=</span> <span class="fu">merge</span>(kveik_het, tree_annot, <span class="at">by =</span> <span class="st">"ID"</span>)</span>
<span id="cb4-75"><a href="#cb4-75"></a>kveik_het <span class="ot">=</span> kveik_het[, <span class="fu">which</span>(<span class="fu">colnames</span>(kveik_het) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"ID"</span>, <span class="st">"HET_CT"</span>, <span class="st">"Pop"</span>, <span class="st">"Culture"</span>, <span class="st">"Subpopulation"</span>, <span class="st">"Geographic.origin"</span>, <span class="st">"Clade"</span>))]</span>
<span id="cb4-76"><a href="#cb4-76"></a></span>
<span id="cb4-77"><a href="#cb4-77"></a></span>
<span id="cb4-78"><a href="#cb4-78"></a></span>
<span id="cb4-79"><a href="#cb4-79"></a><span class="co">#------------------------------------------------------------------------------#</span></span>
<span id="cb4-80"><a href="#cb4-80"></a><span class="do">## 2.2 - Heterozigous position - all                                        ####</span></span>
<span id="cb4-81"><a href="#cb4-81"></a></span>
<span id="cb4-82"><a href="#cb4-82"></a><span class="co"># t-test</span></span>
<span id="cb4-83"><a href="#cb4-83"></a>tests <span class="ot">=</span> kveik_het <span class="sc">%&gt;%</span></span>
<span id="cb4-84"><a href="#cb4-84"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(ID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"40R20"</span>, <span class="st">"40R1"</span>, <span class="st">"40R14"</span>, <span class="st">"38R16"</span>, <span class="st">"k7R25"</span>, <span class="st">"Muri"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb4-85"><a href="#cb4-85"></a>  rstatix<span class="sc">::</span><span class="fu">t_test</span>(HET_CT <span class="sc">~</span> Clade, <span class="at">ref.group =</span> <span class="st">"Farmhouse"</span>)</span>
<span id="cb4-86"><a href="#cb4-86"></a></span>
<span id="cb4-87"><a href="#cb4-87"></a>kveik_het<span class="sc">$</span>Clade <span class="ot">=</span> <span class="fu">factor</span>(kveik_het<span class="sc">$</span>Clade,</span>
<span id="cb4-88"><a href="#cb4-88"></a>  <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Farmhouse"</span>, <span class="st">"Asia"</span>, <span class="st">"Beer1"</span>, <span class="st">"Beer2"</span>, <span class="st">"Mixed"</span>, <span class="st">"Other"</span>, <span class="st">"Wine"</span>))</span>
<span id="cb4-89"><a href="#cb4-89"></a>kveik_het<span class="sc">$</span>Clade <span class="ot">=</span> <span class="fu">relevel</span>(kveik_het<span class="sc">$</span>Clade, <span class="st">"Farmhouse"</span>)</span>
<span id="cb4-90"><a href="#cb4-90"></a></span>
<span id="cb4-91"><a href="#cb4-91"></a><span class="co"># plot</span></span>
<span id="cb4-92"><a href="#cb4-92"></a>p_het <span class="ot">=</span> <span class="fu">ggplot</span>(kveik_het, <span class="fu">aes</span>(<span class="at">x =</span> Clade, <span class="at">y =</span> HET_CT)) <span class="sc">+</span></span>
<span id="cb4-93"><a href="#cb4-93"></a>  <span class="fu">geom_violin</span>(<span class="fu">aes</span>(<span class="at">fill =</span> Clade), <span class="at">alpha =</span> <span class="fl">0.2</span>,</span>
<span id="cb4-94"><a href="#cb4-94"></a>              <span class="at">scale =</span> <span class="st">"width"</span>,</span>
<span id="cb4-95"><a href="#cb4-95"></a>              <span class="at">trim =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb4-96"><a href="#cb4-96"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> <span class="sc">~</span>dplyr<span class="sc">::</span><span class="fu">filter</span>(.x, <span class="sc">!</span>(ID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"40R20"</span>, <span class="st">"40R1"</span>, <span class="st">"40R14"</span>, <span class="st">"38R16"</span>, <span class="st">"k7R25"</span>, <span class="st">"Muri"</span>))),</span>
<span id="cb4-97"><a href="#cb4-97"></a>             <span class="fu">aes</span>(<span class="at">fill =</span> Clade),</span>
<span id="cb4-98"><a href="#cb4-98"></a>             <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.25</span>),</span>
<span id="cb4-99"><a href="#cb4-99"></a>             <span class="at">shape =</span> <span class="dv">21</span>, </span>
<span id="cb4-100"><a href="#cb4-100"></a>             <span class="at">alpha =</span> <span class="fl">0.375</span>,</span>
<span id="cb4-101"><a href="#cb4-101"></a>             <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb4-102"><a href="#cb4-102"></a>  <span class="fu">stat_summary</span>(<span class="at">fun =</span> <span class="st">"median"</span>, <span class="at">colour =</span> <span class="st">"red"</span>, <span class="at">geom =</span> <span class="st">"crossbar"</span>) <span class="sc">+</span></span>
<span id="cb4-103"><a href="#cb4-103"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb4-104"><a href="#cb4-104"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> <span class="sc">~</span>dplyr<span class="sc">::</span><span class="fu">filter</span>(.x, ID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"40R20"</span>, <span class="st">"40R1"</span>, <span class="st">"40R14"</span>, <span class="st">"38R16"</span>, <span class="st">"k7R25"</span>, <span class="st">"Muri"</span>)),</span>
<span id="cb4-105"><a href="#cb4-105"></a>             <span class="fu">aes</span>(<span class="at">fill =</span> <span class="st">"red"</span>),</span>
<span id="cb4-106"><a href="#cb4-106"></a>             <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.25</span>),</span>
<span id="cb4-107"><a href="#cb4-107"></a>             <span class="at">shape =</span> <span class="dv">21</span>, </span>
<span id="cb4-108"><a href="#cb4-108"></a>             <span class="at">alpha =</span> <span class="fl">0.375</span>,</span>
<span id="cb4-109"><a href="#cb4-109"></a>             <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb4-110"><a href="#cb4-110"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> <span class="sc">~</span>dplyr<span class="sc">::</span><span class="fu">filter</span>(.x, ID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"40R20"</span>, <span class="st">"40R1"</span>, <span class="st">"40R14"</span>, <span class="st">"38R16"</span>, <span class="st">"k7R25"</span>, <span class="st">"Muri"</span>)),</span>
<span id="cb4-111"><a href="#cb4-111"></a>            <span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">as.character</span>(ID)),</span>
<span id="cb4-112"><a href="#cb4-112"></a>            <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">vjust =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb4-113"><a href="#cb4-113"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#2297e6"</span>, <span class="st">"#df536b"</span>, <span class="st">"#61d04f"</span>, <span class="st">"#28e2e5"</span>, <span class="st">"#cd0bbc"</span>, <span class="st">"grey75"</span>, <span class="st">"#bcf60c"</span>,</span>
<span id="cb4-114"><a href="#cb4-114"></a>                               <span class="st">"red"</span>)) <span class="sc">+</span></span>
<span id="cb4-115"><a href="#cb4-115"></a>  ggpubr<span class="sc">::</span><span class="fu">stat_compare_means</span>(<span class="at">method =</span> <span class="st">"anova"</span>, <span class="at">label.x =</span> <span class="fl">1.1</span>, <span class="at">label.y =</span> <span class="dv">98000</span>) <span class="sc">+</span></span>
<span id="cb4-116"><a href="#cb4-116"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">2</span>,  <span class="at">y =</span> <span class="dv">95000</span>, <span class="at">label =</span> <span class="fu">c</span>(tests<span class="sc">$</span>p.adj.signif)[[<span class="dv">1</span>]]) <span class="sc">+</span></span>
<span id="cb4-117"><a href="#cb4-117"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">3</span>,  <span class="at">y =</span> <span class="dv">95000</span>, <span class="at">label =</span> <span class="fu">c</span>(tests<span class="sc">$</span>p.adj.signif)[[<span class="dv">2</span>]]) <span class="sc">+</span></span>
<span id="cb4-118"><a href="#cb4-118"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">4</span>,  <span class="at">y =</span> <span class="dv">95000</span>, <span class="at">label =</span> <span class="fu">c</span>(tests<span class="sc">$</span>p.adj.signif)[[<span class="dv">3</span>]]) <span class="sc">+</span></span>
<span id="cb4-119"><a href="#cb4-119"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">5</span>,  <span class="at">y =</span> <span class="dv">95000</span>, <span class="at">label =</span> <span class="fu">c</span>(tests<span class="sc">$</span>p.adj.signif)[[<span class="dv">4</span>]]) <span class="sc">+</span></span>
<span id="cb4-120"><a href="#cb4-120"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">6</span>,  <span class="at">y =</span> <span class="dv">95000</span>, <span class="at">label =</span> <span class="fu">c</span>(tests<span class="sc">$</span>p.adj.signif)[[<span class="dv">5</span>]]) <span class="sc">+</span></span>
<span id="cb4-121"><a href="#cb4-121"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">7</span>,  <span class="at">y =</span> <span class="dv">95000</span>, <span class="at">label =</span> <span class="fu">c</span>(tests<span class="sc">$</span>p.adj.signif)[[<span class="dv">6</span>]]) <span class="sc">+</span></span>
<span id="cb4-122"><a href="#cb4-122"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> scales<span class="sc">::</span>comma,</span>
<span id="cb4-123"><a href="#cb4-123"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">100000</span>)) <span class="sc">+</span></span>
<span id="cb4-124"><a href="#cb4-124"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Heterozygosity in yeast populations"</span>,</span>
<span id="cb4-125"><a href="#cb4-125"></a>       <span class="at">x =</span> <span class="st">"Clade / Niche"</span>,</span>
<span id="cb4-126"><a href="#cb4-126"></a>       <span class="at">y =</span> <span class="st">"# heterozygous</span><span class="sc">\n</span><span class="st">positions"</span>) <span class="sc">+</span></span>
<span id="cb4-127"><a href="#cb4-127"></a>  <span class="fu">theme</span>(<span class="at">title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-128"><a href="#cb4-128"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.95</span>, <span class="at">angle =</span> <span class="dv">60</span>, <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb4-129"><a href="#cb4-129"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb4-130"><a href="#cb4-130"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb4-131"><a href="#cb4-131"></a>        <span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb4-132"><a href="#cb4-132"></a>        <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">fill =</span> <span class="cn">NA</span>),</span>
<span id="cb4-133"><a href="#cb4-133"></a>        <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-134"><a href="#cb4-134"></a>        <span class="at">panel.grid.minor.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-135"><a href="#cb4-135"></a>        <span class="at">panel.grid.major.y =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">"grey75"</span>),</span>
<span id="cb4-136"><a href="#cb4-136"></a>        <span class="at">panel.grid.minor.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb4-137"><a href="#cb4-137"></a></span>
<span id="cb4-138"><a href="#cb4-138"></a></span>
<span id="cb4-139"><a href="#cb4-139"></a></span>
<span id="cb4-140"><a href="#cb4-140"></a><span class="co">#------------------------------------------------------------------------------#</span></span>
<span id="cb4-141"><a href="#cb4-141"></a><span class="do">## 2.3 - Heterozigous position - kveiks                                     ####</span></span>
<span id="cb4-142"><a href="#cb4-142"></a></span>
<span id="cb4-143"><a href="#cb4-143"></a><span class="co"># t-test</span></span>
<span id="cb4-144"><a href="#cb4-144"></a>kveik_comp <span class="ot">=</span> kveik_het[<span class="fu">which</span>(kveik_het<span class="sc">$</span>Pop <span class="sc">==</span> <span class="st">"Farmhouse"</span> <span class="sc">&amp;</span> kveik_het<span class="sc">$</span>Subpopulation <span class="sc">!=</span> <span class="st">"Beer1"</span>), ]</span>
<span id="cb4-145"><a href="#cb4-145"></a>kveik_comp<span class="sc">$</span>Subpopulation <span class="ot">=</span> <span class="st">"All Farmhouse"</span></span>
<span id="cb4-146"><a href="#cb4-146"></a>kveik_comp<span class="sc">$</span>Geographic.origin <span class="ot">=</span> <span class="st">"All Farmhouse"</span></span>
<span id="cb4-147"><a href="#cb4-147"></a><span class="co">#kveik_comp$Culture = "All Farmhouse"</span></span>
<span id="cb4-148"><a href="#cb4-148"></a>kveik_comp <span class="ot">=</span> <span class="fu">rbind</span>(kveik_comp, kveik_het[<span class="fu">which</span>(kveik_het<span class="sc">$</span>Pop <span class="sc">==</span> <span class="st">"Farmhouse"</span> <span class="sc">&amp;</span> kveik_het<span class="sc">$</span>Subpopulation <span class="sc">!=</span> <span class="st">"Beer1"</span>), ])</span>
<span id="cb4-149"><a href="#cb4-149"></a></span>
<span id="cb4-150"><a href="#cb4-150"></a>tests_geo <span class="ot">=</span> kveik_comp <span class="sc">%&gt;%</span> </span>
<span id="cb4-151"><a href="#cb4-151"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(ID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"40R20"</span>, <span class="st">"40R1"</span>, <span class="st">"40R14"</span>, <span class="st">"38R16"</span>, <span class="st">"k7R25"</span>, <span class="st">"Muri"</span>, <span class="st">"39R20"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb4-152"><a href="#cb4-152"></a>  rstatix<span class="sc">::</span><span class="fu">t_test</span>(HET_CT <span class="sc">~</span> Geographic.origin, <span class="at">ref.group =</span> <span class="st">"All Farmhouse"</span>)</span>
<span id="cb4-153"><a href="#cb4-153"></a></span>
<span id="cb4-154"><a href="#cb4-154"></a>tests_geo[<span class="dv">6</span>, ] <span class="ot">=</span> <span class="fu">list</span>(<span class="st">"HET_CT"</span>, <span class="st">"All Farmhouse"</span>, <span class="st">"Russia"</span>, <span class="dv">37</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="st">"ns"</span>)</span>
<span id="cb4-155"><a href="#cb4-155"></a></span>
<span id="cb4-156"><a href="#cb4-156"></a><span class="co"># plot</span></span>
<span id="cb4-157"><a href="#cb4-157"></a>kveik_comp <span class="ot">=</span> kveik_het[<span class="fu">which</span>(kveik_het<span class="sc">$</span>Pop <span class="sc">==</span> <span class="st">"Farmhouse"</span> <span class="sc">&amp;</span> kveik_het<span class="sc">$</span>Subpopulation <span class="sc">!=</span> <span class="st">"Beer1"</span>), ]</span>
<span id="cb4-158"><a href="#cb4-158"></a></span>
<span id="cb4-159"><a href="#cb4-159"></a>p_het2 <span class="ot">=</span> <span class="fu">ggplot</span>(kveik_comp, <span class="fu">aes</span>(<span class="at">x =</span> Geographic.origin, <span class="at">y =</span> HET_CT)) <span class="sc">+</span></span>
<span id="cb4-160"><a href="#cb4-160"></a>  <span class="fu">geom_violin</span>(<span class="fu">aes</span>(<span class="at">fill =</span> Geographic.origin), <span class="at">alpha =</span> <span class="fl">0.2</span>,</span>
<span id="cb4-161"><a href="#cb4-161"></a>              <span class="at">scale =</span> <span class="st">"width"</span>,</span>
<span id="cb4-162"><a href="#cb4-162"></a>              <span class="at">trim =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb4-163"><a href="#cb4-163"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> <span class="sc">~</span>dplyr<span class="sc">::</span><span class="fu">filter</span>(.x, <span class="sc">!</span>(ID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"40R20"</span>, <span class="st">"40R1"</span>, <span class="st">"40R14"</span>, <span class="st">"38R16"</span>, <span class="st">"k7R25"</span>, <span class="st">"Muri"</span>))),</span>
<span id="cb4-164"><a href="#cb4-164"></a>             <span class="fu">aes</span>(<span class="at">fill =</span> Geographic.origin),</span>
<span id="cb4-165"><a href="#cb4-165"></a>             <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.25</span>),</span>
<span id="cb4-166"><a href="#cb4-166"></a>             <span class="at">shape =</span> <span class="dv">21</span>, </span>
<span id="cb4-167"><a href="#cb4-167"></a>             <span class="at">alpha =</span> <span class="fl">0.375</span>,</span>
<span id="cb4-168"><a href="#cb4-168"></a>             <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb4-169"><a href="#cb4-169"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> <span class="sc">~</span>dplyr<span class="sc">::</span><span class="fu">filter</span>(.x, ID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"40R20"</span>, <span class="st">"40R1"</span>, <span class="st">"40R14"</span>, <span class="st">"38R16"</span>, <span class="st">"k7R25"</span>, <span class="st">"Muri"</span>)),</span>
<span id="cb4-170"><a href="#cb4-170"></a>             <span class="fu">aes</span>(<span class="at">fill =</span> <span class="st">"red"</span>),</span>
<span id="cb4-171"><a href="#cb4-171"></a>             <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.25</span>),</span>
<span id="cb4-172"><a href="#cb4-172"></a>             <span class="at">shape =</span> <span class="dv">21</span>, </span>
<span id="cb4-173"><a href="#cb4-173"></a>             <span class="at">alpha =</span> <span class="fl">0.375</span>,</span>
<span id="cb4-174"><a href="#cb4-174"></a>             <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb4-175"><a href="#cb4-175"></a>  <span class="fu">stat_summary</span>(<span class="at">fun =</span> <span class="st">"median"</span>, <span class="at">colour =</span> <span class="st">"red"</span>, <span class="at">geom =</span> <span class="st">"crossbar"</span>) <span class="sc">+</span></span>
<span id="cb4-176"><a href="#cb4-176"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> <span class="sc">~</span>dplyr<span class="sc">::</span><span class="fu">filter</span>(.x, ID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"40R20"</span>, <span class="st">"40R1"</span>, <span class="st">"40R14"</span>, <span class="st">"38R16"</span>, <span class="st">"k7R25"</span>, <span class="st">"Muri"</span>)),</span>
<span id="cb4-177"><a href="#cb4-177"></a>            <span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">as.character</span>(ID)),</span>
<span id="cb4-178"><a href="#cb4-178"></a>            <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">vjust =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb4-179"><a href="#cb4-179"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"green"</span>, <span class="st">"#FFDA00"</span>, <span class="st">"orange"</span>, <span class="st">"#0571B0"</span>, <span class="st">"red"</span>, <span class="st">"#A6611A"</span>, <span class="st">"#92C5DE"</span>)) <span class="sc">+</span></span>
<span id="cb4-180"><a href="#cb4-180"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">label_wrap</span>(<span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb4-181"><a href="#cb4-181"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> scales<span class="sc">::</span>comma,</span>
<span id="cb4-182"><a href="#cb4-182"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">100000</span>)) <span class="sc">+</span></span>
<span id="cb4-183"><a href="#cb4-183"></a>  ggpubr<span class="sc">::</span><span class="fu">stat_compare_means</span>(<span class="at">method =</span> <span class="st">"anova"</span>, <span class="at">label.x =</span> <span class="fl">1.1</span>, <span class="at">label.y =</span> <span class="dv">98000</span>) <span class="sc">+</span></span>
<span id="cb4-184"><a href="#cb4-184"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">1</span>,  <span class="at">y =</span> <span class="dv">95000</span>, <span class="at">label =</span> <span class="fu">c</span>(tests_geo<span class="sc">$</span>p.adj.signif)[[<span class="dv">1</span>]]) <span class="sc">+</span></span>
<span id="cb4-185"><a href="#cb4-185"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">2</span>,  <span class="at">y =</span> <span class="dv">95000</span>, <span class="at">label =</span> <span class="fu">c</span>(tests_geo<span class="sc">$</span>p.adj.signif)[[<span class="dv">2</span>]]) <span class="sc">+</span></span>
<span id="cb4-186"><a href="#cb4-186"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">3</span>,  <span class="at">y =</span> <span class="dv">95000</span>, <span class="at">label =</span> <span class="fu">c</span>(tests_geo<span class="sc">$</span>p.adj.signif)[[<span class="dv">3</span>]]) <span class="sc">+</span></span>
<span id="cb4-187"><a href="#cb4-187"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">4</span>,  <span class="at">y =</span> <span class="dv">95000</span>, <span class="at">label =</span> <span class="fu">c</span>(tests_geo<span class="sc">$</span>p.adj.signif)[[<span class="dv">4</span>]]) <span class="sc">+</span></span>
<span id="cb4-188"><a href="#cb4-188"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">5</span>,  <span class="at">y =</span> <span class="dv">95000</span>, <span class="at">label =</span> <span class="fu">c</span>(tests_geo<span class="sc">$</span>p.adj.signif)[[<span class="dv">5</span>]]) <span class="sc">+</span></span>
<span id="cb4-189"><a href="#cb4-189"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">6</span>,  <span class="at">y =</span> <span class="dv">95000</span>, <span class="at">label =</span> <span class="fu">c</span>(tests_geo<span class="sc">$</span>p.adj.signif)[[<span class="dv">6</span>]]) <span class="sc">+</span></span>
<span id="cb4-190"><a href="#cb4-190"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Geographic origin"</span>) <span class="sc">+</span></span>
<span id="cb4-191"><a href="#cb4-191"></a>  <span class="fu">theme</span>(<span class="at">title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-192"><a href="#cb4-192"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.95</span>, <span class="at">angle =</span> <span class="dv">60</span>, <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb4-193"><a href="#cb4-193"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-194"><a href="#cb4-194"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb4-195"><a href="#cb4-195"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-196"><a href="#cb4-196"></a>        <span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb4-197"><a href="#cb4-197"></a>        <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">fill =</span> <span class="cn">NA</span>),</span>
<span id="cb4-198"><a href="#cb4-198"></a>        <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-199"><a href="#cb4-199"></a>        <span class="at">panel.grid.minor.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-200"><a href="#cb4-200"></a>        <span class="at">panel.grid.major.y =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">"grey75"</span>),</span>
<span id="cb4-201"><a href="#cb4-201"></a>        <span class="at">panel.grid.minor.y =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>ggpubr<span class="sc">::</span><span class="fu">ggarrange</span>(</span>
<span id="cb5-2"><a href="#cb5-2"></a>  p_het, p_het2,</span>
<span id="cb5-3"><a href="#cb5-3"></a>  <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">1</span>,</span>
<span id="cb5-4"><a href="#cb5-4"></a>  <span class="at">align =</span> <span class="st">"h"</span>,</span>
<span id="cb5-5"><a href="#cb5-5"></a>  <span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">0.5</span>)</span>
<span id="cb5-6"><a href="#cb5-6"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="p01-03-variant-calling_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption><span style="color: steelblue;">Figure 1: Boxplot depicting total number of heterozygous sites per genetic clade (right) and geographical origin of farmhouse yeast (right).</span></figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="loss-of-heterozygosity-loh" class="level2" data-number="3.4">
<h2 data-number="3.4" class="anchored" data-anchor-id="loss-of-heterozygosity-loh"><span class="header-section-number">3.4</span> Loss Of Heterozygosity (LOH)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="co"># import LOH matrix</span></span>
<span id="cb6-2"><a href="#cb6-2"></a>LOH_table_10kb <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="st">"./data/p01-03/Vikings.LOH.10kb_win.tab"</span>, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="fu">colnames</span>(LOH_table_10kb) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"strain"</span>, <span class="st">"chr"</span>, <span class="st">"start"</span>, <span class="st">"stop"</span>, <span class="st">"n_var"</span>)</span>
<span id="cb6-4"><a href="#cb6-4"></a></span>
<span id="cb6-5"><a href="#cb6-5"></a><span class="co"># format</span></span>
<span id="cb6-6"><a href="#cb6-6"></a>LOH_table_10kb<span class="sc">$</span>chr <span class="ot">=</span> <span class="fu">factor</span>(</span>
<span id="cb6-7"><a href="#cb6-7"></a>  LOH_table_10kb<span class="sc">$</span>chr,</span>
<span id="cb6-8"><a href="#cb6-8"></a>  <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"I"</span>, <span class="st">"II"</span>, <span class="st">"III"</span>, <span class="st">"IV"</span>, <span class="st">"V"</span>, <span class="st">"VI"</span>, <span class="st">"VII"</span>, <span class="st">"VIII"</span>,</span>
<span id="cb6-9"><a href="#cb6-9"></a>             <span class="st">"IX"</span>, <span class="st">"X"</span>, <span class="st">"XI"</span>, <span class="st">"XII"</span>, <span class="st">"XIII"</span>, <span class="st">"XIV"</span>, <span class="st">"XV"</span>, <span class="st">"XVI"</span>)</span>
<span id="cb6-10"><a href="#cb6-10"></a>)</span>
<span id="cb6-11"><a href="#cb6-11"></a></span>
<span id="cb6-12"><a href="#cb6-12"></a><span class="co"># relevel</span></span>
<span id="cb6-13"><a href="#cb6-13"></a>LOH_table_10kb <span class="ot">=</span> LOH_table_10kb[<span class="fu">which</span>(LOH_table_10kb<span class="sc">$</span>strain <span class="sc">%in%</span> <span class="fu">c</span>(</span>
<span id="cb6-14"><a href="#cb6-14"></a>  <span class="st">"41R10"</span>, <span class="st">"21R38"</span>, <span class="st">"9R40"</span>, <span class="st">"17P5"</span>, <span class="st">"SortdalEbbe1"</span>, <span class="st">"3R11"</span>, <span class="st">"21P1"</span>, <span class="st">"41R15"</span>, <span class="st">"Hornindal1"</span>,</span>
<span id="cb6-15"><a href="#cb6-15"></a>  <span class="st">"Hornindal2"</span>, <span class="st">"1R16"</span>, <span class="st">"2R23"</span>, <span class="st">"8R19"</span>, <span class="st">"Muri"</span>,</span>
<span id="cb6-16"><a href="#cb6-16"></a>  <span class="st">"k7R25"</span>, <span class="st">"38R16"</span>, <span class="st">"44R32"</span>, <span class="st">"19R18"</span>, <span class="st">"44R7"</span>, <span class="st">"6R15"</span>, <span class="st">"Laerdal2"</span>, <span class="st">"7R7"</span>, <span class="st">"14R6"</span>, <span class="st">"14R30"</span>,</span>
<span id="cb6-17"><a href="#cb6-17"></a>  <span class="st">"27R17"</span>, <span class="st">"28P1"</span>, <span class="st">"28P6"</span>, <span class="st">"28R21"</span>, <span class="st">"28R33"</span>, <span class="st">"28R8"</span>,</span>
<span id="cb6-18"><a href="#cb6-18"></a>  <span class="st">"42R20"</span>, <span class="st">"42R31"</span>, <span class="st">"45P5"</span>, <span class="st">"45R11"</span>, </span>
<span id="cb6-19"><a href="#cb6-19"></a>  <span class="st">"46R12"</span>, <span class="st">"46R37"</span>, <span class="st">"16R23"</span>, <span class="st">"16R37"</span>,</span>
<span id="cb6-20"><a href="#cb6-20"></a>  <span class="st">"39R20"</span>, <span class="st">"40R14"</span>, <span class="st">"40R1"</span>, <span class="st">"40R20"</span>,</span>
<span id="cb6-21"><a href="#cb6-21"></a>  <span class="st">"Granvin1"</span>, <span class="st">"Voss1"</span></span>
<span id="cb6-22"><a href="#cb6-22"></a>)), ]</span>
<span id="cb6-23"><a href="#cb6-23"></a></span>
<span id="cb6-24"><a href="#cb6-24"></a>LOH_table_10kb<span class="sc">$</span>strain <span class="ot">=</span> <span class="fu">factor</span>(</span>
<span id="cb6-25"><a href="#cb6-25"></a>  LOH_table_10kb<span class="sc">$</span>strain,</span>
<span id="cb6-26"><a href="#cb6-26"></a>  <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"41R10"</span>, <span class="st">"21R38"</span>, <span class="st">"9R40"</span>, <span class="st">"17P5"</span>, <span class="st">"SortdalEbbe1"</span>, <span class="st">"3R11"</span>, <span class="st">"21P1"</span>, <span class="st">"41R15"</span>, <span class="st">"Hornindal1"</span>,</span>
<span id="cb6-27"><a href="#cb6-27"></a>             <span class="st">"Hornindal2"</span>, <span class="st">"1R16"</span>, <span class="st">"2R23"</span>, <span class="st">"8R19"</span>, <span class="st">"Muri"</span>,</span>
<span id="cb6-28"><a href="#cb6-28"></a>             <span class="st">"k7R25"</span>, <span class="st">"38R16"</span>, <span class="st">"44R32"</span>, <span class="st">"19R18"</span>, <span class="st">"44R7"</span>, <span class="st">"6R15"</span>, <span class="st">"Laerdal2"</span>, <span class="st">"7R7"</span>, <span class="st">"14R6"</span>, <span class="st">"14R30"</span>,</span>
<span id="cb6-29"><a href="#cb6-29"></a>             <span class="st">"27R17"</span>, <span class="st">"28P1"</span>, <span class="st">"28P6"</span>, <span class="st">"28R21"</span>, <span class="st">"28R33"</span>, <span class="st">"28R8"</span>,</span>
<span id="cb6-30"><a href="#cb6-30"></a>             <span class="st">"42R20"</span>, <span class="st">"42R31"</span>, <span class="st">"45P5"</span>, <span class="st">"45R11"</span>, </span>
<span id="cb6-31"><a href="#cb6-31"></a>             <span class="st">"46R12"</span>, <span class="st">"46R37"</span>, <span class="st">"16R23"</span>, <span class="st">"16R37"</span>,</span>
<span id="cb6-32"><a href="#cb6-32"></a>             <span class="st">"39R20"</span>, <span class="st">"40R14"</span>, <span class="st">"40R1"</span>, <span class="st">"40R20"</span>,</span>
<span id="cb6-33"><a href="#cb6-33"></a>             <span class="st">"Granvin1"</span>, <span class="st">"Voss1"</span>)</span>
<span id="cb6-34"><a href="#cb6-34"></a>)</span>
<span id="cb6-35"><a href="#cb6-35"></a></span>
<span id="cb6-36"><a href="#cb6-36"></a></span>
<span id="cb6-37"><a href="#cb6-37"></a><span class="co"># import coverage dataset</span></span>
<span id="cb6-38"><a href="#cb6-38"></a>file_list <span class="ot">=</span> <span class="fu">list.files</span>(</span>
<span id="cb6-39"><a href="#cb6-39"></a>  <span class="at">path =</span> <span class="st">"./data/p01-03/01_start_bed/"</span>,</span>
<span id="cb6-40"><a href="#cb6-40"></a>  <span class="at">pattern =</span> <span class="st">"1kb_cov.bed"</span>,</span>
<span id="cb6-41"><a href="#cb6-41"></a>  <span class="at">full.names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb6-42"><a href="#cb6-42"></a>  <span class="at">ignore.case =</span> <span class="cn">FALSE</span></span>
<span id="cb6-43"><a href="#cb6-43"></a>)</span>
<span id="cb6-44"><a href="#cb6-44"></a></span>
<span id="cb6-45"><a href="#cb6-45"></a><span class="co"># declare average df</span></span>
<span id="cb6-46"><a href="#cb6-46"></a>average_table <span class="ot">=</span> <span class="fu">data.frame</span>(</span>
<span id="cb6-47"><a href="#cb6-47"></a>  <span class="at">strain =</span> <span class="fu">character</span>(),</span>
<span id="cb6-48"><a href="#cb6-48"></a>  <span class="at">average =</span> <span class="fu">double</span>(),</span>
<span id="cb6-49"><a href="#cb6-49"></a>  <span class="at">max =</span> <span class="fu">double</span>(),</span>
<span id="cb6-50"><a href="#cb6-50"></a>  <span class="at">sd =</span> <span class="fu">double</span>(),</span>
<span id="cb6-51"><a href="#cb6-51"></a>  <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb6-52"><a href="#cb6-52"></a>)</span>
<span id="cb6-53"><a href="#cb6-53"></a></span>
<span id="cb6-54"><a href="#cb6-54"></a><span class="co"># declare cov df</span></span>
<span id="cb6-55"><a href="#cb6-55"></a>coverage_table <span class="ot">=</span> <span class="fu">data.frame</span>(</span>
<span id="cb6-56"><a href="#cb6-56"></a>  <span class="at">strain =</span> <span class="fu">character</span>(),</span>
<span id="cb6-57"><a href="#cb6-57"></a>  <span class="at">chr =</span> <span class="fu">character</span>(),</span>
<span id="cb6-58"><a href="#cb6-58"></a>  <span class="at">start =</span> <span class="fu">integer</span>(),</span>
<span id="cb6-59"><a href="#cb6-59"></a>  <span class="at">stop =</span> <span class="fu">integer</span>(),</span>
<span id="cb6-60"><a href="#cb6-60"></a>  <span class="at">cov =</span> <span class="fu">double</span>(),</span>
<span id="cb6-61"><a href="#cb6-61"></a>  <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb6-62"><a href="#cb6-62"></a>)</span>
<span id="cb6-63"><a href="#cb6-63"></a></span>
<span id="cb6-64"><a href="#cb6-64"></a><span class="co"># import coverage counts per strain</span></span>
<span id="cb6-65"><a href="#cb6-65"></a><span class="cf">for</span>(file_to_import <span class="cf">in</span> file_list){</span>
<span id="cb6-66"><a href="#cb6-66"></a>  </span>
<span id="cb6-67"><a href="#cb6-67"></a>  tmp_table <span class="ot">=</span> <span class="fu">read.delim</span>(file_to_import, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb6-68"><a href="#cb6-68"></a>  </span>
<span id="cb6-69"><a href="#cb6-69"></a>  <span class="fu">colnames</span>(tmp_table) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"strain"</span>, <span class="st">"chr"</span>, <span class="st">"start"</span>, <span class="st">"stop"</span>, <span class="st">"cov"</span>)</span>
<span id="cb6-70"><a href="#cb6-70"></a>  tmp_table <span class="ot">=</span> tmp_table[<span class="fu">which</span>(tmp_table<span class="sc">$</span>chr <span class="sc">!=</span> <span class="st">"ref|NC_001224|"</span>), ]</span>
<span id="cb6-71"><a href="#cb6-71"></a>  </span>
<span id="cb6-72"><a href="#cb6-72"></a>  strain_name <span class="ot">=</span> stringr<span class="sc">::</span><span class="fu">str_remove</span>(file_to_import, <span class="st">".align.sort.1kb_cov.bed"</span>)</span>
<span id="cb6-73"><a href="#cb6-73"></a>  strain_name <span class="ot">=</span> stringr<span class="sc">::</span><span class="fu">str_remove</span>(strain_name, <span class="st">"./data/p01-03/01_start_bed//"</span>)</span>
<span id="cb6-74"><a href="#cb6-74"></a>  average_table <span class="ot">=</span> <span class="fu">rbind</span>(</span>
<span id="cb6-75"><a href="#cb6-75"></a>    average_table,</span>
<span id="cb6-76"><a href="#cb6-76"></a>    <span class="fu">data.frame</span>(</span>
<span id="cb6-77"><a href="#cb6-77"></a>      <span class="st">"strain"</span> <span class="ot">=</span> strain_name,</span>
<span id="cb6-78"><a href="#cb6-78"></a>      <span class="st">"average"</span> <span class="ot">=</span> <span class="fu">mean</span>(tmp_table<span class="sc">$</span>cov),</span>
<span id="cb6-79"><a href="#cb6-79"></a>      <span class="st">"max"</span> <span class="ot">=</span> <span class="fu">max</span>(tmp_table<span class="sc">$</span>cov),</span>
<span id="cb6-80"><a href="#cb6-80"></a>      <span class="st">"sd"</span> <span class="ot">=</span> <span class="fu">sd</span>(tmp_table<span class="sc">$</span>cov)</span>
<span id="cb6-81"><a href="#cb6-81"></a>    )</span>
<span id="cb6-82"><a href="#cb6-82"></a>  )</span>
<span id="cb6-83"><a href="#cb6-83"></a>  </span>
<span id="cb6-84"><a href="#cb6-84"></a>  tmp_table <span class="ot">=</span> tmp_table[<span class="fu">which</span>(tmp_table<span class="sc">$</span>cov <span class="sc">&lt;=</span> <span class="fu">mean</span>(tmp_table<span class="sc">$</span>cov)<span class="sc">+</span><span class="fu">sd</span>(tmp_table<span class="sc">$</span>cov)), ]</span>
<span id="cb6-85"><a href="#cb6-85"></a>  tmp_table<span class="sc">$</span>stop_y <span class="ot">=</span> <span class="cn">NA</span></span>
<span id="cb6-86"><a href="#cb6-86"></a>  </span>
<span id="cb6-87"><a href="#cb6-87"></a>  coverage_table <span class="ot">=</span> <span class="fu">rbind</span>(coverage_table, tmp_table)</span>
<span id="cb6-88"><a href="#cb6-88"></a>  </span>
<span id="cb6-89"><a href="#cb6-89"></a>  coverage_table[<span class="fu">which</span>(coverage_table<span class="sc">$</span>strain <span class="sc">==</span> strain_name), <span class="st">"stop_y"</span>] <span class="ot">=</span> <span class="fu">max</span>(coverage_table<span class="sc">$</span>cov)</span>
<span id="cb6-90"><a href="#cb6-90"></a>  </span>
<span id="cb6-91"><a href="#cb6-91"></a>}</span>
<span id="cb6-92"><a href="#cb6-92"></a></span>
<span id="cb6-93"><a href="#cb6-93"></a></span>
<span id="cb6-94"><a href="#cb6-94"></a><span class="co"># format</span></span>
<span id="cb6-95"><a href="#cb6-95"></a>coverage_table<span class="sc">$</span>chr <span class="ot">=</span> <span class="fu">factor</span>(</span>
<span id="cb6-96"><a href="#cb6-96"></a>  coverage_table<span class="sc">$</span>chr,</span>
<span id="cb6-97"><a href="#cb6-97"></a>  <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"I"</span>, <span class="st">"II"</span>, <span class="st">"III"</span>, <span class="st">"IV"</span>, <span class="st">"V"</span>, <span class="st">"VI"</span>, <span class="st">"VII"</span>, <span class="st">"VIII"</span>,</span>
<span id="cb6-98"><a href="#cb6-98"></a>             <span class="st">"IX"</span>, <span class="st">"X"</span>, <span class="st">"XI"</span>, <span class="st">"XII"</span>, <span class="st">"XIII"</span>, <span class="st">"XIV"</span>, <span class="st">"XV"</span>, <span class="st">"XVI"</span>)</span>
<span id="cb6-99"><a href="#cb6-99"></a>)</span>
<span id="cb6-100"><a href="#cb6-100"></a></span>
<span id="cb6-101"><a href="#cb6-101"></a><span class="co"># plot </span></span>
<span id="cb6-102"><a href="#cb6-102"></a>p_LOH1 <span class="ot">=</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb6-103"><a href="#cb6-103"></a>  <span class="fu">geom_rect</span>(<span class="at">data =</span> <span class="fu">subset</span>(</span>
<span id="cb6-104"><a href="#cb6-104"></a>    coverage_table,</span>
<span id="cb6-105"><a href="#cb6-105"></a>    chr <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"I"</span>, <span class="st">"III"</span>, <span class="st">"V"</span>, <span class="st">"VII"</span>, <span class="st">"IX"</span>, <span class="st">"XI"</span>, <span class="st">"XIII"</span>, <span class="st">"XV"</span>)),</span>
<span id="cb6-106"><a href="#cb6-106"></a>    <span class="at">fill =</span> <span class="st">"grey95"</span>,</span>
<span id="cb6-107"><a href="#cb6-107"></a>    <span class="at">xmin =</span> <span class="dv">0</span>, <span class="at">xmax =</span> <span class="fu">max</span>(coverage_table<span class="sc">$</span>stop),</span>
<span id="cb6-108"><a href="#cb6-108"></a>    <span class="at">ymin =</span> <span class="dv">0</span>, <span class="at">ymax =</span> <span class="fu">max</span>(coverage_table<span class="sc">$</span>cov),</span>
<span id="cb6-109"><a href="#cb6-109"></a>    <span class="at">alpha =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb6-110"><a href="#cb6-110"></a>  <span class="fu">geom_hline</span>(<span class="at">data =</span> average_table, </span>
<span id="cb6-111"><a href="#cb6-111"></a>             <span class="fu">aes</span>(<span class="at">yintercept =</span> average),</span>
<span id="cb6-112"><a href="#cb6-112"></a>             <span class="at">color =</span> <span class="st">"firebrick"</span>) <span class="sc">+</span></span>
<span id="cb6-113"><a href="#cb6-113"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> LOH_table_10kb,</span>
<span id="cb6-114"><a href="#cb6-114"></a>            <span class="fu">aes</span>(<span class="at">x =</span> start, <span class="at">y =</span> n_var),</span>
<span id="cb6-115"><a href="#cb6-115"></a>            <span class="at">color =</span> <span class="st">"royalblue4"</span>,</span>
<span id="cb6-116"><a href="#cb6-116"></a>            <span class="at">linewidth =</span> <span class="fl">0.25</span>) <span class="sc">+</span></span>
<span id="cb6-117"><a href="#cb6-117"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"steelblue"</span>, <span class="st">"white"</span>, <span class="st">"salmon"</span>, <span class="st">"grey95"</span>)) <span class="sc">+</span></span>
<span id="cb6-118"><a href="#cb6-118"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> <span class="cf">function</span>(x) <span class="fu">format</span>(x, <span class="at">big.mark =</span> <span class="st">","</span>, <span class="at">scientific =</span> <span class="cn">FALSE</span>)) <span class="sc">+</span></span>
<span id="cb6-119"><a href="#cb6-119"></a>  <span class="co">#scale_y_continuous(limits = c(0, 350)) +</span></span>
<span id="cb6-120"><a href="#cb6-120"></a>  <span class="fu">coord_cartesian</span>(<span class="at">expand =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb6-121"><a href="#cb6-121"></a>  <span class="fu">annotation_custom</span>(grid<span class="sc">::</span><span class="fu">linesGrob</span>(<span class="at">y =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">gp =</span> grid<span class="sc">::</span><span class="fu">gpar</span>(<span class="at">lwd =</span> <span class="dv">3</span>))) <span class="sc">+</span></span>
<span id="cb6-122"><a href="#cb6-122"></a>  <span class="fu">facet_grid2</span>(strain <span class="sc">~</span> chr,</span>
<span id="cb6-123"><a href="#cb6-123"></a>              <span class="at">scales =</span> <span class="st">"free"</span>,</span>
<span id="cb6-124"><a href="#cb6-124"></a>              <span class="at">space =</span> <span class="st">"free_x"</span>,</span>
<span id="cb6-125"><a href="#cb6-125"></a>              <span class="at">switch =</span> <span class="st">"both"</span>,</span>
<span id="cb6-126"><a href="#cb6-126"></a>              <span class="at">strip =</span> <span class="fu">strip_themed</span>(</span>
<span id="cb6-127"><a href="#cb6-127"></a>                <span class="at">background_y =</span> <span class="fu">elem_list_rect</span>(<span class="at">fill =</span> <span class="fu">c</span>(</span>
<span id="cb6-128"><a href="#cb6-128"></a>                  <span class="st">"#0571B0"</span>, <span class="st">"#0571B0"</span>, <span class="st">"#0571B0"</span>, <span class="st">"#0571B0"</span>, <span class="st">"#0571B0"</span>, <span class="st">"#0571B0"</span>, <span class="st">"#0571B0"</span>, <span class="st">"#0571B0"</span>,</span>
<span id="cb6-129"><a href="#cb6-129"></a>                  <span class="st">"#0571B0"</span>, <span class="st">"#0571B0"</span>, <span class="st">"#0571B0"</span>, <span class="st">"#0571B0"</span>, <span class="st">"#0571B0"</span>, <span class="st">"#0571B0"</span>,</span>
<span id="cb6-130"><a href="#cb6-130"></a>                  <span class="st">"#92C5DE"</span>, <span class="st">"#92C5DE"</span>,</span>
<span id="cb6-131"><a href="#cb6-131"></a>                  <span class="st">"#92C5DE"</span>, <span class="st">"#92C5DE"</span>, <span class="st">"#92C5DE"</span>, <span class="st">"#92C5DE"</span>, <span class="st">"#92C5DE"</span>, <span class="st">"#92C5DE"</span>, <span class="st">"#92C5DE"</span>, <span class="st">"#92C5DE"</span>,</span>
<span id="cb6-132"><a href="#cb6-132"></a>                  <span class="st">"#008470"</span>, <span class="st">"#008470"</span>, <span class="st">"#008470"</span>, <span class="st">"#008470"</span>, <span class="st">"#008470"</span>, <span class="st">"#008470"</span>, <span class="st">"#FFDA00"</span>, <span class="st">"#FFDA00"</span>,</span>
<span id="cb6-133"><a href="#cb6-133"></a>                  <span class="st">"#FFDA00"</span>, <span class="st">"#FFDA00"</span>, <span class="st">"#FBA01D"</span>, <span class="st">"#FBA01D"</span>, <span class="st">"#FBA01D"</span>, <span class="st">"#FBA01D"</span>, <span class="st">"#A6611A"</span>, <span class="st">"#A6611A"</span>,</span>
<span id="cb6-134"><a href="#cb6-134"></a>                  <span class="st">"#A6611A"</span>, <span class="st">"#A6611A"</span>, <span class="st">"#51CD34"</span>, <span class="st">"#51CD34"</span></span>
<span id="cb6-135"><a href="#cb6-135"></a>                ))</span>
<span id="cb6-136"><a href="#cb6-136"></a>              )) <span class="sc">+</span></span>
<span id="cb6-137"><a href="#cb6-137"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-138"><a href="#cb6-138"></a>        <span class="at">axis.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-139"><a href="#cb6-139"></a>        <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-140"><a href="#cb6-140"></a>        <span class="at">axis.text =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-141"><a href="#cb6-141"></a>        <span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb6-142"><a href="#cb6-142"></a>        <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-143"><a href="#cb6-143"></a>        <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-144"><a href="#cb6-144"></a>        <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="dv">0</span>, <span class="st">"lines"</span>),</span>
<span id="cb6-145"><a href="#cb6-145"></a>        <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-146"><a href="#cb6-146"></a>        <span class="at">strip.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb6-147"><a href="#cb6-147"></a>        <span class="at">strip.text.y.left =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">angle =</span> <span class="dv">0</span>, <span class="at">face =</span> <span class="st">"bold"</span>))</span>
<span id="cb6-148"><a href="#cb6-148"></a></span>
<span id="cb6-149"><a href="#cb6-149"></a>p_LOH1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="p01-03-variant-calling_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption><span style="color: steelblue;">Figure 2: Loss Of Heterozygotisy events across the genomes of the sequenced Farmhouse yeasts.</span></figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="search-for-signature-of-culture-28-hybridization" class="level2" data-number="3.5">
<h2 data-number="3.5" class="anchored" data-anchor-id="search-for-signature-of-culture-28-hybridization"><span class="header-section-number">3.5</span> Search for signature of Culture #28 hybridization</h2>
<p>We look for variants in common between strains</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1"></a><span class="co"># split vcf in samples</span></span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="cf">for</span> sample <span class="kw">in</span> <span class="kw">`</span><span class="ex">bcftools</span> query <span class="at">-l</span> kveiks.fltr.CE-Norway.vcf<span class="kw">`;</span> <span class="cf">do</span></span>
<span id="cb7-3"><a href="#cb7-3"></a>  <span class="ex">bcftools</span> view <span class="at">-c1</span> <span class="at">-Oz</span> <span class="at">-s</span> <span class="va">$sample</span> <span class="at">-o</span> kveiks.fltr.CE-Norway.<span class="va">$sample</span>.vcf kveiks.fltr.CE-Norway.vcf <span class="kw">&amp;</span></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="cf">done</span></span>
<span id="cb7-5"><a href="#cb7-5"></a></span>
<span id="cb7-6"><a href="#cb7-6"></a><span class="co"># normalize and get positions Culture #28</span></span>
<span id="cb7-7"><a href="#cb7-7"></a><span class="cf">for</span> sample <span class="kw">in</span> 28P1 28P6 28R21 28R33 28R8<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb7-8"><a href="#cb7-8"></a>  <span class="ex">bcftools</span> norm <span class="at">-m-any</span> kveiks.fltr.CE-Norway.<span class="va">$sample</span>.vcf <span class="dt">\</span></span>
<span id="cb7-9"><a href="#cb7-9"></a>  <span class="kw">|</span> <span class="ex">bcftools</span> norm <span class="at">--check-ref</span> w <span class="at">-f</span> ../00_ref_genomes/Scer.fa <span class="dt">\</span></span>
<span id="cb7-10"><a href="#cb7-10"></a>  <span class="kw">|</span> <span class="ex">bcftools</span> annotate <span class="at">-Ob</span> <span class="at">-x</span> <span class="st">'ID'</span> <span class="at">-I</span> +<span class="st">'%CHROM:%POS:%REF:%ALT'</span> <span class="dt">\</span></span>
<span id="cb7-11"><a href="#cb7-11"></a>  <span class="op">&gt;</span> kveiks.fltr.CE-Norway.<span class="va">$sample</span>.bcf <span class="kw">;</span></span>
<span id="cb7-12"><a href="#cb7-12"></a>  <span class="ex">bcftools</span> index kveiks.fltr.CE-Norway.<span class="va">$sample</span>.bcf<span class="kw">;</span></span>
<span id="cb7-13"><a href="#cb7-13"></a>  <span class="ex">bcftools</span> view kveiks.fltr.CE-Norway.<span class="va">$sample</span>.bcf <span class="dt">\</span></span>
<span id="cb7-14"><a href="#cb7-14"></a>  <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f3</span> <span class="kw">|</span> <span class="fu">grep</span> <span class="at">-v</span> <span class="st">"^##"</span> <span class="kw">|</span> <span class="fu">grep</span> <span class="at">-v</span> <span class="st">"^ID"</span> <span class="dt">\</span></span>
<span id="cb7-15"><a href="#cb7-15"></a>  <span class="op">&gt;</span> kveiks.fltr.CE-Norway.<span class="va">$sample</span>.lst</span>
<span id="cb7-16"><a href="#cb7-16"></a><span class="cf">done</span></span>
<span id="cb7-17"><a href="#cb7-17"></a></span>
<span id="cb7-18"><a href="#cb7-18"></a><span class="co"># get Culture #16 positions</span></span>
<span id="cb7-19"><a href="#cb7-19"></a><span class="cf">for</span> sample <span class="kw">in</span> 16R23 16R37 46R12 46R37<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb7-20"><a href="#cb7-20"></a>  <span class="ex">bcftools</span> norm <span class="at">-m-any</span> kveiks.fltr.Lithuania.<span class="va">$sample</span>.vcf <span class="dt">\</span></span>
<span id="cb7-21"><a href="#cb7-21"></a>  <span class="kw">|</span> <span class="ex">bcftools</span> norm <span class="at">--check-ref</span> w <span class="at">-f</span> ../00_ref_genomes/Scer.fa <span class="dt">\</span></span>
<span id="cb7-22"><a href="#cb7-22"></a>  <span class="kw">|</span> <span class="ex">bcftools</span> annotate <span class="at">-Ob</span> <span class="at">-x</span> <span class="st">'ID'</span> <span class="at">-I</span> +<span class="st">'%CHROM:%POS:%REF:%ALT'</span> <span class="dt">\</span></span>
<span id="cb7-23"><a href="#cb7-23"></a>  <span class="op">&gt;</span> kveiks.fltr.Lithuania.<span class="va">$sample</span>.bcf <span class="kw">;</span></span>
<span id="cb7-24"><a href="#cb7-24"></a>  <span class="ex">bcftools</span> index kveiks.fltr.Lithuania.<span class="va">$sample</span>.bcf<span class="kw">;</span></span>
<span id="cb7-25"><a href="#cb7-25"></a>  <span class="ex">bcftools</span> view kveiks.fltr.Lithuania.<span class="va">$sample</span>.bcf <span class="dt">\</span></span>
<span id="cb7-26"><a href="#cb7-26"></a>  <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f3</span> <span class="kw">|</span> <span class="fu">grep</span> <span class="at">-v</span> <span class="st">"^##"</span> <span class="kw">|</span> <span class="fu">grep</span> <span class="at">-v</span> <span class="st">"^ID"</span> <span class="dt">\</span></span>
<span id="cb7-27"><a href="#cb7-27"></a>  <span class="op">&gt;</span> kveiks.fltr.Lithuania.<span class="va">$sample</span>.lst</span>
<span id="cb7-28"><a href="#cb7-28"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>list_28R31 <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="st">"./data/p01-03/kveiks.fltr.CE-Norway.28P1.lst"</span>, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb8-2"><a href="#cb8-2"></a>list_28R1 <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="st">"./data/p01-03/kveiks.fltr.CE-Norway.28P6.lst"</span>, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb8-3"><a href="#cb8-3"></a>list_28R21 <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="st">"./data/p01-03/kveiks.fltr.CE-Norway.28R21.lst"</span>, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb8-4"><a href="#cb8-4"></a>list_28R33 <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="st">"./data/p01-03/kveiks.fltr.CE-Norway.28R33.lst"</span>, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb8-5"><a href="#cb8-5"></a>list_28R8 <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="st">"./data/p01-03/kveiks.fltr.CE-Norway.28R8.lst"</span>, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb8-6"><a href="#cb8-6"></a></span>
<span id="cb8-7"><a href="#cb8-7"></a>UpSetR<span class="sc">::</span><span class="fu">upset</span>(</span>
<span id="cb8-8"><a href="#cb8-8"></a>  UpSetR<span class="sc">::</span><span class="fu">fromList</span>(</span>
<span id="cb8-9"><a href="#cb8-9"></a>    <span class="fu">list</span>(</span>
<span id="cb8-10"><a href="#cb8-10"></a>      <span class="st">`</span><span class="at">28R1</span><span class="st">`</span> <span class="ot">=</span> list_28R1<span class="sc">$</span>V1,</span>
<span id="cb8-11"><a href="#cb8-11"></a>       <span class="st">`</span><span class="at">28R8</span><span class="st">`</span> <span class="ot">=</span> list_28R8<span class="sc">$</span>V1,</span>
<span id="cb8-12"><a href="#cb8-12"></a>       <span class="st">`</span><span class="at">28R21</span><span class="st">`</span> <span class="ot">=</span> list_28R21<span class="sc">$</span>V1,</span>
<span id="cb8-13"><a href="#cb8-13"></a>       <span class="st">`</span><span class="at">28R31</span><span class="st">`</span> <span class="ot">=</span> list_28R31<span class="sc">$</span>V1,</span>
<span id="cb8-14"><a href="#cb8-14"></a>       <span class="st">`</span><span class="at">28R33</span><span class="st">`</span> <span class="ot">=</span> list_28R33<span class="sc">$</span>V1)</span>
<span id="cb8-15"><a href="#cb8-15"></a>  ),</span>
<span id="cb8-16"><a href="#cb8-16"></a>  <span class="at">order.by =</span> <span class="st">"freq"</span>,</span>
<span id="cb8-17"><a href="#cb8-17"></a>  <span class="at">keep.order =</span> <span class="cn">TRUE</span></span>
<span id="cb8-18"><a href="#cb8-18"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="p01-03-variant-calling_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption><span style="color: steelblue;">Figure 3: Culture #28 CE-Norway hybridization event.</span></figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>list_16R23 <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="st">"./data/p01-03/kveiks.fltr.Lithuania.16R23.lst"</span>, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb9-2"><a href="#cb9-2"></a>list_16R37 <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="st">"./data/p01-03/kveiks.fltr.Lithuania.16R37.lst"</span>, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb9-3"><a href="#cb9-3"></a>list_46R12 <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="st">"./data/p01-03/kveiks.fltr.Lithuania.46R12.lst"</span>, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb9-4"><a href="#cb9-4"></a>list_46R37 <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="st">"./data/p01-03/kveiks.fltr.Lithuania.46R37.lst"</span>, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb9-5"><a href="#cb9-5"></a></span>
<span id="cb9-6"><a href="#cb9-6"></a>UpSetR<span class="sc">::</span><span class="fu">upset</span>(</span>
<span id="cb9-7"><a href="#cb9-7"></a>  UpSetR<span class="sc">::</span><span class="fu">fromList</span>(</span>
<span id="cb9-8"><a href="#cb9-8"></a>    <span class="fu">list</span>(</span>
<span id="cb9-9"><a href="#cb9-9"></a>      <span class="st">`</span><span class="at">16R23</span><span class="st">`</span> <span class="ot">=</span> list_16R23<span class="sc">$</span>V1,</span>
<span id="cb9-10"><a href="#cb9-10"></a>      <span class="st">`</span><span class="at">16R37</span><span class="st">`</span> <span class="ot">=</span> list_16R37<span class="sc">$</span>V1,</span>
<span id="cb9-11"><a href="#cb9-11"></a>      <span class="st">`</span><span class="at">46R12</span><span class="st">`</span> <span class="ot">=</span> list_46R12<span class="sc">$</span>V1,</span>
<span id="cb9-12"><a href="#cb9-12"></a>      <span class="st">`</span><span class="at">46R37</span><span class="st">`</span> <span class="ot">=</span> list_46R37<span class="sc">$</span>V1)</span>
<span id="cb9-13"><a href="#cb9-13"></a>  ),</span>
<span id="cb9-14"><a href="#cb9-14"></a>  <span class="at">order.by =</span> <span class="st">"freq"</span>,</span>
<span id="cb9-15"><a href="#cb9-15"></a>  <span class="at">keep.order =</span> <span class="cn">TRUE</span></span>
<span id="cb9-16"><a href="#cb9-16"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="p01-03-variant-calling_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption><span style="color: steelblue;">Figure 4: Culture #16 Lithuania hybridization event.</span></figcaption>
</figure>
</div>
</div>
</div>
<p>Let’s try to phase Culture 28 with the predict correct parents and with the wrong parents. If the phased genome with the correct parents have significant more phased SNPs than the one with the wrong parents, maybe we can infer that Culture #28 is indeed an hybrid.</p>
<p>correct parents 28R21 28P6 28R33</p>
<p>wrong parents 28R21 28P1 28R8 28R21 28P1 27R17 28R21 28R8 27R17 28R21 28P6 27R17</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1"></a></span>
<span id="cb10-2"><a href="#cb10-2"></a><span class="va">DB</span><span class="op">=</span>~/00_Brigida_starting_files/variants/Saccharomyces_cerevisiae.EF4.73.dna.chromosome.all.fa</span>
<span id="cb10-3"><a href="#cb10-3"></a></span>
<span id="cb10-4"><a href="#cb10-4"></a><span class="co"># phase with Whatshap</span></span>
<span id="cb10-5"><a href="#cb10-5"></a><span class="ex">whatshap</span> phase <span class="at">-o</span> phased.vcf.gz <span class="at">--reference</span><span class="op">=</span><span class="va">$DB</span> culture28.vcf.gz 28R21_28P6_28R33.align.bam</span>
<span id="cb10-6"><a href="#cb10-6"></a><span class="ex">whatshap</span> phase <span class="at">-o</span> phased.wrongparents1.vcf.gz <span class="at">--reference</span><span class="op">=</span><span class="va">$DB</span> culture28.wrongparents1.vcf.gz 28R21_28P1_28R8.align.bam</span>
<span id="cb10-7"><a href="#cb10-7"></a><span class="ex">whatshap</span> phase <span class="at">-o</span> phased.wrongparents2.vcf.gz <span class="at">--reference</span><span class="op">=</span><span class="va">$DB</span> culture28.wrongparents2.vcf.gz 28R21_28P1_27R17.align.bam</span>
<span id="cb10-8"><a href="#cb10-8"></a><span class="ex">whatshap</span> phase <span class="at">-o</span> phased.wrongparents3.vcf.gz <span class="at">--reference</span><span class="op">=</span><span class="va">$DB</span> culture28.wrongparents3.vcf.gz 28R21_28R8_27R17.align.bam</span>
<span id="cb10-9"><a href="#cb10-9"></a><span class="ex">whatshap</span> phase <span class="at">-o</span> phased.wrongparents4.vcf.gz <span class="at">--reference</span><span class="op">=</span><span class="va">$DB</span> culture28.wrongparents4.vcf.gz 28R21_28P6_27R17.align.bam</span>
<span id="cb10-10"><a href="#cb10-10"></a></span>
<span id="cb10-11"><a href="#cb10-11"></a><span class="co"># generate tabular output for plotting</span></span>
<span id="cb10-12"><a href="#cb10-12"></a><span class="ex">python3.6</span> Vikings.culture_phasing_summary.py <span class="at">--input</span> culture28.vcf.gz <span class="at">--name</span> 28R21_28P6_28R33</span>
<span id="cb10-13"><a href="#cb10-13"></a><span class="ex">python3.6</span> Vikings.culture_phasing_summary.py <span class="at">--input</span> culture28.wrongparents1.vcf.gz <span class="at">--name</span> 28R21_28P1_28R8</span>
<span id="cb10-14"><a href="#cb10-14"></a><span class="ex">python3.6</span> Vikings.culture_phasing_summary.py <span class="at">--input</span> culture28.wrongparents2.vcf.gz <span class="at">--name</span> 28R21_28P1_27R17</span>
<span id="cb10-15"><a href="#cb10-15"></a><span class="ex">python3.6</span> Vikings.culture_phasing_summary.py <span class="at">--input</span> culture28.wrongparents3.vcf.gz <span class="at">--name</span> 28R21_28R8_27R17</span>
<span id="cb10-16"><a href="#cb10-16"></a><span class="ex">python3.6</span> Vikings.culture_phasing_summary.py <span class="at">--input</span> culture28.wrongparents4.vcf.gz <span class="at">--name</span> 28R21_28P6_27R17</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="co"># import</span></span>
<span id="cb11-2"><a href="#cb11-2"></a>phased <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="st">"./data/p01-03/culture28.phasing.txt"</span>)</span>
<span id="cb11-3"><a href="#cb11-3"></a></span>
<span id="cb11-4"><a href="#cb11-4"></a><span class="co"># melt</span></span>
<span id="cb11-5"><a href="#cb11-5"></a>phased_melted <span class="ot">=</span> reshape2<span class="sc">::</span><span class="fu">melt</span>(phased)</span>
<span id="cb11-6"><a href="#cb11-6"></a></span>
<span id="cb11-7"><a href="#cb11-7"></a><span class="co"># plot</span></span>
<span id="cb11-8"><a href="#cb11-8"></a><span class="fu">ggplot</span>(phased_melted, <span class="fu">aes</span>(<span class="at">x =</span> variable, <span class="at">y =</span> value)) <span class="sc">+</span></span>
<span id="cb11-9"><a href="#cb11-9"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">fill =</span> variable), <span class="at">color =</span> <span class="st">"grey45"</span>, <span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb11-10"><a href="#cb11-10"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> value), <span class="at">vjust =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb11-11"><a href="#cb11-11"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"firebrick"</span>, <span class="st">"steelblue1"</span>, <span class="st">"steelblue3"</span>, <span class="st">"steelblue4"</span>)) <span class="sc">+</span></span>
<span id="cb11-12"><a href="#cb11-12"></a>  <span class="fu">facet_grid</span>(trios <span class="sc">~</span> chr) <span class="sc">+</span></span>
<span id="cb11-13"><a href="#cb11-13"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb11-14"><a href="#cb11-14"></a>  <span class="fu">theme</span>(<span class="at">axis.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb11-15"><a href="#cb11-15"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb11-16"><a href="#cb11-16"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>, <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb11-17"><a href="#cb11-17"></a>        <span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="p01-03-variant-calling_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption><span style="color: steelblue;">Figure 5: Culture #28 Phasing combinations.</span></figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="genome-domestication-signature" class="level2" data-number="3.6">
<h2 data-number="3.6" class="anchored" data-anchor-id="genome-domestication-signature"><span class="header-section-number">3.6</span> Genome Domestication Signature</h2>
<p>Now we try to identify genetic signatures in the Kveiks lineage that hints to genome domestication or potential interesting brewing and industrial phenotypes. The idea is to look for the genes and the respective alleles involved in: copper tolerance, production of Phenolic Off-Flavours, maltose/maltotriose metabolism, ethanol resistance, amilase activity, glycerol production, temperature tolerance, heat-shock proteins, sulfites production, flocculation, desiccation tolerance, presence of Killer Toxins.</p>
<p>To achieve this we follow an approach based on sequence similarity search, variant calling and- based and phylogeny based:</p>
<ul>
<li>we will look for genes involved in industrial phenotypes but absent in S288C with a sequence similarity search.</li>
<li>we will look at loss of function (internal premature stop codons and frame-shifts) for genes involved in industrial phenotypes.</li>
<li>we will look at Kveiks allelic variants and compare them to the variaat sin other industrial yeast strains to discover conserved patterns.</li>
<li>we will cluster phylogenetically kveiks and other industrial strains in a phenotype-wise manner.</li>
</ul>
<section id="identifying-genes-missing-in-s288c" class="level3" data-number="3.6.1">
<h3 data-number="3.6.1" class="anchored" data-anchor-id="identifying-genes-missing-in-s288c"><span class="header-section-number">3.6.1</span> Identifying genes missing in S288C</h3>
<p>Some genes involved in industrial phenotypes are absent in S288C. In order to discover their presence in the sequenced kveiks, we will download the protein sequences of the genes of interest from SGD (https://www.yeastgenome.org/) and perform a sequence similarity search against the predicted proteins for each kveik strain.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1"></a><span class="co"># create protein databases</span></span>
<span id="cb12-2"><a href="#cb12-2"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>fasta<span class="kw">;</span> <span class="cf">do</span> <span class="ex">makeblastdb</span> <span class="at">-dbtype</span> prot <span class="at">-in</span> <span class="va">$file</span><span class="kw">;</span> <span class="cf">done</span></span>
<span id="cb12-3"><a href="#cb12-3"></a></span>
<span id="cb12-4"><a href="#cb12-4"></a><span class="co">## BLASTp</span></span>
<span id="cb12-5"><a href="#cb12-5"></a><span class="co"># EthanolResistance</span></span>
<span id="cb12-6"><a href="#cb12-6"></a><span class="cf">for</span> GENE <span class="kw">in</span> MPR1<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb12-7"><a href="#cb12-7"></a>    <span class="cf">for</span> file <span class="kw">in</span> ../00_prot_DB/<span class="pp">*</span>fasta<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb12-8"><a href="#cb12-8"></a>        <span class="ex">blastp</span> <span class="at">-db</span> <span class="va">$file</span> <span class="at">-query</span> EthanolResistance.<span class="va">$GENE</span>.aa.fa <span class="dt">\</span></span>
<span id="cb12-9"><a href="#cb12-9"></a>            <span class="at">-out</span> <span class="va">$GENE</span>.<span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .fasta<span class="va">)</span>.BLASTp.ER <span class="dt">\</span></span>
<span id="cb12-10"><a href="#cb12-10"></a>            <span class="at">-num_threads</span> 8 <span class="at">-outfmt</span> 6 <span class="at">-threshold</span> 0.9 <span class="at">-qcov_hsp_perc</span> 0.7<span class="kw">;</span></span>
<span id="cb12-11"><a href="#cb12-11"></a>    <span class="cf">done</span><span class="kw">;</span></span>
<span id="cb12-12"><a href="#cb12-12"></a><span class="cf">done</span></span>
<span id="cb12-13"><a href="#cb12-13"></a></span>
<span id="cb12-14"><a href="#cb12-14"></a><span class="co"># Amilase activity</span></span>
<span id="cb12-15"><a href="#cb12-15"></a><span class="cf">for</span> GENE <span class="kw">in</span> STA1 STA2<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb12-16"><a href="#cb12-16"></a>    <span class="cf">for</span> file <span class="kw">in</span> ../00_prot_DB/<span class="pp">*</span>fasta<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb12-17"><a href="#cb12-17"></a>        <span class="ex">blastp</span> <span class="at">-db</span> <span class="va">$file</span> <span class="at">-query</span> AmilaseActivity.<span class="va">$GENE</span>.aa.fa <span class="dt">\</span></span>
<span id="cb12-18"><a href="#cb12-18"></a>            <span class="at">-out</span> <span class="va">$GENE</span>.<span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .fasta<span class="va">)</span>.BLASTp.AA <span class="dt">\</span></span>
<span id="cb12-19"><a href="#cb12-19"></a>            <span class="at">-num_threads</span> 8 <span class="at">-outfmt</span> 6 <span class="at">-threshold</span> 0.9 <span class="at">-qcov_hsp_perc</span> 0.7<span class="kw">;</span></span>
<span id="cb12-20"><a href="#cb12-20"></a>    <span class="cf">done</span><span class="kw">;</span></span>
<span id="cb12-21"><a href="#cb12-21"></a><span class="cf">done</span></span>
<span id="cb12-22"><a href="#cb12-22"></a></span>
<span id="cb12-23"><a href="#cb12-23"></a><span class="co"># KillerToxin</span></span>
<span id="cb12-24"><a href="#cb12-24"></a><span class="cf">for</span> TOXIN <span class="kw">in</span> KHM1 KHR1 KHS1<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb12-25"><a href="#cb12-25"></a>    <span class="cf">for</span> file <span class="kw">in</span> ../00_prot_DB/<span class="pp">*</span>fasta<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb12-26"><a href="#cb12-26"></a>        <span class="ex">blastp</span> <span class="at">-db</span> <span class="va">$file</span> <span class="at">-query</span> KillerToxins.<span class="va">$TOXIN</span>.aa.fa <span class="dt">\</span></span>
<span id="cb12-27"><a href="#cb12-27"></a>            <span class="at">-out</span> <span class="va">$TOXIN</span>.<span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .fasta<span class="va">)</span>.BLASTp.KT <span class="dt">\</span></span>
<span id="cb12-28"><a href="#cb12-28"></a>            <span class="at">-num_threads</span> 8 <span class="at">-outfmt</span> 6 <span class="at">-threshold</span> 0.9 <span class="at">-qcov_hsp_perc</span> 0.7<span class="kw">;</span></span>
<span id="cb12-29"><a href="#cb12-29"></a>    <span class="cf">done</span><span class="kw">;</span></span>
<span id="cb12-30"><a href="#cb12-30"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="annotate-variants" class="level3" data-number="3.6.2">
<h3 data-number="3.6.2" class="anchored" data-anchor-id="annotate-variants"><span class="header-section-number">3.6.2</span> Annotate variants</h3>
<p>We use SnpEff to annotate variants effect.</p>
<p>To run SnpEff, first we need to add reference S288C genome to snpEff. To do so, we modify the config file: <code>~/bin/snpEff/snpEff.config</code>. by adding teh following lines:</p>
<pre><code># S288C GENOME
Scere64K: Scere64K</code></pre>
<p>The, we copy the proteins, transcripts and fasta files to <code>~/bin/snpEff/data</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1"></a><span class="fu">mkdir</span> ~/bin/snpEff/data/Scere64K</span>
<span id="cb14-2"><a href="#cb14-2"></a><span class="fu">cp</span> Scere64K.aa.fa /bin/snpEff/data/S4.consensus/protein.fa</span>
<span id="cb14-3"><a href="#cb14-3"></a><span class="fu">cp</span> Scere64K.genes.gff ~/bin/snpEff/data/S4.consensus.genes.gff</span>
<span id="cb14-4"><a href="#cb14-4"></a><span class="fu">cp</span> Scere64K.cds.fa ~/bin/snpEff/data/S4.consensus/genes.fa</span>
<span id="cb14-5"><a href="#cb14-5"></a><span class="fu">cp</span> Scere64K.fa ~/bin/snpEff/data/S4.consensus/sequences,fa</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>and build the reference database:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1"></a><span class="ex">java</span> <span class="at">-jar</span> ~/bin/snpEff/snpEff.jar build <span class="at">-gff3</span> <span class="at">-v</span> Scere64K</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now we can run SnpEff on a subset of variants composed by the kveiks strains plus the 22 additional industrial strains that we used as comparing in the previous analyses.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1"></a><span class="co"># select species for genome domestication</span></span>
<span id="cb16-2"><a href="#cb16-2"></a><span class="ex">~/bin/bcftools-1.9/bcftools</span> view <span class="dt">\</span></span>
<span id="cb16-3"><a href="#cb16-3"></a>  <span class="at">--samples-file</span> species_list.lst <span class="dt">\</span></span>
<span id="cb16-4"><a href="#cb16-4"></a>    <span class="at">--force-samples</span> <span class="dt">\</span></span>
<span id="cb16-5"><a href="#cb16-5"></a>    <span class="at">--threads</span> 16 <span class="dt">\</span></span>
<span id="cb16-6"><a href="#cb16-6"></a>    <span class="at">--min-alleles</span> 2 <span class="dt">\</span></span>
<span id="cb16-7"><a href="#cb16-7"></a>    <span class="at">--output-type</span> z ../08_GATK_variants3.3/all_plusKV.fltr.backup.vcf.gz <span class="dt">\</span></span>
<span id="cb16-8"><a href="#cb16-8"></a>    <span class="op">&gt;</span> all_plusKV.fltr.backup.reduced.vcf.gz</span>
<span id="cb16-9"><a href="#cb16-9"></a></span>
<span id="cb16-10"><a href="#cb16-10"></a><span class="co"># annotate variants</span></span>
<span id="cb16-11"><a href="#cb16-11"></a><span class="ex">java</span> <span class="at">-jar</span> ~/bin/snpEff/snpEff.jar eff <span class="dt">\</span></span>
<span id="cb16-12"><a href="#cb16-12"></a>    Scere64K <span class="dt">\</span></span>
<span id="cb16-13"><a href="#cb16-13"></a>    all_plusKV.fltr.backup.reduced.vcf.gz <span class="dt">\</span></span>
<span id="cb16-14"><a href="#cb16-14"></a>    <span class="op">&gt;</span> all_plusKV.fltr.backup.reduced.snpEff.vcf<span class="kw">;</span></span>
<span id="cb16-15"><a href="#cb16-15"></a><span class="fu">mv</span> snpEff_genes.txt all_plusKV.fltr.backup.reduced.snpEff_genes.txt<span class="kw">;</span></span>
<span id="cb16-16"><a href="#cb16-16"></a><span class="fu">mv</span> snpEff_summary.html all_plusKV.fltr.backup.reduced.vcf.snpEff_summary.html<span class="kw">;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="parse-variants" class="level3" data-number="3.6.3">
<h3 data-number="3.6.3" class="anchored" data-anchor-id="parse-variants"><span class="header-section-number">3.6.3</span> Parse variants</h3>
<p>Now we can select the subset of variants that falls between gene of interest for industrial applications.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1"></a><span class="co"># select variants</span></span>
<span id="cb17-2"><a href="#cb17-2"></a><span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb17-3"><a href="#cb17-3"></a>    <span class="va">CAT</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$line</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 1 <span class="at">-d</span> <span class="st">" "</span><span class="va">)</span><span class="kw">;</span></span>
<span id="cb17-4"><a href="#cb17-4"></a>    <span class="va">GENE</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$line</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 2 <span class="at">-d</span> <span class="st">" "</span><span class="va">)</span><span class="kw">;</span></span>
<span id="cb17-5"><a href="#cb17-5"></a>    <span class="va">CHR</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$line</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 4 <span class="at">-d</span> <span class="st">" "</span><span class="va">)</span><span class="kw">;</span></span>
<span id="cb17-6"><a href="#cb17-6"></a>    <span class="va">START</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$line</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 5 <span class="at">-d</span> <span class="st">" "</span><span class="va">)</span><span class="kw">;</span></span>
<span id="cb17-7"><a href="#cb17-7"></a>    <span class="va">STOP</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$line</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 6 <span class="at">-d</span> <span class="st">" "</span><span class="va">)</span><span class="kw">;</span></span>
<span id="cb17-8"><a href="#cb17-8"></a>    <span class="ex">tabix</span> all_plusKV.fltr.backup.reduced.snpEff.vcf.gz <span class="dt">\</span></span>
<span id="cb17-9"><a href="#cb17-9"></a>        <span class="st">"</span><span class="va">${CHR}</span><span class="st">"</span>:<span class="st">"</span><span class="va">${START}</span><span class="st">"</span>-<span class="st">"</span><span class="va">${STOP}</span><span class="st">"</span> <span class="op">&gt;</span> <span class="va">$CAT</span>.<span class="va">$GENE</span>.variants.vcf<span class="kw">;</span></span>
<span id="cb17-10"><a href="#cb17-10"></a><span class="cf">done</span> <span class="op">&lt;</span> Domestication.genes.windows.lst</span>
<span id="cb17-11"><a href="#cb17-11"></a></span>
<span id="cb17-12"><a href="#cb17-12"></a><span class="co"># reformat output for plotting</span></span>
<span id="cb17-13"><a href="#cb17-13"></a><span class="cf">for</span> DIR <span class="kw">in</span> 0<span class="pp">[</span><span class="ss">1</span><span class="pp">-</span><span class="ss">9</span><span class="pp">]*</span> 1<span class="pp">[</span><span class="ss">0</span><span class="pp">-</span><span class="ss">2</span><span class="pp">]*</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb17-14"><a href="#cb17-14"></a>    <span class="cf">for</span> file <span class="kw">in</span> <span class="va">$DIR</span>/<span class="pp">*</span>.vcf<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb17-15"><a href="#cb17-15"></a>        <span class="ex">python3.5</span> Vikings_domesticationVCFparser.py <span class="dt">\</span></span>
<span id="cb17-16"><a href="#cb17-16"></a>            <span class="at">--input</span> <span class="va">$file</span> <span class="dt">\</span></span>
<span id="cb17-17"><a href="#cb17-17"></a>            <span class="at">--samples</span> vcf_sample_positions.txt<span class="kw">;</span></span>
<span id="cb17-18"><a href="#cb17-18"></a>    <span class="cf">done</span> <span class="dt">\</span></span>
<span id="cb17-19"><a href="#cb17-19"></a>    <span class="kw">|</span> <span class="fu">sed</span> <span class="st">"s|</span><span class="va">$DIR</span><span class="st">/||g"</span> <span class="dt">\</span></span>
<span id="cb17-20"><a href="#cb17-20"></a>    <span class="kw">|</span> <span class="fu">perl</span> <span class="at">-ne</span> <span class="st">'@a = &lt;&gt;; print sort { length $b &lt;=&gt; length $a } @a'</span> <span class="dt">\</span></span>
<span id="cb17-21"><a href="#cb17-21"></a>  <span class="op">&gt;</span> variant.<span class="st">"</span><span class="va">${DIR}</span><span class="st">"</span>_all.tab <span class="kw">&amp;</span></span>
<span id="cb17-22"><a href="#cb17-22"></a><span class="cf">done</span></span>
<span id="cb17-23"><a href="#cb17-23"></a></span>
<span id="cb17-24"><a href="#cb17-24"></a><span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb17-25"><a href="#cb17-25"></a>    <span class="cf">for</span> file <span class="kw">in</span> variants.<span class="pp">*</span>_all.tab<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb17-26"><a href="#cb17-26"></a>        <span class="va">OLD</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$line</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 2 <span class="at">-d</span> <span class="st">" "</span><span class="va">)</span><span class="kw">;</span></span>
<span id="cb17-27"><a href="#cb17-27"></a>        <span class="va">NEW</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$line</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 3 <span class="at">-d</span> <span class="st">" "</span><span class="va">)</span><span class="kw">;</span></span>
<span id="cb17-28"><a href="#cb17-28"></a>        <span class="fu">sed</span> <span class="at">-i</span> <span class="st">"s/</span><span class="va">$OLD</span><span class="st">/</span><span class="va">$NEW</span><span class="st">/g"</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb17-29"><a href="#cb17-29"></a>    <span class="cf">done</span><span class="kw">;</span></span>
<span id="cb17-30"><a href="#cb17-30"></a><span class="cf">done</span> <span class="op">&lt;</span> Domestication.genes.windows.lst </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Let’s plot them.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a><span class="co"># plot</span></span>
<span id="cb18-2"><a href="#cb18-2"></a>phenotypes <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb18-3"><a href="#cb18-3"></a>  <span class="st">"01_coppertolerance"</span>,</span>
<span id="cb18-4"><a href="#cb18-4"></a>  <span class="st">"02_POF"</span>,</span>
<span id="cb18-5"><a href="#cb18-5"></a>  <span class="st">"03_maltosemaltotriose"</span>,</span>
<span id="cb18-6"><a href="#cb18-6"></a>  <span class="st">"04_ethanolresistance"</span>,</span>
<span id="cb18-7"><a href="#cb18-7"></a>  <span class="st">"05_amilaseactivity"</span>,</span>
<span id="cb18-8"><a href="#cb18-8"></a>  <span class="st">"06_glycerol"</span>,</span>
<span id="cb18-9"><a href="#cb18-9"></a>  <span class="st">"07_temperaturetolerance"</span>,</span>
<span id="cb18-10"><a href="#cb18-10"></a>  <span class="st">"08_heatshock"</span>,</span>
<span id="cb18-11"><a href="#cb18-11"></a>  <span class="st">"09_sulfites"</span>,</span>
<span id="cb18-12"><a href="#cb18-12"></a>  <span class="st">"10_flocculation"</span>,</span>
<span id="cb18-13"><a href="#cb18-13"></a>  <span class="st">"11_dessication"</span>,</span>
<span id="cb18-14"><a href="#cb18-14"></a>  <span class="st">"12_killertoxin"</span></span>
<span id="cb18-15"><a href="#cb18-15"></a>)</span>
<span id="cb18-16"><a href="#cb18-16"></a></span>
<span id="cb18-17"><a href="#cb18-17"></a>color_list <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb18-18"><a href="#cb18-18"></a>  <span class="st">"grey85"</span>, <span class="st">"white"</span>, <span class="st">"#a2d5c6"</span>, <span class="st">"#077b8a"</span>, <span class="st">"#d72631"</span>, <span class="st">"#5c3c92"</span>,</span>
<span id="cb18-19"><a href="#cb18-19"></a>  <span class="st">"#e75874"</span>, <span class="st">"#be1558"</span>, <span class="st">"#fbcbc9"</span>, <span class="st">"#322514"</span>, <span class="st">"#1e3d59"</span>, <span class="st">"#ff6e40"</span>,</span>
<span id="cb18-20"><a href="#cb18-20"></a>  <span class="st">"#ffc13b"</span>, <span class="st">"#ecc19c"</span>, <span class="st">"#1e847f"</span>, <span class="st">"#000000"</span>, <span class="st">"#d9a5b3"</span>, <span class="st">"#1868ae"</span>,</span>
<span id="cb18-21"><a href="#cb18-21"></a>  <span class="st">"#c6d7eb"</span>, <span class="st">"#da68a0"</span>, <span class="st">"#77c593"</span>, <span class="st">"#ed3572"</span>, <span class="st">"#1d3c45"</span>, <span class="st">"#d2601a"</span>,</span>
<span id="cb18-22"><a href="#cb18-22"></a>  <span class="st">"#fff1e1"</span>, <span class="st">"#097770"</span>, <span class="st">"#ffea04"</span>, <span class="st">"#b85042"</span>, <span class="st">"#ffcce7"</span>, <span class="st">"#daf2dc"</span>,</span>
<span id="cb18-23"><a href="#cb18-23"></a>  <span class="st">"#e1dd72"</span>, <span class="st">"#a8c66c"</span>, <span class="st">"#1b6535"</span></span>
<span id="cb18-24"><a href="#cb18-24"></a>)</span>
<span id="cb18-25"><a href="#cb18-25"></a></span>
<span id="cb18-26"><a href="#cb18-26"></a>plot_list <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb18-27"><a href="#cb18-27"></a></span>
<span id="cb18-28"><a href="#cb18-28"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(phenotypes)){</span>
<span id="cb18-29"><a href="#cb18-29"></a>  </span>
<span id="cb18-30"><a href="#cb18-30"></a>  name <span class="ot">=</span> phenotypes[i]</span>
<span id="cb18-31"><a href="#cb18-31"></a>  file <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">"./data/p01-03/variants."</span>, name, <span class="st">"_all.tab"</span>)</span>
<span id="cb18-32"><a href="#cb18-32"></a>  <span class="co"># dataset = read.delim(file, header = FALSE, fill = TRUE)</span></span>
<span id="cb18-33"><a href="#cb18-33"></a>  dataset <span class="ot">=</span> <span class="fu">read.delim</span>(file, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb18-34"><a href="#cb18-34"></a>  </span>
<span id="cb18-35"><a href="#cb18-35"></a>  dataset<span class="sc">$</span>V3 <span class="ot">=</span> <span class="fu">factor</span>(</span>
<span id="cb18-36"><a href="#cb18-36"></a>    dataset<span class="sc">$</span>V3,</span>
<span id="cb18-37"><a href="#cb18-37"></a>    <span class="at">levels =</span> <span class="fu">c</span>(</span>
<span id="cb18-38"><a href="#cb18-38"></a>      <span class="st">"14R30"</span>, <span class="st">"14R6"</span>, <span class="st">"16R23"</span>, <span class="st">"16R37"</span>, <span class="st">"17P5"</span>, <span class="st">"19R18"</span>, <span class="st">"1R16"</span>, <span class="st">"21P1"</span>, <span class="st">"21R38"</span>, <span class="st">"27R17"</span>, <span class="st">"28P1"</span>,</span>
<span id="cb18-39"><a href="#cb18-39"></a>       <span class="st">"28P6"</span>, <span class="st">"28R21"</span>, <span class="st">"28R33"</span>, <span class="st">"28R8"</span>, <span class="st">"2R23"</span>, <span class="st">"38R16"</span>, <span class="st">"39R20"</span>, <span class="st">"3R11"</span>, <span class="st">"40R14"</span>, <span class="st">"40R1"</span>, <span class="st">"40R20"</span>,</span>
<span id="cb18-40"><a href="#cb18-40"></a>       <span class="st">"41R10"</span>, <span class="st">"41R15"</span>, <span class="st">"42R20"</span>, <span class="st">"42R31"</span>, <span class="st">"44R32"</span>, <span class="st">"44R7"</span>, <span class="st">"45P5"</span>, <span class="st">"45R11"</span>, <span class="st">"46R12"</span>, <span class="st">"46R37"</span>, <span class="st">"6R15"</span>,</span>
<span id="cb18-41"><a href="#cb18-41"></a>       <span class="st">"7R7"</span>, <span class="st">"8R19"</span>, <span class="st">"9R40"</span>, <span class="st">"Granvin1"</span>, <span class="st">"Hornindal1"</span>, <span class="st">"Hornindal2"</span>, <span class="st">"k7R25"</span>, <span class="st">"Laerdal2"</span>, <span class="st">"Muri"</span>, <span class="st">"SortdalEbbe1"</span>,</span>
<span id="cb18-42"><a href="#cb18-42"></a>       <span class="st">"Voss1"</span>, <span class="st">"X5000"</span>, <span class="st">"X5009"</span>, <span class="st">"X6004"</span>, <span class="st">"x97"</span>, <span class="st">"X1002"</span>, <span class="st">"X1014"</span>, <span class="st">"X1030"</span>, <span class="st">"X1129"</span>, <span class="st">"X5508"</span>, <span class="st">"X1007"</span>,</span>
<span id="cb18-43"><a href="#cb18-43"></a>       <span class="st">"X1118"</span>, <span class="st">"X1148"</span>, <span class="st">"X7078"</span>, <span class="st">"X1011"</span>, <span class="st">"X1091"</span>, <span class="st">"X1111"</span>, <span class="st">"X1119"</span>, <span class="st">"X8003"</span>, <span class="st">"X35"</span>, <span class="st">"X1046"</span>, <span class="st">"X4037"</span>, <span class="st">"X4051"</span></span>
<span id="cb18-44"><a href="#cb18-44"></a>    )</span>
<span id="cb18-45"><a href="#cb18-45"></a>  )</span>
<span id="cb18-46"><a href="#cb18-46"></a>  </span>
<span id="cb18-47"><a href="#cb18-47"></a>  dataset<span class="sc">$</span>V6 <span class="ot">=</span> <span class="fu">factor</span>(</span>
<span id="cb18-48"><a href="#cb18-48"></a>    dataset<span class="sc">$</span>V6,</span>
<span id="cb18-49"><a href="#cb18-49"></a>    <span class="at">levels =</span> <span class="fu">c</span>(</span>
<span id="cb18-50"><a href="#cb18-50"></a>      <span class="st">"NO_COV"</span>, <span class="st">"REF"</span>, <span class="st">"0/1"</span>, <span class="st">"1/1"</span>, <span class="st">"0/2"</span>, <span class="st">"0/3"</span>, </span>
<span id="cb18-51"><a href="#cb18-51"></a>      <span class="st">"0/4"</span>, <span class="st">"0/6"</span>, <span class="st">"1/2"</span>, <span class="st">"1/3"</span>, <span class="st">"1/4"</span>, <span class="st">"1/5"</span>,</span>
<span id="cb18-52"><a href="#cb18-52"></a>      <span class="st">"1/6"</span>, <span class="st">"1/8"</span>, <span class="st">"2/1"</span>, <span class="st">"2/2"</span>, <span class="st">"2/3"</span>, <span class="st">"2/4"</span>,</span>
<span id="cb18-53"><a href="#cb18-53"></a>      <span class="st">"2/5"</span>, <span class="st">"3/1"</span>, <span class="st">"3/2"</span>, <span class="st">"3/3"</span>, <span class="st">"3/4"</span>, <span class="st">"4/1"</span>,</span>
<span id="cb18-54"><a href="#cb18-54"></a>      <span class="st">"4/3"</span>, <span class="st">"4/4"</span>, <span class="st">"5/1"</span>, <span class="st">"5/5"</span>, <span class="st">"5/6"</span>, <span class="st">"7/7"</span>,</span>
<span id="cb18-55"><a href="#cb18-55"></a>      <span class="st">"7/8"</span>, <span class="st">"8/8"</span>, <span class="st">"./."</span></span>
<span id="cb18-56"><a href="#cb18-56"></a>    )</span>
<span id="cb18-57"><a href="#cb18-57"></a>  )</span>
<span id="cb18-58"><a href="#cb18-58"></a>  </span>
<span id="cb18-59"><a href="#cb18-59"></a>  <span class="do">### create the plot</span></span>
<span id="cb18-60"><a href="#cb18-60"></a>  p1 <span class="ot">=</span> <span class="fu">ggplot</span>(dataset, <span class="fu">aes</span>(V3, V5)) <span class="sc">+</span> </span>
<span id="cb18-61"><a href="#cb18-61"></a>    <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">fill =</span> V6)) <span class="sc">+</span></span>
<span id="cb18-62"><a href="#cb18-62"></a>    <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fl">44.5</span>), <span class="at">color =</span> <span class="st">"grey50"</span>, <span class="at">size =</span> <span class="fl">0.125</span>) <span class="sc">+</span></span>
<span id="cb18-63"><a href="#cb18-63"></a>    <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fl">48.5</span>), <span class="at">color =</span> <span class="st">"grey50"</span>, <span class="at">size =</span> <span class="fl">0.125</span>) <span class="sc">+</span></span>
<span id="cb18-64"><a href="#cb18-64"></a>    <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fl">53.5</span>), <span class="at">color =</span> <span class="st">"grey50"</span>, <span class="at">size =</span> <span class="fl">0.125</span>) <span class="sc">+</span></span>
<span id="cb18-65"><a href="#cb18-65"></a>    <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fl">57.5</span>), <span class="at">color =</span> <span class="st">"grey50"</span>, <span class="at">size =</span> <span class="fl">0.125</span>) <span class="sc">+</span></span>
<span id="cb18-66"><a href="#cb18-66"></a>    <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fl">61.5</span>), <span class="at">color =</span> <span class="st">"grey50"</span>, <span class="at">size =</span> <span class="fl">0.125</span>) <span class="sc">+</span></span>
<span id="cb18-67"><a href="#cb18-67"></a>    <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fl">62.5</span>), <span class="at">color =</span> <span class="st">"grey50"</span>, <span class="at">size =</span> <span class="fl">0.125</span>) <span class="sc">+</span></span>
<span id="cb18-68"><a href="#cb18-68"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> color_list) <span class="sc">+</span></span>
<span id="cb18-69"><a href="#cb18-69"></a>    <span class="fu">coord_cartesian</span>(<span class="at">expand =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb18-70"><a href="#cb18-70"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>V2, <span class="at">nrow =</span> <span class="dv">1</span>, <span class="at">scales =</span> <span class="st">"free"</span>,</span>
<span id="cb18-71"><a href="#cb18-71"></a>               <span class="at">strip.position =</span> <span class="st">"bottom"</span>) <span class="sc">+</span></span>
<span id="cb18-72"><a href="#cb18-72"></a>    <span class="fu">labs</span>(<span class="at">fill =</span> <span class="st">"Genotype"</span>,</span>
<span id="cb18-73"><a href="#cb18-73"></a>         <span class="at">y =</span> <span class="st">"Positions"</span>) <span class="sc">+</span></span>
<span id="cb18-74"><a href="#cb18-74"></a>    <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-75"><a href="#cb18-75"></a>          <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-76"><a href="#cb18-76"></a>          <span class="at">axis.ticks.x =</span> <span class="fu">element_line</span>(</span>
<span id="cb18-77"><a href="#cb18-77"></a>            <span class="at">size =</span> <span class="dv">2</span>,</span>
<span id="cb18-78"><a href="#cb18-78"></a>            <span class="at">colour =</span> <span class="fu">c</span>(</span>
<span id="cb18-79"><a href="#cb18-79"></a>              <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, </span>
<span id="cb18-80"><a href="#cb18-80"></a>              <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, </span>
<span id="cb18-81"><a href="#cb18-81"></a>              <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, </span>
<span id="cb18-82"><a href="#cb18-82"></a>              <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, </span>
<span id="cb18-83"><a href="#cb18-83"></a>              <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, </span>
<span id="cb18-84"><a href="#cb18-84"></a>              <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#3288BD"</span>, <span class="st">"#3288BD"</span>, <span class="st">"#3288BD"</span>, <span class="st">"#3288BD"</span>,</span>
<span id="cb18-85"><a href="#cb18-85"></a>              <span class="st">"#FC8D59"</span>, <span class="st">"#FC8D59"</span>, <span class="st">"#FC8D59"</span>, <span class="st">"#FC8D59"</span>, <span class="st">"#FC8D59"</span>, <span class="st">"#99D594"</span>, <span class="st">"#99D594"</span>, <span class="st">"#99D594"</span>,</span>
<span id="cb18-86"><a href="#cb18-86"></a>              <span class="st">"#99D594"</span>, <span class="st">"#FFFFBF"</span>, <span class="st">"#FFFFBF"</span>, <span class="st">"#FFFFBF"</span>, <span class="st">"#FFFFBF"</span>, <span class="st">"#7BCEB8"</span>, <span class="st">"#FEE08B"</span>, <span class="st">"#FEE08B"</span>,</span>
<span id="cb18-87"><a href="#cb18-87"></a>              <span class="st">"#FEE08B"</span></span>
<span id="cb18-88"><a href="#cb18-88"></a>            )</span>
<span id="cb18-89"><a href="#cb18-89"></a>          ),</span>
<span id="cb18-90"><a href="#cb18-90"></a>          <span class="at">axis.ticks.length.x =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">"lines"</span>),</span>
<span id="cb18-91"><a href="#cb18-91"></a>          <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-92"><a href="#cb18-92"></a>          <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb18-93"><a href="#cb18-93"></a>          <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-94"><a href="#cb18-94"></a>          <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-95"><a href="#cb18-95"></a>          <span class="at">legend.box.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">colour =</span> <span class="st">"black"</span>),</span>
<span id="cb18-96"><a href="#cb18-96"></a>          <span class="at">legend.key =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">colour =</span> <span class="st">"black"</span>),</span>
<span id="cb18-97"><a href="#cb18-97"></a>          <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb18-98"><a href="#cb18-98"></a>          <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-99"><a href="#cb18-99"></a>          <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="sc">-</span><span class="fl">0.2</span>, <span class="st">"lines"</span>),</span>
<span id="cb18-100"><a href="#cb18-100"></a>          <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">fill =</span> <span class="cn">NA</span>),</span>
<span id="cb18-101"><a href="#cb18-101"></a>          <span class="at">strip.background =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">fill =</span> <span class="st">"grey90"</span>),</span>
<span id="cb18-102"><a href="#cb18-102"></a>          <span class="at">strip.placement =</span> <span class="st">"outside"</span>,</span>
<span id="cb18-103"><a href="#cb18-103"></a>          <span class="at">strip.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>))</span>
<span id="cb18-104"><a href="#cb18-104"></a></span>
<span id="cb18-105"><a href="#cb18-105"></a>  <span class="do">### Construct species tables</span></span>
<span id="cb18-106"><a href="#cb18-106"></a>  <span class="co"># Construct the 14 grobs - 7 symbols and 7 labels</span></span>
<span id="cb18-107"><a href="#cb18-107"></a>  L1 <span class="ot">=</span> <span class="fu">rectGrob</span>(<span class="at">height =</span> .<span class="dv">5</span>, <span class="at">width =</span> .<span class="dv">5</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#D53E4F"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb18-108"><a href="#cb18-108"></a>  L2 <span class="ot">=</span> <span class="fu">rectGrob</span>(<span class="at">height =</span> .<span class="dv">5</span>, <span class="at">width =</span> .<span class="dv">5</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#3288BD"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb18-109"><a href="#cb18-109"></a>  L3 <span class="ot">=</span> <span class="fu">rectGrob</span>(<span class="at">height =</span> .<span class="dv">5</span>, <span class="at">width =</span> .<span class="dv">5</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#FC8D59"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb18-110"><a href="#cb18-110"></a>  L4 <span class="ot">=</span> <span class="fu">rectGrob</span>(<span class="at">height =</span> .<span class="dv">5</span>, <span class="at">width =</span> .<span class="dv">5</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#99D594"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb18-111"><a href="#cb18-111"></a>  L5 <span class="ot">=</span> <span class="fu">rectGrob</span>(<span class="at">height =</span> .<span class="dv">5</span>, <span class="at">width =</span> .<span class="dv">5</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#FFFFBF"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb18-112"><a href="#cb18-112"></a>  L6 <span class="ot">=</span> <span class="fu">rectGrob</span>(<span class="at">height =</span> .<span class="dv">5</span>, <span class="at">width =</span> .<span class="dv">5</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#7BCEB8"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb18-113"><a href="#cb18-113"></a>  L7 <span class="ot">=</span> <span class="fu">rectGrob</span>(<span class="at">height =</span> .<span class="dv">5</span>, <span class="at">width =</span> .<span class="dv">5</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#FEE08B"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb18-114"><a href="#cb18-114"></a>  T1 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Farmhouse"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb18-115"><a href="#cb18-115"></a>  T2 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Asia"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb18-116"><a href="#cb18-116"></a>  T3 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Beer 1"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb18-117"><a href="#cb18-117"></a>  T4 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Beer 2"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb18-118"><a href="#cb18-118"></a>  T5 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Mixed"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb18-119"><a href="#cb18-119"></a>  T6 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Mosaic"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb18-120"><a href="#cb18-120"></a>  T7 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Wine"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb18-121"><a href="#cb18-121"></a>  </span>
<span id="cb18-122"><a href="#cb18-122"></a>  <span class="co"># Construct a gtable - 2 columns X 8 rows</span></span>
<span id="cb18-123"><a href="#cb18-123"></a>  leg <span class="ot">=</span> <span class="fu">gtable</span>(<span class="at">width =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="st">"cm"</span>),</span>
<span id="cb18-124"><a href="#cb18-124"></a>               <span class="at">height =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>), <span class="st">"cm"</span>))</span>
<span id="cb18-125"><a href="#cb18-125"></a>  leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, <span class="fu">rectGrob</span>(<span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">col =</span> <span class="st">"black"</span>)),</span>
<span id="cb18-126"><a href="#cb18-126"></a>                        <span class="at">t =</span> <span class="dv">1</span>, <span class="at">l =</span> <span class="dv">1</span>, <span class="at">b =</span> <span class="dv">8</span>, <span class="at">r =</span> <span class="dv">2</span>)</span>
<span id="cb18-127"><a href="#cb18-127"></a></span>
<span id="cb18-128"><a href="#cb18-128"></a>  <span class="co"># gtable_show_layout(leg)</span></span>
<span id="cb18-129"><a href="#cb18-129"></a>  <span class="co"># Place the 14 grob into the table</span></span>
<span id="cb18-130"><a href="#cb18-130"></a>  leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L1, <span class="at">t=</span><span class="dv">2</span>, <span class="at">l=</span><span class="dv">1</span>)</span>
<span id="cb18-131"><a href="#cb18-131"></a>  leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L2, <span class="at">t=</span><span class="dv">3</span>, <span class="at">l=</span><span class="dv">1</span>)</span>
<span id="cb18-132"><a href="#cb18-132"></a>  leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L3, <span class="at">t=</span><span class="dv">4</span>, <span class="at">l=</span><span class="dv">1</span>)</span>
<span id="cb18-133"><a href="#cb18-133"></a>  leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L4, <span class="at">t=</span><span class="dv">5</span>, <span class="at">l=</span><span class="dv">1</span>)</span>
<span id="cb18-134"><a href="#cb18-134"></a>  leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L5, <span class="at">t=</span><span class="dv">6</span>, <span class="at">l=</span><span class="dv">1</span>)</span>
<span id="cb18-135"><a href="#cb18-135"></a>  leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L6, <span class="at">t=</span><span class="dv">7</span>, <span class="at">l=</span><span class="dv">1</span>)</span>
<span id="cb18-136"><a href="#cb18-136"></a>  leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L7, <span class="at">t=</span><span class="dv">8</span>, <span class="at">l=</span><span class="dv">1</span>)  </span>
<span id="cb18-137"><a href="#cb18-137"></a>  leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T1, <span class="at">t=</span><span class="dv">2</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb18-138"><a href="#cb18-138"></a>  leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T2, <span class="at">t=</span><span class="dv">3</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb18-139"><a href="#cb18-139"></a>  leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T3, <span class="at">t=</span><span class="dv">4</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb18-140"><a href="#cb18-140"></a>  leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T4, <span class="at">t=</span><span class="dv">5</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb18-141"><a href="#cb18-141"></a>  leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T5, <span class="at">t=</span><span class="dv">6</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb18-142"><a href="#cb18-142"></a>  leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T6, <span class="at">t=</span><span class="dv">7</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb18-143"><a href="#cb18-143"></a>  leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T7, <span class="at">t=</span><span class="dv">8</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb18-144"><a href="#cb18-144"></a>  </span>
<span id="cb18-145"><a href="#cb18-145"></a>  <span class="co"># Give it a title (if needed)</span></span>
<span id="cb18-146"><a href="#cb18-146"></a>  leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, <span class="fu">textGrob</span>(<span class="st">"Population"</span>), <span class="at">t =</span> <span class="dv">1</span>, <span class="at">l =</span> <span class="dv">1</span>, <span class="at">r =</span> <span class="dv">2</span>)</span>
<span id="cb18-147"><a href="#cb18-147"></a>  </span>
<span id="cb18-148"><a href="#cb18-148"></a>  <span class="co"># Get the ggplot grob for plot1</span></span>
<span id="cb18-149"><a href="#cb18-149"></a>  g <span class="ot">=</span> <span class="fu">ggplotGrob</span>(p1)</span>
<span id="cb18-150"><a href="#cb18-150"></a>  </span>
<span id="cb18-151"><a href="#cb18-151"></a>  <span class="co"># Get the position of the panel,</span></span>
<span id="cb18-152"><a href="#cb18-152"></a>  <span class="co"># add a column to the right of the panel, </span></span>
<span id="cb18-153"><a href="#cb18-153"></a>  <span class="co"># put the legend into that column, </span></span>
<span id="cb18-154"><a href="#cb18-154"></a>  <span class="co"># and then add another spacing column</span></span>
<span id="cb18-155"><a href="#cb18-155"></a>  pos <span class="ot">=</span> g<span class="sc">$</span>layout[<span class="fu">grepl</span>(<span class="st">"panel"</span>, g<span class="sc">$</span>layout<span class="sc">$</span>name), <span class="fu">c</span>(<span class="st">'t'</span>, <span class="st">'l'</span>)]</span>
<span id="cb18-156"><a href="#cb18-156"></a>  g <span class="ot">=</span> <span class="fu">gtable_add_cols</span>(g, <span class="fu">sum</span>(leg<span class="sc">$</span>widths), <span class="fu">tail</span>(pos<span class="sc">$</span>l, <span class="at">n =</span><span class="dv">1</span>))</span>
<span id="cb18-157"><a href="#cb18-157"></a>  g <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(g, leg, <span class="at">t =</span> pos<span class="sc">$</span>t[<span class="dv">1</span>], <span class="at">l =</span> <span class="fu">tail</span>(pos<span class="sc">$</span>l, <span class="at">n =</span><span class="dv">1</span>) <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb18-158"><a href="#cb18-158"></a>  g <span class="ot">=</span> <span class="fu">gtable_add_cols</span>(g, <span class="fu">unit</span>(<span class="dv">6</span>, <span class="st">"pt"</span>), <span class="fu">tail</span>(pos<span class="sc">$</span>l, <span class="at">n =</span><span class="dv">1</span>))</span>
<span id="cb18-159"><a href="#cb18-159"></a>  </span>
<span id="cb18-160"><a href="#cb18-160"></a>  <span class="co"># Draw it</span></span>
<span id="cb18-161"><a href="#cb18-161"></a>  plot_list[[i]] <span class="ot">=</span> g</span>
<span id="cb18-162"><a href="#cb18-162"></a>  </span>
<span id="cb18-163"><a href="#cb18-163"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true" aria-current="page">Copper Tolerance</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false">Phenolic Off-Flavours</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-3" role="tab" aria-controls="tabset-1-3" aria-selected="false">Maltose/Maltotriose Metabolism</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-4" role="tab" aria-controls="tabset-1-4" aria-selected="false">Ethanol Resistance</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-5-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-5" role="tab" aria-controls="tabset-1-5" aria-selected="false">Amilase Activity</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-6-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-6" role="tab" aria-controls="tabset-1-6" aria-selected="false">Glycerol Production</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-7-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-7" role="tab" aria-controls="tabset-1-7" aria-selected="false">Temperature Tolerance</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-8-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-8" role="tab" aria-controls="tabset-1-8" aria-selected="false">Heat-Shock Proteins</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-9-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-9" role="tab" aria-controls="tabset-1-9" aria-selected="false">Sulfites Production</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-10-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-10" role="tab" aria-controls="tabset-1-10" aria-selected="false">Flocculation</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-11-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-11" role="tab" aria-controls="tabset-1-11" aria-selected="false">Dessication Tolerance</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-12-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-12" role="tab" aria-controls="tabset-1-12" aria-selected="false">Killer Toxin</a></li></ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="p01-03-variant-calling_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption><span style="color: steelblue;">Figure 6: Variants on Copper Tolerance genes.</span></figcaption>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="p01-03-variant-calling_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption><span style="color: steelblue;">Figure 7: Variants on Phenolic Off-Flavours genes.</span></figcaption>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-3-tab">
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="p01-03-variant-calling_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption><span style="color: steelblue;">Figure 8: Variants on Maltose/Maltotriose Metabolism genes.</span></figcaption>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-4-tab">
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="p01-03-variant-calling_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption><span style="color: steelblue;">Figure 9: Variants on Ethanol Resistance genes.</span></figcaption>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-5" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-5-tab">
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="p01-03-variant-calling_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption><span style="color: steelblue;">Figure 10: Variants on Amilase Activity genes.</span></figcaption>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-6" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-6-tab">
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="p01-03-variant-calling_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption><span style="color: steelblue;">Figure 11: Variants on Glycerol Production genes.</span></figcaption>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-7" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-7-tab">
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="p01-03-variant-calling_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption><span style="color: steelblue;">Figure 12: Variants on Temperature Tolerance genes.</span></figcaption>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-8" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-8-tab">
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="p01-03-variant-calling_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption><span style="color: steelblue;">Figure 13: Variants on Heat-Shock Proteins genes.</span></figcaption>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-9" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-9-tab">
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="p01-03-variant-calling_files/figure-html/unnamed-chunk-28-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption><span style="color: steelblue;">Figure 14: Variants on Sulfites Production genes.</span></figcaption>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-10" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-10-tab">
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="p01-03-variant-calling_files/figure-html/unnamed-chunk-29-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption><span style="color: steelblue;">Figure 15: Variants on Flocculation genes.</span></figcaption>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-11" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-11-tab">
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="p01-03-variant-calling_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption><span style="color: steelblue;">Figure 16: Variants on Dessication Tolerance genes.</span></figcaption>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-12" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-12-tab">
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="p01-03-variant-calling_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption><span style="color: steelblue;">Figure 17: Variants on Killer Toxin genes.</span></figcaption>
</figure>
</div>
</div>
</div>
</div>
</div>
</div>
<p>TPS3 238G&gt;A : 41 / 44 Farmhouse yeasts 2500A&gt;G : 42 / 44 Farmhouse yeasts</p>
<p>AGT1 / MAL11 1175_1176insT 1772delA : 38 / 44 Farmhouse yeasts 491AAG&gt;TTA 1236C&gt;G</p>
<p>NTH1 1213C&gt;T (Leu405Phe) : 38 / 44 Farmhouse yeasts</p>
<p>ATH1 3394A&gt;G, Asn1132Asp : 3434A&gt;G, Ser1145Asn : 3596A&gt;G; Asn1199Ser :</p>
</section>
<section id="identify-disrupted-genes" class="level3" data-number="3.6.4">
<h3 data-number="3.6.4" class="anchored" data-anchor-id="identify-disrupted-genes"><span class="header-section-number">3.6.4</span> Identify disrupted genes</h3>
<p>From the annotate variants, we can easily filter disrupted genes (variants with HIGH effect: i.e.&nbsp;premature stop codons and frameshifts), and look for patterns of disrupted “pathways” among kveiks and other industrial yeasts.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a><span class="co"># import</span></span>
<span id="cb19-2"><a href="#cb19-2"></a>dataset <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="st">"./data/p01-03/variants.stopcodons.ready.tab"</span>, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb19-3"><a href="#cb19-3"></a></span>
<span id="cb19-4"><a href="#cb19-4"></a><span class="co"># reformat dataset</span></span>
<span id="cb19-5"><a href="#cb19-5"></a>dataset<span class="sc">$</span>V4 <span class="ot">=</span> <span class="fu">factor</span>(</span>
<span id="cb19-6"><a href="#cb19-6"></a>  dataset<span class="sc">$</span>V4,</span>
<span id="cb19-7"><a href="#cb19-7"></a>  <span class="at">levels =</span> <span class="fu">c</span>(</span>
<span id="cb19-8"><a href="#cb19-8"></a>    <span class="st">"X4051"</span>, <span class="st">"X4037"</span>, <span class="st">"X1046"</span>, <span class="st">"X35"</span>, <span class="st">"X8003"</span>, <span class="st">"x1119"</span>, <span class="st">"X1111"</span>, <span class="st">"X1091"</span>, <span class="st">"X1011"</span>, <span class="st">"x7078-B"</span>, <span class="st">"X1148"</span>,</span>
<span id="cb19-9"><a href="#cb19-9"></a>    <span class="st">"X1118"</span>, <span class="st">"X1007"</span>, <span class="st">"X5508"</span>, <span class="st">"x1129"</span>, <span class="st">"X1030"</span>, <span class="st">"X1014"</span>, <span class="st">"X1002"</span>, <span class="st">"x97"</span>, <span class="st">"X6004"</span>, <span class="st">"X5009"</span>, <span class="st">"X5000"</span>, <span class="st">"Voss1"</span>,</span>
<span id="cb19-10"><a href="#cb19-10"></a>    <span class="st">"SortdalEbbe1"</span>, <span class="st">"Muri"</span>, <span class="st">"Laerdal2"</span>, <span class="st">"k7R25"</span>, <span class="st">"Hornindal2"</span>, <span class="st">"Hornindal1"</span>, <span class="st">"Granvin1"</span>, <span class="st">"9R40"</span>, <span class="st">"8R19"</span>, <span class="st">"7R7"</span>,</span>
<span id="cb19-11"><a href="#cb19-11"></a>    <span class="st">"6R15"</span>, <span class="st">"46R37"</span>, <span class="st">"46R12"</span>, <span class="st">"45R11"</span>, <span class="st">"45P5"</span>, <span class="st">"44R7"</span>, <span class="st">"44R32"</span>, <span class="st">"42R31"</span>, <span class="st">"42R20"</span>, <span class="st">"41R15"</span>, <span class="st">"41R10"</span>, <span class="st">"40R20"</span>,</span>
<span id="cb19-12"><a href="#cb19-12"></a>    <span class="st">"40R1"</span>, <span class="st">"40R14"</span>, <span class="st">"3R11"</span>, <span class="st">"39R20"</span>, <span class="st">"38R16"</span>, <span class="st">"2R23"</span>, <span class="st">"28R8"</span>, <span class="st">"28R33"</span>, <span class="st">"28R21"</span>, <span class="st">"28P6"</span>, <span class="st">"28P1"</span>, <span class="st">"27R17"</span>,</span>
<span id="cb19-13"><a href="#cb19-13"></a>    <span class="st">"21R38"</span>, <span class="st">"21P1"</span>, <span class="st">"1R16"</span>, <span class="st">"19R18"</span>, <span class="st">"17P5"</span>, <span class="st">"16R37"</span>, <span class="st">"16R23"</span>, <span class="st">"14R6"</span>, <span class="st">"14R30"</span></span>
<span id="cb19-14"><a href="#cb19-14"></a>  )</span>
<span id="cb19-15"><a href="#cb19-15"></a>)</span>
<span id="cb19-16"><a href="#cb19-16"></a></span>
<span id="cb19-17"><a href="#cb19-17"></a>dataset<span class="sc">$</span>V1 <span class="ot">=</span> <span class="fu">factor</span>(</span>
<span id="cb19-18"><a href="#cb19-18"></a>  dataset<span class="sc">$</span>V1,</span>
<span id="cb19-19"><a href="#cb19-19"></a>  <span class="at">levels =</span> <span class="fu">c</span>(</span>
<span id="cb19-20"><a href="#cb19-20"></a>    <span class="st">"Copper tolerance"</span>, <span class="st">"POF"</span>, <span class="st">"maltose/maltotriose"</span>, <span class="st">"ethanol tolerance"</span>, <span class="st">"amylase activity"</span>,</span>
<span id="cb19-21"><a href="#cb19-21"></a>    <span class="st">"glycerol"</span>, <span class="st">"temperature tolerance"</span>, <span class="st">"heat-shock"</span>, <span class="st">"sulfites"</span>, <span class="st">"flocculation"</span>, <span class="st">"dessication"</span>,</span>
<span id="cb19-22"><a href="#cb19-22"></a>    <span class="st">"killer toxin"</span></span>
<span id="cb19-23"><a href="#cb19-23"></a>  )</span>
<span id="cb19-24"><a href="#cb19-24"></a>)</span>
<span id="cb19-25"><a href="#cb19-25"></a></span>
<span id="cb19-26"><a href="#cb19-26"></a><span class="co"># label wrapper function</span></span>
<span id="cb19-27"><a href="#cb19-27"></a>label_wrap_gen <span class="ot">=</span> <span class="cf">function</span>(<span class="at">width =</span> <span class="dv">25</span>) {</span>
<span id="cb19-28"><a href="#cb19-28"></a>  <span class="cf">function</span>(variable, value) {</span>
<span id="cb19-29"><a href="#cb19-29"></a>    <span class="fu">laply</span>(<span class="fu">strwrap</span>(<span class="fu">as.character</span>(value), <span class="at">width=</span>width, <span class="at">simplify=</span><span class="cn">FALSE</span>), </span>
<span id="cb19-30"><a href="#cb19-30"></a>          paste, <span class="at">collapse=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb19-31"><a href="#cb19-31"></a>  }</span>
<span id="cb19-32"><a href="#cb19-32"></a>}</span>
<span id="cb19-33"><a href="#cb19-33"></a></span>
<span id="cb19-34"><a href="#cb19-34"></a><span class="co"># plot</span></span>
<span id="cb19-35"><a href="#cb19-35"></a>p1 <span class="ot">=</span> <span class="fu">ggplot</span>(dataset) <span class="sc">+</span></span>
<span id="cb19-36"><a href="#cb19-36"></a>  <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">x =</span> V3, <span class="at">y =</span> V4, <span class="at">fill =</span> V6), <span class="at">color =</span> <span class="st">"grey75"</span>) <span class="sc">+</span></span>
<span id="cb19-37"><a href="#cb19-37"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">"white"</span>, <span class="at">high =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb19-38"><a href="#cb19-38"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>V1, <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free_x"</span>,</span>
<span id="cb19-39"><a href="#cb19-39"></a>             <span class="at">labeller=</span><span class="fu">label_wrap_gen</span>(<span class="at">width =</span> <span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb19-40"><a href="#cb19-40"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Premature stop codons distribution"</span>,</span>
<span id="cb19-41"><a href="#cb19-41"></a>       <span class="at">x =</span> <span class="st">"Genes"</span>,</span>
<span id="cb19-42"><a href="#cb19-42"></a>       <span class="at">y =</span> <span class="st">"Strains"</span>) <span class="sc">+</span></span>
<span id="cb19-43"><a href="#cb19-43"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-44"><a href="#cb19-44"></a>        <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-45"><a href="#cb19-45"></a>        <span class="at">axis.ticks.y =</span> <span class="fu">element_line</span>(</span>
<span id="cb19-46"><a href="#cb19-46"></a>          <span class="at">size =</span> <span class="fl">2.5</span>,</span>
<span id="cb19-47"><a href="#cb19-47"></a>          <span class="at">colour =</span> <span class="fu">c</span>(</span>
<span id="cb19-48"><a href="#cb19-48"></a>            <span class="st">"#FEE08B"</span>, <span class="st">"#FEE08B"</span>, <span class="st">"#FEE08B"</span>, <span class="st">"#7BCEB8"</span>, <span class="st">"#FFFFBF"</span>, <span class="st">"#FFFFBF"</span>, <span class="st">"#FFFFBF"</span>, <span class="st">"#FFFFBF"</span>,</span>
<span id="cb19-49"><a href="#cb19-49"></a>            <span class="st">"#FFFFBF"</span>, <span class="st">"#99D594"</span>, <span class="st">"#99D594"</span>, <span class="st">"#99D594"</span>, <span class="st">"#99D594"</span>, <span class="st">"#FC8D59"</span>, <span class="st">"#FC8D59"</span>, <span class="st">"#FC8D59"</span>, </span>
<span id="cb19-50"><a href="#cb19-50"></a>            <span class="st">"#FC8D59"</span>, <span class="st">"#FC8D59"</span>, <span class="st">"#3288BD"</span>, <span class="st">"#3288BD"</span>, <span class="st">"#3288BD"</span>, <span class="st">"#3288BD"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>,</span>
<span id="cb19-51"><a href="#cb19-51"></a>            <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>,</span>
<span id="cb19-52"><a href="#cb19-52"></a>            <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>,</span>
<span id="cb19-53"><a href="#cb19-53"></a>            <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>,</span>
<span id="cb19-54"><a href="#cb19-54"></a>            <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>,</span>
<span id="cb19-55"><a href="#cb19-55"></a>            <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span>,</span>
<span id="cb19-56"><a href="#cb19-56"></a>            <span class="st">"#D53E4F"</span>, <span class="st">"#D53E4F"</span></span>
<span id="cb19-57"><a href="#cb19-57"></a>          )</span>
<span id="cb19-58"><a href="#cb19-58"></a>        ),</span>
<span id="cb19-59"><a href="#cb19-59"></a>        <span class="at">axis.ticks.length.y =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">"lines"</span>),</span>
<span id="cb19-60"><a href="#cb19-60"></a>        <span class="at">axis.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-61"><a href="#cb19-61"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb19-62"><a href="#cb19-62"></a>        <span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb19-63"><a href="#cb19-63"></a>        <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-64"><a href="#cb19-64"></a>        <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="sc">-</span><span class="fl">0.2</span>, <span class="st">"lines"</span>),</span>
<span id="cb19-65"><a href="#cb19-65"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">fill =</span> <span class="cn">NA</span>),</span>
<span id="cb19-66"><a href="#cb19-66"></a>        <span class="at">strip.background =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">"NA"</span>, <span class="at">fill =</span> <span class="st">"NA"</span>),</span>
<span id="cb19-67"><a href="#cb19-67"></a>        <span class="at">strip.placement =</span> <span class="st">"outside"</span>,</span>
<span id="cb19-68"><a href="#cb19-68"></a>        <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">0</span>))</span>
<span id="cb19-69"><a href="#cb19-69"></a></span>
<span id="cb19-70"><a href="#cb19-70"></a><span class="do">### Construct species tables</span></span>
<span id="cb19-71"><a href="#cb19-71"></a><span class="co"># Construct the 14 grobs - 7 symbols and 7 labels</span></span>
<span id="cb19-72"><a href="#cb19-72"></a>L1 <span class="ot">=</span> <span class="fu">rectGrob</span>(<span class="at">height =</span> .<span class="dv">5</span>, <span class="at">width =</span> .<span class="dv">5</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#D53E4F"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb19-73"><a href="#cb19-73"></a>L2 <span class="ot">=</span> <span class="fu">rectGrob</span>(<span class="at">height =</span> .<span class="dv">5</span>, <span class="at">width =</span> .<span class="dv">5</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#3288BD"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb19-74"><a href="#cb19-74"></a>L3 <span class="ot">=</span> <span class="fu">rectGrob</span>(<span class="at">height =</span> .<span class="dv">5</span>, <span class="at">width =</span> .<span class="dv">5</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#FC8D59"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb19-75"><a href="#cb19-75"></a>L4 <span class="ot">=</span> <span class="fu">rectGrob</span>(<span class="at">height =</span> .<span class="dv">5</span>, <span class="at">width =</span> .<span class="dv">5</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#99D594"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb19-76"><a href="#cb19-76"></a>L5 <span class="ot">=</span> <span class="fu">rectGrob</span>(<span class="at">height =</span> .<span class="dv">5</span>, <span class="at">width =</span> .<span class="dv">5</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#FFFFBF"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb19-77"><a href="#cb19-77"></a>L6 <span class="ot">=</span> <span class="fu">rectGrob</span>(<span class="at">height =</span> .<span class="dv">5</span>, <span class="at">width =</span> .<span class="dv">5</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#7BCEB8"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb19-78"><a href="#cb19-78"></a>L7 <span class="ot">=</span> <span class="fu">rectGrob</span>(<span class="at">height =</span> .<span class="dv">5</span>, <span class="at">width =</span> .<span class="dv">5</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#FEE08B"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb19-79"><a href="#cb19-79"></a>T1 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Kveik"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb19-80"><a href="#cb19-80"></a>T2 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Asia"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb19-81"><a href="#cb19-81"></a>T3 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Beer 1"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb19-82"><a href="#cb19-82"></a>T4 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Beer 2"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb19-83"><a href="#cb19-83"></a>T5 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Mixed"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb19-84"><a href="#cb19-84"></a>T6 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Mosaic"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb19-85"><a href="#cb19-85"></a>T7 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Wine"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb19-86"><a href="#cb19-86"></a> </span>
<span id="cb19-87"><a href="#cb19-87"></a><span class="co"># Construct a gtable - 2 columns X 8 rows</span></span>
<span id="cb19-88"><a href="#cb19-88"></a>leg <span class="ot">=</span> <span class="fu">gtable</span>(<span class="at">width =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="st">"cm"</span>),</span>
<span id="cb19-89"><a href="#cb19-89"></a>             <span class="at">height =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>), <span class="st">"cm"</span>))</span>
<span id="cb19-90"><a href="#cb19-90"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, <span class="fu">rectGrob</span>(<span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">col =</span> <span class="st">"black"</span>)),</span>
<span id="cb19-91"><a href="#cb19-91"></a>                      <span class="at">t =</span> <span class="dv">1</span>, <span class="at">l =</span> <span class="dv">1</span>, <span class="at">b =</span> <span class="dv">8</span>, <span class="at">r =</span> <span class="dv">2</span>)</span>
<span id="cb19-92"><a href="#cb19-92"></a></span>
<span id="cb19-93"><a href="#cb19-93"></a><span class="co"># gtable_show_layout(leg)</span></span>
<span id="cb19-94"><a href="#cb19-94"></a><span class="co"># Place the 14 grob into the table</span></span>
<span id="cb19-95"><a href="#cb19-95"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L1, <span class="at">t=</span><span class="dv">2</span>, <span class="at">l=</span><span class="dv">1</span>)</span>
<span id="cb19-96"><a href="#cb19-96"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L2, <span class="at">t=</span><span class="dv">3</span>, <span class="at">l=</span><span class="dv">1</span>)</span>
<span id="cb19-97"><a href="#cb19-97"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L3, <span class="at">t=</span><span class="dv">4</span>, <span class="at">l=</span><span class="dv">1</span>)</span>
<span id="cb19-98"><a href="#cb19-98"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L4, <span class="at">t=</span><span class="dv">5</span>, <span class="at">l=</span><span class="dv">1</span>)</span>
<span id="cb19-99"><a href="#cb19-99"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L5, <span class="at">t=</span><span class="dv">6</span>, <span class="at">l=</span><span class="dv">1</span>)</span>
<span id="cb19-100"><a href="#cb19-100"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L6, <span class="at">t=</span><span class="dv">7</span>, <span class="at">l=</span><span class="dv">1</span>)</span>
<span id="cb19-101"><a href="#cb19-101"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L7, <span class="at">t=</span><span class="dv">8</span>, <span class="at">l=</span><span class="dv">1</span>)  </span>
<span id="cb19-102"><a href="#cb19-102"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T1, <span class="at">t=</span><span class="dv">2</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb19-103"><a href="#cb19-103"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T2, <span class="at">t=</span><span class="dv">3</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb19-104"><a href="#cb19-104"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T3, <span class="at">t=</span><span class="dv">4</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb19-105"><a href="#cb19-105"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T4, <span class="at">t=</span><span class="dv">5</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb19-106"><a href="#cb19-106"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T5, <span class="at">t=</span><span class="dv">6</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb19-107"><a href="#cb19-107"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T6, <span class="at">t=</span><span class="dv">7</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb19-108"><a href="#cb19-108"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T7, <span class="at">t=</span><span class="dv">8</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb19-109"><a href="#cb19-109"></a> </span>
<span id="cb19-110"><a href="#cb19-110"></a><span class="co"># Give it a title (if needed)</span></span>
<span id="cb19-111"><a href="#cb19-111"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, <span class="fu">textGrob</span>(<span class="st">"Population"</span>), <span class="at">t =</span> <span class="dv">1</span>, <span class="at">l =</span> <span class="dv">1</span>, <span class="at">r =</span> <span class="dv">2</span>)</span>
<span id="cb19-112"><a href="#cb19-112"></a> </span>
<span id="cb19-113"><a href="#cb19-113"></a><span class="co"># Get the ggplot grob for plot1</span></span>
<span id="cb19-114"><a href="#cb19-114"></a>g <span class="ot">=</span> <span class="fu">ggplotGrob</span>(p1)</span>
<span id="cb19-115"><a href="#cb19-115"></a> </span>
<span id="cb19-116"><a href="#cb19-116"></a><span class="co"># Get the position of the panel,</span></span>
<span id="cb19-117"><a href="#cb19-117"></a><span class="co"># add a column to the right of the panel,</span></span>
<span id="cb19-118"><a href="#cb19-118"></a><span class="co"># put the legend into that column,</span></span>
<span id="cb19-119"><a href="#cb19-119"></a><span class="co"># and then add another spacing column</span></span>
<span id="cb19-120"><a href="#cb19-120"></a>pos <span class="ot">=</span> g<span class="sc">$</span>layout[<span class="fu">grepl</span>(<span class="st">"panel"</span>, g<span class="sc">$</span>layout<span class="sc">$</span>name), <span class="fu">c</span>(<span class="st">'t'</span>, <span class="st">'l'</span>)]</span>
<span id="cb19-121"><a href="#cb19-121"></a>g <span class="ot">=</span> <span class="fu">gtable_add_cols</span>(g, <span class="fu">sum</span>(leg<span class="sc">$</span>widths), <span class="fu">tail</span>(pos<span class="sc">$</span>l, <span class="at">n =</span><span class="dv">1</span>))</span>
<span id="cb19-122"><a href="#cb19-122"></a>g <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(g, leg, <span class="at">t =</span> pos<span class="sc">$</span>t[<span class="dv">1</span>], <span class="at">l =</span> <span class="fu">tail</span>(pos<span class="sc">$</span>l, <span class="at">n =</span><span class="dv">1</span>) <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb19-123"><a href="#cb19-123"></a>g <span class="ot">=</span> <span class="fu">gtable_add_cols</span>(g, <span class="fu">unit</span>(<span class="dv">6</span>, <span class="st">"pt"</span>), <span class="fu">tail</span>(pos<span class="sc">$</span>l, <span class="at">n =</span><span class="dv">1</span>))</span>
<span id="cb19-124"><a href="#cb19-124"></a> </span>
<span id="cb19-125"><a href="#cb19-125"></a><span class="co"># Draw it</span></span>
<span id="cb19-126"><a href="#cb19-126"></a><span class="fu">grid.newpage</span>()</span>
<span id="cb19-127"><a href="#cb19-127"></a><span class="fu">grid.draw</span>(g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="p01-03-variant-calling_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption><span style="color: steelblue;">Figure 18: Premature stop codons distribution.</span></figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="phylogenetic-relationships" class="level3" data-number="3.6.5">
<h3 data-number="3.6.5" class="anchored" data-anchor-id="phylogenetic-relationships"><span class="header-section-number">3.6.5</span> Phylogenetic relationships</h3>
<p>Since it is difficult to eye-ball patterns of distributions of variants among strains by eye, we can use a phylogeny approach to visualize these patterns and genetic distances. The idea behind this, is to infer phylogenetic relationships based only ina small set of genes involved in a particular phenotype. While these signals are lost in a massive multigene phylogenetic analysis, they still can be informative and give hints on how to cluster strains based on their industrial phenotypes.</p>
<p>The protocol we will apply is the following:</p>
<ul>
<li>For each strain of interest (kviks + 22 industrial), generate the consensus fasta file from the aligned variants.</li>
<li>Generate gene alignments.</li>
<li>Concatenate all the alignments for each of the 12 industrial phenotypes of interest.</li>
<li></li>
<li>Build a Maximum-Likelihood phylogenetic tree with IQtree.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1"></a><span class="va">SAMPLES</span><span class="op">=</span><span class="st">"14R30 14R6 16R23 16R37 17P5 19R18 1R16 21P1 21R38 27R17 28P1 28P6 28R21 28R33 28R8 2R23 38R16 39R20 3R11 40R14 40R1 40R20 41R10 41R15 42R20 42R31 44R32 44R7 45P5 45R11 46R12 46R37 6R15 7R7 8R19 9R40 Granvin1 Hornindal1 Hornindal2 k7R25 Laerdal2 Muri SortdalEbbe1 Voss1 X1002_all_sorted_picard_dedup_all X1007_all_sorted_picard_dedup_all X1011_all_sorted_picard_dedup_realigned_recalibrated_all X1014_all_sorted_picard_dedup_all X1030_all_sorted_picard_dedup_all X1046_all_sorted_picard_dedup_realigned_recalibrated_all X1091_all_sorted_picard_dedup_all X1111_all_sorted_picard_dedup_all X1118_all_sorted_picard_dedup_all X1148_all_sorted_picard_dedup_all X35_all_sorted_picard_dedup_all X4037_all_sorted_picard_dedup_all X4051_all_sorted_picard_dedup_all X5000_all_sorted_picard_dedup_all X5009_all_sorted_picard_dedup_all X5508_all_sorted_picard_dedup_all X6004_all_sorted_picard_dedup_all X8003_all_sorted_picard_dedup_all x1119_all_sorted_picard_dedup_all x1129_all_sorted_picard_dedup_all x7078-B x97"</span></span>
<span id="cb20-2"><a href="#cb20-2"></a></span>
<span id="cb20-3"><a href="#cb20-3"></a><span class="co"># generate the fasta files for the genes of interest</span></span>
<span id="cb20-4"><a href="#cb20-4"></a><span class="ex">~/bin/bcftools-1.9/bcftools</span> view <span class="dt">\</span></span>
<span id="cb20-5"><a href="#cb20-5"></a>    <span class="at">-h</span> all_plusKV.fltr.backup.reduced.snpEff.vcf.gz <span class="dt">\</span></span>
<span id="cb20-6"><a href="#cb20-6"></a>    <span class="op">&gt;</span> all_plusKV.fltr.backup.header</span>
<span id="cb20-7"><a href="#cb20-7"></a>    </span>
<span id="cb20-8"><a href="#cb20-8"></a><span class="cf">for</span> file <span class="kw">in</span> 0<span class="pp">[</span><span class="ss">1</span><span class="pp">-</span><span class="ss">9</span><span class="pp">]*</span>/<span class="pp">*</span>vcf 1<span class="pp">[</span><span class="ss">0</span><span class="pp">-</span><span class="ss">2</span><span class="pp">]*</span>/<span class="pp">*</span>vcf<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb20-9"><a href="#cb20-9"></a></span>
<span id="cb20-10"><a href="#cb20-10"></a>    <span class="fu">cat</span> all_plusKV.fltr.backup.header <span class="va">$file</span> <span class="op">&gt;</span> <span class="va">$file</span>.1<span class="kw">;</span></span>
<span id="cb20-11"><a href="#cb20-11"></a>    <span class="fu">mv</span> <span class="va">$file</span>.1 <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb20-12"><a href="#cb20-12"></a>    <span class="ex">bgzip</span> <span class="at">-c</span> <span class="va">$file</span> <span class="op">&gt;</span> <span class="va">$file</span>.gz<span class="kw">;</span></span>
<span id="cb20-13"><a href="#cb20-13"></a>    <span class="ex">tabix</span> <span class="va">$file</span>.gz<span class="kw">;</span></span>
<span id="cb20-14"><a href="#cb20-14"></a>    <span class="va">GENE</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$file</span> <span class="kw">|</span> <span class="bu">echo</span> <span class="va">$file</span> <span class="kw">|</span> <span class="fu">tr</span> <span class="st">"."</span> <span class="st">"\t"</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 2<span class="va">)</span><span class="kw">;</span></span>
<span id="cb20-15"><a href="#cb20-15"></a>    <span class="va">CHR</span><span class="op">=</span><span class="va">$(</span><span class="fu">grep</span> <span class="va">$GENE</span> Domestication.genes.windows.lst <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 4<span class="va">)</span><span class="kw">;</span></span>
<span id="cb20-16"><a href="#cb20-16"></a>    <span class="va">START</span><span class="op">=</span><span class="va">$(</span><span class="fu">grep</span> <span class="va">$GENE</span> Domestication.genes.windows.lst <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 5<span class="va">)</span><span class="kw">;</span></span>
<span id="cb20-17"><a href="#cb20-17"></a>    <span class="va">STOP</span><span class="op">=</span><span class="va">$(</span><span class="fu">grep</span> <span class="va">$GENE</span> Domestication.genes.windows.lst <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 6<span class="va">)</span><span class="kw">;</span></span>
<span id="cb20-18"><a href="#cb20-18"></a>    </span>
<span id="cb20-19"><a href="#cb20-19"></a>    <span class="cf">for</span> SAMPLE <span class="kw">in</span> <span class="va">$SAMPLES</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb20-20"><a href="#cb20-20"></a>        <span class="ex">~/bin/samtools-1.9/samtools</span> faidx <span class="dt">\</span></span>
<span id="cb20-21"><a href="#cb20-21"></a>            ../09_CNVs/00_refgen/Saccharomyces_cerevisiae.EF4.73.dna.chromosome.all.fa <span class="dt">\</span></span>
<span id="cb20-22"><a href="#cb20-22"></a>            <span class="st">"</span><span class="va">${CHR}</span><span class="st">"</span>:<span class="st">"</span><span class="va">${START}</span><span class="st">"</span>-<span class="st">"</span><span class="va">${STOP}</span><span class="st">"</span> <span class="kw">|</span><span class="dt">\</span></span>
<span id="cb20-23"><a href="#cb20-23"></a>            <span class="ex">~/bin/bcftools-1.9/bcftools</span> consensus <span class="dt">\</span></span>
<span id="cb20-24"><a href="#cb20-24"></a>                <span class="at">--haplotype</span> A <span class="dt">\</span></span>
<span id="cb20-25"><a href="#cb20-25"></a>                <span class="at">--missing</span> <span class="at">-</span> <span class="dt">\</span></span>
<span id="cb20-26"><a href="#cb20-26"></a>                <span class="at">--sample</span> <span class="va">$SAMPLE</span> <span class="va">$file</span>.gz <span class="dt">\</span></span>
<span id="cb20-27"><a href="#cb20-27"></a>              <span class="kw">|</span> <span class="fu">sed</span> <span class="st">"s/&gt;.*/&gt;</span><span class="va">$SAMPLE</span><span class="st">/g"</span><span class="kw">;</span></span>
<span id="cb20-28"><a href="#cb20-28"></a>    <span class="cf">done</span> <span class="op">&gt;</span> <span class="va">$(</span><span class="bu">echo</span> <span class="va">$file</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/vcf/fa/g'</span><span class="va">)</span><span class="kw">;</span></span>
<span id="cb20-29"><a href="#cb20-29"></a>    </span>
<span id="cb20-30"><a href="#cb20-30"></a><span class="cf">done</span> </span>
<span id="cb20-31"><a href="#cb20-31"></a></span>
<span id="cb20-32"><a href="#cb20-32"></a><span class="co"># alignments</span></span>
<span id="cb20-33"><a href="#cb20-33"></a><span class="cf">for</span> file <span class="kw">in</span> 00_nt_genes/<span class="pp">*</span>.fa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb20-34"><a href="#cb20-34"></a>    <span class="ex">mafft</span> <span class="at">--localpair</span> <span class="dt">\</span></span>
<span id="cb20-35"><a href="#cb20-35"></a>      <span class="at">--maxiterate</span> 1000 <span class="dt">\</span></span>
<span id="cb20-36"><a href="#cb20-36"></a>        <span class="at">--nuc</span> <span class="dt">\</span></span>
<span id="cb20-37"><a href="#cb20-37"></a>        <span class="at">--ep</span> 0.123 <span class="dt">\</span></span>
<span id="cb20-38"><a href="#cb20-38"></a>        <span class="at">--thread</span> 32 <span class="dt">\</span></span>
<span id="cb20-39"><a href="#cb20-39"></a>        <span class="va">$file</span> <span class="dt">\</span></span>
<span id="cb20-40"><a href="#cb20-40"></a>        <span class="op">&gt;</span> 01_mafft_align/<span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .fa<span class="va">)</span>.align.fa<span class="kw">;</span></span>
<span id="cb20-41"><a href="#cb20-41"></a><span class="cf">done</span></span>
<span id="cb20-42"><a href="#cb20-42"></a></span>
<span id="cb20-43"><a href="#cb20-43"></a><span class="cf">for</span> file <span class="kw">in</span> 01_mafft_align/<span class="pp">*</span>.fa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb20-44"><a href="#cb20-44"></a>    <span class="fu">sed</span> <span class="at">-i</span> <span class="st">'s/_all_sorted_picard_dedup_all//g'</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb20-45"><a href="#cb20-45"></a>    <span class="fu">sed</span> <span class="at">-i</span> <span class="st">'s/_all_sorted_picard_dedup_realigned_recalibrated_all//g'</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb20-46"><a href="#cb20-46"></a><span class="cf">done</span></span>
<span id="cb20-47"><a href="#cb20-47"></a></span>
<span id="cb20-48"><a href="#cb20-48"></a><span class="co"># concatenate and ML tree</span></span>
<span id="cb20-49"><a href="#cb20-49"></a><span class="co"># I filter out Muri and k7R25, since they are hybrid and </span></span>
<span id="cb20-50"><a href="#cb20-50"></a><span class="cf">for</span> WORD <span class="kw">in</span> amylase_activity <span class="dt">\</span></span>
<span id="cb20-51"><a href="#cb20-51"></a>  Copper_tolerance <span class="dt">\</span></span>
<span id="cb20-52"><a href="#cb20-52"></a>  dessication <span class="dt">\</span></span>
<span id="cb20-53"><a href="#cb20-53"></a>    ethanol_tolerance <span class="dt">\</span></span>
<span id="cb20-54"><a href="#cb20-54"></a>    flocculation <span class="dt">\</span></span>
<span id="cb20-55"><a href="#cb20-55"></a>    glycerol <span class="dt">\</span></span>
<span id="cb20-56"><a href="#cb20-56"></a>    heat-shock <span class="dt">\</span></span>
<span id="cb20-57"><a href="#cb20-57"></a>    killer_toxin <span class="dt">\</span></span>
<span id="cb20-58"><a href="#cb20-58"></a>    maltose_maltotriose <span class="dt">\</span></span>
<span id="cb20-59"><a href="#cb20-59"></a>    POF <span class="dt">\</span></span>
<span id="cb20-60"><a href="#cb20-60"></a>    sulfites <span class="dt">\</span></span>
<span id="cb20-61"><a href="#cb20-61"></a>    temperature_rolerance<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb20-62"><a href="#cb20-62"></a>    <span class="fu">perl</span> ~/scripts/catfasta2phyml.pl <span class="dt">\</span></span>
<span id="cb20-63"><a href="#cb20-63"></a>        <span class="at">--concatenate</span> <span class="dt">\</span></span>
<span id="cb20-64"><a href="#cb20-64"></a>        <span class="at">--sequential</span> 01_mafft_align/<span class="va">$WORD</span>.<span class="pp">*</span>fa <span class="dt">\</span></span>
<span id="cb20-65"><a href="#cb20-65"></a>        <span class="op">&gt;</span> 03_concatenated_align/<span class="va">$WORD</span>.allgenes.fa<span class="kw">;</span></span>
<span id="cb20-66"><a href="#cb20-66"></a><span class="cf">done</span></span>
<span id="cb20-67"><a href="#cb20-67"></a></span>
<span id="cb20-68"><a href="#cb20-68"></a><span class="cf">for</span> file <span class="kw">in</span> 03_concatenated_align/<span class="pp">*</span>.fa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb20-69"><a href="#cb20-69"></a>    <span class="ex">iqtree-omp</span> <span class="at">-s</span> <span class="va">$file</span> <span class="dt">\</span></span>
<span id="cb20-70"><a href="#cb20-70"></a>      <span class="at">-st</span> DNA <span class="dt">\</span></span>
<span id="cb20-71"><a href="#cb20-71"></a>        <span class="at">-pre</span> <span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .fa<span class="va">)</span> <span class="dt">\</span></span>
<span id="cb20-72"><a href="#cb20-72"></a>        <span class="at">-nt</span> 72 <span class="dt">\</span></span>
<span id="cb20-73"><a href="#cb20-73"></a>        <span class="at">-bb</span> 1000 <span class="dt">\</span></span>
<span id="cb20-74"><a href="#cb20-74"></a>        <span class="at">-wbt</span> <span class="dt">\</span></span>
<span id="cb20-75"><a href="#cb20-75"></a>        <span class="at">-alrt</span> 1000 <span class="dt">\</span></span>
<span id="cb20-76"><a href="#cb20-76"></a>        <span class="at">-m</span> GTR+G<span class="kw">;</span></span>
<span id="cb20-77"><a href="#cb20-77"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a><span class="co"># upload files</span></span>
<span id="cb21-2"><a href="#cb21-2"></a>phenotypes <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb21-3"><a href="#cb21-3"></a>  <span class="st">"Copper_tolerance"</span>,</span>
<span id="cb21-4"><a href="#cb21-4"></a>  <span class="st">"POF"</span>,</span>
<span id="cb21-5"><a href="#cb21-5"></a>  <span class="st">"maltose_maltotriose"</span>,</span>
<span id="cb21-6"><a href="#cb21-6"></a>  <span class="st">"ethanol_tolerance"</span>,</span>
<span id="cb21-7"><a href="#cb21-7"></a>  <span class="st">"amylase_activity"</span>,</span>
<span id="cb21-8"><a href="#cb21-8"></a>  <span class="st">"glycerol"</span>,</span>
<span id="cb21-9"><a href="#cb21-9"></a>  <span class="st">"temperature_rolerance"</span>,</span>
<span id="cb21-10"><a href="#cb21-10"></a>  <span class="st">"heat-shock"</span>,</span>
<span id="cb21-11"><a href="#cb21-11"></a>  <span class="st">"sulfites"</span>,</span>
<span id="cb21-12"><a href="#cb21-12"></a>  <span class="st">"flocculation"</span>,</span>
<span id="cb21-13"><a href="#cb21-13"></a>  <span class="st">"dessication"</span>,</span>
<span id="cb21-14"><a href="#cb21-14"></a>  <span class="st">"killer_toxin"</span></span>
<span id="cb21-15"><a href="#cb21-15"></a>)</span>
<span id="cb21-16"><a href="#cb21-16"></a></span>
<span id="cb21-17"><a href="#cb21-17"></a>headers <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb21-18"><a href="#cb21-18"></a>  <span class="st">"Copper Tolerance"</span>,</span>
<span id="cb21-19"><a href="#cb21-19"></a>  <span class="st">"Phenolic Off-flavours"</span>,</span>
<span id="cb21-20"><a href="#cb21-20"></a>  <span class="st">"Maltose Utilization"</span>,</span>
<span id="cb21-21"><a href="#cb21-21"></a>  <span class="st">"Ethanol Resistance"</span>,</span>
<span id="cb21-22"><a href="#cb21-22"></a>  <span class="st">"Amilase Activity"</span>,</span>
<span id="cb21-23"><a href="#cb21-23"></a>  <span class="st">"Glycerol Production"</span>,</span>
<span id="cb21-24"><a href="#cb21-24"></a>  <span class="st">"Temperature Tolerance"</span>,</span>
<span id="cb21-25"><a href="#cb21-25"></a>  <span class="st">"Heat-shock Proteins"</span>,</span>
<span id="cb21-26"><a href="#cb21-26"></a>  <span class="st">"Sulfites Production"</span>,</span>
<span id="cb21-27"><a href="#cb21-27"></a>  <span class="st">"Flocculation"</span>,</span>
<span id="cb21-28"><a href="#cb21-28"></a>  <span class="st">"Dessication Tolerance"</span>,</span>
<span id="cb21-29"><a href="#cb21-29"></a>  <span class="st">"Killer Toxins"</span></span>
<span id="cb21-30"><a href="#cb21-30"></a>)</span>
<span id="cb21-31"><a href="#cb21-31"></a></span>
<span id="cb21-32"><a href="#cb21-32"></a>info <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">"./data/p01-03/variants.tree.tiplabels.csv"</span>)</span>
<span id="cb21-33"><a href="#cb21-33"></a></span>
<span id="cb21-34"><a href="#cb21-34"></a>p <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb21-35"><a href="#cb21-35"></a></span>
<span id="cb21-36"><a href="#cb21-36"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">12</span>){</span>
<span id="cb21-37"><a href="#cb21-37"></a>  </span>
<span id="cb21-38"><a href="#cb21-38"></a>  <span class="co"># read the tree and root it</span></span>
<span id="cb21-39"><a href="#cb21-39"></a>  name <span class="ot">=</span> phenotypes[i]</span>
<span id="cb21-40"><a href="#cb21-40"></a>  file <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">"./data/p01-03/"</span>, name, <span class="st">".allgenes.contree"</span>)</span>
<span id="cb21-41"><a href="#cb21-41"></a>  tree <span class="ot">=</span> <span class="fu">read.newick</span>(file)</span>
<span id="cb21-42"><a href="#cb21-42"></a>  tree <span class="ot">=</span> <span class="fu">root</span>(tree, <span class="fu">which</span>(tree<span class="sc">$</span>tip.label <span class="sc">==</span> <span class="st">"S288C"</span>))</span>
<span id="cb21-43"><a href="#cb21-43"></a></span>
<span id="cb21-44"><a href="#cb21-44"></a>  max_axis <span class="ot">=</span> <span class="fu">max</span>(<span class="fu">nodeHeights</span>(tree))<span class="sc">*</span><span class="fl">1.09</span></span>
<span id="cb21-45"><a href="#cb21-45"></a>  </span>
<span id="cb21-46"><a href="#cb21-46"></a>  p[[i]] <span class="ot">=</span> <span class="fu">ggtree</span>(tree, <span class="at">ladderize =</span> <span class="cn">FALSE</span>) <span class="sc">%&lt;+%</span> info <span class="sc">+</span></span>
<span id="cb21-47"><a href="#cb21-47"></a>    <span class="fu">geom_tiplab</span>(<span class="at">align =</span> <span class="cn">TRUE</span>, <span class="at">size =</span> <span class="fl">1.75</span>) <span class="sc">+</span></span>
<span id="cb21-48"><a href="#cb21-48"></a>    <span class="fu">geom_tippoint</span>(<span class="fu">aes</span>(<span class="at">fill =</span> population, <span class="at">size =</span> <span class="fl">2.5</span>), <span class="at">shape =</span> <span class="dv">21</span>) <span class="sc">+</span></span>
<span id="cb21-49"><a href="#cb21-49"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="at">Kveik =</span> <span class="st">"#D53E4F"</span>, <span class="at">Mixed =</span> <span class="st">"#FFFFBF"</span>, <span class="at">Mosaic =</span> <span class="st">"#7BCEB8"</span>,</span>
<span id="cb21-50"><a href="#cb21-50"></a>                                  <span class="at">Wine =</span> <span class="st">"#FEE08B"</span>, <span class="at">Asia =</span> <span class="st">"#3288BD"</span>, <span class="at">Beer1 =</span> <span class="st">"#FC8D59"</span>,</span>
<span id="cb21-51"><a href="#cb21-51"></a>                                  <span class="at">Beer2 =</span> <span class="st">"#99D594"</span>, <span class="at">Lab =</span> <span class="st">"grey"</span>)) <span class="sc">+</span></span>
<span id="cb21-52"><a href="#cb21-52"></a>    <span class="fu">labs</span>(<span class="at">title =</span> headers[[i]]) <span class="sc">+</span></span>
<span id="cb21-53"><a href="#cb21-53"></a>    <span class="fu">xlim</span>(<span class="dv">0</span>, max_axis) <span class="sc">+</span></span>
<span id="cb21-54"><a href="#cb21-54"></a>    <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb21-55"><a href="#cb21-55"></a>    <span class="fu">theme_tree</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb21-56"><a href="#cb21-56"></a>               <span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb21-57"><a href="#cb21-57"></a>  </span>
<span id="cb21-58"><a href="#cb21-58"></a>}</span>
<span id="cb21-59"><a href="#cb21-59"></a></span>
<span id="cb21-60"><a href="#cb21-60"></a>myplots <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb21-61"><a href="#cb21-61"></a>  p[[<span class="dv">1</span>]], p[[<span class="dv">2</span>]], p[[<span class="dv">3</span>]], p[[<span class="dv">4</span>]],</span>
<span id="cb21-62"><a href="#cb21-62"></a>  p[[<span class="dv">5</span>]], p[[<span class="dv">6</span>]], p[[<span class="dv">7</span>]], p[[<span class="dv">8</span>]],</span>
<span id="cb21-63"><a href="#cb21-63"></a>  p[[<span class="dv">9</span>]], p[[<span class="dv">10</span>]], p[[<span class="dv">11</span>]], p[[<span class="dv">12</span>]]</span>
<span id="cb21-64"><a href="#cb21-64"></a>)</span>
<span id="cb21-65"><a href="#cb21-65"></a></span>
<span id="cb21-66"><a href="#cb21-66"></a>p1 <span class="ot">=</span> <span class="fu">ggarrange</span>(<span class="at">plotlist =</span> myplots, <span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">3</span>))</span>
<span id="cb21-67"><a href="#cb21-67"></a></span>
<span id="cb21-68"><a href="#cb21-68"></a><span class="do">### Construct species tables</span></span>
<span id="cb21-69"><a href="#cb21-69"></a><span class="co"># Construct the 16 grobs - 8 symbols and 8 labels</span></span>
<span id="cb21-70"><a href="#cb21-70"></a>L1 <span class="ot">=</span> <span class="fu">circleGrob</span>(<span class="at">x=</span><span class="fl">0.5</span>, <span class="at">y=</span><span class="fl">0.5</span>, <span class="at">r=</span><span class="fl">0.25</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#D53E4F"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb21-71"><a href="#cb21-71"></a>L2 <span class="ot">=</span> <span class="fu">circleGrob</span>(<span class="at">x=</span><span class="fl">0.5</span>, <span class="at">y=</span><span class="fl">0.5</span>, <span class="at">r=</span><span class="fl">0.25</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#3288BD"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb21-72"><a href="#cb21-72"></a>L3 <span class="ot">=</span> <span class="fu">circleGrob</span>(<span class="at">x=</span><span class="fl">0.5</span>, <span class="at">y=</span><span class="fl">0.5</span>, <span class="at">r=</span><span class="fl">0.25</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#FC8D59"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb21-73"><a href="#cb21-73"></a>L4 <span class="ot">=</span> <span class="fu">circleGrob</span>(<span class="at">x=</span><span class="fl">0.5</span>, <span class="at">y=</span><span class="fl">0.5</span>, <span class="at">r=</span><span class="fl">0.25</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#99D594"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb21-74"><a href="#cb21-74"></a>L5 <span class="ot">=</span> <span class="fu">circleGrob</span>(<span class="at">x=</span><span class="fl">0.5</span>, <span class="at">y=</span><span class="fl">0.5</span>, <span class="at">r=</span><span class="fl">0.25</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#FFFFBF"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb21-75"><a href="#cb21-75"></a>L6 <span class="ot">=</span> <span class="fu">circleGrob</span>(<span class="at">x=</span><span class="fl">0.5</span>, <span class="at">y=</span><span class="fl">0.5</span>, <span class="at">r=</span><span class="fl">0.25</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#7BCEB8"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb21-76"><a href="#cb21-76"></a>L7 <span class="ot">=</span> <span class="fu">circleGrob</span>(<span class="at">x=</span><span class="fl">0.5</span>, <span class="at">y=</span><span class="fl">0.5</span>, <span class="at">r=</span><span class="fl">0.25</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"#FEE08B"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb21-77"><a href="#cb21-77"></a>L8 <span class="ot">=</span> <span class="fu">circleGrob</span>(<span class="at">x=</span><span class="fl">0.5</span>, <span class="at">y=</span><span class="fl">0.5</span>, <span class="at">r=</span><span class="fl">0.25</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"grey"</span>, <span class="at">col =</span> <span class="st">"black"</span>))</span>
<span id="cb21-78"><a href="#cb21-78"></a>T1 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Farmhouse"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb21-79"><a href="#cb21-79"></a>T2 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Asia"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb21-80"><a href="#cb21-80"></a>T3 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Beer 1"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb21-81"><a href="#cb21-81"></a>T4 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Beer 2"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb21-82"><a href="#cb21-82"></a>T5 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Mixed"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb21-83"><a href="#cb21-83"></a>T6 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Mosaic"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb21-84"><a href="#cb21-84"></a>T7 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Wine"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb21-85"><a href="#cb21-85"></a>T8 <span class="ot">=</span> <span class="fu">textGrob</span>(<span class="st">"Lab"</span>, <span class="at">x =</span> .<span class="dv">2</span>, <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb21-86"><a href="#cb21-86"></a></span>
<span id="cb21-87"><a href="#cb21-87"></a><span class="co"># Construct a gtable - 2 columns X 8 rows</span></span>
<span id="cb21-88"><a href="#cb21-88"></a>leg <span class="ot">=</span> <span class="fu">gtable</span>(<span class="at">width =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="st">"cm"</span>),</span>
<span id="cb21-89"><a href="#cb21-89"></a>             <span class="at">height =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>), <span class="st">"cm"</span>))</span>
<span id="cb21-90"><a href="#cb21-90"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, <span class="fu">rectGrob</span>(<span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">col =</span> <span class="st">"black"</span>)),</span>
<span id="cb21-91"><a href="#cb21-91"></a>                      <span class="at">t =</span> <span class="dv">1</span>, <span class="at">l =</span> <span class="dv">1</span>, <span class="at">b =</span> <span class="dv">9</span>, <span class="at">r =</span> <span class="dv">2</span>)</span>
<span id="cb21-92"><a href="#cb21-92"></a></span>
<span id="cb21-93"><a href="#cb21-93"></a><span class="co"># gtable_show_layout(leg)</span></span>
<span id="cb21-94"><a href="#cb21-94"></a><span class="co"># Place the 14 grob into the table</span></span>
<span id="cb21-95"><a href="#cb21-95"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L1, <span class="at">t=</span><span class="dv">2</span>, <span class="at">l=</span><span class="dv">1</span>)</span>
<span id="cb21-96"><a href="#cb21-96"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L2, <span class="at">t=</span><span class="dv">3</span>, <span class="at">l=</span><span class="dv">1</span>)</span>
<span id="cb21-97"><a href="#cb21-97"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L3, <span class="at">t=</span><span class="dv">4</span>, <span class="at">l=</span><span class="dv">1</span>)</span>
<span id="cb21-98"><a href="#cb21-98"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L4, <span class="at">t=</span><span class="dv">5</span>, <span class="at">l=</span><span class="dv">1</span>)</span>
<span id="cb21-99"><a href="#cb21-99"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L5, <span class="at">t=</span><span class="dv">6</span>, <span class="at">l=</span><span class="dv">1</span>)</span>
<span id="cb21-100"><a href="#cb21-100"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L6, <span class="at">t=</span><span class="dv">7</span>, <span class="at">l=</span><span class="dv">1</span>)</span>
<span id="cb21-101"><a href="#cb21-101"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L7, <span class="at">t=</span><span class="dv">8</span>, <span class="at">l=</span><span class="dv">1</span>)</span>
<span id="cb21-102"><a href="#cb21-102"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, L8, <span class="at">t=</span><span class="dv">9</span>, <span class="at">l=</span><span class="dv">1</span>)</span>
<span id="cb21-103"><a href="#cb21-103"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T1, <span class="at">t=</span><span class="dv">2</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb21-104"><a href="#cb21-104"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T2, <span class="at">t=</span><span class="dv">3</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb21-105"><a href="#cb21-105"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T3, <span class="at">t=</span><span class="dv">4</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb21-106"><a href="#cb21-106"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T4, <span class="at">t=</span><span class="dv">5</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb21-107"><a href="#cb21-107"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T5, <span class="at">t=</span><span class="dv">6</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb21-108"><a href="#cb21-108"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T6, <span class="at">t=</span><span class="dv">7</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb21-109"><a href="#cb21-109"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T7, <span class="at">t=</span><span class="dv">8</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb21-110"><a href="#cb21-110"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, T8, <span class="at">t=</span><span class="dv">9</span>, <span class="at">l=</span><span class="dv">2</span>)</span>
<span id="cb21-111"><a href="#cb21-111"></a></span>
<span id="cb21-112"><a href="#cb21-112"></a><span class="co"># Give it a title (if needed)</span></span>
<span id="cb21-113"><a href="#cb21-113"></a>leg <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(leg, <span class="fu">textGrob</span>(<span class="st">"Population"</span>), <span class="at">t =</span> <span class="dv">1</span>, <span class="at">l =</span> <span class="dv">1</span>, <span class="at">r =</span> <span class="dv">2</span>)</span>
<span id="cb21-114"><a href="#cb21-114"></a></span>
<span id="cb21-115"><a href="#cb21-115"></a><span class="co"># Get the ggplot grob for plot1</span></span>
<span id="cb21-116"><a href="#cb21-116"></a>g <span class="ot">=</span> <span class="fu">ggplotGrob</span>(p1)</span>
<span id="cb21-117"><a href="#cb21-117"></a></span>
<span id="cb21-118"><a href="#cb21-118"></a><span class="co"># Get the position of the panel,</span></span>
<span id="cb21-119"><a href="#cb21-119"></a><span class="co"># add a column to the right of the panel, </span></span>
<span id="cb21-120"><a href="#cb21-120"></a><span class="co"># put the legend into that column, </span></span>
<span id="cb21-121"><a href="#cb21-121"></a><span class="co"># and then add another spacing column</span></span>
<span id="cb21-122"><a href="#cb21-122"></a>pos <span class="ot">=</span> g<span class="sc">$</span>layout[<span class="fu">grepl</span>(<span class="st">"panel"</span>, g<span class="sc">$</span>layout<span class="sc">$</span>name), <span class="fu">c</span>(<span class="st">'t'</span>, <span class="st">'l'</span>)]</span>
<span id="cb21-123"><a href="#cb21-123"></a>g <span class="ot">=</span> <span class="fu">gtable_add_cols</span>(g, <span class="fu">sum</span>(leg<span class="sc">$</span>widths), <span class="fu">tail</span>(pos<span class="sc">$</span>l, <span class="at">n =</span><span class="dv">1</span>))</span>
<span id="cb21-124"><a href="#cb21-124"></a>g <span class="ot">=</span> <span class="fu">gtable_add_grob</span>(g, leg, <span class="at">t =</span> pos<span class="sc">$</span>t[<span class="dv">1</span>], <span class="at">l =</span> <span class="fu">tail</span>(pos<span class="sc">$</span>l, <span class="at">n =</span><span class="dv">1</span>) <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb21-125"><a href="#cb21-125"></a>g <span class="ot">=</span> <span class="fu">gtable_add_cols</span>(g, <span class="fu">unit</span>(<span class="dv">6</span>, <span class="st">"pt"</span>), <span class="fu">tail</span>(pos<span class="sc">$</span>l, <span class="at">n =</span><span class="dv">1</span>))</span>
<span id="cb21-126"><a href="#cb21-126"></a></span>
<span id="cb21-127"><a href="#cb21-127"></a><span class="co"># Draw it</span></span>
<span id="cb21-128"><a href="#cb21-128"></a><span class="fu">grid.newpage</span>()</span>
<span id="cb21-129"><a href="#cb21-129"></a><span class="fu">grid.draw</span>(g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="p01-03-variant-calling_files/figure-html/unnamed-chunk-34-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption><span style="color: steelblue;">Figure 19: Domestication gene trees.</span></figcaption>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="sec-lesson-learnt-feature-selection" class="level2" data-number="3.7">
<h2 data-number="3.7" class="anchored" data-anchor-id="sec-lesson-learnt-feature-selection"><span class="header-section-number">3.7</span> Lessons Learnt</h2>
<p>Based on the we have learnt:</p>
<ul>
<li>Fr</li>
</ul>
</section>
<section id="session-information" class="level2" data-number="3.8">
<h2 data-number="3.8" class="anchored" data-anchor-id="session-information"><span class="header-section-number">3.8</span> Session Information</h2>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-1-contents" aria-controls="callout-1" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-1" class="callout-1-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.3.2 (2023-10-31)
Platform: x86_64-conda-linux-gnu (64-bit)
Running under: openSUSE Tumbleweed

Matrix products: default
BLAS/LAPACK: /home/andrea/miniforge3/envs/moai/lib/libmkl_rt.so.2;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=it_IT.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Europe/Brussels
tzcode source: system (glibc)

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] UpSetR_1.4.0           treeio_1.26.0          stringr_1.5.1         
 [4] reshape2_1.4.4         reshape_0.8.9          RColorBrewer_1.1-3    
 [7] plyr_1.8.9             phytools_2.4-4         maps_3.4.2.1          
[10] org.Sc.sgd.db_3.18.0   AnnotationDbi_1.64.1   IRanges_2.36.0        
[13] S4Vectors_0.40.2       Biobase_2.62.0         BiocGenerics_0.48.1   
[16] naturalsort_0.1.3      magrittr_2.0.3         gtable_0.3.6          
[19] gridExtra_2.3          ggtreeExtra_1.12.0     ggtree_3.10.1         
[22] ggpubr_0.6.0           ggnewscale_0.5.0       ggh4x_0.3.0           
[25] ggplot2_3.5.1          DT_0.33                dplyr_1.1.4           
[28] ComplexHeatmap_2.18.0  clusterProfiler_4.10.1 circlize_0.4.16       
[31] aplot_0.2.4            ape_5.8-1             

loaded via a namespace (and not attached):
  [1] jsonlite_1.8.9          shape_1.4.6.1           farver_2.1.2           
  [4] rmarkdown_2.29          GlobalOptions_0.1.2     fs_1.6.5               
  [7] zlibbioc_1.48.2         vctrs_0.6.5             memoise_2.0.1          
 [10] RCurl_1.98-1.16         rstatix_0.7.2           htmltools_0.5.8.1      
 [13] DEoptim_2.2-8           broom_1.0.7             Formula_1.2-5          
 [16] gridGraphics_0.5-1      htmlwidgets_1.6.4       cachem_1.1.0           
 [19] igraph_2.1.4            lifecycle_1.0.4         iterators_1.0.14       
 [22] pkgconfig_2.0.3         Matrix_1.6-5            R6_2.5.1               
 [25] fastmap_1.2.0           gson_0.1.0              GenomeInfoDbData_1.2.11
 [28] clue_0.3-66             numDeriv_2016.8-1.1     digest_0.6.37          
 [31] enrichplot_1.22.0       colorspace_2.1-1        patchwork_1.3.0        
 [34] RSQLite_2.3.9           labeling_0.4.3          clusterGeneration_1.3.8
 [37] httr_1.4.7              polyclip_1.10-7         abind_1.4-8            
 [40] compiler_4.3.2          bit64_4.6.0-1           withr_3.0.2            
 [43] doParallel_1.0.17       backports_1.5.0         optimParallel_1.0-2    
 [46] BiocParallel_1.36.0     carData_3.0-5           viridis_0.6.5          
 [49] DBI_1.2.3               ggforce_0.4.2           ggsignif_0.6.4         
 [52] MASS_7.3-60.0.1         scatterplot3d_0.3-44    rjson_0.2.23           
 [55] HDO.db_0.99.1           tools_4.3.2             scatterpie_0.2.4       
 [58] quadprog_1.5-8          glue_1.8.0              nlme_3.1-167           
 [61] GOSemSim_2.28.1         shadowtext_0.1.4        cluster_2.1.8          
 [64] fgsea_1.28.0            generics_0.1.3          tidyr_1.3.1            
 [67] data.table_1.16.4       tidygraph_1.3.1         car_3.1-3              
 [70] XVector_0.42.0          ggrepel_0.9.6           foreach_1.5.2          
 [73] pillar_1.10.1           yulab.utils_0.2.0       splines_4.3.2          
 [76] tweenr_2.0.3            lattice_0.22-6          bit_4.5.0.1            
 [79] tidyselect_1.2.1        GO.db_3.18.0            Biostrings_2.70.3      
 [82] knitr_1.49              xfun_0.50               expm_1.0-0             
 [85] graphlayouts_1.2.2      matrixStats_1.5.0       stringi_1.8.4          
 [88] lazyeval_0.2.2          ggfun_0.1.8             yaml_2.3.10            
 [91] evaluate_1.0.3          codetools_0.2-20        ggraph_2.2.1           
 [94] tibble_3.2.1            qvalue_2.34.0           ggplotify_0.1.2        
 [97] cli_3.6.3               munsell_0.5.1           Rcpp_1.0.14            
[100] GenomeInfoDb_1.38.8     coda_0.19-4.1           png_0.1-8              
[103] parallel_4.3.2          blob_1.2.4              DOSE_3.28.2            
[106] bitops_1.0-9            phangorn_2.12.1         viridisLite_0.4.2      
[109] tidytree_0.4.6          scales_1.3.0            combinat_0.0-8         
[112] purrr_1.0.2             crayon_1.5.3            GetoptLong_1.0.5       
[115] rlang_1.1.5             mnormt_2.1.1            cowplot_1.1.3          
[118] fastmatch_1.1-6         KEGGREST_1.42.0        </code></pre>
</div>
</div>
</div>
</div>
</div>


</section>

</main> <!-- /main -->
<script data-goatcounter="https://19adc99.goatcounter.com/count" async="" src="//gc.zgo.at/count.js"></script>
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./p01-02-genome-assembly-and-annotation.html" class="pagination-link" aria-label="Genome Assembly and Annotation">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Genome Assembly and Annotation</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./p01-04-copy-number-variants.html" class="pagination-link" aria-label="Copy Number Variants">
        <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Copy Number Variants</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      <ul class="footer-items list-unstyled">
    <li class="nav-item">
 Copyright © 2025 Andrea Del Cortona.
  </li>  
</ul>
    </div>   
    <div class="nav-footer-center">
      <ul class="footer-items list-unstyled">
    <li class="nav-item">
    <a class="nav-link" href="https://quarto.org/">
<p>This page is built with ❤️ and <em>Quarto</em>.</p>
</a>
  </li>  
</ul>
    </div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/19ADC99">
      <i class="bi bi-github" role="img" aria-label="GitHub">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://www.linkedin.com/in/andrea-del-cortona-86a75474/">
      <i class="bi bi-linkedin" role="img" aria-label="Lindekin">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>




</body></html>