<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Phylogenetic reconstruction | Vikings Project</title>
  <meta name="description" content="Exploring yeast phenotypic diversity in traditional European farmhouse ale cultures as a source of novel industrial traits" />
  <meta name="generator" content="bookdown 0.32 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Phylogenetic reconstruction | Vikings Project" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Exploring yeast phenotypic diversity in traditional European farmhouse ale cultures as a source of novel industrial traits" />
  <meta name="github-repo" content="19ADC99/VikingsProject" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Phylogenetic reconstruction | Vikings Project" />
  
  <meta name="twitter:description" content="Exploring yeast phenotypic diversity in traditional European farmhouse ale cultures as a source of novel industrial traits" />
  

<meta name="author" content="Andrea Del Cortona" />


<meta name="date" content="2023-03-21" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="horizontal-gene-transfer.html"/>
<link rel="next" href="variant-calling.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.24/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-fixedcolumns-1.11.3/css/fixedColumns.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-fixedcolumns-1.11.3/js/dataTables.fixedColumns.min.js"></script>
<link href="libs/dt-ext-fixedheader-1.11.3/css/fixedHeader.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-fixedheader-1.11.3/js/dataTables.fixedHeader.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Vikings Project</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="1" data-path="qc-and-genome-composition-assessment.html"><a href="qc-and-genome-composition-assessment.html"><i class="fa fa-check"></i><b>1</b> QC and Genome Composition assessment</a>
<ul>
<li class="chapter" data-level="1.1" data-path="qc-and-genome-composition-assessment.html"><a href="qc-and-genome-composition-assessment.html#reads-qc-and-filtering"><i class="fa fa-check"></i><b>1.1</b> Reads QC and Filtering</a></li>
<li class="chapter" data-level="1.2" data-path="qc-and-genome-composition-assessment.html"><a href="qc-and-genome-composition-assessment.html#genome-composition"><i class="fa fa-check"></i><b>1.2</b> Genome Composition</a></li>
<li class="chapter" data-level="1.3" data-path="qc-and-genome-composition-assessment.html"><a href="qc-and-genome-composition-assessment.html#ploydies-estimation"><i class="fa fa-check"></i><b>1.3</b> Ploydies estimation</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="copy-number-variants.html"><a href="copy-number-variants.html"><i class="fa fa-check"></i><b>2</b> Copy Number Variants</a>
<ul>
<li class="chapter" data-level="2.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#call-cnvs"><i class="fa fa-check"></i><b>2.1</b> Call CNVs</a></li>
<li class="chapter" data-level="2.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#identify-common-cnvs-in-farmhouse-yeasts"><i class="fa fa-check"></i><b>2.2</b> Identify common CNVs in Farmhouse yeasts</a></li>
<li class="chapter" data-level="2.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv-functional-enrichment"><i class="fa fa-check"></i><b>2.3</b> CNV Functional Enrichment</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---i1-160000"><i class="fa fa-check"></i><b>2.3.1</b> CNV - I:1-160,000</a>
<ul>
<li class="chapter" data-level="2.3.1.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms"><i class="fa fa-check"></i><b>2.3.1.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.1.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms"><i class="fa fa-check"></i><b>2.3.1.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.1.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms"><i class="fa fa-check"></i><b>2.3.1.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.1.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways"><i class="fa fa-check"></i><b>2.3.1.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---i160501-230500"><i class="fa fa-check"></i><b>2.3.2</b> CNV - I:160,501-230,500</a>
<ul>
<li class="chapter" data-level="2.3.2.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-1"><i class="fa fa-check"></i><b>2.3.2.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.2.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-1"><i class="fa fa-check"></i><b>2.3.2.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.2.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-1"><i class="fa fa-check"></i><b>2.3.2.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.2.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-1"><i class="fa fa-check"></i><b>2.3.2.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---iii1001-317000"><i class="fa fa-check"></i><b>2.3.3</b> CNV - III:1,001-317,000</a>
<ul>
<li class="chapter" data-level="2.3.3.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-2"><i class="fa fa-check"></i><b>2.3.3.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.3.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-2"><i class="fa fa-check"></i><b>2.3.3.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.3.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-2"><i class="fa fa-check"></i><b>2.3.3.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.3.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-2"><i class="fa fa-check"></i><b>2.3.3.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---iv1-18500"><i class="fa fa-check"></i><b>2.3.4</b> CNV - IV:1-18,500</a>
<ul>
<li class="chapter" data-level="2.3.4.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-3"><i class="fa fa-check"></i><b>2.3.4.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.4.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-3"><i class="fa fa-check"></i><b>2.3.4.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.4.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-3"><i class="fa fa-check"></i><b>2.3.4.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.4.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-3"><i class="fa fa-check"></i><b>2.3.4.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.5" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---iv527001-540000"><i class="fa fa-check"></i><b>2.3.5</b> CNV - IV:527,001-540,000</a>
<ul>
<li class="chapter" data-level="2.3.5.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-4"><i class="fa fa-check"></i><b>2.3.5.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.5.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-4"><i class="fa fa-check"></i><b>2.3.5.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.5.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-4"><i class="fa fa-check"></i><b>2.3.5.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.5.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-4"><i class="fa fa-check"></i><b>2.3.5.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.6" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---vi4001-270500"><i class="fa fa-check"></i><b>2.3.6</b> CNV - VI:4,001-270,500</a>
<ul>
<li class="chapter" data-level="2.3.6.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-5"><i class="fa fa-check"></i><b>2.3.6.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.6.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-5"><i class="fa fa-check"></i><b>2.3.6.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.6.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-5"><i class="fa fa-check"></i><b>2.3.6.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.6.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-5"><i class="fa fa-check"></i><b>2.3.6.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.7" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---viii85501-556000"><i class="fa fa-check"></i><b>2.3.7</b> CNV - VIII:85,501-556,000</a>
<ul>
<li class="chapter" data-level="2.3.7.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-6"><i class="fa fa-check"></i><b>2.3.7.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.7.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-6"><i class="fa fa-check"></i><b>2.3.7.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.7.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-6"><i class="fa fa-check"></i><b>2.3.7.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.7.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-6"><i class="fa fa-check"></i><b>2.3.7.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.8" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---xi645001-667000"><i class="fa fa-check"></i><b>2.3.8</b> CNV - XI:645,001-667,000</a>
<ul>
<li class="chapter" data-level="2.3.8.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-7"><i class="fa fa-check"></i><b>2.3.8.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.8.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-7"><i class="fa fa-check"></i><b>2.3.8.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.8.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-7"><i class="fa fa-check"></i><b>2.3.8.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.8.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-7"><i class="fa fa-check"></i><b>2.3.8.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.9" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---xii451001-490000"><i class="fa fa-check"></i><b>2.3.9</b> CNV - XII:451,001-490,000</a>
<ul>
<li class="chapter" data-level="2.3.9.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-8"><i class="fa fa-check"></i><b>2.3.9.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.9.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-8"><i class="fa fa-check"></i><b>2.3.9.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.9.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-8"><i class="fa fa-check"></i><b>2.3.9.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.9.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-8"><i class="fa fa-check"></i><b>2.3.9.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.10" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---xv1-38000"><i class="fa fa-check"></i><b>2.3.10</b> CNV - XV:1-38,000</a>
<ul>
<li class="chapter" data-level="2.3.10.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-9"><i class="fa fa-check"></i><b>2.3.10.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.10.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-9"><i class="fa fa-check"></i><b>2.3.10.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.10.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-9"><i class="fa fa-check"></i><b>2.3.10.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.10.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-9"><i class="fa fa-check"></i><b>2.3.10.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.11" data-path="copy-number-variants.html"><a href="copy-number-variants.html#summary-of-enriched-or-depleted-functions"><i class="fa fa-check"></i><b>2.3.11</b> Summary of enriched or depleted functions</a>
<ul>
<li class="chapter" data-level="2.3.11.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#enriched-go-terms"><i class="fa fa-check"></i><b>2.3.11.1</b> Enriched GO terms</a></li>
<li class="chapter" data-level="2.3.11.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#enriched-kegg-pathways"><i class="fa fa-check"></i><b>2.3.11.2</b> Enriched KEGG pathways</a></li>
<li class="chapter" data-level="2.3.11.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#enriched-reactome-pathways"><i class="fa fa-check"></i><b>2.3.11.3</b> Enriched Reactome pathways</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="genome-assembly-and-annotation.html"><a href="genome-assembly-and-annotation.html"><i class="fa fa-check"></i><b>3</b> Genome Assembly and Annotation</a>
<ul>
<li class="chapter" data-level="3.1" data-path="genome-assembly-and-annotation.html"><a href="genome-assembly-and-annotation.html#genome-assembly"><i class="fa fa-check"></i><b>3.1</b> Genome Assembly</a></li>
<li class="chapter" data-level="3.2" data-path="genome-assembly-and-annotation.html"><a href="genome-assembly-and-annotation.html#genome-annotation"><i class="fa fa-check"></i><b>3.2</b> Genome Annotation</a></li>
<li class="chapter" data-level="3.3" data-path="genome-assembly-and-annotation.html"><a href="genome-assembly-and-annotation.html#identification-of-farmhouse-specific-gene-families"><i class="fa fa-check"></i><b>3.3</b> Identification of Farmhouse-specific Gene Families</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html"><i class="fa fa-check"></i><b>4</b> Horizontal Gene Transfer</a>
<ul>
<li class="chapter" data-level="4.1" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#taxonomic-annotation-of-predicted-genes"><i class="fa fa-check"></i><b>4.1</b> Taxonomic annotation of predicted genes</a></li>
<li class="chapter" data-level="4.2" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#evidence-for-bacterial-hgt"><i class="fa fa-check"></i><b>4.2</b> Evidence for bacterial HGT</a></li>
<li class="chapter" data-level="4.3" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#evidence-for-hgt-from-ascomycota"><i class="fa fa-check"></i><b>4.3</b> Evidence for HGT from Ascomycota</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#zygosaccharomyces-parabailli-transferred-operon"><i class="fa fa-check"></i><b>4.3.1</b> <em>Zygosaccharomyces parabailli</em> transferred operon</a>
<ul>
<li class="chapter" data-level="4.3.1.1" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#overrepresented-go-bp-terms"><i class="fa fa-check"></i><b>4.3.1.1</b> Overrepresented GO BP terms</a></li>
<li class="chapter" data-level="4.3.1.2" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#overrepresented-go-cc-terms"><i class="fa fa-check"></i><b>4.3.1.2</b> Overrepresented GO CC terms</a></li>
<li class="chapter" data-level="4.3.1.3" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#overrepresented-go-mf-terms"><i class="fa fa-check"></i><b>4.3.1.3</b> Overrepresented GO MF terms</a></li>
<li class="chapter" data-level="4.3.1.4" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#overrepresented-reactome-pathways"><i class="fa fa-check"></i><b>4.3.1.4</b> Overrepresented Reactome Pathways</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="phylogenetic-reconstruction.html"><a href="phylogenetic-reconstruction.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic reconstruction</a>
<ul>
<li class="chapter" data-level="5.1" data-path="phylogenetic-reconstruction.html"><a href="phylogenetic-reconstruction.html#identify-quasi-single-copy-genes"><i class="fa fa-check"></i><b>5.1</b> Identify quasi single-copy genes</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="phylogenetic-reconstruction.html"><a href="phylogenetic-reconstruction.html#create-animo-acid-and-codon-alignments"><i class="fa fa-check"></i><b>5.1.1</b> Create animo acid and codon alignments</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="phylogenetic-reconstruction.html"><a href="phylogenetic-reconstruction.html#run-phylogenetic-reconstruction"><i class="fa fa-check"></i><b>5.2</b> Run phylogenetic reconstruction</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="variant-calling.html"><a href="variant-calling.html"><i class="fa fa-check"></i><b>6</b> Variant Calling</a>
<ul>
<li class="chapter" data-level="6.1" data-path="variant-calling.html"><a href="variant-calling.html#variant-calling-1"><i class="fa fa-check"></i><b>6.1</b> Variant Calling</a></li>
<li class="chapter" data-level="6.2" data-path="variant-calling.html"><a href="variant-calling.html#heterozigous-positions"><i class="fa fa-check"></i><b>6.2</b> Heterozigous positions</a></li>
<li class="chapter" data-level="6.3" data-path="variant-calling.html"><a href="variant-calling.html#loh"><i class="fa fa-check"></i><b>6.3</b> LOH</a></li>
<li class="chapter" data-level="6.4" data-path="variant-calling.html"><a href="variant-calling.html#population-structure"><i class="fa fa-check"></i><b>6.4</b> Population Structure</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Vikings Project</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="phylogenetic-reconstruction" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Phylogenetic reconstruction<a href="phylogenetic-reconstruction.html#phylogenetic-reconstruction" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="identify-quasi-single-copy-genes" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Identify quasi single-copy genes<a href="phylogenetic-reconstruction.html#identify-quasi-single-copy-genes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Uniform header sequences, since they come from multiple sources and they are all messed up and incosistent.</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb104-1"><a href="phylogenetic-reconstruction.html#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="co"># rename protein sequences</span></span>
<span id="cb104-2"><a href="phylogenetic-reconstruction.html#cb104-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>maker.aa.fa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb104-3"><a href="phylogenetic-reconstruction.html#cb104-3" aria-hidden="true" tabindex="-1"></a>  <span class="va">NAME</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .maker.aa.fa<span class="va">)</span>_<span class="kw">;</span></span>
<span id="cb104-4"><a href="phylogenetic-reconstruction.html#cb104-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&quot;s/&gt;/&gt;</span><span class="va">$NAME</span><span class="st">/g&quot;</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb104-5"><a href="phylogenetic-reconstruction.html#cb104-5" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb104-6"><a href="phylogenetic-reconstruction.html#cb104-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-7"><a href="phylogenetic-reconstruction.html#cb104-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*[</span><span class="ss">0</span><span class="pp">-</span><span class="ss">9</span><span class="pp">]</span>.aa.fa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb104-8"><a href="phylogenetic-reconstruction.html#cb104-8" aria-hidden="true" tabindex="-1"></a>  <span class="va">NAME</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .aa.fa<span class="va">)</span>_<span class="kw">;</span></span>
<span id="cb104-9"><a href="phylogenetic-reconstruction.html#cb104-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&quot;s/&gt;/&gt;</span><span class="va">$NAME</span><span class="st">/g&quot;</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb104-10"><a href="phylogenetic-reconstruction.html#cb104-10" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb104-11"><a href="phylogenetic-reconstruction.html#cb104-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-12"><a href="phylogenetic-reconstruction.html#cb104-12" aria-hidden="true" tabindex="-1"></a><span class="co"># concatenate all files together</span></span>
<span id="cb104-13"><a href="phylogenetic-reconstruction.html#cb104-13" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="pp">*</span>.aa.fa <span class="op">&gt;</span> all_plusKV.aa.fa</span></code></pre></div>
<p>We will create a set of non-redundant <em>S. cerevisiae</em> proteins, by clustering them based on a 70% sequence similarity groups.</p>
<div class="sourceCode" id="cb105"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb105-1"><a href="phylogenetic-reconstruction.html#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="co"># cluster sequences</span></span>
<span id="cb105-2"><a href="phylogenetic-reconstruction.html#cb105-2" aria-hidden="true" tabindex="-1"></a><span class="ex">cd-hit</span> <span class="at">-i</span> all_plusKV.aa.fa <span class="at">-o</span> all_plusKV.aa.0.7.fa <span class="at">-T</span> 24 <span class="at">-M</span> 16000 <span class="at">-c</span> 0.7 <span class="at">-d</span> 60</span></code></pre></div>
<p>We now check protein clusters having only a single copy gene per yeast strain.</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb106-1"><a href="phylogenetic-reconstruction.html#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="co"># filter clusters into small one gene ortholog clusters</span></span>
<span id="cb106-2"><a href="phylogenetic-reconstruction.html#cb106-2" aria-hidden="true" tabindex="-1"></a><span class="ex">python3.5</span> Vikings.filterCDHITclusters.py <span class="at">--clstr</span> all_plusKV.aa.0.7.fa.clstr</span>
<span id="cb106-3"><a href="phylogenetic-reconstruction.html#cb106-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-4"><a href="phylogenetic-reconstruction.html#cb106-4" aria-hidden="true" tabindex="-1"></a><span class="co"># clean folder</span></span>
<span id="cb106-5"><a href="phylogenetic-reconstruction.html#cb106-5" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> 01_filter_clusters/01_clusters_311-350</span>
<span id="cb106-6"><a href="phylogenetic-reconstruction.html#cb106-6" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> 01_filter_clusters/02_clusters_311-350_noDuplicates</span>
<span id="cb106-7"><a href="phylogenetic-reconstruction.html#cb106-7" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> all_plusKV.aa.0.7.fa.clstr.fltr.clstr.<span class="pp">*</span> 01_filter_clusters</span>
<span id="cb106-8"><a href="phylogenetic-reconstruction.html#cb106-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-9"><a href="phylogenetic-reconstruction.html#cb106-9" aria-hidden="true" tabindex="-1"></a><span class="co"># remove duplicated genes from orthologs clusters</span></span>
<span id="cb106-10"><a href="phylogenetic-reconstruction.html#cb106-10" aria-hidden="true" tabindex="-1"></a><span class="ex">python3.5</span> Vikings.filterCDHITclustersDeduplicates.py <span class="dt">\</span></span>
<span id="cb106-11"><a href="phylogenetic-reconstruction.html#cb106-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">--indir</span> 01_filter_clusters/01_clusters_311-350 <span class="dt">\</span></span>
<span id="cb106-12"><a href="phylogenetic-reconstruction.html#cb106-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">--outdir</span> 01_filter_clusters/02_clusters_311-350_noDuplicate</span>
<span id="cb106-13"><a href="phylogenetic-reconstruction.html#cb106-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-14"><a href="phylogenetic-reconstruction.html#cb106-14" aria-hidden="true" tabindex="-1"></a><span class="co"># retrieve amino acid sequences</span></span>
<span id="cb106-15"><a href="phylogenetic-reconstruction.html#cb106-15" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> 01_filter_clusters/03_clusters_aa</span>
<span id="cb106-16"><a href="phylogenetic-reconstruction.html#cb106-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-17"><a href="phylogenetic-reconstruction.html#cb106-17" aria-hidden="true" tabindex="-1"></a><span class="co"># format headers</span></span>
<span id="cb106-18"><a href="phylogenetic-reconstruction.html#cb106-18" aria-hidden="true" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/ .*//g&#39;</span> ../all_plusKV.aa.fa</span>
<span id="cb106-19"><a href="phylogenetic-reconstruction.html#cb106-19" aria-hidden="true" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/|.*//g&#39;</span> ../all_plusKV.aa.fa <span class="kw">&amp;</span></span>
<span id="cb106-20"><a href="phylogenetic-reconstruction.html#cb106-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-21"><a href="phylogenetic-reconstruction.html#cb106-21" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> 02_clusters_311-350_noDuplicates/<span class="pp">*</span><span class="kw">;</span> <span class="cf">do</span> <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/|.*//g&#39;</span> <span class="va">$file</span> <span class="kw">;</span> <span class="cf">done</span></span>
<span id="cb106-22"><a href="phylogenetic-reconstruction.html#cb106-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-23"><a href="phylogenetic-reconstruction.html#cb106-23" aria-hidden="true" tabindex="-1"></a><span class="ex">python3.5</span> Vikings.CDHITclustersList2ntSeq.py <span class="dt">\</span></span>
<span id="cb106-24"><a href="phylogenetic-reconstruction.html#cb106-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">--fasta</span> all_plusKV.aa.fa <span class="dt">\</span></span>
<span id="cb106-25"><a href="phylogenetic-reconstruction.html#cb106-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">--indir</span> ./01_filter_clusters/02_clusters_311-350_noDuplicates/ <span class="dt">\</span></span>
<span id="cb106-26"><a href="phylogenetic-reconstruction.html#cb106-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">--outdir</span> ./01_filter_clusters/03_clusters_aa/</span></code></pre></div>
<div id="create-animo-acid-and-codon-alignments" class="section level3 hasAnchor" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> Create animo acid and codon alignments<a href="phylogenetic-reconstruction.html#create-animo-acid-and-codon-alignments" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Prepare nucleotidic sequences</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb107-1"><a href="phylogenetic-reconstruction.html#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> 02_nt_start/</span>
<span id="cb107-2"><a href="phylogenetic-reconstruction.html#cb107-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-3"><a href="phylogenetic-reconstruction.html#cb107-3" aria-hidden="true" tabindex="-1"></a><span class="co"># reformat the header to match aa</span></span>
<span id="cb107-4"><a href="phylogenetic-reconstruction.html#cb107-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.fa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb107-5"><a href="phylogenetic-reconstruction.html#cb107-5" aria-hidden="true" tabindex="-1"></a>    <span class="va">STRAIN</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> <span class="kw">|</span> <span class="fu">tr</span> <span class="st">&#39;.&#39;</span> <span class="st">&#39; &#39;</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 1 <span class="at">-d</span> <span class="st">&quot; &quot;</span><span class="va">)</span><span class="kw">;</span></span>
<span id="cb107-6"><a href="phylogenetic-reconstruction.html#cb107-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/ .*//g&#39;</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb107-7"><a href="phylogenetic-reconstruction.html#cb107-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/|.*//g&#39;</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb107-8"><a href="phylogenetic-reconstruction.html#cb107-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/&gt;/&gt;_/g&#39;</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb107-9"><a href="phylogenetic-reconstruction.html#cb107-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&quot;s/&gt;/&gt;</span><span class="va">$STRAIN</span><span class="st">/g&quot;</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb107-10"><a href="phylogenetic-reconstruction.html#cb107-10" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span> </span>
<span id="cb107-11"><a href="phylogenetic-reconstruction.html#cb107-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-12"><a href="phylogenetic-reconstruction.html#cb107-12" aria-hidden="true" tabindex="-1"></a><span class="co"># select nt per cluster</span></span>
<span id="cb107-13"><a href="phylogenetic-reconstruction.html#cb107-13" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="pp">*</span>.fa <span class="op">&gt;</span> ../all_plusKV.nt.fa</span>
<span id="cb107-14"><a href="phylogenetic-reconstruction.html#cb107-14" aria-hidden="true" tabindex="-1"></a><span class="co"># correct worng headers</span></span>
<span id="cb107-15"><a href="phylogenetic-reconstruction.html#cb107-15" aria-hidden="true" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/Skudr_Skudr/Skudr/g&#39;</span> all_plusKV.nt.fa<span class="kw">;</span></span>
<span id="cb107-16"><a href="phylogenetic-reconstruction.html#cb107-16" aria-hidden="true" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/Suvar_Suvar/Suvar/g&#39;</span> all_plusKV.nt.fa<span class="kw">;</span></span>
<span id="cb107-17"><a href="phylogenetic-reconstruction.html#cb107-17" aria-hidden="true" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/Seuba_Seuba/Seuba/g&#39;</span> all_plusKV.nt.fa</span>
<span id="cb107-18"><a href="phylogenetic-reconstruction.html#cb107-18" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> 03_nt_clusters</span>
<span id="cb107-19"><a href="phylogenetic-reconstruction.html#cb107-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-20"><a href="phylogenetic-reconstruction.html#cb107-20" aria-hidden="true" tabindex="-1"></a><span class="co"># cluster NT</span></span>
<span id="cb107-21"><a href="phylogenetic-reconstruction.html#cb107-21" aria-hidden="true" tabindex="-1"></a><span class="ex">cd-hit</span> <span class="at">-T</span> 72 <span class="at">-c</span> 0.98 <span class="at">-aL</span> 0.95 <span class="at">-d</span> 0 <span class="at">-i</span> all_plusKV.nt.fa <span class="at">-o</span> all_plusKV.nt.098.fa</span>
<span id="cb107-22"><a href="phylogenetic-reconstruction.html#cb107-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-23"><a href="phylogenetic-reconstruction.html#cb107-23" aria-hidden="true" tabindex="-1"></a><span class="co"># extract cluster gene lists</span></span>
<span id="cb107-24"><a href="phylogenetic-reconstruction.html#cb107-24" aria-hidden="true" tabindex="-1"></a><span class="ex">python3.5</span> Vikings.filterCDHITclusters.nt.py <span class="at">--clstr</span> all_plusKV.nt.098.fa.clstr</span>
<span id="cb107-25"><a href="phylogenetic-reconstruction.html#cb107-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-26"><a href="phylogenetic-reconstruction.html#cb107-26" aria-hidden="true" tabindex="-1"></a><span class="co"># select genes</span></span>
<span id="cb107-27"><a href="phylogenetic-reconstruction.html#cb107-27" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> gene.lst <span class="dt">\</span></span>
<span id="cb107-28"><a href="phylogenetic-reconstruction.html#cb107-28" aria-hidden="true" tabindex="-1"></a>  <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-n</span> 1 <span class="at">-P</span> 64 <span class="at">-I</span> {} sh <span class="at">-c</span> <span class="dt">\</span></span>
<span id="cb107-29"><a href="phylogenetic-reconstruction.html#cb107-29" aria-hidden="true" tabindex="-1"></a>    <span class="st">&#39;echo {}; \</span></span>
<span id="cb107-30"><a href="phylogenetic-reconstruction.html#cb107-30" aria-hidden="true" tabindex="-1"></a><span class="st">    for file in 00_nt_clusters_lst/*.clstr.*; do </span></span>
<span id="cb107-31"><a href="phylogenetic-reconstruction.html#cb107-31" aria-hidden="true" tabindex="-1"></a><span class="st">      if grep -q {} $file; </span></span>
<span id="cb107-32"><a href="phylogenetic-reconstruction.html#cb107-32" aria-hidden="true" tabindex="-1"></a><span class="st">        then cp $file 00_nt_clusters_lst_select;</span></span>
<span id="cb107-33"><a href="phylogenetic-reconstruction.html#cb107-33" aria-hidden="true" tabindex="-1"></a><span class="st">    fi; done&#39;</span></span>
<span id="cb107-34"><a href="phylogenetic-reconstruction.html#cb107-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-35"><a href="phylogenetic-reconstruction.html#cb107-35" aria-hidden="true" tabindex="-1"></a><span class="co"># add Hittinger IDs</span></span>
<span id="cb107-36"><a href="phylogenetic-reconstruction.html#cb107-36" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> 99_Hittinger/coding/<span class="pp">*</span>.fsa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb107-37"><a href="phylogenetic-reconstruction.html#cb107-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grep</span> <span class="st">&quot;&gt;&quot;</span> <span class="va">$file</span> <span class="dt">\</span></span>
<span id="cb107-38"><a href="phylogenetic-reconstruction.html#cb107-38" aria-hidden="true" tabindex="-1"></a>    <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&#39;s/&gt;//g&#39;</span> <span class="dt">\</span></span>
<span id="cb107-39"><a href="phylogenetic-reconstruction.html#cb107-39" aria-hidden="true" tabindex="-1"></a>    <span class="op">&gt;</span> 99_Hittinger/geneList/<span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .fsa<span class="va">)</span>.lst<span class="kw">;</span></span>
<span id="cb107-40"><a href="phylogenetic-reconstruction.html#cb107-40" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb107-41"><a href="phylogenetic-reconstruction.html#cb107-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-42"><a href="phylogenetic-reconstruction.html#cb107-42" aria-hidden="true" tabindex="-1"></a><span class="ex">python3.5</span> Vikings.addHittingerID.py <span class="dt">\</span></span>
<span id="cb107-43"><a href="phylogenetic-reconstruction.html#cb107-43" aria-hidden="true" tabindex="-1"></a>  <span class="at">--CDlists</span> 01_nt_clusters_lst_select <span class="dt">\</span></span>
<span id="cb107-44"><a href="phylogenetic-reconstruction.html#cb107-44" aria-hidden="true" tabindex="-1"></a>  <span class="at">--Hit</span> 99_Hittinger/geneList/ <span class="dt">\</span></span>
<span id="cb107-45"><a href="phylogenetic-reconstruction.html#cb107-45" aria-hidden="true" tabindex="-1"></a>  <span class="at">--outdir</span> 02_nt_clusters_lst_select_Hittinger</span>
<span id="cb107-46"><a href="phylogenetic-reconstruction.html#cb107-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-47"><a href="phylogenetic-reconstruction.html#cb107-47" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> all_plusKV.aa.fa 99_Hittinger/coding/<span class="pp">*</span>.aa.mfa <span class="op">&gt;</span> all_plusKV.Hit.aa.fa</span>
<span id="cb107-48"><a href="phylogenetic-reconstruction.html#cb107-48" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> all_plusKV.nt.fa 99_Hittinger/coding/<span class="pp">*</span>.fsa <span class="op">&gt;</span> all_plusKV.Hit.nt.fa</span>
<span id="cb107-49"><a href="phylogenetic-reconstruction.html#cb107-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-50"><a href="phylogenetic-reconstruction.html#cb107-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-51"><a href="phylogenetic-reconstruction.html#cb107-51" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> 02_nt_clusters_lst_select_Hittinger/<span class="pp">*</span>.Hit<span class="kw">;</span> <span class="cf">do</span> <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/ .*//g&#39;</span> <span class="va">$file</span> <span class="kw">;</span> <span class="cf">done</span></span>
<span id="cb107-52"><a href="phylogenetic-reconstruction.html#cb107-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-53"><a href="phylogenetic-reconstruction.html#cb107-53" aria-hidden="true" tabindex="-1"></a><span class="co"># select aa for each cluster</span></span>
<span id="cb107-54"><a href="phylogenetic-reconstruction.html#cb107-54" aria-hidden="true" tabindex="-1"></a><span class="ex">python3.5</span> Vikings.CDHITclustersList2ntSeq.py <span class="dt">\</span></span>
<span id="cb107-55"><a href="phylogenetic-reconstruction.html#cb107-55" aria-hidden="true" tabindex="-1"></a>  <span class="at">--fasta</span> all_plusKV.Hit.aa.fa <span class="dt">\</span></span>
<span id="cb107-56"><a href="phylogenetic-reconstruction.html#cb107-56" aria-hidden="true" tabindex="-1"></a>  <span class="at">--indir</span> ./02_nt_clusters_lst_select_Hittinger <span class="dt">\</span></span>
<span id="cb107-57"><a href="phylogenetic-reconstruction.html#cb107-57" aria-hidden="true" tabindex="-1"></a>  <span class="at">--outdir</span> ./03_aa_clusters/</span>
<span id="cb107-58"><a href="phylogenetic-reconstruction.html#cb107-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-59"><a href="phylogenetic-reconstruction.html#cb107-59" aria-hidden="true" tabindex="-1"></a><span class="co"># align aa clusters</span></span>
<span id="cb107-60"><a href="phylogenetic-reconstruction.html#cb107-60" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> ./03_aa_clusters/ <span class="dt">\</span></span>
<span id="cb107-61"><a href="phylogenetic-reconstruction.html#cb107-61" aria-hidden="true" tabindex="-1"></a>  <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-n</span> 8 <span class="at">-P</span> 8 <span class="at">-I</span> {} sh <span class="at">-c</span> <span class="dt">\</span></span>
<span id="cb107-62"><a href="phylogenetic-reconstruction.html#cb107-62" aria-hidden="true" tabindex="-1"></a>  <span class="st">&#39;mafft --localpair --maxiterate 1000 --ep 0.123 --thread 8 ./03_aa_clusters/{} &gt; ./04_aa_clusters_aln/{}.aln.fa&#39;</span></span>
<span id="cb107-63"><a href="phylogenetic-reconstruction.html#cb107-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-64"><a href="phylogenetic-reconstruction.html#cb107-64" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> ./04_aa_clusters_aln/<span class="pp">*</span>.fa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb107-65"><a href="phylogenetic-reconstruction.html#cb107-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">perl</span> ~/rubbish/00_Brigida_starting_files/pal2nal.v14/pal2nal.pl <span class="dt">\</span></span>
<span id="cb107-66"><a href="phylogenetic-reconstruction.html#cb107-66" aria-hidden="true" tabindex="-1"></a>    <span class="va">$file</span> 05_nt_clusters/<span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .aln.fa<span class="va">)</span><span class="dt">\</span></span>
<span id="cb107-67"><a href="phylogenetic-reconstruction.html#cb107-67" aria-hidden="true" tabindex="-1"></a>    <span class="at">-output</span> fasta <span class="dt">\</span></span>
<span id="cb107-68"><a href="phylogenetic-reconstruction.html#cb107-68" aria-hidden="true" tabindex="-1"></a>    <span class="op">&gt;</span> 06_nt_clusters_aln/<span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .fa<span class="va">)</span>.fa<span class="kw">;</span></span>
<span id="cb107-69"><a href="phylogenetic-reconstruction.html#cb107-69" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb107-70"><a href="phylogenetic-reconstruction.html#cb107-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-71"><a href="phylogenetic-reconstruction.html#cb107-71" aria-hidden="true" tabindex="-1"></a><span class="co"># rename headers to keep only strain name</span></span>
<span id="cb107-72"><a href="phylogenetic-reconstruction.html#cb107-72" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> 05_nt_align_long/<span class="pp">*</span> <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-n</span> 1 <span class="at">-P</span> 64 sed <span class="at">-i</span> <span class="st">&#39;s/_.*//g&#39;</span> </span>
<span id="cb107-73"><a href="phylogenetic-reconstruction.html#cb107-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-74"><a href="phylogenetic-reconstruction.html#cb107-74" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> ../07_nt_aln_ready<span class="kw">;</span></span>
<span id="cb107-75"><a href="phylogenetic-reconstruction.html#cb107-75" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> ./<span class="pp">*</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb107-76"><a href="phylogenetic-reconstruction.html#cb107-76" aria-hidden="true" tabindex="-1"></a>  <span class="va">NAME</span><span class="op">=</span><span class="va">$(</span><span class="fu">grep</span> S288C <span class="va">$file</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&#39;s/&gt;S288C_//g&#39;</span><span class="va">)</span><span class="kw">;</span></span>
<span id="cb107-77"><a href="phylogenetic-reconstruction.html#cb107-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cp</span> <span class="va">$file</span> ../07_nt_aln_ready/<span class="va">$NAME</span>.nt.aln.fa<span class="kw">;</span></span>
<span id="cb107-78"><a href="phylogenetic-reconstruction.html#cb107-78" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb107-79"><a href="phylogenetic-reconstruction.html#cb107-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-80"><a href="phylogenetic-reconstruction.html#cb107-80" aria-hidden="true" tabindex="-1"></a><span class="co"># rename headers to keep only strain name</span></span>
<span id="cb107-81"><a href="phylogenetic-reconstruction.html#cb107-81" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> 07_nt_aln_ready/<span class="pp">*</span> <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-n</span> 1 <span class="at">-P</span> 64 sed <span class="at">-i</span> <span class="st">&#39;s/_.*//g&#39;</span> </span></code></pre></div>
</div>
</div>
<div id="run-phylogenetic-reconstruction" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Run phylogenetic reconstruction<a href="phylogenetic-reconstruction.html#run-phylogenetic-reconstruction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This resulted in 155 single-copy genes identified across 151 yeast strains of Gallone et al.2016 and the 44 kveiks strains in this study.
Lets build a Maximum Likelihood tree to reconstruct the phylogenetic relationships between these <em>S. cerevisiae</em> yeast strains!</p>
<pre class="bach"><code># concatenate alignmnets
mkdir 08_concatenated_ML
perl catfasta2phyml.pl \
  --fasta \
  --concatenate 07_nt_aln_ready/*.fa \
  &gt; 08_concatenated_ML/clstr.all.concat.nt.align.fa

# ML analysis on concatenated alignment
~/bin/iqtree-1.6.8-Linux/bin/iqtree \
  -s clstr.all.concat.nt.align.fa \
  -st CODON \
  -pre clstr.all.concat.nt.align \
  -nt 52 \
  -wbt \
  -bb 1000 \
  -alrt 1000 \
  -m MG+F1X4</code></pre>
<p>Plot the reconstructed phylogenetic tree.</p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="phylogenetic-reconstruction.html#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load tree</span></span>
<span id="cb109-2"><a href="phylogenetic-reconstruction.html#cb109-2" aria-hidden="true" tabindex="-1"></a>best_tree <span class="ot">=</span> <span class="fu">read.tree</span>(<span class="fu">paste</span>(mainDir, <span class="st">&quot;genetrees.output.BS.best.tre&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb109-3"><a href="phylogenetic-reconstruction.html#cb109-3" aria-hidden="true" tabindex="-1"></a>best_tree <span class="ot">=</span> <span class="fu">as.phylo</span>(best_tree)</span>
<span id="cb109-4"><a href="phylogenetic-reconstruction.html#cb109-4" aria-hidden="true" tabindex="-1"></a>best_tree<span class="sc">$</span>node.label <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(best_tree<span class="sc">$</span>node.label))</span>
<span id="cb109-5"><a href="phylogenetic-reconstruction.html#cb109-5" aria-hidden="true" tabindex="-1"></a>best_tree <span class="ot">=</span> ape<span class="sc">::</span><span class="fu">root</span>(best_tree,</span>
<span id="cb109-6"><a href="phylogenetic-reconstruction.html#cb109-6" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">which</span>(best_tree<span class="sc">$</span>tip.label <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Skud&quot;</span>)),</span>
<span id="cb109-7"><a href="phylogenetic-reconstruction.html#cb109-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">edgelabel =</span> <span class="cn">TRUE</span>)</span>
<span id="cb109-8"><a href="phylogenetic-reconstruction.html#cb109-8" aria-hidden="true" tabindex="-1"></a>best_tree <span class="ot">=</span> <span class="fu">ladderize</span>(best_tree, <span class="at">right =</span> <span class="cn">TRUE</span>)</span>
<span id="cb109-9"><a href="phylogenetic-reconstruction.html#cb109-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-10"><a href="phylogenetic-reconstruction.html#cb109-10" aria-hidden="true" tabindex="-1"></a><span class="co"># load annotations</span></span>
<span id="cb109-11"><a href="phylogenetic-reconstruction.html#cb109-11" aria-hidden="true" tabindex="-1"></a>tree_annot_raw <span class="ot">=</span> <span class="fu">read.table</span>(<span class="fu">paste</span>(mainDir, <span class="st">&quot;genetrees.output.BS.annot.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>),</span>
<span id="cb109-12"><a href="phylogenetic-reconstruction.html#cb109-12" aria-hidden="true" tabindex="-1"></a>                            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb109-13"><a href="phylogenetic-reconstruction.html#cb109-13" aria-hidden="true" tabindex="-1"></a>                            <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb109-14"><a href="phylogenetic-reconstruction.html#cb109-14" aria-hidden="true" tabindex="-1"></a>                            <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb109-15"><a href="phylogenetic-reconstruction.html#cb109-15" aria-hidden="true" tabindex="-1"></a>tree_annot_raw <span class="ot">=</span> <span class="fu">data.frame</span>(tree_annot_raw)</span>
<span id="cb109-16"><a href="phylogenetic-reconstruction.html#cb109-16" aria-hidden="true" tabindex="-1"></a>tree_annot_raw<span class="sc">$</span>Origin <span class="ot">=</span> <span class="fu">factor</span>(tree_annot_raw<span class="sc">$</span>Origin,</span>
<span id="cb109-17"><a href="phylogenetic-reconstruction.html#cb109-17" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;North-West Norway&quot;</span>, <span class="st">&quot;Eastern Norway&quot;</span>, <span class="st">&quot;South-West Norway&quot;</span>,</span>
<span id="cb109-18"><a href="phylogenetic-reconstruction.html#cb109-18" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">&quot;Latvia&quot;</span>, <span class="st">&quot;Lithuania&quot;</span>, <span class="st">&quot;Russia&quot;</span>, <span class="st">&quot; &quot;</span>))</span>
<span id="cb109-19"><a href="phylogenetic-reconstruction.html#cb109-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-20"><a href="phylogenetic-reconstruction.html#cb109-20" aria-hidden="true" tabindex="-1"></a><span class="co"># plot</span></span>
<span id="cb109-21"><a href="phylogenetic-reconstruction.html#cb109-21" aria-hidden="true" tabindex="-1"></a>p <span class="ot">=</span> <span class="fu">ggtree</span>(best_tree,</span>
<span id="cb109-22"><a href="phylogenetic-reconstruction.html#cb109-22" aria-hidden="true" tabindex="-1"></a>           <span class="at">layout =</span> <span class="st">&quot;fan&quot;</span>,</span>
<span id="cb109-23"><a href="phylogenetic-reconstruction.html#cb109-23" aria-hidden="true" tabindex="-1"></a>           <span class="at">open.angle =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb109-24"><a href="phylogenetic-reconstruction.html#cb109-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tiplab2</span>(<span class="at">offset =</span> <span class="dv">3</span>,</span>
<span id="cb109-25"><a href="phylogenetic-reconstruction.html#cb109-25" aria-hidden="true" tabindex="-1"></a>               <span class="at">size =</span> <span class="fl">2.75</span>) <span class="sc">+</span></span>
<span id="cb109-26"><a href="phylogenetic-reconstruction.html#cb109-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_nodepoint</span>(<span class="fu">aes</span>(<span class="at">fill =</span> <span class="fu">cut</span>(<span class="fu">as.numeric</span>(label), <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">70</span>, <span class="dv">90</span>, <span class="dv">100</span>))),</span>
<span id="cb109-27"><a href="phylogenetic-reconstruction.html#cb109-27" aria-hidden="true" tabindex="-1"></a>                 <span class="at">shape =</span> <span class="dv">21</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb109-28"><a href="phylogenetic-reconstruction.html#cb109-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_tree</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.55</span>)) <span class="sc">+</span></span>
<span id="cb109-29"><a href="phylogenetic-reconstruction.html#cb109-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;black&quot;</span>),</span>
<span id="cb109-30"><a href="phylogenetic-reconstruction.html#cb109-30" aria-hidden="true" tabindex="-1"></a>                    <span class="at">guide =</span> <span class="st">&quot;legend&quot;</span>,</span>
<span id="cb109-31"><a href="phylogenetic-reconstruction.html#cb109-31" aria-hidden="true" tabindex="-1"></a>                    <span class="at">name =</span> <span class="st">&quot;Bootstrap&quot;</span>,</span>
<span id="cb109-32"><a href="phylogenetic-reconstruction.html#cb109-32" aria-hidden="true" tabindex="-1"></a>                    <span class="at">breaks =</span> <span class="fu">c</span>(<span class="st">&quot;(90,100]&quot;</span>, <span class="st">&quot;(70,90]&quot;</span>, <span class="st">&quot;(0,70]&quot;</span>),</span>
<span id="cb109-33"><a href="phylogenetic-reconstruction.html#cb109-33" aria-hidden="true" tabindex="-1"></a>                    <span class="at">labels =</span> <span class="fu">expression</span>(BP<span class="sc">&gt;=</span><span class="dv">90</span>,<span class="dv">70</span> <span class="sc">&lt;=</span> BP <span class="sc">*</span> <span class="st">&quot; &lt; 90&quot;</span>, BP <span class="sc">&lt;</span> <span class="dv">70</span>))</span>
<span id="cb109-34"><a href="phylogenetic-reconstruction.html#cb109-34" aria-hidden="true" tabindex="-1"></a>p <span class="ot">=</span> <span class="fu">flip</span>(p, <span class="dv">250</span>, <span class="dv">317</span>)</span>
<span id="cb109-35"><a href="phylogenetic-reconstruction.html#cb109-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-36"><a href="phylogenetic-reconstruction.html#cb109-36" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">=</span> p <span class="sc">+</span></span>
<span id="cb109-37"><a href="phylogenetic-reconstruction.html#cb109-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">new_scale_fill</span>() <span class="sc">+</span></span>
<span id="cb109-38"><a href="phylogenetic-reconstruction.html#cb109-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_fruit</span>(<span class="at">data =</span> tree_annot_raw,</span>
<span id="cb109-39"><a href="phylogenetic-reconstruction.html#cb109-39" aria-hidden="true" tabindex="-1"></a>             <span class="at">geom =</span> geom_tile,</span>
<span id="cb109-40"><a href="phylogenetic-reconstruction.html#cb109-40" aria-hidden="true" tabindex="-1"></a>             <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">y =</span> Strain, <span class="at">fill =</span> Clade),</span>
<span id="cb109-41"><a href="phylogenetic-reconstruction.html#cb109-41" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="cn">NA</span>, <span class="at">offset =</span> <span class="fl">0.04</span>, <span class="at">size =</span> <span class="fl">0.02</span>) <span class="sc">+</span></span>
<span id="cb109-42"><a href="phylogenetic-reconstruction.html#cb109-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&#39;#e6194b&#39;</span>, <span class="st">&#39;#3cb44b&#39;</span>, <span class="st">&#39;#ffe119&#39;</span>, <span class="st">&#39;#4363d8&#39;</span>, <span class="st">&#39;#f58231&#39;</span>,</span>
<span id="cb109-43"><a href="phylogenetic-reconstruction.html#cb109-43" aria-hidden="true" tabindex="-1"></a>                               <span class="st">&#39;#911eb4&#39;</span>, <span class="st">&#39;#46f0f0&#39;</span>, <span class="st">&#39;#f032e6&#39;</span>, <span class="st">&#39;#bcf60c&#39;</span>, <span class="st">&#39;#fabebe&#39;</span>,</span>
<span id="cb109-44"><a href="phylogenetic-reconstruction.html#cb109-44" aria-hidden="true" tabindex="-1"></a>                               <span class="st">&#39;#008080&#39;</span>, <span class="st">&#39;#e6beff&#39;</span>, <span class="st">&#39;#9a6324&#39;</span>, <span class="st">&#39;#fffac8&#39;</span>, <span class="st">&#39;#800000&#39;</span>,</span>
<span id="cb109-45"><a href="phylogenetic-reconstruction.html#cb109-45" aria-hidden="true" tabindex="-1"></a>                               <span class="st">&#39;#aaffc3&#39;</span>, <span class="st">&#39;#808000&#39;</span>, <span class="st">&#39;#ffd8b1&#39;</span>, <span class="st">&#39;#000075&#39;</span>, <span class="st">&#39;#808080&#39;</span>,</span>
<span id="cb109-46"><a href="phylogenetic-reconstruction.html#cb109-46" aria-hidden="true" tabindex="-1"></a>                               <span class="st">&#39;#ffffff&#39;</span>, <span class="st">&#39;#000000&#39;</span>)) </span>
<span id="cb109-47"><a href="phylogenetic-reconstruction.html#cb109-47" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">=</span> p1 <span class="sc">+</span></span>
<span id="cb109-48"><a href="phylogenetic-reconstruction.html#cb109-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">new_scale_fill</span>() <span class="sc">+</span></span>
<span id="cb109-49"><a href="phylogenetic-reconstruction.html#cb109-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_fruit</span>(<span class="at">data =</span> tree_annot_raw,</span>
<span id="cb109-50"><a href="phylogenetic-reconstruction.html#cb109-50" aria-hidden="true" tabindex="-1"></a>             <span class="at">geom =</span> geom_tile,</span>
<span id="cb109-51"><a href="phylogenetic-reconstruction.html#cb109-51" aria-hidden="true" tabindex="-1"></a>             <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">y =</span> Strain, <span class="at">fill =</span> Origin),</span>
<span id="cb109-52"><a href="phylogenetic-reconstruction.html#cb109-52" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="cn">NA</span>, <span class="at">offset =</span> <span class="fl">0.06</span>, <span class="at">size =</span> <span class="fl">0.02</span>) <span class="sc">+</span></span>
<span id="cb109-53"><a href="phylogenetic-reconstruction.html#cb109-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&#39;#03045e&#39;</span>, <span class="st">&#39;#0077b6&#39;</span>, <span class="st">&#39;#00b4d8&#39;</span>, </span>
<span id="cb109-54"><a href="phylogenetic-reconstruction.html#cb109-54" aria-hidden="true" tabindex="-1"></a>                               <span class="st">&#39;#808000&#39;</span>, <span class="st">&#39;#aaffc3&#39;</span>, <span class="st">&#39;#800000&#39;</span>,</span>
<span id="cb109-55"><a href="phylogenetic-reconstruction.html#cb109-55" aria-hidden="true" tabindex="-1"></a>                               <span class="st">&#39;#ffffff&#39;</span>)) <span class="sc">+</span></span>
<span id="cb109-56"><a href="phylogenetic-reconstruction.html#cb109-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">1.1</span>, <span class="fl">0.5</span>))</span>
<span id="cb109-57"><a href="phylogenetic-reconstruction.html#cb109-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-58"><a href="phylogenetic-reconstruction.html#cb109-58" aria-hidden="true" tabindex="-1"></a>p2</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-133"></span>
<img src="_main_files/figure-html/unnamed-chunk-133-1.png" alt="Farmhouse yeasts phynogenetic recontruction" width="672" />
<p class="caption">
Figure 5.1: Farmhouse yeasts phynogenetic recontruction
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="horizontal-gene-transfer.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="variant-calling.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/05-Phylogenetic-Reconstruction.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
