<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 QC and Genome Composition assessment | Vikings Project</title>
  <meta name="description" content="Exploring yeast phenotypic diversity in traditional European farmhouse ale cultures as a source of novel industrial traits" />
  <meta name="generator" content="bookdown 0.32 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 QC and Genome Composition assessment | Vikings Project" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Exploring yeast phenotypic diversity in traditional European farmhouse ale cultures as a source of novel industrial traits" />
  <meta name="github-repo" content="19ADC99/VikingsProject" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 QC and Genome Composition assessment | Vikings Project" />
  
  <meta name="twitter:description" content="Exploring yeast phenotypic diversity in traditional European farmhouse ale cultures as a source of novel industrial traits" />
  

<meta name="author" content="Andrea Del Cortona" />


<meta name="date" content="2023-03-21" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="copy-number-variants.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.24/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-fixedcolumns-1.11.3/css/fixedColumns.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-fixedcolumns-1.11.3/js/dataTables.fixedColumns.min.js"></script>
<link href="libs/dt-ext-fixedheader-1.11.3/css/fixedHeader.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-fixedheader-1.11.3/js/dataTables.fixedHeader.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Vikings Project</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="1" data-path="qc-and-genome-composition-assessment.html"><a href="qc-and-genome-composition-assessment.html"><i class="fa fa-check"></i><b>1</b> QC and Genome Composition assessment</a>
<ul>
<li class="chapter" data-level="1.1" data-path="qc-and-genome-composition-assessment.html"><a href="qc-and-genome-composition-assessment.html#reads-qc-and-filtering"><i class="fa fa-check"></i><b>1.1</b> Reads QC and Filtering</a></li>
<li class="chapter" data-level="1.2" data-path="qc-and-genome-composition-assessment.html"><a href="qc-and-genome-composition-assessment.html#genome-composition"><i class="fa fa-check"></i><b>1.2</b> Genome Composition</a></li>
<li class="chapter" data-level="1.3" data-path="qc-and-genome-composition-assessment.html"><a href="qc-and-genome-composition-assessment.html#ploydies-estimation"><i class="fa fa-check"></i><b>1.3</b> Ploydies estimation</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="copy-number-variants.html"><a href="copy-number-variants.html"><i class="fa fa-check"></i><b>2</b> Copy Number Variants</a>
<ul>
<li class="chapter" data-level="2.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#call-cnvs"><i class="fa fa-check"></i><b>2.1</b> Call CNVs</a></li>
<li class="chapter" data-level="2.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#identify-common-cnvs-in-farmhouse-yeasts"><i class="fa fa-check"></i><b>2.2</b> Identify common CNVs in Farmhouse yeasts</a></li>
<li class="chapter" data-level="2.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv-functional-enrichment"><i class="fa fa-check"></i><b>2.3</b> CNV Functional Enrichment</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---i1-160000"><i class="fa fa-check"></i><b>2.3.1</b> CNV - I:1-160,000</a>
<ul>
<li class="chapter" data-level="2.3.1.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms"><i class="fa fa-check"></i><b>2.3.1.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.1.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms"><i class="fa fa-check"></i><b>2.3.1.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.1.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms"><i class="fa fa-check"></i><b>2.3.1.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.1.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways"><i class="fa fa-check"></i><b>2.3.1.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---i160501-230500"><i class="fa fa-check"></i><b>2.3.2</b> CNV - I:160,501-230,500</a>
<ul>
<li class="chapter" data-level="2.3.2.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-1"><i class="fa fa-check"></i><b>2.3.2.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.2.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-1"><i class="fa fa-check"></i><b>2.3.2.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.2.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-1"><i class="fa fa-check"></i><b>2.3.2.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.2.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-1"><i class="fa fa-check"></i><b>2.3.2.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---iii1001-317000"><i class="fa fa-check"></i><b>2.3.3</b> CNV - III:1,001-317,000</a>
<ul>
<li class="chapter" data-level="2.3.3.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-2"><i class="fa fa-check"></i><b>2.3.3.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.3.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-2"><i class="fa fa-check"></i><b>2.3.3.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.3.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-2"><i class="fa fa-check"></i><b>2.3.3.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.3.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-2"><i class="fa fa-check"></i><b>2.3.3.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---iv1-18500"><i class="fa fa-check"></i><b>2.3.4</b> CNV - IV:1-18,500</a>
<ul>
<li class="chapter" data-level="2.3.4.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-3"><i class="fa fa-check"></i><b>2.3.4.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.4.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-3"><i class="fa fa-check"></i><b>2.3.4.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.4.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-3"><i class="fa fa-check"></i><b>2.3.4.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.4.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-3"><i class="fa fa-check"></i><b>2.3.4.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.5" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---iv527001-540000"><i class="fa fa-check"></i><b>2.3.5</b> CNV - IV:527,001-540,000</a>
<ul>
<li class="chapter" data-level="2.3.5.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-4"><i class="fa fa-check"></i><b>2.3.5.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.5.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-4"><i class="fa fa-check"></i><b>2.3.5.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.5.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-4"><i class="fa fa-check"></i><b>2.3.5.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.5.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-4"><i class="fa fa-check"></i><b>2.3.5.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.6" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---vi4001-270500"><i class="fa fa-check"></i><b>2.3.6</b> CNV - VI:4,001-270,500</a>
<ul>
<li class="chapter" data-level="2.3.6.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-5"><i class="fa fa-check"></i><b>2.3.6.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.6.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-5"><i class="fa fa-check"></i><b>2.3.6.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.6.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-5"><i class="fa fa-check"></i><b>2.3.6.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.6.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-5"><i class="fa fa-check"></i><b>2.3.6.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.7" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---viii85501-556000"><i class="fa fa-check"></i><b>2.3.7</b> CNV - VIII:85,501-556,000</a>
<ul>
<li class="chapter" data-level="2.3.7.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-6"><i class="fa fa-check"></i><b>2.3.7.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.7.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-6"><i class="fa fa-check"></i><b>2.3.7.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.7.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-6"><i class="fa fa-check"></i><b>2.3.7.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.7.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-6"><i class="fa fa-check"></i><b>2.3.7.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.8" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---xi645001-667000"><i class="fa fa-check"></i><b>2.3.8</b> CNV - XI:645,001-667,000</a>
<ul>
<li class="chapter" data-level="2.3.8.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-7"><i class="fa fa-check"></i><b>2.3.8.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.8.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-7"><i class="fa fa-check"></i><b>2.3.8.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.8.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-7"><i class="fa fa-check"></i><b>2.3.8.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.8.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-7"><i class="fa fa-check"></i><b>2.3.8.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.9" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---xii451001-490000"><i class="fa fa-check"></i><b>2.3.9</b> CNV - XII:451,001-490,000</a>
<ul>
<li class="chapter" data-level="2.3.9.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-8"><i class="fa fa-check"></i><b>2.3.9.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.9.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-8"><i class="fa fa-check"></i><b>2.3.9.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.9.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-8"><i class="fa fa-check"></i><b>2.3.9.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.9.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-8"><i class="fa fa-check"></i><b>2.3.9.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.10" data-path="copy-number-variants.html"><a href="copy-number-variants.html#cnv---xv1-38000"><i class="fa fa-check"></i><b>2.3.10</b> CNV - XV:1-38,000</a>
<ul>
<li class="chapter" data-level="2.3.10.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---bp-terms-9"><i class="fa fa-check"></i><b>2.3.10.1</b> Over-represented / Enriched GO - BP terms</a></li>
<li class="chapter" data-level="2.3.10.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---cc-terms-9"><i class="fa fa-check"></i><b>2.3.10.2</b> Over-represented / Enriched GO - CC terms</a></li>
<li class="chapter" data-level="2.3.10.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-go---mf-terms-9"><i class="fa fa-check"></i><b>2.3.10.3</b> Over-represented / Enriched GO - MF terms</a></li>
<li class="chapter" data-level="2.3.10.4" data-path="copy-number-variants.html"><a href="copy-number-variants.html#over-represented-enriched-reactome-pathways-9"><i class="fa fa-check"></i><b>2.3.10.4</b> Over-represented / Enriched Reactome Pathways</a></li>
</ul></li>
<li class="chapter" data-level="2.3.11" data-path="copy-number-variants.html"><a href="copy-number-variants.html#summary-of-enriched-or-depleted-functions"><i class="fa fa-check"></i><b>2.3.11</b> Summary of enriched or depleted functions</a>
<ul>
<li class="chapter" data-level="2.3.11.1" data-path="copy-number-variants.html"><a href="copy-number-variants.html#enriched-go-terms"><i class="fa fa-check"></i><b>2.3.11.1</b> Enriched GO terms</a></li>
<li class="chapter" data-level="2.3.11.2" data-path="copy-number-variants.html"><a href="copy-number-variants.html#enriched-kegg-pathways"><i class="fa fa-check"></i><b>2.3.11.2</b> Enriched KEGG pathways</a></li>
<li class="chapter" data-level="2.3.11.3" data-path="copy-number-variants.html"><a href="copy-number-variants.html#enriched-reactome-pathways"><i class="fa fa-check"></i><b>2.3.11.3</b> Enriched Reactome pathways</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="genome-assembly-and-annotation.html"><a href="genome-assembly-and-annotation.html"><i class="fa fa-check"></i><b>3</b> Genome Assembly and Annotation</a>
<ul>
<li class="chapter" data-level="3.1" data-path="genome-assembly-and-annotation.html"><a href="genome-assembly-and-annotation.html#genome-assembly"><i class="fa fa-check"></i><b>3.1</b> Genome Assembly</a></li>
<li class="chapter" data-level="3.2" data-path="genome-assembly-and-annotation.html"><a href="genome-assembly-and-annotation.html#genome-annotation"><i class="fa fa-check"></i><b>3.2</b> Genome Annotation</a></li>
<li class="chapter" data-level="3.3" data-path="genome-assembly-and-annotation.html"><a href="genome-assembly-and-annotation.html#identification-of-farmhouse-specific-gene-families"><i class="fa fa-check"></i><b>3.3</b> Identification of Farmhouse-specific Gene Families</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html"><i class="fa fa-check"></i><b>4</b> Horizontal Gene Transfer</a>
<ul>
<li class="chapter" data-level="4.1" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#taxonomic-annotation-of-predicted-genes"><i class="fa fa-check"></i><b>4.1</b> Taxonomic annotation of predicted genes</a></li>
<li class="chapter" data-level="4.2" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#evidence-for-bacterial-hgt"><i class="fa fa-check"></i><b>4.2</b> Evidence for bacterial HGT</a></li>
<li class="chapter" data-level="4.3" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#evidence-for-hgt-from-ascomycota"><i class="fa fa-check"></i><b>4.3</b> Evidence for HGT from Ascomycota</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#zygosaccharomyces-parabailli-transferred-operon"><i class="fa fa-check"></i><b>4.3.1</b> <em>Zygosaccharomyces parabailli</em> transferred operon</a>
<ul>
<li class="chapter" data-level="4.3.1.1" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#overrepresented-go-bp-terms"><i class="fa fa-check"></i><b>4.3.1.1</b> Overrepresented GO BP terms</a></li>
<li class="chapter" data-level="4.3.1.2" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#overrepresented-go-cc-terms"><i class="fa fa-check"></i><b>4.3.1.2</b> Overrepresented GO CC terms</a></li>
<li class="chapter" data-level="4.3.1.3" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#overrepresented-go-mf-terms"><i class="fa fa-check"></i><b>4.3.1.3</b> Overrepresented GO MF terms</a></li>
<li class="chapter" data-level="4.3.1.4" data-path="horizontal-gene-transfer.html"><a href="horizontal-gene-transfer.html#overrepresented-reactome-pathways"><i class="fa fa-check"></i><b>4.3.1.4</b> Overrepresented Reactome Pathways</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="phylogenetic-reconstruction.html"><a href="phylogenetic-reconstruction.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic reconstruction</a>
<ul>
<li class="chapter" data-level="5.1" data-path="phylogenetic-reconstruction.html"><a href="phylogenetic-reconstruction.html#identify-quasi-single-copy-genes"><i class="fa fa-check"></i><b>5.1</b> Identify quasi single-copy genes</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="phylogenetic-reconstruction.html"><a href="phylogenetic-reconstruction.html#create-animo-acid-and-codon-alignments"><i class="fa fa-check"></i><b>5.1.1</b> Create animo acid and codon alignments</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="phylogenetic-reconstruction.html"><a href="phylogenetic-reconstruction.html#run-phylogenetic-reconstruction"><i class="fa fa-check"></i><b>5.2</b> Run phylogenetic reconstruction</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="variant-calling.html"><a href="variant-calling.html"><i class="fa fa-check"></i><b>6</b> Variant Calling</a>
<ul>
<li class="chapter" data-level="6.1" data-path="variant-calling.html"><a href="variant-calling.html#variant-calling-1"><i class="fa fa-check"></i><b>6.1</b> Variant Calling</a></li>
<li class="chapter" data-level="6.2" data-path="variant-calling.html"><a href="variant-calling.html#heterozigous-positions"><i class="fa fa-check"></i><b>6.2</b> Heterozigous positions</a></li>
<li class="chapter" data-level="6.3" data-path="variant-calling.html"><a href="variant-calling.html#loh"><i class="fa fa-check"></i><b>6.3</b> LOH</a></li>
<li class="chapter" data-level="6.4" data-path="variant-calling.html"><a href="variant-calling.html#population-structure"><i class="fa fa-check"></i><b>6.4</b> Population Structure</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Vikings Project</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="qc-and-genome-composition-assessment" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> QC and Genome Composition assessment<a href="qc-and-genome-composition-assessment.html#qc-and-genome-composition-assessment" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="reads-qc-and-filtering" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Reads QC and Filtering<a href="qc-and-genome-composition-assessment.html#reads-qc-and-filtering" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Just trim low quality bases with Trimmomatic and discard short reads. I keep only properly paired reads and I discard the unpaired, since the latter are just a tiny fraction of my total data, they complicate the analysis and there is not much added value in them (at least for the analysis we run).</p>
<p>The file sample.lst just contains the names of my kveiks samples.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="qc-and-genome-composition-assessment.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># generate FastQC report</span></span>
<span id="cb1-2"><a href="qc-and-genome-composition-assessment.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.fastq<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb1-3"><a href="qc-and-genome-composition-assessment.html#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="ex">~/bin/FastQC/fastqc</span> <span class="va">$file</span> <span class="kw">&amp;</span></span>
<span id="cb1-4"><a href="qc-and-genome-composition-assessment.html#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb1-5"><a href="qc-and-genome-composition-assessment.html#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="qc-and-genome-composition-assessment.html#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co"># LowQ base trimming and filtering</span></span>
<span id="cb1-7"><a href="qc-and-genome-composition-assessment.html#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb1-8"><a href="qc-and-genome-composition-assessment.html#cb1-8" aria-hidden="true" tabindex="-1"></a>    <span class="ex">java</span> <span class="at">-jar</span> ~/bin/trimmomatic/trimmomatic.jar PE <span class="dt">\</span></span>
<span id="cb1-9"><a href="qc-and-genome-composition-assessment.html#cb1-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">-threads</span> 72 <span class="dt">\</span></span>
<span id="cb1-10"><a href="qc-and-genome-composition-assessment.html#cb1-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">-phred33</span> <span class="dt">\</span></span>
<span id="cb1-11"><a href="qc-and-genome-composition-assessment.html#cb1-11" aria-hidden="true" tabindex="-1"></a>        <span class="va">$line</span>.R1.fq.gz <span class="va">$line</span>.R2.fq.gz <span class="dt">\</span></span>
<span id="cb1-12"><a href="qc-and-genome-composition-assessment.html#cb1-12" aria-hidden="true" tabindex="-1"></a>        <span class="va">$line</span>.R1.tr.fq.gz <span class="va">$line</span>.R1.tr.un.fq.gz <span class="va">$line</span>.R2.tr.fq.gz <span class="va">$line</span>.R2.tr.un.fq.gz <span class="dt">\</span></span>
<span id="cb1-13"><a href="qc-and-genome-composition-assessment.html#cb1-13" aria-hidden="true" tabindex="-1"></a>        SLIDINGWINDOW:10:30 <span class="dt">\</span></span>
<span id="cb1-14"><a href="qc-and-genome-composition-assessment.html#cb1-14" aria-hidden="true" tabindex="-1"></a>        TRAILING:30 <span class="dt">\</span></span>
<span id="cb1-15"><a href="qc-and-genome-composition-assessment.html#cb1-15" aria-hidden="true" tabindex="-1"></a>        MINLEN:50</span>
<span id="cb1-16"><a href="qc-and-genome-composition-assessment.html#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span> <span class="op">&lt;</span> sample.lst</span></code></pre></div>
</div>
<div id="genome-composition" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Genome Composition<a href="qc-and-genome-composition-assessment.html#genome-composition" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We check the composition of the sequenced kveiks by aligning the reads to a set of Saccharomyces reference genomes.</p>
<table>
<caption><span id="tab:unnamed-chunk-3">Table 1.1: </span>Yeast Reference Genomes</caption>
<thead>
<tr class="header">
<th align="left">Species</th>
<th align="left">Genome Accession</th>
<th align="left">Label</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">S. cerevisiae</td>
<td align="left">S288C_R64-2-1</td>
<td align="left">Scere</td>
</tr>
<tr class="even">
<td align="left">S. eubayanus</td>
<td align="left">GCF_001298625.1</td>
<td align="left">Seuba</td>
</tr>
<tr class="odd">
<td align="left">S. kudriavzevi</td>
<td align="left">gca_000167075</td>
<td align="left">Skudr</td>
</tr>
<tr class="even">
<td align="left">S. mikatae</td>
<td align="left">SRX055454</td>
<td align="left">Smika</td>
</tr>
<tr class="odd">
<td align="left">S. paradoxus</td>
<td align="left">ASM207914v1</td>
<td align="left">Spara</td>
</tr>
<tr class="even">
<td align="left">S. uvarum</td>
<td align="left">ASM224264v1</td>
<td align="left">Suvar</td>
</tr>
</tbody>
</table>
<p>First, we align the reads to the six <em>Saccharomyces</em> reference genomes. Then we create 10kb windows with <code>bedtools</code> and we calculate the average coverage for each window.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="qc-and-genome-composition-assessment.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">## index the reference genome</span></span>
<span id="cb2-2"><a href="qc-and-genome-composition-assessment.html#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="ex">~/bin/bwa/bwa</span> index Saccharomyces_RefGen.fa</span>
<span id="cb2-3"><a href="qc-and-genome-composition-assessment.html#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="qc-and-genome-composition-assessment.html#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co">## reads alignment</span></span>
<span id="cb2-5"><a href="qc-and-genome-composition-assessment.html#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span> <span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb2-6"><a href="qc-and-genome-composition-assessment.html#cb2-6" aria-hidden="true" tabindex="-1"></a>    <span class="ex">bwa</span> mem <span class="at">-t</span> 72 <span class="dt">\</span></span>
<span id="cb2-7"><a href="qc-and-genome-composition-assessment.html#cb2-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">-K</span> 100000000 <span class="dt">\</span></span>
<span id="cb2-8"><a href="qc-and-genome-composition-assessment.html#cb2-8" aria-hidden="true" tabindex="-1"></a>        Saccharomyces_RefGen.fa <span class="dt">\</span></span>
<span id="cb2-9"><a href="qc-and-genome-composition-assessment.html#cb2-9" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;</span><span class="va">${line}</span><span class="st">&quot;</span>.R1.tr.fq.gz <span class="st">&quot;</span><span class="va">${line}</span><span class="st">&quot;</span>.R2.tr.fq.gz <span class="dt">\</span></span>
<span id="cb2-10"><a href="qc-and-genome-composition-assessment.html#cb2-10" aria-hidden="true" tabindex="-1"></a>    <span class="op">&gt;</span> <span class="st">&quot;</span><span class="va">${line}</span><span class="st">&quot;</span>.align.sam</span>
<span id="cb2-11"><a href="qc-and-genome-composition-assessment.html#cb2-11" aria-hidden="true" tabindex="-1"></a>  <span class="ex">samtools</span> view <span class="at">-@</span> 72 <span class="at">-Sb</span> <span class="dt">\</span></span>
<span id="cb2-12"><a href="qc-and-genome-composition-assessment.html#cb2-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;</span><span class="va">${line}</span><span class="st">&quot;</span>.align.sam <span class="op">&gt;</span> <span class="st">&quot;</span><span class="va">${line}</span><span class="st">&quot;</span>.align.bam</span>
<span id="cb2-13"><a href="qc-and-genome-composition-assessment.html#cb2-13" aria-hidden="true" tabindex="-1"></a>  <span class="ex">samtools</span> sort <span class="at">-@</span> 72 </span>
<span id="cb2-14"><a href="qc-and-genome-composition-assessment.html#cb2-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;</span><span class="va">${line}</span><span class="st">&quot;</span><span class="ex">.align.bam</span> <span class="st">&quot;</span><span class="va">${line}</span><span class="st">&quot;</span>.align.sort</span>
<span id="cb2-15"><a href="qc-and-genome-composition-assessment.html#cb2-15" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span> <span class="op">&lt;</span> ../sample.lst </span>
<span id="cb2-16"><a href="qc-and-genome-composition-assessment.html#cb2-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-17"><a href="qc-and-genome-composition-assessment.html#cb2-17" aria-hidden="true" tabindex="-1"></a><span class="co">## bam2bed</span></span>
<span id="cb2-18"><a href="qc-and-genome-composition-assessment.html#cb2-18" aria-hidden="true" tabindex="-1"></a><span class="co"># prepare reference genomes</span></span>
<span id="cb2-19"><a href="qc-and-genome-composition-assessment.html#cb2-19" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools-1.9/samtools</span> faidx Saccharomyces_RefGen.fa</span>
<span id="cb2-20"><a href="qc-and-genome-composition-assessment.html#cb2-20" aria-hidden="true" tabindex="-1"></a><span class="ex">gatk</span> CreateSequenceDictionary <span class="at">--R</span> Saccharomyces_RefGen.fa</span>
<span id="cb2-21"><a href="qc-and-genome-composition-assessment.html#cb2-21" aria-hidden="true" tabindex="-1"></a><span class="fu">cut</span> <span class="at">-f</span> 2,3 Saccharomyces_RefGen.dict <span class="dt">\</span></span>
<span id="cb2-22"><a href="qc-and-genome-composition-assessment.html#cb2-22" aria-hidden="true" tabindex="-1"></a>  <span class="kw">|</span> <span class="fu">tail</span> <span class="at">-n</span> +2 <span class="dt">\</span></span>
<span id="cb2-23"><a href="qc-and-genome-composition-assessment.html#cb2-23" aria-hidden="true" tabindex="-1"></a>  <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&#39;s/LN://g&#39;</span> <span class="dt">\</span></span>
<span id="cb2-24"><a href="qc-and-genome-composition-assessment.html#cb2-24" aria-hidden="true" tabindex="-1"></a>  <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&#39;s/SN://g&#39;</span> <span class="dt">\</span></span>
<span id="cb2-25"><a href="qc-and-genome-composition-assessment.html#cb2-25" aria-hidden="true" tabindex="-1"></a>  <span class="op">&gt;</span> Saccharomyces_RefGen.bedchor</span>
<span id="cb2-26"><a href="qc-and-genome-composition-assessment.html#cb2-26" aria-hidden="true" tabindex="-1"></a><span class="ex">bedtools</span> makewindows <span class="at">-g</span> Saccharomyces_RefGen.bedchor <span class="dt">\</span></span>
<span id="cb2-27"><a href="qc-and-genome-composition-assessment.html#cb2-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">-w</span> 1000 <span class="op">&gt;</span> Saccharomyces_RefGen.1kb_win.tab</span>
<span id="cb2-28"><a href="qc-and-genome-composition-assessment.html#cb2-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-29"><a href="qc-and-genome-composition-assessment.html#cb2-29" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate mean coverage</span></span>
<span id="cb2-30"><a href="qc-and-genome-composition-assessment.html#cb2-30" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb2-31"><a href="qc-and-genome-composition-assessment.html#cb2-31" aria-hidden="true" tabindex="-1"></a>    <span class="ex">bedtools</span> coverage <span class="at">-a</span> Saccharomyces_RefGen.1kb_win.tab <span class="dt">\</span></span>
<span id="cb2-32"><a href="qc-and-genome-composition-assessment.html#cb2-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">-b</span> <span class="st">&quot;</span><span class="va">${line}</span><span class="st">&quot;</span>.align.sort.bam <span class="at">-mean</span> <span class="op">&gt;</span> <span class="st">&quot;</span><span class="va">${line}</span><span class="st">&quot;</span>.align.sort.1kb_cov.bed<span class="kw">;</span></span>
<span id="cb2-33"><a href="qc-and-genome-composition-assessment.html#cb2-33" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb2-34"><a href="qc-and-genome-composition-assessment.html#cb2-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-35"><a href="qc-and-genome-composition-assessment.html#cb2-35" aria-hidden="true" tabindex="-1"></a><span class="co"># format output</span></span>
<span id="cb2-36"><a href="qc-and-genome-composition-assessment.html#cb2-36" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.1kb_cov.bed <span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb2-37"><a href="qc-and-genome-composition-assessment.html#cb2-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/Scere_/Scere\t/g&#39;</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb2-38"><a href="qc-and-genome-composition-assessment.html#cb2-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/Seuba_/Seuba\t/g&#39;</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb2-39"><a href="qc-and-genome-composition-assessment.html#cb2-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/Skudr_/Skudr\t/g&#39;</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb2-40"><a href="qc-and-genome-composition-assessment.html#cb2-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/Smika_/Smika\t/g&#39;</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb2-41"><a href="qc-and-genome-composition-assessment.html#cb2-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/Spara_/Spara\t/g&#39;</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb2-42"><a href="qc-and-genome-composition-assessment.html#cb2-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/Suvar_/Suvar\t/g&#39;</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb2-43"><a href="qc-and-genome-composition-assessment.html#cb2-43" aria-hidden="true" tabindex="-1"></a>    <span class="ex">python</span> bed_remove0coverage.py <span class="at">--input</span> <span class="va">$file</span> <span class="kw">|</span><span class="dt">\</span></span>
<span id="cb2-44"><a href="qc-and-genome-composition-assessment.html#cb2-44" aria-hidden="true" tabindex="-1"></a>        <span class="fu">sed</span> <span class="st">&#39;/^[[:space:]]*$/d&#39;</span> <span class="op">&gt;</span> <span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .bed<span class="va">)</span>.no0.bed<span class="kw">;</span></span>
<span id="cb2-45"><a href="qc-and-genome-composition-assessment.html#cb2-45" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb2-46"><a href="qc-and-genome-composition-assessment.html#cb2-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-47"><a href="qc-and-genome-composition-assessment.html#cb2-47" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.no0.bed<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb2-48"><a href="qc-and-genome-composition-assessment.html#cb2-48" aria-hidden="true" tabindex="-1"></a>    <span class="va">NAME</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .align.sort.1kb_cov.no0.bed<span class="va">)</span><span class="kw">;</span></span>
<span id="cb2-49"><a href="qc-and-genome-composition-assessment.html#cb2-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&quot;s/^/</span><span class="va">$NAME</span><span class="st">\t/&quot;</span> <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb2-50"><a href="qc-and-genome-composition-assessment.html#cb2-50" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb2-51"><a href="qc-and-genome-composition-assessment.html#cb2-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-52"><a href="qc-and-genome-composition-assessment.html#cb2-52" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="pp">*</span>.1kb_cov.no0.bed <span class="op">&gt;</span> Viking.species.cov.bed</span>
<span id="cb2-53"><a href="qc-and-genome-composition-assessment.html#cb2-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-54"><a href="qc-and-genome-composition-assessment.html#cb2-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-55"><a href="qc-and-genome-composition-assessment.html#cb2-55" aria-hidden="true" tabindex="-1"></a><span class="co"># generate per chromosomes coverages</span></span>
<span id="cb2-56"><a href="qc-and-genome-composition-assessment.html#cb2-56" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb2-57"><a href="qc-and-genome-composition-assessment.html#cb2-57" aria-hidden="true" tabindex="-1"></a>    <span class="va">OLD</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$line</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-d</span> <span class="st">&quot; &quot;</span> <span class="at">-f</span> 1<span class="va">)</span></span>
<span id="cb2-58"><a href="qc-and-genome-composition-assessment.html#cb2-58" aria-hidden="true" tabindex="-1"></a>    <span class="va">NEW</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$line</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-d</span> <span class="st">&quot; &quot;</span> <span class="at">-f</span> 2<span class="va">)</span></span>
<span id="cb2-59"><a href="qc-and-genome-composition-assessment.html#cb2-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&quot;s/</span><span class="va">$OLD</span><span class="st">\t/</span><span class="va">$NEW</span><span class="st">\t/g&quot;</span> Viking.species.cov.chr.bed</span>
<span id="cb2-60"><a href="qc-and-genome-composition-assessment.html#cb2-60" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span> <span class="op">&lt;</span> ref_chr_names.lst </span>
<span id="cb2-61"><a href="qc-and-genome-composition-assessment.html#cb2-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-62"><a href="qc-and-genome-composition-assessment.html#cb2-62" aria-hidden="true" tabindex="-1"></a><span class="co"># clean output</span></span>
<span id="cb2-63"><a href="qc-and-genome-composition-assessment.html#cb2-63" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.no0.bed<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb2-64"><a href="qc-and-genome-composition-assessment.html#cb2-64" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cp</span> <span class="va">$file</span> <span class="va">$file</span>.1<span class="kw">;</span></span>
<span id="cb2-65"><a href="qc-and-genome-composition-assessment.html#cb2-65" aria-hidden="true" tabindex="-1"></a>    <span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb2-66"><a href="qc-and-genome-composition-assessment.html#cb2-66" aria-hidden="true" tabindex="-1"></a>        <span class="va">OLD</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$line</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-d</span> <span class="st">&quot; &quot;</span> <span class="at">-f</span> 1<span class="va">)</span><span class="kw">;</span></span>
<span id="cb2-67"><a href="qc-and-genome-composition-assessment.html#cb2-67" aria-hidden="true" tabindex="-1"></a>        <span class="va">NEW</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$line</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-d</span> <span class="st">&quot; &quot;</span> <span class="at">-f</span> 2<span class="va">)</span><span class="kw">;</span></span>
<span id="cb2-68"><a href="qc-and-genome-composition-assessment.html#cb2-68" aria-hidden="true" tabindex="-1"></a>        <span class="fu">sed</span> <span class="at">-i</span> <span class="st">&quot;s/</span><span class="va">$OLD</span><span class="st">\t/</span><span class="va">$NEW</span><span class="st">\t/g&quot;</span> <span class="va">$file</span>.1<span class="kw">;</span></span>
<span id="cb2-69"><a href="qc-and-genome-composition-assessment.html#cb2-69" aria-hidden="true" tabindex="-1"></a>    <span class="cf">done</span> <span class="op">&lt;</span> ../00_ref_genomes/ref_chr_names.lst<span class="kw">;</span></span>
<span id="cb2-70"><a href="qc-and-genome-composition-assessment.html#cb2-70" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grep</span> Chr <span class="va">$file</span>.1 <span class="op">&gt;</span> <span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .bed<span class="va">)</span>.chr.bed<span class="kw">;</span></span>
<span id="cb2-71"><a href="qc-and-genome-composition-assessment.html#cb2-71" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rm</span> <span class="va">$file</span>.1 <span class="kw">&amp;</span></span>
<span id="cb2-72"><a href="qc-and-genome-composition-assessment.html#cb2-72" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>We then remove windows with 0 coverage to have a clearer figure, and we plot the coverages with transforming the y axis in log10 scale, since Krogerus kveiks have much higher coverage than the strains we sequenced.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="qc-and-genome-composition-assessment.html#cb3-1" aria-hidden="true" tabindex="-1"></a>data_cov <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="fu">paste</span>(mainDir, <span class="st">&quot;Viking.species.cov.chr.bed&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb3-2"><a href="qc-and-genome-composition-assessment.html#cb3-2" aria-hidden="true" tabindex="-1"></a>data_cov<span class="sc">$</span>V7 <span class="ot">=</span> <span class="fu">ifelse</span>(data_cov<span class="sc">$</span>V6 <span class="sc">&gt;=</span> <span class="dv">1</span>, <span class="st">&quot;Up&quot;</span>, <span class="st">&quot;Down&quot;</span>)</span>
<span id="cb3-3"><a href="qc-and-genome-composition-assessment.html#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="qc-and-genome-composition-assessment.html#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co"># ylim = 1500</span></span>
<span id="cb3-5"><a href="qc-and-genome-composition-assessment.html#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data_cov, <span class="fu">aes</span>(<span class="at">x =</span> V2, <span class="at">y =</span> V6)) <span class="sc">+</span></span>
<span id="cb3-6"><a href="qc-and-genome-composition-assessment.html#cb3-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">color =</span> V7),</span>
<span id="cb3-7"><a href="qc-and-genome-composition-assessment.html#cb3-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">show.legend =</span> <span class="cn">FALSE</span>,</span>
<span id="cb3-8"><a href="qc-and-genome-composition-assessment.html#cb3-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">alpha =</span> <span class="fl">0.035</span>,</span>
<span id="cb3-9"><a href="qc-and-genome-composition-assessment.html#cb3-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">shape =</span> <span class="st">&quot;.&quot;</span>,</span>
<span id="cb3-10"><a href="qc-and-genome-composition-assessment.html#cb3-10" aria-hidden="true" tabindex="-1"></a>              <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="fl">0.475</span>)) <span class="sc">+</span></span>
<span id="cb3-11"><a href="qc-and-genome-composition-assessment.html#cb3-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">linetype =</span> <span class="dv">2</span>, <span class="at">color =</span> <span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb3-12"><a href="qc-and-genome-composition-assessment.html#cb3-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">expand =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb3-13"><a href="qc-and-genome-composition-assessment.html#cb3-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values  =</span> <span class="fu">c</span>(<span class="st">&quot;grey75&quot;</span>, <span class="st">&quot;cyan4&quot;</span>)) <span class="sc">+</span> </span>
<span id="cb3-14"><a href="qc-and-genome-composition-assessment.html#cb3-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">4</span>, <span class="dv">9000</span>),</span>
<span id="cb3-15"><a href="qc-and-genome-composition-assessment.html#cb3-15" aria-hidden="true" tabindex="-1"></a>                     <span class="at">trans =</span> <span class="st">&quot;log2&quot;</span>,</span>
<span id="cb3-16"><a href="qc-and-genome-composition-assessment.html#cb3-16" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">1000</span>),</span>
<span id="cb3-17"><a href="qc-and-genome-composition-assessment.html#cb3-17" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> scales<span class="sc">::</span>comma) <span class="sc">+</span></span>
<span id="cb3-18"><a href="qc-and-genome-composition-assessment.html#cb3-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Coverage of Reference Genomes&quot;</span>,</span>
<span id="cb3-19"><a href="qc-and-genome-composition-assessment.html#cb3-19" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;Reference Species&quot;</span>,</span>
<span id="cb3-20"><a href="qc-and-genome-composition-assessment.html#cb3-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Coverage&quot;</span>) <span class="sc">+</span></span>
<span id="cb3-21"><a href="qc-and-genome-composition-assessment.html#cb3-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>V1, <span class="at">scales =</span> <span class="st">&quot;free_y&quot;</span>) <span class="sc">+</span></span>
<span id="cb3-22"><a href="qc-and-genome-composition-assessment.html#cb3-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">20</span>),</span>
<span id="cb3-23"><a href="qc-and-genome-composition-assessment.html#cb3-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb3-24"><a href="qc-and-genome-composition-assessment.html#cb3-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb3-25"><a href="qc-and-genome-composition-assessment.html#cb3-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major.y =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">&quot;gray75&quot;</span>, <span class="at">size =</span> <span class="fl">0.5</span>),</span>
<span id="cb3-26"><a href="qc-and-genome-composition-assessment.html#cb3-26" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.background =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="at">fill =</span> <span class="cn">NA</span>),</span>
<span id="cb3-27"><a href="qc-and-genome-composition-assessment.html#cb3-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb3-28"><a href="qc-and-genome-composition-assessment.html#cb3-28" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb3-29"><a href="qc-and-genome-composition-assessment.html#cb3-29" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.65</span>))</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-5"></span>
<img src="_main_files/figure-html/unnamed-chunk-5-1.png" alt="Coverage of Reference Genomes" width="672" />
<p class="caption">
Figure 1.1: Coverage of Reference Genomes
</p>
</div>
<p>All sequenced farmhouse yeasts are pure <em>S. cerevisiae</em> strains, except for Muri and k7R25 that are <em>S. cerevisiae</em>, <em>S. eubayanus</em> and <em>S. uvarum</em> triple hybrids.</p>
<p>We can dig in into the coverage of the single chromosomes and visualize the extent of hybridization and eventual aneuploidies.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="qc-and-genome-composition-assessment.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="do">### PER SAMPLE COVERAGE</span></span>
<span id="cb4-2"><a href="qc-and-genome-composition-assessment.html#cb4-2" aria-hidden="true" tabindex="-1"></a>sample_list <span class="ot">=</span> <span class="fu">unique</span>(data_cov<span class="sc">$</span>V1)</span>
<span id="cb4-3"><a href="qc-and-genome-composition-assessment.html#cb4-3" aria-hidden="true" tabindex="-1"></a>r <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb4-4"><a href="qc-and-genome-composition-assessment.html#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="qc-and-genome-composition-assessment.html#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co"># I iterate through the samples</span></span>
<span id="cb4-6"><a href="qc-and-genome-composition-assessment.html#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(sample_list)){</span>
<span id="cb4-7"><a href="qc-and-genome-composition-assessment.html#cb4-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-8"><a href="qc-and-genome-composition-assessment.html#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># I read the sample name and restrict the plot to that sample</span></span>
<span id="cb4-9"><a href="qc-and-genome-composition-assessment.html#cb4-9" aria-hidden="true" tabindex="-1"></a>  name <span class="ot">=</span> sample_list[i]</span>
<span id="cb4-10"><a href="qc-and-genome-composition-assessment.html#cb4-10" aria-hidden="true" tabindex="-1"></a>  file <span class="ot">=</span> <span class="fu">paste</span>(mainDir, <span class="st">&quot;/01_start_bed/&quot;</span>,name,<span class="st">&quot;.align.sort.1kb_cov.no0.chr.bed&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb4-11"><a href="qc-and-genome-composition-assessment.html#cb4-11" aria-hidden="true" tabindex="-1"></a>  dataset <span class="ot">=</span> <span class="fu">read.delim</span>(file, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb4-12"><a href="qc-and-genome-composition-assessment.html#cb4-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-13"><a href="qc-and-genome-composition-assessment.html#cb4-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># plot</span></span>
<span id="cb4-14"><a href="qc-and-genome-composition-assessment.html#cb4-14" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">=</span> <span class="fu">ggplot</span>(dataset, <span class="fu">aes</span>(V5, V6, <span class="at">colour =</span> V2)) <span class="sc">+</span></span>
<span id="cb4-15"><a href="qc-and-genome-composition-assessment.html#cb4-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb4-16"><a href="qc-and-genome-composition-assessment.html#cb4-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">1.5</span>, <span class="dv">9000</span>),</span>
<span id="cb4-17"><a href="qc-and-genome-composition-assessment.html#cb4-17" aria-hidden="true" tabindex="-1"></a>                       <span class="at">trans =</span> <span class="st">&quot;log2&quot;</span>,</span>
<span id="cb4-18"><a href="qc-and-genome-composition-assessment.html#cb4-18" aria-hidden="true" tabindex="-1"></a>                       <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">1000</span>),</span>
<span id="cb4-19"><a href="qc-and-genome-composition-assessment.html#cb4-19" aria-hidden="true" tabindex="-1"></a>                       <span class="at">labels =</span> scales<span class="sc">::</span>comma) <span class="sc">+</span></span>
<span id="cb4-20"><a href="qc-and-genome-composition-assessment.html#cb4-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">breaks =</span> (<span class="fu">seq</span>(<span class="dv">0</span>,<span class="fl">2e6</span>,<span class="dv">10000</span>))) <span class="sc">+</span></span>
<span id="cb4-21"><a href="qc-and-genome-composition-assessment.html#cb4-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_cartesian</span>(<span class="at">expand =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb4-22"><a href="qc-and-genome-composition-assessment.html#cb4-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(V2<span class="sc">~</span>dataset<span class="sc">$</span>V3,</span>
<span id="cb4-23"><a href="qc-and-genome-composition-assessment.html#cb4-23" aria-hidden="true" tabindex="-1"></a>               <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>,</span>
<span id="cb4-24"><a href="qc-and-genome-composition-assessment.html#cb4-24" aria-hidden="true" tabindex="-1"></a>               <span class="at">space =</span> <span class="st">&quot;free_x&quot;</span>,</span>
<span id="cb4-25"><a href="qc-and-genome-composition-assessment.html#cb4-25" aria-hidden="true" tabindex="-1"></a>               <span class="at">switch =</span> <span class="st">&quot;y&quot;</span>) <span class="sc">+</span></span>
<span id="cb4-26"><a href="qc-and-genome-composition-assessment.html#cb4-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> name,</span>
<span id="cb4-27"><a href="qc-and-genome-composition-assessment.html#cb4-27" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">&quot;Chromosome&quot;</span>,</span>
<span id="cb4-28"><a href="qc-and-genome-composition-assessment.html#cb4-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">&quot;Genome coverage&quot;</span>,</span>
<span id="cb4-29"><a href="qc-and-genome-composition-assessment.html#cb4-29" aria-hidden="true" tabindex="-1"></a>         <span class="at">color =</span> <span class="st">&quot;Genomes&quot;</span>) <span class="sc">+</span></span>
<span id="cb4-30"><a href="qc-and-genome-composition-assessment.html#cb4-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">20</span>),</span>
<span id="cb4-31"><a href="qc-and-genome-composition-assessment.html#cb4-31" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb4-32"><a href="qc-and-genome-composition-assessment.html#cb4-32" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-33"><a href="qc-and-genome-composition-assessment.html#cb4-33" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.spacing.x =</span> <span class="fu">unit</span>(<span class="dv">0</span>, <span class="st">&quot;lines&quot;</span>),</span>
<span id="cb4-34"><a href="qc-and-genome-composition-assessment.html#cb4-34" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>, <span class="at">fill =</span> <span class="cn">NA</span>),</span>
<span id="cb4-35"><a href="qc-and-genome-composition-assessment.html#cb4-35" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.background =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>, <span class="at">fill =</span> <span class="cn">NA</span>))</span>
<span id="cb4-36"><a href="qc-and-genome-composition-assessment.html#cb4-36" aria-hidden="true" tabindex="-1"></a>  r[[name]] <span class="ot">=</span> p</span>
<span id="cb4-37"><a href="qc-and-genome-composition-assessment.html#cb4-37" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-38"><a href="qc-and-genome-composition-assessment.html#cb4-38" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;14R30&quot;</span>, <span class="st">&quot;Muri&quot;</span>, <span class="st">&quot;k7R25&quot;</span>)){</span>
<span id="cb4-39"><a href="qc-and-genome-composition-assessment.html#cb4-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(p)</span>
<span id="cb4-40"><a href="qc-and-genome-composition-assessment.html#cb4-40" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-41"><a href="qc-and-genome-composition-assessment.html#cb4-41" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-6-1"></span>
<img src="_main_files/figure-html/unnamed-chunk-6-1.png" alt="Sub-genomes coverages" width="672" />
<p class="caption">
Figure 1.2: Sub-genomes coverages
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-6-2"></span>
<img src="_main_files/figure-html/unnamed-chunk-6-2.png" alt="Sub-genomes coverages" width="672" />
<p class="caption">
Figure 1.3: Sub-genomes coverages
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-6-3"></span>
<img src="_main_files/figure-html/unnamed-chunk-6-3.png" alt="Sub-genomes coverages" width="672" />
<p class="caption">
Figure 1.4: Sub-genomes coverages
</p>
</div>
<p>Muri and k7R25 are triple hybrids <em>S. cerevisiae</em> X <em>S. eubayanus</em> X <em>S. uvarum</em>. What is extremely interesting is that they share the same genomic composition in terms of origin of the chromosomes, but they have distinct CNVs patterns. I would suggest a common origin for such hybrid.</p>
</div>
<div id="ploydies-estimation" class="section level2 hasAnchor" number="1.3">
<h2><span class="header-section-number">1.3</span> Ploydies estimation<a href="qc-and-genome-composition-assessment.html#ploydies-estimation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We can estimates Kveiks ploidy by looking at the k-mer frequency distributions of the sequenced reads obtained from whole genome sequencing.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="qc-and-genome-composition-assessment.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate kmer freq</span></span>
<span id="cb5-2"><a href="qc-and-genome-composition-assessment.html#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> 01_alignments/<span class="pp">*</span>bam<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb5-3"><a href="qc-and-genome-composition-assessment.html#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="ex">~/bin/ntCard/bin/ntcard</span> <span class="at">--threads</span><span class="op">=</span>64 <span class="at">--kmer</span><span class="op">=</span>41 <span class="dt">\</span></span>
<span id="cb5-4"><a href="qc-and-genome-composition-assessment.html#cb5-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">--pref</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .align.sort.md.r.bam<span class="va">)</span>.kmer41freq <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb5-5"><a href="qc-and-genome-composition-assessment.html#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb5-6"><a href="qc-and-genome-composition-assessment.html#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="co"># merge outputs</span></span>
<span id="cb5-7"><a href="qc-and-genome-composition-assessment.html#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.hist<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb5-8"><a href="qc-and-genome-composition-assessment.html#cb5-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tail</span> <span class="at">-n</span> +3 <span class="va">$file</span> <span class="op">&gt;</span> <span class="va">$file</span>.1<span class="kw">;</span></span>
<span id="cb5-9"><a href="qc-and-genome-composition-assessment.html#cb5-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mv</span> <span class="va">$file</span>.1 <span class="va">$file</span><span class="kw">;</span></span>
<span id="cb5-10"><a href="qc-and-genome-composition-assessment.html#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span> <span class="va">$file</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&quot;s/^/</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">$file</span> .kmer41freq_k41.hist<span class="va">)</span><span class="st">\t/g&quot;</span><span class="kw">;</span></span>
<span id="cb5-11"><a href="qc-and-genome-composition-assessment.html#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span> <span class="op">&gt;</span> Viking.species.kmer41freq.txt</span></code></pre></div>
<p>Kmer frequency distribution plots can suggest us the degree of heterozygosity of kveiks genomes. All Kveiks seems to have a high degree of heterozygosity, except for hybrid strains Muri and k7R25. This support the idea that the sequenced kveiks so far are tetraploid <em>S. cerevisiae</em>, while Muri and k7R25 are triploid hybrids <em>S. cerevisiae</em> X <em>Seubayanus</em> X <em>S. uvarum</em>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="qc-and-genome-composition-assessment.html#cb6-1" aria-hidden="true" tabindex="-1"></a>data_k <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="fu">paste</span>(mainDir, <span class="st">&quot;Viking.species.kmer41freq.mod.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb6-2"><a href="qc-and-genome-composition-assessment.html#cb6-2" aria-hidden="true" tabindex="-1"></a>data_k<span class="sc">$</span>V2 <span class="ot">=</span> <span class="fu">as.numeric</span>(data_k<span class="sc">$</span>V2)</span>
<span id="cb6-3"><a href="qc-and-genome-composition-assessment.html#cb6-3" aria-hidden="true" tabindex="-1"></a>data_k<span class="sc">$</span>V3 <span class="ot">=</span> <span class="fu">as.numeric</span>(data_k<span class="sc">$</span>V3)</span>
<span id="cb6-4"><a href="qc-and-genome-composition-assessment.html#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="qc-and-genome-composition-assessment.html#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="co"># plot kmer distributions</span></span>
<span id="cb6-6"><a href="qc-and-genome-composition-assessment.html#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data_k, <span class="fu">aes</span>(<span class="at">x =</span> V2, <span class="at">y =</span> V3)) <span class="sc">+</span></span>
<span id="cb6-7"><a href="qc-and-genome-composition-assessment.html#cb6-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">&quot;steelblue&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb6-8"><a href="qc-and-genome-composition-assessment.html#cb6-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> <span class="cf">function</span>(x) <span class="fu">format</span>(x, <span class="at">big.mark =</span> <span class="st">&quot;,&quot;</span>, <span class="at">scientific =</span> <span class="cn">FALSE</span>)) <span class="sc">+</span></span>
<span id="cb6-9"><a href="qc-and-genome-composition-assessment.html#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>V1, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>) <span class="sc">+</span></span>
<span id="cb6-10"><a href="qc-and-genome-composition-assessment.html#cb6-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;K-mer frequencies distribution&quot;</span>,</span>
<span id="cb6-11"><a href="qc-and-genome-composition-assessment.html#cb6-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;k-mer depth&quot;</span>,</span>
<span id="cb6-12"><a href="qc-and-genome-composition-assessment.html#cb6-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;k-mer frequency&quot;</span>) <span class="sc">+</span></span>
<span id="cb6-13"><a href="qc-and-genome-composition-assessment.html#cb6-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb6-14"><a href="qc-and-genome-composition-assessment.html#cb6-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb6-15"><a href="qc-and-genome-composition-assessment.html#cb6-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-16"><a href="qc-and-genome-composition-assessment.html#cb6-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-17"><a href="qc-and-genome-composition-assessment.html#cb6-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-18"><a href="qc-and-genome-composition-assessment.html#cb6-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="at">fill =</span> <span class="cn">NA</span>),</span>
<span id="cb6-19"><a href="qc-and-genome-composition-assessment.html#cb6-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.background =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="at">fill =</span> <span class="cn">NA</span>),</span>
<span id="cb6-20"><a href="qc-and-genome-composition-assessment.html#cb6-20" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>))</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-8"></span>
<img src="_main_files/figure-html/unnamed-chunk-8-1.png" alt="K-mer frequencies distribution" width="672" />
<p class="caption">
Figure 1.5: K-mer frequencies distribution
</p>
</div>
<p>We compare the predicted ploidy results with the ones of 18 <em>S. cerevisiae</em> industrial strains that will be used across this whole study as comparison.</p>
<div id="htmlwidget-a2aec5193653e8eb59c4" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a2aec5193653e8eb59c4">{"x":{"filter":"none","vertical":false,"extensions":["FixedColumns","FixedHeader"],"caption":"<caption>Yeast Reference Genomes<\/caption>","data":[["X1002","X1014","X1018","X1030","X1129","X5508","X1005","X1007","X1118","X1131","X1148","X7078-B","X1046","X4009","X4037","X4040","X4042","X4051"],["Beer1","Beer1","Beer1","Beer1","Beer1","Beer1","Beer2","Beer2","Beer2","Beer2","Beer2","Beer2","Wine","Wine","Wine","Wine","Wine","Wine"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Strain<\/th>\n      <th>Industrial niche<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"paging":true,"fixedHeader":false,"columnDefs":[],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="qc-and-genome-composition-assessment.html#cb7-1" aria-hidden="true" tabindex="-1"></a>data_ay <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="fu">paste</span>(mainDir, <span class="st">&quot;additional_yeasts.kmer41freq.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb7-2"><a href="qc-and-genome-composition-assessment.html#cb7-2" aria-hidden="true" tabindex="-1"></a>data_ay<span class="sc">$</span>V2 <span class="ot">=</span> <span class="fu">as.numeric</span>(data_ay<span class="sc">$</span>V2)</span>
<span id="cb7-3"><a href="qc-and-genome-composition-assessment.html#cb7-3" aria-hidden="true" tabindex="-1"></a>data_ay<span class="sc">$</span>V3 <span class="ot">=</span> <span class="fu">as.numeric</span>(data_ay<span class="sc">$</span>V3)</span>
<span id="cb7-4"><a href="qc-and-genome-composition-assessment.html#cb7-4" aria-hidden="true" tabindex="-1"></a>data_ay<span class="sc">$</span>V4 <span class="ot">=</span> <span class="fu">ifelse</span>(data_ay<span class="sc">$</span>V1 <span class="sc">==</span> <span class="st">&quot;X1002&quot;</span> <span class="sc">|</span> data_ay<span class="sc">$</span>V1 <span class="sc">==</span> <span class="st">&quot;X1014&quot;</span> <span class="sc">|</span> data_ay<span class="sc">$</span>V1 <span class="sc">==</span> <span class="st">&quot;X1018&quot;</span> <span class="sc">|</span> data_ay<span class="sc">$</span>V1 <span class="sc">==</span> <span class="st">&quot;X1030&quot;</span> <span class="sc">|</span> data_ay<span class="sc">$</span>V1 <span class="sc">==</span> <span class="st">&quot;X1129&quot;</span> <span class="sc">|</span> data_ay<span class="sc">$</span>V1 <span class="sc">==</span> <span class="st">&quot;X5508&quot;</span>, <span class="st">&quot;Beer1&quot;</span>,</span>
<span id="cb7-5"><a href="qc-and-genome-composition-assessment.html#cb7-5" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">ifelse</span>(data_ay<span class="sc">$</span>V1 <span class="sc">==</span> <span class="st">&quot;X1005&quot;</span> <span class="sc">|</span> data_ay<span class="sc">$</span>V1 <span class="sc">==</span> <span class="st">&quot;X1007&quot;</span> <span class="sc">|</span> data_ay<span class="sc">$</span>V1 <span class="sc">==</span> <span class="st">&quot;X1118&quot;</span> <span class="sc">|</span> data_ay<span class="sc">$</span>V1 <span class="sc">==</span> <span class="st">&quot;X1131&quot;</span> <span class="sc">|</span> data_ay<span class="sc">$</span>V1 <span class="sc">==</span> <span class="st">&quot;X1148&quot;</span> <span class="sc">|</span> data_ay<span class="sc">$</span>V1 <span class="sc">==</span><span class="st">&quot;x7078-B&quot;</span>, <span class="st">&quot;Beer2&quot;</span>, <span class="st">&quot;Wine&quot;</span>))</span>
<span id="cb7-6"><a href="qc-and-genome-composition-assessment.html#cb7-6" aria-hidden="true" tabindex="-1"></a>data_ay<span class="sc">$</span>V1 <span class="ot">=</span> <span class="fu">factor</span>(data_ay<span class="sc">$</span>V1,</span>
<span id="cb7-7"><a href="qc-and-genome-composition-assessment.html#cb7-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;X1002&quot;</span>, <span class="st">&quot;X1014&quot;</span>, <span class="st">&quot;X1018&quot;</span>, <span class="st">&quot;X1030&quot;</span>, <span class="st">&quot;X1129&quot;</span>, <span class="st">&quot;X5508&quot;</span>,</span>
<span id="cb7-8"><a href="qc-and-genome-composition-assessment.html#cb7-8" aria-hidden="true" tabindex="-1"></a>                               <span class="st">&quot;X1005&quot;</span>, <span class="st">&quot;X1007&quot;</span>, <span class="st">&quot;X1118&quot;</span>, <span class="st">&quot;X1131&quot;</span>, <span class="st">&quot;X1148&quot;</span>, <span class="st">&quot;x7078-B&quot;</span>,</span>
<span id="cb7-9"><a href="qc-and-genome-composition-assessment.html#cb7-9" aria-hidden="true" tabindex="-1"></a>                               <span class="st">&quot;X1046&quot;</span>, <span class="st">&quot;X4009&quot;</span>, <span class="st">&quot;X4037&quot;</span>, <span class="st">&quot;X4040&quot;</span>, <span class="st">&quot;X4042&quot;</span>, <span class="st">&quot;X4051&quot;</span>))</span>
<span id="cb7-10"><a href="qc-and-genome-composition-assessment.html#cb7-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-11"><a href="qc-and-genome-composition-assessment.html#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="co"># plot kmer distributions</span></span>
<span id="cb7-12"><a href="qc-and-genome-composition-assessment.html#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data_ay, <span class="fu">aes</span>(<span class="at">x =</span> V2, <span class="at">y =</span> V3)) <span class="sc">+</span></span>
<span id="cb7-13"><a href="qc-and-genome-composition-assessment.html#cb7-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">color =</span> V4), <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb7-14"><a href="qc-and-genome-composition-assessment.html#cb7-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> <span class="cf">function</span>(x) <span class="fu">format</span>(x, <span class="at">big.mark =</span> <span class="st">&quot;,&quot;</span>, <span class="at">scientific =</span> <span class="cn">FALSE</span>)) <span class="sc">+</span></span>
<span id="cb7-15"><a href="qc-and-genome-composition-assessment.html#cb7-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">10</span>, <span class="dv">200</span>) <span class="sc">+</span></span>
<span id="cb7-16"><a href="qc-and-genome-composition-assessment.html#cb7-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">515000</span>) <span class="sc">+</span></span>
<span id="cb7-17"><a href="qc-and-genome-composition-assessment.html#cb7-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>V1, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">ncol =</span> <span class="dv">6</span>) <span class="sc">+</span></span>
<span id="cb7-18"><a href="qc-and-genome-composition-assessment.html#cb7-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;K-mer frequencies distribution&quot;</span>,</span>
<span id="cb7-19"><a href="qc-and-genome-composition-assessment.html#cb7-19" aria-hidden="true" tabindex="-1"></a>       <span class="at">colour =</span> <span class="st">&quot;Industrial</span><span class="sc">\n</span><span class="st"> niche&quot;</span>,</span>
<span id="cb7-20"><a href="qc-and-genome-composition-assessment.html#cb7-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;k-mer depth&quot;</span>,</span>
<span id="cb7-21"><a href="qc-and-genome-composition-assessment.html#cb7-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;k-mer frequency&quot;</span>) <span class="sc">+</span></span>
<span id="cb7-22"><a href="qc-and-genome-composition-assessment.html#cb7-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb7-23"><a href="qc-and-genome-composition-assessment.html#cb7-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb7-24"><a href="qc-and-genome-composition-assessment.html#cb7-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb7-25"><a href="qc-and-genome-composition-assessment.html#cb7-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb7-26"><a href="qc-and-genome-composition-assessment.html#cb7-26" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb7-27"><a href="qc-and-genome-composition-assessment.html#cb7-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="at">fill =</span> <span class="cn">NA</span>),</span>
<span id="cb7-28"><a href="qc-and-genome-composition-assessment.html#cb7-28" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.background =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="at">fill =</span> <span class="cn">NA</span>),</span>
<span id="cb7-29"><a href="qc-and-genome-composition-assessment.html#cb7-29" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>))</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-9"></span>
<img src="_main_files/figure-html/unnamed-chunk-9-1.png" alt="K-mer frequencies distribution 
 Industrail strains" width="672" />
<p class="caption">
Figure 1.6: K-mer frequencies distribution
Industrail strains
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="copy-number-variants.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/01-SequencingQC.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
